# Manipulating Tree with Data {#chapter2}


```{r include=FALSE}
library(knitr)
opts_chunk$set(message=FALSE, warning=FALSE, eval=TRUE, echo=TRUE, cache=TRUE)
```



```{r echo=FALSE, results="hide", message=FALSE}
library("ape")
library("tidytree")
library(treeio)
```

## Manipulating tree data using tidy interface {#tidytree}

All the tree data parsed/merged
by [treeio](https://bioconductor.org/packages/treeio/) can be converted to tidy
data frame using the [tidytree](https://cran.r-project.org/package=tidytree)
package. The [tidytree](https://cran.r-project.org/package=tidytree) package
provides tidy interfaces to manipulate tree with associated data. For instances,
external data can be linked to phylogeny or evolutionary data obtained from
different sources can be merged using tidyverse verbs. After the tree data was
manipulated, it can be converted back to `treedata` object and exported to a
single tree file, further analyzed in R or visualized using [ggtree](https://bioconductor.org/packages/ggtree/) [@yu_ggtree:_2017].


### `phylo` object


The `phylo` class defined in [ape](https://cran.r-project.org/package=ape) is
fundamental for phylogenetic analysis in R. Most of the R packages
in [this field](https://CRAN.R-project.org/view=Phylogenetics) rely
extensively on `phylo` object. The tidytree package provides `as_tibble`
method to convert the `phylo` object to tidy data frame, a `tbl_tree` object.

```{r}
library(ape)
set.seed(2017)
tree <- rtree(4)
tree
x <- as_tibble(tree)
x
```

The `tbl_tree` object can be converted back to a `phylo` object.


```{r}
as.phylo(x)
```


Using `tbl_tree` object makes tree and data manipulation more effective and
easier. For example, we can link evolutionary trait to phylogeny using the verbs `full_join`

```{r}
d <- tibble(label = paste0('t', 1:4),
            trait = rnorm(4))

y <- full_join(x, d, by = 'label')
y
```

### treedata object


The tidytree package defines a `treedata` class to store phylogenetic tree with
associated data. After mapping external data to the tree structure, the
`tbl_tree` object can be converted to a `treedata` object.


```{r}
as.treedata(y)
```

The `treedata` class is also used
in [treeio](https://bioconductor.org/packages/treeio/) package to store
evolutionary evidences inferred by commonly used software (BEAST, EPA, HYPHY,
MrBayes, PAML, PHYLODOG, pplacer, r8s, RAxML and RevBayes) (see details in [Chapter 1](#chapter1)).

The tidytree package also provides `as_tibble` to convert `treedata` object
to a tidy data frame. The phylogentic tree structure and the evolutionary
inferences were stored in the `tbl_tree` object, making it consistent and easier
for manipulating evolutionary statistics inferred by different software as well
as linking external data to the same tree structure.


```{r}
y %>% as.treedata %>% as_tibble
```



### Access related nodes {#accesor-tidytree}

[dplyr](https://cran.r-project.org/package=dplyr) verbs can be applied to `tbl_tree` directly to manipulate tree data. In
addition, tidytree provides several verbs to filter related nodes, including
`child`, `parent`, `offspring`, `ancestor`, `sibling` and `MRCA`.

These verbs accept a `tbl_tree` and a selected node which can be node number or label.


```{r}
child(y, 5)
parent(y, 2)
offspring(y, 5)
ancestor(y, 2)
sibling(y, 2)
MRCA(y, 2, 3)
```

## Data Integration

### Combining tree data


The [treeio](https://bioconductor.org/packages/treeio/) package serves as an
infrastructure that enables various types of phylogenetic data inferred from
common analysis programs to be imported and used in R. For instance *d~N~/d~S~*
or ancestral sequences estimated
by [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html),
and clade support values (posterior) inferred
by [BEAST](http://beast2.org/)/[MrBayes](http://nbisweden.github.io/MrBayes/).
In addition, [treeio](https://bioconductor.org/packages/treeio/) package
supports linking external data to phylogeny. It brings these external
phylogenetic data (either from software output or exteranl sources) to the R
community and make it available for further analysis in R.
Furthermore, [treeio](https://bioconductor.org/packages/treeio/) can combine
multiple phylogenetic trees together into one with their node/branch-specific
attribute data. Essentially, as a result, one such attribute (*e.g.*,
substitution rate) can be mapped to another attribute (*e.g.*, *d~N~/d~S~*) of
the same node/branch for comparison and further computations.


A previously published data set, seventy-six H3 hemagglutinin gene sequences of
a lineage containing swine and human influenza A viruses
[@liang_expansion_2014], was here to demonstrate the utilities of comparing
evolutionary statistics inferred by different software. The dataset was
re-analyzed by [BEAST](http://beast2.org/) for timescale estimation
and [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) for synonymous and
non-synonymous substitution estimation. In this example, we first parsed the
outputs from [BEAST](http://beast2.org/) using `read.beast` and
from [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) using
`read.codeml` into two `treedata` objects. Then the two objects containing
separate sets of node/branch-specific data were merged via the `merge_tree` function.



```{r}
beast_file <- system.file("examples/MCC_FluA_H3.tree", package="ggtree")
rst_file <- system.file("examples/rst", package="ggtree")
mlc_file <- system.file("examples/mlc", package="ggtree")
beast_tree <- read.beast(beast_file)
codeml_tree <- read.codeml(rst_file, mlc_file)

merged_tree <- merge_tree(beast_tree, codeml_tree)
merged_tree
```

After merging the `beast_tree` and `codeml_tree` objects, all
node/branch-specific data imported from [BEAST](http://beast2.org/)
and [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) output files are
all available in the `merged_tree` object. The tree object was converted to
tidy data frame using [tidytree](https://cran.r-project.org/package=tidytree)
package and visualized as hexbin scatterplot of *d~N~/d~S~*, *d~N~* and *d~S~* inferred
by [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) versus *rate*
(substitution rate in unit of substitutions/site/year) inferred
by [BEAST](http://beast2.org/) on the same branches.


(ref:correlationscap) Correlation of *d~N~/d~S~*, *d~N~* and *d~S~* versus substitution rate.

(ref:correlationcap) **Correlation of *d~N~/d~S~*, *d~N~* and *d~S~* versus substitution rate.** After merging the *BEAST* and *CodeML* outputs, the branch-specific estimates (substitution rate, *d~N~/d~S~* , *d~N~* and *d~S~*) from the two analysis programs are compared on the same branch basis. The associations of *d~N~/d~S~*, *d~N~* and *d~S~* vs. *rate* are visualized in hexbin scatter plots.


```{r include=FALSE}
ggplot2::theme_set(theme_grey())
```

```{r correlations, fig.width=9, fig.height=4, warning=FALSE, fig.cap="(ref:correlationcap)", fig.scap="(ref:correlationscap)", out.extra=''}
library(dplyr)
df <- fortify(merged_tree) %>%
  select(dN_vs_dS, dN, dS, rate) %>%
  subset(dN_vs_dS >=0 & dN_vs_dS <= 1.5) %>%
  tidyr::gather(type, value, dN_vs_dS:dS)
df$type[df$type == 'dN_vs_dS'] <- 'dN/dS'
df$type <- factor(df$type, levels=c("dN/dS", "dN", "dS"))
ggplot(df, aes(rate, value)) + geom_hex() + 
  facet_wrap(~type, scale='free_y') 
```


The output is illustrated in Fig. \@ref(fig:correlations). We can then test the association of these node/branch-specific data using Pearson correlation, which in this case showed that *d~N~* and *d~S~*, but not *d~N~/d~S~*\index{d\textsubscript{N}/d\textsubscript{S}}  are significantly (*p*-values) associated with *rate*.


Using `merge_tree`, we are able to compare analysis results using identical
model from different software packages or different models using different or
identical software. It also allows users to integrate different analysis finding
from different software packages. Merging tree data is not restricted to
software findings, associating external data to analysis findings is also
granted. The `merge_tree` function is chainable and allows several tree objects
to be merged into one.


```{r}
phylo <- as.phylo(beast_tree)
N <- Nnode2(phylo)
d <- tibble(node = 1:N, fake_trait = rnorm(N), another_trait = runif(N))
fake_tree <- treedata(phylo = phylo, data = d)
triple_tree <- merge_tree(merged_tree, fake_tree)
triple_tree
```

The `triple_tree` object showed above contains analysis results obtained from [BEAST](http://beast2.org/)
and [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html), and evolutionary
trait from external sources. All these information can be used to annotate the
tree using [ggtree](https://bioconductor.org/packages/ggtree/) [@yu_ggtree:_2017].




### Linking external data to phylogeny


In addition to analysis findings that are associated with the tree as we showed
above, there is a wide range of heterogeneous data, including phenotypic data,
experimental data and clinical data *etc.*, that need to be integrated and
linked to phylogeny. For example, in the study of viral evolution, tree nodes may
associated with epidemiological information, such as location, age and subtype.
Functional annotations may need to be mapped on gene trees for comparative
genomics studies. To facilitate data
integration, [treeio](https://bioconductor.org/packages/treeio) provides
`full_join` method to link external data to phylogeny and stored in `treedata` object.


Here are examples of linking external data to a phylogenetic tree. After that,
we can use [exporter](Exporter.html) to combine the tree and the data to a
single tree file. The data that mapped on the phylogenetic tree can also be used to visualize or
annotate the tree using [ggtree](https://bioconductor.org/packages/ggtree/)
[@yu_ggtree:_2017].


```{r}
file <- system.file("extdata/BEAST", "beast_mcc.tree", package="treeio")
beast <- read.beast(file)
x <- data_frame(label = as.phylo(beast)$tip.label, trait = rnorm(Ntip(beast)))
full_join(beast, x, by="label")

N <- Nnode2(beast)
y <- data_frame(node = 1:N, fake_trait = rnorm(N), another_trait = runif(N))
full_join(beast, y, by="node")
```


Manipulating tree object is frustrated with the fragmented functions available
for working with `phylo` object, not to mention linking external data to the
phylogeny structure. With `r Biocpkg("treeio")` package, it is easy to combine tree data from various sources. 
In addition, with `r CRANpkg("tidytree")` package ([session 1.2](#tidytree)), manipulating tree is more easier using 
[tidy data principles](https://www.jstatsoft.org/article/view/v059i10) and 
consistent with tools already in wide use, including
[dplyr](https://cran.r-project.org/package=dplyr),
[tidyr](https://cran.r-project.org/package=tidyr),
[ggplot2](https://cran.r-project.org/package=ggplot2)
and [ggtree](https://bioconductor.org/packages/ggtree/).




### Grouping taxa

tidytree implemented `groupOTU` and `groupClade` for adding taxa grouping
information to the input `tbl_tree` object. These grouping information can be
used directly in tree visualization (*e.g.* coloring tree based on grouping)
with [ggtree](https://bioconductor.org/packages/ggtree/).

#### groupClade

The `groupClade` method accepts an internal node or a vector of internal nodes
to add grouping information of clade/clades.



```{r}
nwk <- '(((((((A:4,B:4):6,C:5):8,D:6):3,E:21):10,((F:4,G:12):14,H:8):13):13,((I:5,J:2):30,(K:11,L:11):2):17):4,M:56);'
tree <- read.tree(text=nwk)

groupClade(as_tibble(tree), c(17, 21))
```


#### groupOTU


```{r}
set.seed(2017)
x <- as_tibble(rtree(4))
## the input nodes can be node ID or label
groupOTU(x, c('t1', 't4'), group_name = "fake_group")
```

The `groupOTU` will trace back from input nodes to most recent common ancestor.
In this example, nodes 2, 3, 7 and 6 (`2 (t1) -> 7 -> 6` and `3 (t4) -> 6`) are
grouping together.

Related OTUs are grouping together and they are not necessarily within a clade.
They can be monophyletic (clade), polyphyletic or paraphyletic.

```{r}
cls <- list(c1=c("A", "B", "C", "D", "E"),
            c2=c("F", "G", "H"),
            c3=c("L", "K", "I", "J"),
            c4="M")

as_tibble(tree) %>% groupOTU(cls)
```

If there are conflicts when tracing back to mrca, user can set `overlap`
parameter to "origin" (the first one counts), "overwrite" (default, the last one
counts) or "abandon" (un-selected for grouping), see
also
[discussion here](https://groups.google.com/forum/#!msg/bioc-ggtree/Q4LnwoTf1DM/uqYdYB_VBAAJ).


## Rescaling Tree Branches {#rescale-treeio}

Phylogenetic data can be merged for joint analysis (Figure \@ref(fig:correlations)). They can be displayed on the same tree structure as more complex annotation to help visually inspection of their evolutionary patterns. All the numerical data stored in `treedata` object can be used to re-scale tree branches. For example, CodeML infers d~N~/d~S~, d~N~ and d~S~, all these statistics can be used as branch lengths.



(ref:rescalescap) Re-scaling tree branches.

(ref:rescalecap) **Re-scaling tree branches.** The tree with branches scaled in time (year from the root) (A). The tree was re-scaled using *d~N~* as branch lengths (B). The tree was re-scaled using substitution rates (C).

```{r rescale, fig.width=12, fig.height=4.5, message=F, echo=T, fig.cap="(ref:rescalecap)", fig.scap="(ref:rescalescap)", out.extra=''}
p1 <- ggtree(merged_tree) + theme_tree2()
p2 <- ggtree(rescale_tree(merged_tree, 'dN')) + theme_tree2()
p3 <- ggtree(rescale_tree(merged_tree, 'rate')) + theme_tree2()

cowplot::plot_grid(p1, p2, p3, ncol=3, labels = LETTERS)
```


## Subsetting Tree with Data

### Remove tips in a phylogenetic tree

### Subset tree by tip label


## Manipulating tree data for visualization {#ggtree-fortify}

Tree visualization is supported by [ggtree](#chapter4). Although *ggtree* implemented several methods for [visual exploration of tree with data](#chapter6), you may want to do something that is not supported directly. In this case, you need to manipulate tree data with node coordination positions that used for visualization. This is quite easy with *ggtree*. User can use `foritfy` method which internally call `tidytree::as_tibble` to convert the tree to tidy data frame and add columns of coordination positions (*i.e.* x, y, branch and angle) that are used to plot the tree. You can also access the data via `ggtree(tree)$data`.

Here is an example to plot two trees face to face that is similar to a `cophyloplot`.

```{r}
library(dplyr)
library(ggtree)

x <- rtree(30)
y <- rtree(30)
p1 <- ggtree(x)
p2 <- ggtree(y)

d1 <- p1$data
d2 <- p2$data

## reverse x-axis and 
## set offset to make the tree in the right hand side of the first tree
d2$x <- max(d2$x) - d2$x + max(d1$x) + 1

pp <- p1 + geom_tiplab() + geom_tree(data=d2) + geom_tiplab(data = d2, hjust=1)

dd <- bind_rows(d1, d2) %>% 
  filter(!is.na(label))

pp + geom_line(aes(x, y, group=label), data=dd, color='grey')
```

It is quite easy to plot multiple trees and connect taxa in one figure. For instance, plotting trees contructed from all internal gene segments of influenza virus and connecting equivalent strans across the trees [@venkatesh_avian_2018]. 


```{r}
z <- rtree(30)
d2 <- fortify(y)
d3 <- fortify(z)
d2$x <- d2$x + max(d1$x) + 1
d3$x <- d3$x + max(d2$x) + 1

pp <- p1 + geom_tree(data = d2) + geom_tree(data = d3)

dd = bind_rows(d1, d2, d3) %>% 
  filter(!is.na(label))

pp + geom_line(aes(x, y, group=label, color=node < 15), data=dd, alpha=.3)
```



## Summary



