<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.8/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.5.3/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.5.3/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.4/tabs.js"></script><script src="libs/bs3compat-0.2.4/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="libs/font-awesome-5.3.1/css/fontawesome-all.min.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="libs/pathseg-1.0.2/pathseg.js"></script><script src="libs/comic-0.961/comic.min.js"></script><script src="libs/comicR-binding-0.2/comicR.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome</a></li>
<li><a class="" href="introduction.html">Introduction</a></li>
<li class="book-part">Part I: Tree data input, output and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="active" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Tree</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for other tree-like objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for presenting data on circular layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of reproducible examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="session-info.html"><span class="header-section-number">D</span> Session Info</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="chapter5" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Phylogenetic Tree Annotation<a class="anchor" aria-label="anchor" href="#chapter5"><i class="fas fa-link"></i></a>
</h1>
<div id="visualizing-and-annotating-tree-using-grammar-of-graphics" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> Visualizing and Annotating Tree using Grammar of Graphics<a class="anchor" aria-label="anchor" href="#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al. 2017</a>)</span> is designed for more general purpose or specific type of tree visualization and annotation. It supports grammar of graphics implemented in <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> and users can freely visualize/annotate a tree by combining several annotation layers.</p>


<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/treedata-book/">ggtree</a></span><span class="op">)</span>
<span class="va">treetext</span> <span class="op">=</span> <span class="st">"(((ADH2:0.1[&amp;&amp;NHX:S=human], ADH1:0.11[&amp;&amp;NHX:S=human]):
0.05 [&amp;&amp;NHX:S=primates:D=Y:B=100],ADHY:
0.1[&amp;&amp;NHX:S=nematode],ADHX:0.12 [&amp;&amp;NHX:S=insect]):
0.1[&amp;&amp;NHX:S=metazoa:D=N],(ADH4:0.09[&amp;&amp;NHX:S=yeast],
ADH3:0.13[&amp;&amp;NHX:S=yeast], ADH2:0.12[&amp;&amp;NHX:S=yeast],
ADH1:0.11[&amp;&amp;NHX:S=yeast]):0.1[&amp;&amp;NHX:S=Fungi])[&amp;&amp;NHX:D=N];"</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.nhx.html">read.nhx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/textconnections.html">textConnection</a></span><span class="op">(</span><span class="va">treetext</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_label</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">branch</span>, label<span class="op">=</span><span class="va">S</span><span class="op">)</span>, fill<span class="op">=</span><span class="st">'lightgreen'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_label</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">D</span><span class="op">)</span>, fill<span class="op">=</span><span class="st">'steelblue'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="va">B</span><span class="op">)</span>, hjust<span class="op">=</span><span class="op">-</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:ggtreeNHX"></span>
<img src="treedata_files/figure-html/ggtreeNHX-1.svg" alt="Annotating tree using grammar of graphics. The NHX tree was annotated using grammar of graphic syntax by combining different layers using + operator. Species information were labelled on the middle of the branches, Duplication events were shown on most recent common ancestor and clade bootstrap value were dispalyed near to it." width="672"><p class="caption">
Figure 5.1: <strong>Annotating tree using grammar of graphics.</strong> The NHX tree was annotated using grammar of graphic syntax by combining different layers using <code><a href="https://rdrr.io/r/base/Arithmetic.html">+</a></code> operator. Species information were labelled on the middle of the branches, Duplication events were shown on most recent common ancestor and clade bootstrap value were dispalyed near to it.
</p>
</div>
<p>Here, as an example, we visualized the tree with several layers to display annotation stored in NHX tags, including a layer of <code>geom_tiplab</code> to display tip labels (gene name in this case), a layer using <code>geom_label</code> to show species information (<code>S</code> tag) colored by lightgreen, a layer of duplication event information (<code>D</code> tag) colored by steelblue and another layer using <em>geom_text</em> to show bootstrap value (<code>B</code> tag).</p>
<p>Layers defined in <em>ggplot2</em> can be applied to <em>ggtree</em> directly as demonstrated in Figure <a href="chapter5.html#fig:ggtreeNHX">5.1</a> of using <em>geom_label</em> and <em>geom_text</em>. But <em>ggplot2</em> does not provide graphic layers that are specific designed for phylogenetic tree annotation. For instance, layers for tip labels, tree branch scale legend, highlight or labeling clade are all unavailable. To make tree annotation more flexible, a number of layers have been implemented in <em>ggtree</em> (Table <a href="chapter5.html#tab:geoms">5.1</a>), enabling different ways of annotation on various parts/components of a phylogenetic tree.</p>
<div class="inline-table"><table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:geoms">Table 5.1: </span>Geom layers defined in ggtree.
</caption>
<thead><tr>
<th style="text-align:left;">
Layer
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
geom_balance
</td>
<td style="text-align:left;">
highlights the two direct descendant clades of an internal node
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_cladelabel
</td>
<td style="text-align:left;">
annotate a clade with bar and text label
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_facet
</td>
<td style="text-align:left;">
plot associated data in specific panel (facet) and align the plot with the tree
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_hilight
</td>
<td style="text-align:left;">
highlight selected clade with rectanglar or round shape
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_inset
</td>
<td style="text-align:left;">
add insets (subplots) to tree nodes
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_label2
</td>
<td style="text-align:left;">
modified version of geom_label, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_nodepoint
</td>
<td style="text-align:left;">
annotate internal nodes with symbolic points
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_point2
</td>
<td style="text-align:left;">
modified version of geom_point, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_range
</td>
<td style="text-align:left;">
bar layer to present uncertainty of evolutionary inference
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_rootpoint
</td>
<td style="text-align:left;">
annotate root node with symbolic point
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_rootedge
</td>
<td style="text-align:left;">
add root edge to a tree
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_segment2
</td>
<td style="text-align:left;">
modified version of geom_segment, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_strip
</td>
<td style="text-align:left;">
annotate associated taxa with bar and (optional) text label
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_taxalink
</td>
<td style="text-align:left;">
Linking related taxa
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_text2
</td>
<td style="text-align:left;">
modified version of geom_text, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tiplab
</td>
<td style="text-align:left;">
layer of tip labels
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tippoint
</td>
<td style="text-align:left;">
annotate external nodes with symbolic points
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tree
</td>
<td style="text-align:left;">
tree structure layer, with multiple layout supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_treescale
</td>
<td style="text-align:left;">
tree branch scale legend
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="layers-for-tree-annotation" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> Layers for Tree Annotation<a class="anchor" aria-label="anchor" href="#layers-for-tree-annotation"><i class="fas fa-link"></i></a>
</h2>
<div id="colored-strips" class="section level3" number="5.2.1">
<h3>
<span class="header-section-number">5.2.1</span> Colored strips<a class="anchor" aria-label="anchor" href="#colored-strips"><i class="fas fa-link"></i></a>
</h3>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(<a href="references.html#ref-yu_ggtree:_2017" role="doc-biblioref">Yu et al. 2017</a>)</span> implements <code><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel()</a></code> layer to annotate a selected clade with a bar indicating the clade with a corresponding label.</p>
<p>The <code><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel()</a></code> layer accepts a selected internal node number and label corresponding clade automatically (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>A). To get the internal node number, please refer to <a href="chapter2.html#accesor-tidytree">Chapter 2</a>.</p>
<div class="sourceCode" id="cb158"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">2015</span><span class="op">-</span><span class="fl">12</span><span class="op">-</span><span class="fl">21</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/rtree.html">rtree</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">6</span><span class="op">)</span>

<span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">45</span>, label<span class="op">=</span><span class="st">"test label"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">34</span>, label<span class="op">=</span><span class="st">"another clade"</span><span class="op">)</span></code></pre></div>
<p>Users can set the parameter, <code>align = TRUE</code>, to align the clade label, <code>offset</code>, to adjust the position and color to set the color of bar and label text <em>etc</em> (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>B).</p>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">45</span>, label<span class="op">=</span><span class="st">"test label"</span>, align<span class="op">=</span><span class="cn">TRUE</span>,  offset <span class="op">=</span> <span class="fl">.2</span>, color<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">34</span>, label<span class="op">=</span><span class="st">"another clade"</span>, align<span class="op">=</span><span class="cn">TRUE</span>, offset <span class="op">=</span> <span class="fl">.2</span>, color<span class="op">=</span><span class="st">'blue'</span><span class="op">)</span></code></pre></div>
<p>Users can change the <code>angle</code> of the clade label text and relative position from text to bar via the parameter <code>offset.text</code>. The size of the bar and text can be changed via the parameters <code>barsize</code> and <code>fontsize</code> respectively (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>C).</p>
<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">45</span>, label<span class="op">=</span><span class="st">"test label"</span>, align<span class="op">=</span><span class="cn">T</span>, angle<span class="op">=</span><span class="fl">270</span>, hjust<span class="op">=</span><span class="st">'center'</span>, offset.text<span class="op">=</span><span class="fl">.5</span>, barsize<span class="op">=</span><span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">34</span>, label<span class="op">=</span><span class="st">"another clade"</span>, align<span class="op">=</span><span class="cn">T</span>, angle<span class="op">=</span><span class="fl">45</span>, fontsize<span class="op">=</span><span class="fl">8</span><span class="op">)</span></code></pre></div>
<p>Users can also use <code>geom_label</code> to label the text and can set the background color by <code>fill</code> parameter (Figure <a href="chapter5.html#fig:cladelabel">5.2</a>D).</p>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">34</span>, label<span class="op">=</span><span class="st">"another clade"</span>, align<span class="op">=</span><span class="cn">T</span>, geom<span class="op">=</span><span class="st">'label'</span>, fill<span class="op">=</span><span class="st">'lightblue'</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span id="fig:cladelabel"></span>
<img src="treedata_files/figure-html/cladelabel-1.svg" alt="Labeling clades." width="100%"><p class="caption">
Figure 5.2: <strong>Labeling clades.</strong>
</p>
</div>
<p><code>geom_cladelabel</code> also supports unrooted tree layouts (Figure <a href="chapter5.html#fig:striplabel">5.3</a>A).</p>
<div class="sourceCode" id="cb162"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"daylight"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">35</span>, label<span class="op">=</span><span class="st">"test label"</span>, angle<span class="op">=</span><span class="fl">0</span>, 
                  fontsize<span class="op">=</span><span class="fl">8</span>, offset<span class="op">=</span><span class="fl">.5</span>, vjust<span class="op">=</span><span class="fl">.5</span><span class="op">)</span>  <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_cladelabel.html">geom_cladelabel</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">55</span>, label<span class="op">=</span><span class="st">'another clade'</span>, 
                  angle<span class="op">=</span><span class="op">-</span><span class="fl">95</span>, hjust<span class="op">=</span><span class="fl">.5</span>, fontsize<span class="op">=</span><span class="fl">8</span><span class="op">)</span></code></pre></div>
<p><code>geom_cladelabel</code> is designed for labeling Monophyletic (Clade) while there are related taxa that are not form a clade. <code>ggtree</code> provides <code>geom_strip</code> to add a strip/bar to indicate the association with optional label for Polyphyletic or Paraphyletic (Figure <a href="chapter5.html#fig:striplabel">5.3</a>B).</p>
<div class="sourceCode" id="cb163"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_strip.html">geom_strip</a></span><span class="op">(</span><span class="st">'t10'</span>, <span class="st">'t30'</span>, barsize<span class="op">=</span><span class="fl">2</span>, color<span class="op">=</span><span class="st">'red'</span>, 
            label<span class="op">=</span><span class="st">"associated taxa"</span>, offset.text<span class="op">=</span><span class="fl">.1</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_strip.html">geom_strip</a></span><span class="op">(</span><span class="st">'t1'</span>, <span class="st">'t18'</span>, barsize<span class="op">=</span><span class="fl">2</span>, color<span class="op">=</span><span class="st">'blue'</span>, 
            label <span class="op">=</span> <span class="st">"another label"</span>, offset.text<span class="op">=</span><span class="fl">.1</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span id="fig:striplabel"></span>
<img src="treedata_files/figure-html/striplabel-1.svg" alt="Labeling associated taxa. geom_cladelabel is for labeling Monophyletic and it also supports unrooted layout (A). geom_strip is designed for labeling associated taxa (Monophyletic, Polyphyletic or Paraphyletic) (B)." width="100%"><p class="caption">
Figure 5.3: <strong>Labeling associated taxa.</strong> <code>geom_cladelabel</code> is for labeling Monophyletic and it also supports unrooted layout (A). <code>geom_strip</code> is designed for labeling associated taxa (Monophyletic, Polyphyletic or Paraphyletic) (B).
</p>
</div>
</div>
<div id="highlight-clades" class="section level3" number="5.2.2">
<h3>
<span class="header-section-number">5.2.2</span> Highlight clades<a class="anchor" aria-label="anchor" href="#highlight-clades"><i class="fas fa-link"></i></a>
</h3>
<p><code>ggtree</code> implements <em><code>geom_hilight</code></em> layer, that accepts an internal node number and add a layer of rectangle to highlight the selected clade Figure (<a href="chapter5.html#fig:hilight">5.4</a>).<a class="footnote-ref" tabindex="0" data-toggle="popover" data-content='&lt;p&gt;If you want to plot the tree above the highlighting area, visit &lt;a href="faq.html#faq-under-the-tree"&gt;FAQ&lt;/a&gt; for details.&lt;/p&gt;'><sup>8</sup></a></p>
<div class="sourceCode" id="cb164"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nwk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.nwk"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span><span class="va">nwk</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">21</span>, fill<span class="op">=</span><span class="st">"steelblue"</span>, alpha<span class="op">=</span><span class="fl">.6</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">17</span>, fill<span class="op">=</span><span class="st">"darkgreen"</span>, alpha<span class="op">=</span><span class="fl">.6</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">21</span>, fill<span class="op">=</span><span class="st">"steelblue"</span>, alpha<span class="op">=</span><span class="fl">.6</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">23</span>, fill<span class="op">=</span><span class="st">"darkgreen"</span>, alpha<span class="op">=</span><span class="fl">.6</span><span class="op">)</span></code></pre></div>
<p>The <code>geom_hilight</code> layer also support highlighting clades for unrooted layout trees with round (‘encircle’) or rectangular (‘rect’) shape (Figure <a href="chapter5.html#fig:hilight">5.4</a>C).</p>
<div class="sourceCode" id="cb166"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## type can be 'encircle' or 'rect'</span>
<span class="va">pg</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">55</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">35</span>, fill<span class="op">=</span><span class="st">'darkgreen'</span>, type<span class="op">=</span><span class="st">"rect"</span><span class="op">)</span></code></pre></div>
<p>Another way to highlight selected clades is setting the clades with different colors and/or line types as demonstrated in Figure <a href="chapter6.html#fig:scaleClade">6.2</a>.</p>
<p>In addition to <code>geom_hilight</code>, <code>ggtree</code> also implements <code>geom_balance</code>
which is designed to highlight neighboring subclades of a given internal node (Figure <a href="chapter5.html#fig:hilight">5.4</a>D).</p>
<div class="sourceCode" id="cb167"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_balance.html">geom_balance</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">16</span>, fill<span class="op">=</span><span class="st">'steelblue'</span>, color<span class="op">=</span><span class="st">'white'</span>, alpha<span class="op">=</span><span class="fl">0.6</span>, extend<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_balance.html">geom_balance</a></span><span class="op">(</span>node<span class="op">=</span><span class="fl">19</span>, fill<span class="op">=</span><span class="st">'darkgreen'</span>, color<span class="op">=</span><span class="st">'white'</span>, alpha<span class="op">=</span><span class="fl">0.6</span>, extend<span class="op">=</span><span class="fl">1</span><span class="op">)</span> </code></pre></div>
<p>The <code>geom_hilight</code> layer supports using aesthetic mapping to automatically highlight clades as demonstrated in Figure <a href="chapter5.html#fig:hilight">5.4</a>E and <a href="chapter5.html#fig:hilight">5.4</a>F.</p>
<div class="sourceCode" id="cb168"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## using external data</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>node<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">17</span>, <span class="fl">21</span><span class="op">)</span>, type<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">d</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>node<span class="op">=</span><span class="va">node</span>, fill<span class="op">=</span><span class="va">type</span><span class="op">)</span><span class="op">)</span>

<span class="co">## using data stored in tree object</span>
<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.nhx.html">read.nhx</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/NHX/ADH.nhx"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom-hilight.html">geom_hilight</a></span><span class="op">(</span>mapping<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>subset <span class="op">=</span> <span class="va">node</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">12</span><span class="op">)</span>, fill <span class="op">=</span> <span class="va">S</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"steelblue"</span>, <span class="st">"darkgreen"</span><span class="op">)</span><span class="op">)</span></code></pre></div>


<div class="figure">
<span id="fig:hilight"></span>
<img src="treedata_files/figure-html/hilight-1.svg" alt="Highlight selected clades. Rectangular layout (A), circular/fan (B) and unrooted layouts. Highlight neighboring subclades simultaneously (D). Highlight selected clades using associated data (E and F)." width="100%"><p class="caption">
Figure 5.4: <strong>Highlight selected clades.</strong> Rectangular layout (A), circular/fan (B) and unrooted layouts. Highlight neighboring subclades simultaneously (D). Highlight selected clades using associated data (E and F).
</p>
</div>
</div>
<div id="taxa-connection" class="section level3" number="5.2.3">
<h3>
<span class="header-section-number">5.2.3</span> Taxa connection<a class="anchor" aria-label="anchor" href="#taxa-connection"><i class="fas fa-link"></i></a>
</h3>
<p>Some evolutionary events (e.g. reassortment, horizontal gene transfer) cannot be modeled by a simple tree. <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>geom_taxalink</code> layer that allows drawing straight or curved lines between any of two nodes in the tree, allow it to represent evolutionary events by connecting taxa. It works with rectangular (Figure <a href="chapter5.html#fig:taxalink">5.5</a>A), circular (Figure <a href="chapter5.html#fig:taxalink">5.5</a>B) and inward circular (Figure <a href="chapter5.html#fig:taxalink">5.5</a>C) layouts.</p>
<p>The <code>geom_taxalink</code> layout supports aesthetic mapping, which requires a <code>data.frame</code> that stored association information with/without meta data as input (Figure <a href="chapter5.html#fig:taxalink">5.5</a>D).</p>


<div class="sourceCode" id="cb169"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">'A'</span>, taxa2<span class="op">=</span><span class="st">'E'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">'F'</span>, taxa2<span class="op">=</span><span class="st">'K'</span>, color<span class="op">=</span><span class="st">'red'</span>, linetype <span class="op">=</span> <span class="st">'dashed'</span>,
    arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.02</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"circular"</span><span class="op">)</span> <span class="op">+</span> 
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">'A'</span>, taxa2<span class="op">=</span><span class="st">'E'</span>, color<span class="op">=</span><span class="st">"grey"</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, offset<span class="op">=</span><span class="fl">0.05</span>,
                    arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">'F'</span>, taxa2<span class="op">=</span><span class="st">'K'</span>, color<span class="op">=</span><span class="st">'red'</span>, linetype <span class="op">=</span> <span class="st">'dashed'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, offset<span class="op">=</span><span class="fl">0.05</span>,
                    arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">"L"</span>, taxa2<span class="op">=</span><span class="st">"M"</span>, color<span class="op">=</span><span class="st">"blue"</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, offset<span class="op">=</span><span class="fl">0.05</span>,
                    hratio<span class="op">=</span><span class="fl">0.8</span>, arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># when the tree was created using reverse x, </span>
<span class="co"># we can set outward to FALSE, which will generate the inward curve lines.</span>
<span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"inward_circular"</span>, xlim<span class="op">=</span><span class="fl">150</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">'A'</span>, taxa2<span class="op">=</span><span class="st">'E'</span>, color<span class="op">=</span><span class="st">"grey"</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, offset<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span>, 
                    outward<span class="op">=</span><span class="cn">FALSE</span>,
                    arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">'F'</span>, taxa2<span class="op">=</span><span class="st">'K'</span>, color<span class="op">=</span><span class="st">'red'</span>, linetype <span class="op">=</span> <span class="st">'dashed'</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, offset<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span>,
                    outward<span class="op">=</span><span class="cn">FALSE</span>,
                    arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="st">"L"</span>, taxa2<span class="op">=</span><span class="st">"M"</span>, color<span class="op">=</span><span class="st">"blue"</span>, alpha<span class="op">=</span><span class="fl">0.5</span>, offset<span class="op">=</span><span class="op">-</span><span class="fl">0.2</span>, 
                    outward<span class="op">=</span><span class="cn">FALSE</span>,
                    arrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/arrow.html">arrow</a></span><span class="op">(</span>length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.01</span>, <span class="st">"npc"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
      <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span> 

<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>from<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"F"</span>, <span class="st">"L"</span><span class="op">)</span>, 
                  to<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"E"</span>, <span class="st">"K"</span>, <span class="st">"M"</span><span class="op">)</span>, 
                  h<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">0.1</span><span class="op">)</span>, 
                  type<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"t1"</span>, <span class="st">"t2"</span>, <span class="st">"t3"</span><span class="op">)</span>, 
                  s<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">tree</span>, layout<span class="op">=</span><span class="st">"inward_circular"</span>, xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">150</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_taxalink.html">geom_taxalink</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">dat</span>, 
                         mapping<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>taxa1<span class="op">=</span><span class="va">from</span>, 
                                     taxa2<span class="op">=</span><span class="va">to</span>, 
                                     color<span class="op">=</span><span class="va">type</span>, 
                                     size<span class="op">=</span><span class="va">s</span><span class="op">)</span>, 
                         ncp<span class="op">=</span><span class="fl">10</span>,
                         offset<span class="op">=</span><span class="fl">0.15</span><span class="op">)</span> <span class="op">+</span> 
          <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_tiplab.html">geom_tiplab</a></span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">scale_size_continuous</span><span class="op">(</span>range<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, <span class="va">p3</span>, <span class="va">p4</span>, ncol<span class="op">=</span><span class="fl">2</span>, labels<span class="op">=</span><span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:taxalink"></span>
<img src="treedata_files/figure-html/taxalink-1.svg" alt="Linking related taxa. This can be used to indicate evolutionary events such as reassortment and horizontal gene transfer." width="100%"><p class="caption">
Figure 5.5: <strong>Linking related taxa.</strong> This can be used to indicate evolutionary events such as reassortment and horizontal gene transfer.
</p>
</div>
</div>
<div id="uncertainty-of-evolutionary-inference" class="section level3" number="5.2.4">
<h3>
<span class="header-section-number">5.2.4</span> Uncertainty of evolutionary inference<a class="anchor" aria-label="anchor" href="#uncertainty-of-evolutionary-inference"><i class="fas fa-link"></i></a>
</h3>
<p>The <code>geom_range</code> layer supports displaying interval (highest posterior density, confidence interval, range) as horizontal bars on tree nodes. The center of the interval will anchor to corresponding node. The center by default is the mean value of the interval (Figure <a href="chapter5.html#fig:geomRange">5.6</a>A). We can set the <code>center</code> to estimated mean or median value (Figure <a href="chapter5.html#fig:geomRange">5.6</a>B), or observed value. As the tree branch and the interval may not be in the same scale, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>scale_x_range</code> to add second x axis for the range (Figure <a href="chapter5.html#fig:geomRange">5.6</a>C). Note that x axis is disable by default theme and we need to enable it if we want to dispaly it (<em>e.g.</em> <code>theme_tree2</code>).</p>


<pre eval="FALSE"><code>file &lt;- system.file("extdata/MEGA7", "mtCDNA_timetree.nex", package = "treeio")
x &lt;- read.mega(file)
p1 &lt;- ggtree(x) + geom_range('reltime_0.95_CI', color='red', size=3, alpha=.3)
p2 &lt;- ggtree(x) + geom_range('reltime_0.95_CI', color='red', size=3, alpha=.3, center='reltime')  
p3 &lt;- p2 + scale_x_range() + theme_tree2()</code></pre>
<div class="figure">
<span id="fig:geomRange"></span>
<img src="treedata_files/figure-html/geomRange-1.svg" alt="Displaying uncertainty of evolutoinary inference. The center (mean value of the range (A) or estimated value (B)) is anchor to the tree nodes. A second x axis was used for range scaling (C)." width="100%"><p class="caption">
Figure 5.6: <strong>Displaying uncertainty of evolutoinary inference.</strong> The center (mean value of the range (A) or estimated value (B)) is anchor to the tree nodes. A second x axis was used for range scaling (C).
</p>
</div>
</div>
</div>
<div id="tree-annotation-with-output-from-evolution-software" class="section level2" number="5.3">
<h2>
<span class="header-section-number">5.3</span> Tree annotation with output from evolution software<a class="anchor" aria-label="anchor" href="#tree-annotation-with-output-from-evolution-software"><i class="fas fa-link"></i></a>
</h2>
<div id="tree-annotation-using-data-from-evolutionary-analysis-software" class="section level3" number="5.3.1">
<h3>
<span class="header-section-number">5.3.1</span> Tree annotation using data from evolutionary analysis software<a class="anchor" aria-label="anchor" href="#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fas fa-link"></i></a>
</h3>
<p><a href="#chapter">Chapter 1</a> introduced using <a href="http://bioconductor.org/packages/treeio">treeio</a> package <span class="citation">(<a href="references.html#ref-wang_treeio_2020" role="doc-biblioref">Wang et al. 2020</a>)</span> to parse different tree formats and commonly used software outputs to obtain phylogeny-associated data. These imported data as <code>S4</code> objects can be visualized directly using <a href="http://bioconductor.org/packages/ggtree">ggtree</a>. Figure <a href="chapter5.html#fig:ggtreeNHX">5.1</a> demonstrates a tree annotated using the information (species classification, duplication event and bootstrap value) stored in NHX file. <em>PHYLDOG</em> and <em>RevBayes</em> output NHX files that can be parsed by <a href="http://bioconductor.org/packages/treeio">treeio</a> and visualized by <a href="http://bioconductor.org/packages/ggtree">ggtree</a> with annotation using their inference data.</p>
<p>Furthermore, the evolutionary data from the inference of <em>BEAST</em>, <em>MrBayes</em> and <em>RevBayes</em>, <em>d<sub>N</sub>/d<sub>S</sub></em> values inferred by <em>CodeML</em>, ancestral sequences inferred by <em>HyPhy</em>, <em>CodeML</em> or <em>BaseML</em> and short read placement by <em>EPA</em> and <em>pplacer</em> can be used to annotate the tree directly.</p>


<div class="sourceCode" id="cb171"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/BEAST"</span>, <span class="st">"beast_mcc.tree"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">beast</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/beast-parser.html">read.beast</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">beast</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">rate</span><span class="op">)</span><span class="op">)</span>  <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_range.html">geom_range</a></span><span class="op">(</span>range<span class="op">=</span><span class="st">'length_0.95_HPD'</span>, color<span class="op">=</span><span class="st">'red'</span>, alpha<span class="op">=</span><span class="fl">.6</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/geom_nodelab.html">geom_nodelab</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">branch</span>, label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">posterior</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>, vjust<span class="op">=</span><span class="op">-</span><span class="fl">.5</span>, size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">scale_color_continuous</span><span class="op">(</span>low<span class="op">=</span><span class="st">"darkgreen"</span>, high<span class="op">=</span><span class="st">"red"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.1</span>, <span class="fl">.8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:beast"></span>
<img src="treedata_files/figure-html/beast-1.svg" alt="Annotating BEAST tree with length_95%_HPD and posterior. Branch length credible intervals (95% HPD) were displayed as red horizontal bars and clade posterior values were shown on the middle of branches." width="100%"><p class="caption">
Figure 5.7: <strong>Annotating <em>BEAST</em> tree with <em>length_95%_HPD</em> and posterior.</strong> Branch length credible intervals (95% HPD) were displayed as red horizontal bars and clade posterior values were shown on the middle of branches.
</p>
</div>
<p>In Figure <a href="chapter5.html#fig:beast">5.7</a>, the tree was visualized and annotated with posterior &gt; 0.9 and demonstrated length uncertainty (95% Highest Posterior Density (HPD) interval).</p>
<p>Ancestral sequences inferred by <em>HyPhy</em> can be parsed using <a href="chapter1.html#parsing-hyphy-output"><em>treeio</em></a>, whereas the substitutions along each tree branch was automatically computed and stored inside the phylogenetic tree object (<em>i.e.</em>, <code>S4</code> object). The <em>ggtree</em> can utilize this information in the object to annotate the tree, as demonstrated in Figure <a href="chapter5.html#fig:hyphy">5.8</a>.</p>


<div class="sourceCode" id="cb172"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nwk</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/HYPHY"</span>, <span class="st">"labelledtree.tree"</span>, 
                   package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">ancseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/HYPHY"</span>, <span class="st">"ancseq.nex"</span>, 
                      package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">tipfas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pa.fas"</span>, package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">hy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.hyphy.html">read.hyphy</a></span><span class="op">(</span><span class="va">nwk</span>, <span class="va">ancseq</span>, <span class="va">tipfas</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">hy</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">branch</span>, label<span class="op">=</span><span class="va">AA_subs</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">2</span>, 
            vjust<span class="op">=</span><span class="op">-</span><span class="fl">.3</span>, color<span class="op">=</span><span class="st">"firebrick"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:hyphy"></span>
<img src="treedata_files/figure-html/hyphy-1.svg" alt="Annotating tree with amino acid substitution determined by ancestral sequences inferred by HYPHY. Amino acid substitutions were displayed on the middle of branches." width="100%"><p class="caption">
Figure 5.8: <strong>Annotating tree with amino acid substitution determined by ancestral sequences inferred by HYPHY.</strong> Amino acid substitutions were displayed on the middle of branches.
</p>
</div>
<p><em>PAML</em>’s <em>BaseML</em> and <em>CodeML</em> can be also used to infer ancestral sequences, whereas <em>CodeML</em> can infer selection pressure. After parsing this information using <a href="chapter1.html#parsing-paml-output"><em>treeio</em></a>, <em>ggtree</em> can integrate this information into the same tree structure and used for annotation as illustrated in Figure <a href="chapter5.html#fig:codeml">5.9</a>.</p>


<div class="sourceCode" id="cb173"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rstfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/PAML_Codeml"</span>, <span class="st">"rst"</span>, 
                       package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">mlcfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/PAML_Codeml"</span>, <span class="st">"mlc"</span>, 
                       package<span class="op">=</span><span class="st">"treeio"</span><span class="op">)</span>
<span class="va">ml</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/treeio/reference/read.codeml.html">read.codeml</a></span><span class="op">(</span><span class="va">rstfile</span>, <span class="va">mlcfile</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/ggtree.html">ggtree</a></span><span class="op">(</span><span class="va">ml</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>color<span class="op">=</span><span class="va">dN_vs_dS</span><span class="op">)</span>, branch.length<span class="op">=</span><span class="st">'dN_vs_dS'</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">scale_color_continuous</span><span class="op">(</span>name<span class="op">=</span><span class="st">'dN/dS'</span>, limits<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.5</span><span class="op">)</span>,
                         oob<span class="op">=</span><span class="fu">scales</span><span class="fu">::</span><span class="va"><a href="https://scales.r-lib.org/reference/oob.html">squish</a></span>,
                         low<span class="op">=</span><span class="st">'darkgreen'</span>, high<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">branch</span>, label<span class="op">=</span><span class="va">AA_subs</span><span class="op">)</span>, 
            vjust<span class="op">=</span><span class="op">-</span><span class="fl">.5</span>, color<span class="op">=</span><span class="st">'steelblue'</span>, size<span class="op">=</span><span class="fl">2</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html">theme_tree2</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.9</span>, <span class="fl">.3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<span id="fig:codeml"></span>
<img src="treedata_files/figure-html/codeml-1.svg" alt="Annotating tree with animo acid substitution and dN/dS inferred by CodeML. Branches were rescaled and colored by dN/dS values and amino acid substitutions were displayed on the middle of branches." width="100%"><p class="caption">
Figure 5.9: <strong>Annotating tree with animo acid substitution and <em>d<sub>N</sub>/d<sub>S</sub></em> inferred by <em>CodeML</em>.</strong> Branches were rescaled and colored by <em>d<sub>N</sub>/d<sub>S</sub></em> values and amino acid substitutions were displayed on the middle of branches.
</p>
</div>
<p>Not only all the tree data that parsed by <a href="http://bioconductor.org/packages/treeio">treeio</a> can be used to visualize and annotate phylogenetic tree using <code>ggtree</code>, but also other tree and tree-like objects defined in R community are supported. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> plays an unique role in R ecosystem to facilitate phylogenetic analysis and it can be easily integrated into other packages and pipelines. For more details, please refer to <a href="chapter9.html#chapter9">chapter 9</a>. In addition to direct support of tree objects, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> also allow users to plot tree with different types of external data (see also <a href="chapter7.html#chapter7">chapter 7</a> and <span class="citation">(<a href="references.html#ref-yu_two_2018" role="doc-biblioref">Yu et al. 2018</a>)</span>).</p>
</div>
</div>
<div id="summary5" class="section level2" number="5.4">
<h2>
<span class="header-section-number">5.4</span> Summary<a class="anchor" aria-label="anchor" href="#summary5"><i class="fas fa-link"></i></a>
</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package implements grammar of graphics for annotating phylogenetic trees. Users can use ggplot2 syntax to combine different annotation layers to produce complex tree annotation. If you are familiar with <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>, tree annotation with high level of customization can be intuitive and flexible using <a href="http://bioconductor.org/packages/ggtree">ggtree</a>.</p>

</div>
</div>

  <div class="chapter-nav">
<div class="prev"><a href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></div>
<div class="next"><a href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Tree</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#chapter5"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="nav-link" href="#visualizing-and-annotating-tree-using-grammar-of-graphics"><span class="header-section-number">5.1</span> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li>
<a class="nav-link" href="#layers-for-tree-annotation"><span class="header-section-number">5.2</span> Layers for Tree Annotation</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#colored-strips"><span class="header-section-number">5.2.1</span> Colored strips</a></li>
<li><a class="nav-link" href="#highlight-clades"><span class="header-section-number">5.2.2</span> Highlight clades</a></li>
<li><a class="nav-link" href="#taxa-connection"><span class="header-section-number">5.2.3</span> Taxa connection</a></li>
<li><a class="nav-link" href="#uncertainty-of-evolutionary-inference"><span class="header-section-number">5.2.4</span> Uncertainty of evolutionary inference</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#tree-annotation-with-output-from-evolution-software"><span class="header-section-number">5.3</span> Tree annotation with output from evolution software</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#tree-annotation-using-data-from-evolutionary-analysis-software"><span class="header-section-number">5.3.1</span> Tree annotation using data from evolutionary analysis software</a></li></ul>
</li>
<li><a class="nav-link" href="#summary5"><span class="header-section-number">5.4</span> Summary</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2021-05-14.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
