<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>References | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<meta name="description" content="Amer, Abdrazak, Sheila Galvin, Claire M Healy, and Gary P Moran. 2017. “The Microbiome of Potentially Malignant Oral Leukoplakia Exhibits Enrichment for Fusobacterium, Leptotrichia, Campylobacter,...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="References | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta property="og:type" content="book">
<meta property="og:image" content="/book-cover.png">
<meta property="og:description" content="Amer, Abdrazak, Sheila Galvin, Claire M Healy, and Gary P Moran. 2017. “The Microbiome of Potentially Malignant Oral Leukoplakia Exhibits Enrichment for Fusobacterium, Leptotrichia, Campylobacter,...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="References | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
<meta name="twitter:description" content="Amer, Abdrazak, Sheila Galvin, Claire M Healy, and Gary P Moran. 2017. “The Microbiome of Potentially Malignant Oral Leukoplakia Exhibits Enrichment for Fusobacterium, Leptotrichia, Campylobacter,...">
<meta name="twitter:image" content="/book-cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs/header-attrs.js"></script><script src="libs/jquery/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap/bootstrap.bundle.min.js"></script><script src="libs/bs3compat/transition.js"></script><script src="libs/bs3compat/tabs.js"></script><script src="libs/bs3compat/bs3compat.js"></script><link href="libs/bs4_book/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book/bs4_book.js"></script><script src="libs/kePrint/kePrint.js"></script><link href="libs/lightable/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets/htmlwidgets.js"></script><script src="libs/pathseg/pathseg.js"></script><script src="libs/comic/comic.min.js"></script><script src="libs/comicR-binding/comicR.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS -->
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="author.html">About the Author</a></li>
<li class="book-part">Part I: Tree data input, output, and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Tree</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for other tree-like objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for presenting data on a circular layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of reproducible examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree package suite</a></li>
<li><a class="active" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references" class="section level1 unnumbered">
<h1>References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-amer2017microbiome" class="csl-entry">
Amer, Abdrazak, Sheila Galvin, Claire M Healy, and Gary P Moran. 2017. <span>“The Microbiome of Potentially Malignant Oral Leukoplakia Exhibits Enrichment for Fusobacterium, Leptotrichia, Campylobacter, and Rothia Species.”</span> <em>Frontiers in Microbiology</em> 8: 2391. <a href="https://doi.org/10.3389/fmicb.2017.02391">https://doi.org/10.3389/fmicb.2017.02391</a>.
</div>
<div id="ref-arenas_trends_2015" class="csl-entry">
Arenas, Miguel. 2015. <span>“Trends in Substitution Models of Molecular Evolution.”</span> <em>Frontiers in Genetics</em> 6 (October). <a href="https://doi.org/10.3389/fgene.2015.00319">https://doi.org/10.3389/fgene.2015.00319</a>.
</div>
<div id="ref-GraPhlAn" class="csl-entry">
Asnicar, Francesco, George Weingart, Timothy L Tickle, Curtis Huttenhower, and Nicola Segata. 2015. <span>“Compact Graphical Representation of Phylogenetic Data and Metadata with GraPhlAn.”</span> <em>PeerJ</em> 3: e1029. <a href="https://doi.org/10.7717/peerj.1029">https://doi.org/10.7717/peerj.1029</a>.
</div>
<div id="ref-barton_broad_2016" class="csl-entry">
Barton, Kirston, Bonnie Hiener, Anni Winckelmann, Thomas Aagaard Rasmussen, Wei Shao, Karen Byth, Robert Lanfear, et al. 2016. <span>“Broad Activation of Latent <span>HIV</span>-1 in Vivo.”</span> <em>Nature Communications</em> 7 (September): 12731. <a href="https://doi.org/10.1038/ncomms12731">https://doi.org/10.1038/ncomms12731</a>.
</div>
<div id="ref-berger_performance_2011" class="csl-entry">
Berger, Simon A., Denis Krompass, and Alexandros Stamatakis. 2011. <span>“Performance, <span>Accuracy</span>, and <span>Web</span> <span>Server</span> for <span>Evolutionary</span> <span>Placement</span> of <span>Short</span> <span>Sequence</span> <span>Reads</span> Under <span>Maximum</span> <span>Likelihood</span>.”</span> <em>Systematic Biology</em>, March, 291–302. <a href="https://doi.org/10.1093/sysbio/syr010">https://doi.org/10.1093/sysbio/syr010</a>.
</div>
<div id="ref-bosi_comparative_2016" class="csl-entry">
Bosi, Emanuele, Jonathan M. Monk, Ramy K. Aziz, Marco Fondi, Victor Nizet, and Bernhard Ø. Palsson. 2016. <span>“Comparative Genome-Scale Modelling of <span>Staphylococcus</span> Aureus Strains Identifies Strain-Specific Metabolic Capabilities Linked to Pathogenicity.”</span> <em>Proceedings of the National Academy of Sciences of the United States of America</em> 113 (26): E3801–9. <a href="https://doi.org/10.1073/pnas.1523199113">https://doi.org/10.1073/pnas.1523199113</a>.
</div>
<div id="ref-bouckaert_beast_2014" class="csl-entry">
Bouckaert, Remco, Joseph Heled, Denise Kühnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. <span>“<span>BEAST</span> 2: <span>A</span> <span>Software</span> <span>Platform</span> for <span>Bayesian</span> <span>Evolutionary</span> <span>Analysis</span>.”</span> <em>PLoS Comput Biol</em> 10 (4): e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>.
</div>
<div id="ref-boussau_genome-scale_2013" class="csl-entry">
Boussau, Bastien, Gergely J. Szöllősi, Laurent Duret, Manolo Gouy, Eric Tannier, and Vincent Daubin. 2013. <span>“Genome-Scale Coestimation of Species and Gene Trees.”</span> <em>Genome Research</em> 23 (2): 323–30. <a href="https://doi.org/10.1101/gr.141978.112">https://doi.org/10.1101/gr.141978.112</a>.
</div>
<div id="ref-callahan_dada2_2016" class="csl-entry">
Callahan, Benjamin J., Paul J. McMurdie, Michael J. Rosen, Andrew W. Han, Amy Jo A. Johnson, and Susan P. Holmes. 2016. <span>“<span>DADA</span>2: <span>High</span>-Resolution Sample Inference from <span>Illumina</span> Amplicon Data.”</span> <em>Nature Methods</em> 13 (7): 581–83. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>.
</div>
<div id="ref-chen_ancient_2017" class="csl-entry">
Chen, Zigui, Wendy C. S. Ho, Siaw Shi Boon, Priscilla T. Y. Law, Martin C. W. Chan, Rob DeSalle, Robert D. Burk, and Paul K. S. Chan. 2017. <span>“Ancient Evolution and Dispersion of Human Papillomavirus 58 Variants.”</span> <em>Journal of Virology</em> 91 (21): e01285–17. <a href="https://doi.org/10.1128/JVI.01285-17">https://doi.org/10.1128/JVI.01285-17</a>.
</div>
<div id="ref-chevenet_treedyn:_2006" class="csl-entry">
Chevenet, François, Christine Brun, Anne-Laure Bañuls, Bernard Jacq, and Richard Christen. 2006. <span>“<span>TreeDyn</span>: Towards Dynamic Graphics and Annotations for Analyses of Trees.”</span> <em>BMC Bioinformatics</em> 7 (October): 439. <a href="https://doi.org/10.1186/1471-2105-7-439">https://doi.org/10.1186/1471-2105-7-439</a>.
</div>
<div id="ref-Chowe:_2020" class="csl-entry">
Chow, Nancy A., José F. Muñoz, Lalitha Gade, Elizabeth L. Berkow, Xiao Li, Rory M. Welsh, Kaitlin Forsberg, et al. 2020. <span>“Tracing the Evolutionary History and Global Expansion of Candida Auris Using Population Genomic Analyses.”</span> Edited by Geraldine Butler and Kirsten Nielsen. <em>mBio</em> 11 (2). <a href="https://doi.org/10.1128/mBio.03364-19">https://doi.org/10.1128/mBio.03364-19</a>.
</div>
<div id="ref-reroot_review" class="csl-entry">
Czech, Lucas, Jaime Huerta-Cepas, and Alexandros Stamatakis. 2017. <span>“<span class="nocase">A Critical Review on the Use of Support Values in Tree Viewers and Bioinformatics Toolkits</span>.”</span> <em>Molecular Biology and Evolution</em> 34 (6): 1535–42. <a href="https://doi.org/10.1093/molbev/msx055">https://doi.org/10.1093/molbev/msx055</a>.
</div>
<div id="ref-escudero_grand_2020" class="csl-entry">
Escudero, Marcial, and Jonathan F. Wendel. 2020. <span>“The Grand Sweep of Chromosomal Evolution in Angiosperms.”</span> <em>New Phytologist</em> 228 (3): 805–8. <a href="https://doi.org/10.1111/nph.16802">https://doi.org/10.1111/nph.16802</a>.
</div>
<div id="ref-felsenstein_evolutionary_1981" class="csl-entry">
Felsenstein, J. 1981. <span>“Evolutionary Trees from <span>DNA</span> Sequences: A Maximum Likelihood Approach.”</span> <em>Journal of Molecular Evolution</em> 17 (6): 368–76.
</div>
<div id="ref-felsenstein_cases_1978" class="csl-entry">
Felsenstein, Joseph. 1978. <span>“Cases in Which <span>Parsimony</span> or <span>Compatibility</span> <span>Methods</span> Will Be <span>Positively</span> <span>Misleading</span>.”</span> <em>Systematic Biology</em> 27 (4): 401–10. <a href="https://doi.org/10.1093/sysbio/27.4.401">https://doi.org/10.1093/sysbio/27.4.401</a>.
</div>
<div id="ref-felsenstein_phylip_1989" class="csl-entry">
———. 1989. <span>“<span>PHYLIP</span> - <span>Phylogeny</span> <span>Inference</span> <span>Package</span> (<span>Version</span> 3.2).”</span> <em>Cladistics</em> 5: 164–66.
</div>
<div id="ref-fitch_toward_1971" class="csl-entry">
Fitch, Walter M. 1971. <span>“Toward <span>Defining</span> the <span>Course</span> of <span>Evolution</span>: <span>Minimum</span> <span>Change</span> for a <span>Specific</span> <span>Tree</span> <span>Topology</span>.”</span> <em>Systematic Zoology</em> 20 (4): 406–16. <a href="https://doi.org/10.2307/2412116">https://doi.org/10.2307/2412116</a>.
</div>
<div id="ref-gentleman_bioconductor_2004" class="csl-entry">
Gentleman, Robert C, Vincent J Carey, Douglas M Bates, Ben Bolstad, Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004. <span>“Bioconductor: Open Software Development for Computational Biology and Bioinformatics.”</span> <em>Genome Biology</em> 5 (10): R80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80">https://doi.org/10.1186/gb-2004-5-10-r80</a>.
</div>
<div id="ref-goldman_codon-based_1994" class="csl-entry">
Goldman, N., and Z. Yang. 1994. <span>“A Codon-Based Model of Nucleotide Substitution for Protein-Coding <span>DNA</span> Sequences.”</span> <em>Molecular Biology and Evolution</em> 11 (5): 725–36.
</div>
<div id="ref-grubaugh_genomic_2017" class="csl-entry">
Grubaugh, Nathan D., Jason T. Ladner, Moritz U. G. Kraemer, Gytis Dudas, Amanda L. Tan, Karthik Gangavarapu, Michael R. Wiley, et al. 2017. <span>“Genomic Epidemiology Reveals Multiple Introductions of Zika Virus into the United States.”</span> <em>Nature</em> 546 (7658): 401–5. <a href="https://doi.org/10.1038/nature22400">https://doi.org/10.1038/nature22400</a>.
</div>
<div id="ref-gupta_using_2015" class="csl-entry">
Gupta, Ankit, and Vineet K Sharma. 2015. <span>“Using the Taxon-Specific Genes for the Taxonomic Classification of Bacterial Genomes.”</span> <em>BMC Genomics</em> 16 (1). <a href="https://doi.org/10.1186/s12864-015-1542-0">https://doi.org/10.1186/s12864-015-1542-0</a>.
</div>
<div id="ref-he_emergence_2013" class="csl-entry">
He, Ya-Qing, Long Chen, Wen-Bo Xu, Hong Yang, Han-Zhong Wang, Wen-Ping Zong, Hui-Xia Xian, et al. 2013. <span>“Emergence, <span>Circulation</span>, and <span>Spatiotemporal</span> <span>Phylogenetic</span> <span>Analysis</span> of <span>Coxsackievirus</span> <span>A</span>6- and <span>Coxsackievirus</span> <span>A</span>10-<span>Associated</span> <span>Hand</span>, <span>Foot</span>, and <span>Mouth</span> <span>Disease</span> <span>Infections</span> from 2008 to 2012 in <span>Shenzhen</span>, <span>China</span>.”</span> <em>Journal of Clinical Microbiology</em> 51 (11): 3560–66. <a href="https://doi.org/10.1128/JCM.01231-13">https://doi.org/10.1128/JCM.01231-13</a>.
</div>
<div id="ref-He289" class="csl-entry">
He, Zhen, Raad Z Gharaibeh, Rachel C Newsome, Jllian L Pope, Michael W Dougherty, Sarah Tomkovich, Benoit Pons, et al. 2019. <span>“Campylobacter Jejuni Promotes Colorectal Tumorigenesis Through the Action of Cytolethal Distending Toxin.”</span> <em>Gut</em> 68 (2): 289–300. <a href="https://doi.org/10.1136/gutjnl-2018-317200">https://doi.org/10.1136/gutjnl-2018-317200</a>.
</div>
<div id="ref-he_evolview_2016" class="csl-entry">
He, Zilong, Huangkai Zhang, Shenghan Gao, Martin J. Lercher, Wei-Hua Chen, and Songnian Hu. 2016. <span>“Evolview V2: An Online Visualization and Management Tool for Customized and Annotated Phylogenetic Trees.”</span> <em>Nucleic Acids Research</em> 44 (W1): W236–241. <a href="https://doi.org/10.1093/nar/gkw370">https://doi.org/10.1093/nar/gkw370</a>.
</div>
<div id="ref-hohna_probabilistic_2014" class="csl-entry">
Höhna, Sebastian, Tracy A. Heath, Bastien Boussau, Michael J. Landis, Fredrik Ronquist, and John P. Huelsenbeck. 2014. <span>“Probabilistic Graphical Model Representation in Phylogenetics.”</span> <em>Systematic Biology</em> 63 (5): 753–71. <a href="https://doi.org/10.1093/sysbio/syu039">https://doi.org/10.1093/sysbio/syu039</a>.
</div>
<div id="ref-hohna_revbayes:_2016" class="csl-entry">
Höhna, Sebastian, Michael J. Landis, Tracy A. Heath, Bastien Boussau, Nicolas Lartillot, Brian R. Moore, John P. Huelsenbeck, and Fredrik Ronquist. 2016. <span>“<span>RevBayes</span>: <span>Bayesian</span> <span>Phylogenetic</span> <span>Inference</span> <span>Using</span> <span>Graphical</span> <span>Models</span> and an <span>Interactive</span> <span>Model</span>-<span>Specification</span> <span>Language</span>.”</span> <em>Systematic Biology</em> 65 (4): 726–36. <a href="https://doi.org/10.1093/sysbio/syw021">https://doi.org/10.1093/sysbio/syw021</a>.
</div>
<div id="ref-huelsenbeck_mrbayes_2001" class="csl-entry">
Huelsenbeck, J. P., and F. Ronquist. 2001. <span>“<span>MRBAYES</span>: <span>Bayesian</span> Inference of Phylogenetic Trees.”</span> <em>Bioinformatics (Oxford, England)</em> 17 (8): 754–55.
</div>
<div id="ref-huson_dendroscope_2012" class="csl-entry">
Huson, Daniel H., and Celine Scornavacca. 2012. <span>“Dendroscope 3: An Interactive Tool for Rooted Phylogenetic Trees and Networks.”</span> <em>Systematic Biology</em> 61 (6): 1061–67. <a href="https://doi.org/10.1093/sysbio/sys062">https://doi.org/10.1093/sysbio/sys062</a>.
</div>
<div id="ref-jombart_outbreaktools_2014" class="csl-entry">
Jombart, Thibaut, David M. Aanensen, Marc Baguelin, Paul Birrell, Simon Cauchemez, Anton Camacho, Caroline Colijn, et al. 2014. <span>“<span>OutbreakTools</span>: A New Platform for Disease Outbreak Analysis Using the <span>R</span> Software.”</span> <em>Epidemics</em> 7 (June): 28–34. <a href="https://doi.org/10.1016/j.epidem.2014.04.003">https://doi.org/10.1016/j.epidem.2014.04.003</a>.
</div>
<div id="ref-kostic2012genomic" class="csl-entry">
Kostic, Aleksandar D, Dirk Gevers, Chandra Sekhar Pedamallu, Monia Michaud, Fujiko Duke, Ashlee M Earl, Akinyemi I Ojesina, et al. 2012. <span>“Genomic Analysis Identifies Association of Fusobacterium with Colorectal Carcinoma.”</span> <em>Genome Research</em> 22 (2): 292–98. <a href="https://doi.org/10.1101/gr.126573.111">https://doi.org/10.1101/gr.126573.111</a>.
</div>
<div id="ref-kuczynski_using_2011" class="csl-entry">
Kuczynski, Justin, Jesse Stombaugh, William Anton Walters, Antonio González, J. Gregory Caporaso, and Rob Knight. 2011. <span>“Using <span>QIIME</span> to Analyze 16s <span class="nocase">rRNA</span> Gene Sequences from <span>Microbial</span> <span>Communities</span>.”</span> <em>Current Protocols in Bioinformatics / Editoral Board, Andreas D. Baxevanis ... [Et Al.]</em> CHAPTER (December): Unit10.7. <a href="https://doi.org/10.1002/0471250953.bi1007s36">https://doi.org/10.1002/0471250953.bi1007s36</a>.
</div>
<div id="ref-kumar_mega7_2016" class="csl-entry">
Kumar, Sudhir, Glen Stecher, and Koichiro Tamura. 2016. <span>“<span>MEGA</span>7: Molecular Evolutionary Genetics Analysis Version 7.0 for Bigger Datasets.”</span> <em>Molecular Biology and Evolution</em> 33 (7): 1870–74. <a href="https://doi.org/10.1093/molbev/msw054">https://doi.org/10.1093/molbev/msw054</a>.
</div>
<div id="ref-kunin_pyrotagger_2010" class="csl-entry">
Kunin, Victor, and Philip Hugenholtz. 2010. <span>“<span>PyroTagger</span> : <span>A</span> Fast , Accurate Pipeline for Analysis of <span class="nocase">rRNA</span> Amplicon Pyrosequence Data.”</span> <em>The Open Journal</em>, 1–8. <a href="http://www.theopenjournal.org/toj_articles/1">http://www.theopenjournal.org/toj_articles/1</a>.
</div>
<div id="ref-lam_phylodynamics_2012" class="csl-entry">
Lam, Tommy Tsan-Yuk, Chung-Chau Hon, Philippe Lemey, Oliver G. Pybus, Mang Shi, Hein Min Tun, Jun Li, Jingwei Jiang, Edward C. Holmes, and Frederick Chi-Ching Leung. 2012. <span>“Phylodynamics of <span>H</span>5N1 Avian Influenza Virus in <span>Indonesia</span>.”</span> <em>Molecular Ecology</em> 21 (12): 3062–77. <a href="https://doi.org/10.1111/j.1365-294X.2012.05577.x">https://doi.org/10.1111/j.1365-294X.2012.05577.x</a>.
</div>
<div id="ref-lam_dissemination_2015" class="csl-entry">
Lam, Tommy Tsan-Yuk, Boping Zhou, Jia Wang, Yujuan Chai, Yongyi Shen, Xinchun Chen, Chi Ma, et al. 2015. <span>“Dissemination, Divergence and Establishment of <span>H</span>7N9 Influenza Viruses in <span>China</span>.”</span> <em>Nature</em> 522 (7554): 102–5. <a href="https://doi.org/10.1038/nature14348">https://doi.org/10.1038/nature14348</a>.
</div>
<div id="ref-larsen_identification_2019" class="csl-entry">
Larsen, Frederik T., Bertrand Bed’Hom, Bernt Guldbrandtsen, and Tina S. Dalgaard. 2019. <span>“Identification and Tissue-Expression Profiling of Novel Chicken c-Type Lectin-Like Domain Containing Proteins as Potential Targets for Carbohydrate-Based Vaccine Strategies.”</span> <em>Molecular Immunology</em> 114 (October): 216–25. <a href="https://doi.org/10.1016/j.molimm.2019.07.022">https://doi.org/10.1016/j.molimm.2019.07.022</a>.
</div>
<div id="ref-lemmon_importance_2004" class="csl-entry">
Lemmon, Alan R., and Emily C. Moriarty. 2004. <span>“The Importance of Proper Model Assumption in Bayesian Phylogenetics.”</span> <em>Systematic Biology</em> 53 (2): 265–77. <a href="https://doi.org/10.1080/10635150490423520">https://doi.org/10.1080/10635150490423520</a>.
</div>
<div id="ref-letunic_interactive_2007" class="csl-entry">
Letunic, Ivica, and Peer Bork. 2007. <span>“Interactive <span>Tree</span> <span>Of</span> <span>Life</span> (<span class="nocase">iTOL</span>): An Online Tool for Phylogenetic Tree Display and Annotation.”</span> <em>Bioinformatics</em> 23 (1): 127–28. <a href="https://doi.org/10.1093/bioinformatics/btl529">https://doi.org/10.1093/bioinformatics/btl529</a>.
</div>
<div id="ref-liang_expansion_2014" class="csl-entry">
Liang, Huyi, Tommy Tsan-Yuk Lam, Xiaohui Fan, Xinchun Chen, Yu Zeng, Ji Zhou, Lian Duan, et al. 2014. <span>“Expansion of Genotypic Diversity and Establishment of 2009 <span>H</span>1N1 Pandemic-Origin Internal Genes in Pigs in <span>China</span>.”</span> <em>Journal of Virology</em>, July, JVI.01327–14. <a href="https://doi.org/10.1128/JVI.01327-14">https://doi.org/10.1128/JVI.01327-14</a>.
</div>
<div id="ref-lott_covenntree:_2015" class="csl-entry">
Lott, Steffen C., Björn Voß, Wolfgang R. Hess, and Claudia Steglich. 2015. <span>“<span>CoVennTree</span>: A New Method for the Comparative Analysis of Large Datasets.”</span> <em>Frontiers in Genetics</em> 6: 43. <a href="https://doi.org/10.3389/fgene.2015.00043">https://doi.org/10.3389/fgene.2015.00043</a>.
</div>
<div id="ref-maddison_nexus:_1997" class="csl-entry">
Maddison, David R., David L. Swofford, Wayne P. Maddison, and David Cannatella. 1997. <span>“Nexus: <span>An</span> <span>Extensible</span> <span>File</span> <span>Format</span> for <span>Systematic</span> <span>Information</span>.”</span> <em>Systematic Biology</em> 46 (4): 590–621. <a href="https://doi.org/10.1093/sysbio/46.4.590">https://doi.org/10.1093/sysbio/46.4.590</a>.
</div>
<div id="ref-matsen_format_2012" class="csl-entry">
Matsen, Frederick A., Noah G. Hoffman, Aaron Gallagher, and Alexandros Stamatakis. 2012. <span>“A <span>Format</span> for <span>Phylogenetic</span> <span>Placements</span>.”</span> <em>PLOS ONE</em> 7 (2): e31009. <a href="https://doi.org/10.1371/journal.pone.0031009">https://doi.org/10.1371/journal.pone.0031009</a>.
</div>
<div id="ref-matsen_pplacer:_2010" class="csl-entry">
Matsen, Frederick A., Robin B. Kodner, and E Virginia Armbrust. 2010a. <span>“Pplacer: Linear Time Maximum-Likelihood and <span>Bayesian</span> Phylogenetic Placement of Sequences onto a Fixed Reference Tree.”</span> <em>BMC Bioinformatics</em> 11: 538. <a href="https://doi.org/10.1186/1471-2105-11-538">https://doi.org/10.1186/1471-2105-11-538</a>.
</div>
<div id="ref-matsen_pplacer_2010" class="csl-entry">
Matsen, Frederick A, Robin B Kodner, and E Virginia Armbrust. 2010b. <span>“Pplacer: Linear Time Maximum-Likelihood and Bayesian Phylogenetic Placement of Sequences onto a Fixed Reference Tree.”</span> <em><span>BMC</span> Bioinformatics</em> 11 (1): 538. <a href="https://doi.org/10.1186/1471-2105-11-538">https://doi.org/10.1186/1471-2105-11-538</a>.
</div>
<div id="ref-mcmurdie_phyloseq_2013" class="csl-entry">
McMurdie, Paul J., and Susan Holmes. 2013. <span>“Phyloseq: An <span>R</span> Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.”</span> <em>PloS One</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.
</div>
<div id="ref-michonneau_rotl:_2016" class="csl-entry">
Michonneau, François, Joseph W. Brown, and David J. Winter. 2016. <span>“Rotl: An r Package to Interact with the Open Tree of Life Data.”</span> <em>Methods in Ecology and Evolution</em> 7 (12): 1476–81. <a href="https://doi.org/10.1111/2041-210X.12593">https://doi.org/10.1111/2041-210X.12593</a>.
</div>
<div id="ref-morgan2013HMP" class="csl-entry">
Morgan, Xochitl C., Nicola Segata, and Curtis Huttenhower. 2013. <span>“Biodiversity and Functional Genomics in the Human Microbiome.”</span> <em>Trends in Genetics</em> 29 (1): 51–58. <a href="https://doi.org/10.1016/J.TIG.2012.09.005">https://doi.org/10.1016/J.TIG.2012.09.005</a>.
</div>
<div id="ref-neher_prediction_2016" class="csl-entry">
Neher, Richard A., Trevor Bedford, Rodney S. Daniels, Colin A. Russell, and Boris I. Shraiman. 2016. <span>“Prediction, Dynamics, and Visualization of Antigenic Phenotypes of Seasonal Influenza Viruses.”</span> <em>Proceedings of the National Academy of Sciences</em> 113 (12): E1701–9. <a href="https://doi.org/10.1073/pnas.1525578113">https://doi.org/10.1073/pnas.1525578113</a>.
</div>
<div id="ref-page_visualizing_2002" class="csl-entry">
Page, Roderic D. M. 2002. <span>“Visualizing Phylogenetic Trees Using <span>TreeView</span>.”</span> <em>Current Protocols in Bioinformatics</em> Chapter 6 (August): Unit 6.2. <a href="https://doi.org/10.1002/0471250953.bi0602s01">https://doi.org/10.1002/0471250953.bi0602s01</a>.
</div>
<div id="ref-paradis_ape_2004" class="csl-entry">
Paradis, Emmanuel, Julien Claude, and Korbinian Strimmer. 2004. <span>“<span>APE</span>: <span>Analyses</span> of <span>Phylogenetics</span> and <span>Evolution</span> in <span>R</span> Language.”</span> <em>Bioinformatics</em> 20 (2): 289–90. <a href="https://doi.org/10.1093/bioinformatics/btg412">https://doi.org/10.1093/bioinformatics/btg412</a>.
</div>
<div id="ref-pond_hyphy:_2005" class="csl-entry">
Pond, Sergei L. Kosakovsky, Simon D. W. Frost, and Spencer V. Muse. 2005. <span>“<span>HyPhy</span>: Hypothesis Testing Using Phylogenies.”</span> <em>Bioinformatics (Oxford, England)</em> 21 (5): 676–79. <a href="https://doi.org/10.1093/bioinformatics/bti079">https://doi.org/10.1093/bioinformatics/bti079</a>.
</div>
<div id="ref-rstats" class="csl-entry">
R Core Team. 2016. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-rannala_probability_1996" class="csl-entry">
Rannala, B., and Z. Yang. 1996. <span>“Probability Distribution of Molecular Evolutionary Trees: A New Method of Phylogenetic Inference.”</span> <em>Journal of Molecular Evolution</em> 43 (3): 304–11.
</div>
<div id="ref-retief_phylogenetic_2000" class="csl-entry">
Retief, J. D. 2000. <span>“Phylogenetic Analysis Using <span>PHYLIP</span>.”</span> <em>Methods in Molecular Biology (Clifton, N.J.)</em> 132: 243–58.
</div>
<div id="ref-revell_phytools_2012" class="csl-entry">
Revell, Liam J. 2012. <span>“Phytools: An <span>R</span> Package for Phylogenetic Comparative Biology (and Other Things).”</span> <em>Methods in Ecology and Evolution</em> 3 (2): 217–23. <a href="https://doi.org/10.1111/j.2041-210X.2011.00169.x">https://doi.org/10.1111/j.2041-210X.2011.00169.x</a>.
</div>
<div id="ref-sanderson_r8s:_2003" class="csl-entry">
Sanderson, Michael J. 2003. <span>“R8s: Inferring Absolute Rates of Molecular Evolution and Divergence Times in the Absence of a Molecular Clock.”</span> <em>Bioinformatics</em> 19 (2): 301–2. <a href="https://doi.org/10.1093/bioinformatics/19.2.301">https://doi.org/10.1093/bioinformatics/19.2.301</a>.
</div>
<div id="ref-schliep_phangorn_2011" class="csl-entry">
Schliep, Klaus Peter. 2011. <span>“Phangorn: Phylogenetic Analysis in <span>R</span>.”</span> <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706">https://doi.org/10.1093/bioinformatics/btq706</a>.
</div>
<div id="ref-schloss_introducing_2009" class="csl-entry">
Schloss, Patrick D., Sarah L. Westcott, Thomas Ryabin, Justine R. Hall, Martin Hartmann, Emily B. Hollister, Ryan A. Lesniewski, et al. 2009. <span>“Introducing Mothur: Open-Source, Platform-Independent, Community-Supported Software for Describing and Comparing Microbial Communities.”</span> <em>Applied and Environmental Microbiology</em> 75 (23): 7537–41. <a href="https://doi.org/10.1128/AEM.01541-09">https://doi.org/10.1128/AEM.01541-09</a>.
</div>
<div id="ref-schmidt_tree-puzzle:_2002" class="csl-entry">
Schmidt, Heiko A., Korbinian Strimmer, Martin Vingron, and Arndt von Haeseler. 2002. <span>“<span>TREE</span>-<span>PUZZLE</span>: Maximum Likelihood Phylogenetic Analysis Using Quartets and Parallel Computing.”</span> <em>Bioinformatics (Oxford, England)</em> 18 (3): 502–4.
</div>
<div id="ref-schon_age_2015" class="csl-entry">
Schön, Isa, Rylan Shearn, Koen Martens, Annette Koenders, and Stuart Halse. 2015. <span>“Age and Origin of <span>Australian</span> <span>Bennelongia</span> (<span>Crustacea</span>, <span>Ostracoda</span>).”</span> <em>Hydrobiologia</em> 750 (1): 125–46. <a href="https://doi.org/10.1007/s10750-014-2159-z">https://doi.org/10.1007/s10750-014-2159-z</a>.
</div>
<div id="ref-segata_metagenomic_2011" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>“Metagenomic Biomarker Discovery and Explanation.”</span> <em>Genome Biology</em> 12 (6): R60. <a href="https://doi.org/10.1186/gb-2011-12-6-r60">https://doi.org/10.1186/gb-2011-12-6-r60</a>.
</div>
<div id="ref-shoemaker_evidence_1989" class="csl-entry">
Shoemaker, J. S., and W. M. Fitch. 1989. <span>“Evidence from Nuclear Sequences That Invariable Sites Should Be Considered When Sequence Divergence Is Calculated.”</span> <em>Molecular Biology and Evolution</em> 6 (3): 270–89.
</div>
<div id="ref-stamatakis_raxml_2014" class="csl-entry">
Stamatakis, Alexandros. 2014. <span>“<span>RAxML</span> Version 8: A Tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies.”</span> <em>Bioinformatics</em> 30 (9): 1312–13. <a href="https://doi.org/10.1093/bioinformatics/btu033">https://doi.org/10.1093/bioinformatics/btu033</a>.
</div>
<div id="ref-venkatesh_avian_2018" class="csl-entry">
Venkatesh, Divya, Marjolein J. Poen, Theo M. Bestebroer, Rachel D. Scheuer, Oanh Vuong, Mzia Chkhaidze, Anna Machablishvili, et al. 2018. <span>“Avian Influenza Viruses in Wild Birds: Virus Evolution in a Multihost Ecosystem.”</span> <em>Journal of Virology</em> 92 (15). <a href="https://doi.org/10.1128/JVI.00433-18">https://doi.org/10.1128/JVI.00433-18</a>.
</div>
<div id="ref-vos_nexml:_2012" class="csl-entry">
Vos, Rutger A., James P. Balhoff, Jason A. Caravas, Mark T. Holder, Hilmar Lapp, Wayne P. Maddison, Peter E. Midford, et al. 2012. <span>“<span>NeXML</span>: Rich, Extensible, and Verifiable Representation of Comparative Data and Metadata.”</span> <em>Systematic Biology</em> 61 (4): 675–89. <a href="https://doi.org/10.1093/sysbio/sys025">https://doi.org/10.1093/sysbio/sys025</a>.
</div>
<div id="ref-wang_treeio_2020" class="csl-entry">
Wang, Li-Gen, Tommy Tsan-Yuk Lam, Shuangbin Xu, Zehan Dai, Lang Zhou, Tingze Feng, Pingfan Guo, et al. 2020. <span>“Treeio: An r Package for Phylogenetic Tree Input and Output with Richly Annotated and Associated Data.”</span> <em>Molecular Biology and Evolution</em> 37 (2): 599–603. <a href="https://doi.org/10.1093/molbev/msz240">https://doi.org/10.1093/molbev/msz240</a>.
</div>
<div id="ref-wickham_ggplot2_2016" class="csl-entry">
Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer. <a href="http://ggplot2.org">http://ggplot2.org</a>.
</div>
<div id="ref-wilgenbusch_inferring_2003" class="csl-entry">
Wilgenbusch, James C., and David Swofford. 2003. <span>“Inferring Evolutionary Trees with <span>PAUP</span>*.”</span> <em>Current Protocols in Bioinformatics</em> Chapter 6 (February): Unit 6.4. <a href="https://doi.org/10.1002/0471250953.bi0604s00">https://doi.org/10.1002/0471250953.bi0604s00</a>.
</div>
<div id="ref-wilkinson_grammar_2005" class="csl-entry">
Wilkinson, Leland, D. Wills, D. Rope, A. Norton, and R. Dubbs. 2005. <em>The <span>Grammar</span> of <span>Graphics</span></em>. 2nd edition. New York: Springer.
</div>
<div id="ref-RN46:_2015" class="csl-entry">
Wong, Vanessa K., Stephen Baker, Derek J. Pickard, Julian Parkhill, Andrew J. Page, Nicholas A. Feasey, Robert A. Kingsley, et al. 2015. <span>“Phylogeographical Analysis of the Dominant Multidrug-Resistant H58 Clade of Salmonella Typhi Identifies Inter- and Intracontinental Transmission Events.”</span> Journal Article. <em>Nature Genetics</em> 47 (6): 632–39. <a href="https://doi.org/10.1038/ng.3281">https://doi.org/10.1038/ng.3281</a>.
</div>
<div id="ref-wu2013dysbiosis" class="csl-entry">
Wu, Na, Xi Yang, Ruifen Zhang, Jun Li, Xue Xiao, Yongfei Hu, Yanfei Chen, et al. 2013. <span>“Dysbiosis Signature of Fecal Microbiota in Colorectal Cancer Patients.”</span> <em>Microbial Ecology</em> 66 (2): 462–70. <a href="https://doi.org/10.1007/s00248-013-0245-9">https://doi.org/10.1007/s00248-013-0245-9</a>.
</div>
<div id="ref-ggtreeExtra_2021" class="csl-entry">
Xu, Shuangbin, Zehan Dai, Pingfan Guo, Xiaocong Fu, Shanshan Liu, Lang Zhou, Wenli Tang, et al. 2021. <span>“<span class="nocase">ggtreeExtra: Compact Visualization of Richly Annotated Phylogenetic Data</span>.”</span> <em>Molecular Biology and Evolution</em> 38 (9): 4039–42. <a href="https://doi.org/10.1093/molbev/msab166">https://doi.org/10.1093/molbev/msab166</a>.
</div>
<div id="ref-yang_maximum_1994" class="csl-entry">
Yang, Z. 1994. <span>“Maximum Likelihood Phylogenetic Estimation from <span>DNA</span> Sequences with Variable Rates over Sites: Approximate Methods.”</span> <em>Journal of Molecular Evolution</em> 39 (3): 306–14.
</div>
<div id="ref-yang_paml_2007" class="csl-entry">
Yang, Ziheng. 2007. <span>“<span>PAML</span> 4: <span>Phylogenetic</span> <span>Analysis</span> by <span>Maximum</span> <span>Likelihood</span>.”</span> <em>Molecular Biology and Evolution</em> 24 (8): 1586–91. <a href="https://doi.org/10.1093/molbev/msm088">https://doi.org/10.1093/molbev/msm088</a>.
</div>
<div id="ref-yu_cp_2020" class="csl-entry">
Yu, Guangchuang. 2020. <span>“Using Ggtree to Visualize Data on Tree-Like Structures.”</span> <em>Current Protocols in Bioinformatics</em> 69 (1): e96. <a href="https://doi.org/10.1002/cpbi.96">https://doi.org/10.1002/cpbi.96</a>.
</div>
<div id="ref-yu_reactomepa_2016" class="csl-entry">
Yu, Guangchuang, and Qing-Yu He. 2016. <span>“<span>ReactomePA</span>: An r/Bioconductor Package for Reactome Pathway Analysis and Visualization.”</span> <em>Molecular <span>BioSystems</span></em> 12 (2): 477–79. <a href="https://doi.org/10.1039/C5MB00663E">https://doi.org/10.1039/C5MB00663E</a>.
</div>
<div id="ref-yu_two_2018" class="csl-entry">
Yu, Guangchuang, Tommy Tsan-Yuk Lam, Huachen Zhu, and Yi Guan. 2018. <span>“Two Methods for Mapping and Visualizing Associated Data on Phylogeny Using Ggtree.”</span> <em>Molecular Biology and Evolution</em> 35 (12): 3041–43. <a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>.
</div>
<div id="ref-yu_ggtree:_2017" class="csl-entry">
Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. <span>“Ggtree: An r Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.”</span> <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.
</div>
<div id="ref-yu2012" class="csl-entry">
Yu, Guangchuang, Li-Gen Wang, Yanyan Han, and Qing-Yu He. 2012. <span>“<span class="nocase">clusterProfiler:</span> An r Package for Comparing Biological Themes Among Gene Clusters.”</span> <em><span>OMICS:</span> A Journal of Integrative Biology</em> 16 (5): 284–87. <a href="https://doi.org/10.1089/omi.2011.0118">https://doi.org/10.1089/omi.2011.0118</a>.
</div>
<div id="ref-yu_chipseeker:_2015" class="csl-entry">
Yu, Guangchuang, Li-Gen Wang, and Qing-Yu He. 2015. <span>“<span>ChIPseeker</span>: An <span>R</span>/<span>Bioconductor</span> Package for <span>ChIP</span> Peak Annotation, Comparison and Visualization.”</span> <em>Bioinformatics</em> 31 (14): 2382–83. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>.
</div>
<div id="ref-yu_dose_2015" class="csl-entry">
Yu, Guangchuang, Li-Gen Wang, Guang-Rong Yan, and Qing-Yu He. 2015. <span>“<span>DOSE</span>: An r/Bioconductor Package for Disease Ontology Semantic and Enrichment Analysis.”</span> <em>Bioinformatics</em> 31 (4): 608–9. <a href="https://doi.org/10.1093/bioinformatics/btu684">https://doi.org/10.1093/bioinformatics/btu684</a>.
</div>
<div id="ref-zmasek_atv:_2001" class="csl-entry">
Zmasek, Christian M., and Sean R. Eddy. 2001. <span>“<span>ATV</span>: Display and Manipulation of Annotated Phylogenetic Trees.”</span> <em>Bioinformatics</em> 17 (4): 383–84. <a href="https://doi.org/10.1093/bioinformatics/17.4.383">https://doi.org/10.1093/bioinformatics/17.4.383</a>.
</div>
</div>
</div>





























  <div class="chapter-nav">
<div class="prev"><a href="publications-of-the-ggtree-package-suite.html"><span class="header-section-number">D</span> Publications of the ggtree package suite</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references">References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2021-11-09.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
