<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>References | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
<meta name="author" content="Guangchuang Yu">
<!-- CSS --><!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.6/header-attrs.js"></script><script src="libs/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.5.2/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.5.2/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.1.9000/tabs.js"></script><script src="libs/bs3compat-0.2.1.9000/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="libs/font-awesome-5.3.1/css/fontawesome-all.min.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="libs/pathseg-1.0.2/pathseg.js"></script><script src="libs/comic-0.961/comic.min.js"></script><script src="libs/comicR-binding-0.2/comicR.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome</a></li>
<li><a class="" href="introduction.html">Introduction</a></li>
<li class="book-part">Part I: Tree data input, output and manipulation</li>
<li><a class="" href="chapter1.html"><span class="header-section-number">1</span> Importing Tree with Data</a></li>
<li><a class="" href="chapter2.html"><span class="header-section-number">2</span> Manipulating Tree with Data</a></li>
<li><a class="" href="chapter3.html"><span class="header-section-number">3</span> Exporting tree with data</a></li>
<li class="book-part">Part II: Tree data visualization and annotation</li>
<li><a class="" href="chapter4.html"><span class="header-section-number">4</span> Phylogenetic Tree Visualization</a></li>
<li><a class="" href="chapter5.html"><span class="header-section-number">5</span> Phylogenetic Tree Annotation</a></li>
<li><a class="" href="chapter6.html"><span class="header-section-number">6</span> Visual Exploration of Phylogenetic Tree</a></li>
<li><a class="" href="chapter7.html"><span class="header-section-number">7</span> Plotting tree with data</a></li>
<li><a class="" href="chapter8.html"><span class="header-section-number">8</span> Annotating Tree with Silhouette Images and Sub-plots</a></li>
<li><a class="" href="chapter9.html"><span class="header-section-number">9</span> ggtree for other tree-like objects</a></li>
<li class="book-part">Part III: ggtree extensions</li>
<li><a class="" href="chapter10.html"><span class="header-section-number">10</span> ggtreeExtra for presenting data on circular layout</a></li>
<li><a class="" href="chapter11.html"><span class="header-section-number">11</span> Other ggtree extensions</a></li>
<li class="book-part">Part IV: Miscellaneous topics</li>
<li><a class="" href="chapter12.html"><span class="header-section-number">12</span> ggtree utilities</a></li>
<li><a class="" href="chapter13.html"><span class="header-section-number">13</span> Gallery of reproducible examples</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="faq.html"><span class="header-section-number">A</span> Frequently asked questions</a></li>
<li><a class="" href="related-tools.html"><span class="header-section-number">B</span> Related tools</a></li>
<li><a class="" href="figures-and-tables.html"><span class="header-section-number">C</span> Figures and Tables</a></li>
<li><a class="" href="session-info.html"><span class="header-section-number">D</span> Session Info</a></li>
<li><a class="active" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references" class="section level1 unnumbered">
<h1>References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h1>

<div id="refs" class="references hanging-indent">
<div id="ref-amer2017microbiome">
<p>Amer, Abdrazak, Sheila Galvin, Claire M Healy, and Gary P Moran. 2017. “The Microbiome of Potentially Malignant Oral Leukoplakia Exhibits Enrichment for Fusobacterium, Leptotrichia, Campylobacter, and Rothia Species.” <em>Frontiers in Microbiology</em> 8: 2391. <a href="https://doi.org/10.3389/fmicb.2017.02391">https://doi.org/10.3389/fmicb.2017.02391</a>.</p>
</div>
<div id="ref-arenas_trends_2015">
<p>Arenas, Miguel. 2015. “Trends in Substitution Models of Molecular Evolution.” <em>Frontiers in Genetics</em> 6 (October). <a href="https://doi.org/10.3389/fgene.2015.00319">https://doi.org/10.3389/fgene.2015.00319</a>.</p>
</div>
<div id="ref-GraPhlAn">
<p>Asnicar, Francesco, George Weingart, Timothy L Tickle, Curtis Huttenhower, and Nicola Segata. 2015. “Compact Graphical Representation of Phylogenetic Data and Metadata with Graphlan.” <em>PeerJ</em> 3: e1029. <a href="https://doi.org/10.7717/peerj.1029">https://doi.org/10.7717/peerj.1029</a>.</p>
</div>
<div id="ref-berger_performance_2011">
<p>Berger, Simon A., Denis Krompass, and Alexandros Stamatakis. 2011. “Performance, Accuracy, and Web Server for Evolutionary Placement of Short Sequence Reads Under Maximum Likelihood.” <em>Systematic Biology</em>, March, 291–302. <a href="https://doi.org/10.1093/sysbio/syr010">https://doi.org/10.1093/sysbio/syr010</a>.</p>
</div>
<div id="ref-bosi_comparative_2016">
<p>Bosi, Emanuele, Jonathan M. Monk, Ramy K. Aziz, Marco Fondi, Victor Nizet, and Bernhard Ø. Palsson. 2016. “Comparative Genome-Scale Modelling of Staphylococcus Aureus Strains Identifies Strain-Specific Metabolic Capabilities Linked to Pathogenicity.” <em>Proceedings of the National Academy of Sciences of the United States of America</em> 113 (26): E3801–E3809. <a href="https://doi.org/10.1073/pnas.1523199113">https://doi.org/10.1073/pnas.1523199113</a>.</p>
</div>
<div id="ref-bouckaert_beast_2014">
<p>Bouckaert, Remco, Joseph Heled, Denise Kühnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. “BEAST 2: A Software Platform for Bayesian Evolutionary Analysis.” <em>PLoS Comput Biol</em> 10 (4): e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>.</p>
</div>
<div id="ref-boussau_genome-scale_2013">
<p>Boussau, Bastien, Gergely J. Szöllősi, Laurent Duret, Manolo Gouy, Eric Tannier, and Vincent Daubin. 2013. “Genome-Scale Coestimation of Species and Gene Trees.” <em>Genome Research</em> 23 (2): 323–30. <a href="https://doi.org/10.1101/gr.141978.112">https://doi.org/10.1101/gr.141978.112</a>.</p>
</div>
<div id="ref-callahan_dada2_2016">
<p>Callahan, Benjamin J., Paul J. McMurdie, Michael J. Rosen, Andrew W. Han, Amy Jo A. Johnson, and Susan P. Holmes. 2016. “DADA2: High-Resolution Sample Inference from Illumina Amplicon Data.” <em>Nature Methods</em> 13 (7): 581–83. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>.</p>
</div>
<div id="ref-chen_ancient_2017">
<p>Chen, Zigui, Wendy C. S. Ho, Siaw Shi Boon, Priscilla T. Y. Law, Martin C. W. Chan, Rob DeSalle, Robert D. Burk, and Paul K. S. Chan. 2017. “Ancient Evolution and Dispersion of Human Papillomavirus 58 Variants.” <em>Journal of Virology</em> 91 (21): e01285–17. <a href="https://doi.org/10.1128/JVI.01285-17">https://doi.org/10.1128/JVI.01285-17</a>.</p>
</div>
<div id="ref-chevenet_treedyn:_2006">
<p>Chevenet, François, Christine Brun, Anne-Laure Bañuls, Bernard Jacq, and Richard Christen. 2006. “TreeDyn: Towards Dynamic Graphics and Annotations for Analyses of Trees.” <em>BMC Bioinformatics</em> 7 (October): 439. <a href="https://doi.org/10.1186/1471-2105-7-439">https://doi.org/10.1186/1471-2105-7-439</a>.</p>
</div>
<div id="ref-Chowe:_2020">
<p>Chow, Nancy A., José F. Muñoz, Lalitha Gade, Elizabeth L. Berkow, Xiao Li, Rory M. Welsh, Kaitlin Forsberg, et al. 2020. “Tracing the Evolutionary History and Global Expansion of Candida Auris Using Population Genomic Analyses.” Edited by Geraldine Butler and Kirsten Nielsen. <em>mBio</em> 11 (2). <a href="https://doi.org/10.1128/mBio.03364-19">https://doi.org/10.1128/mBio.03364-19</a>.</p>
</div>
<div id="ref-felsenstein_evolutionary_1981">
<p>Felsenstein, J. 1981. “Evolutionary Trees from DNA Sequences: A Maximum Likelihood Approach.” <em>Journal of Molecular Evolution</em> 17 (6): 368–76.</p>
</div>
<div id="ref-felsenstein_phylip_1989">
<p>Felsenstein, Joseph. 1989. “PHYLIP - Phylogeny Inference Package (Version 3.2).” <em>Cladistics</em> 5: 164–66.</p>
</div>
<div id="ref-felsenstein_cases_1978">
<p>———. 1978. “Cases in Which Parsimony or Compatibility Methods Will Be Positively Misleading.” <em>Systematic Biology</em> 27 (4): 401–10. <a href="https://doi.org/10.1093/sysbio/27.4.401">https://doi.org/10.1093/sysbio/27.4.401</a>.</p>
</div>
<div id="ref-fitch_toward_1971">
<p>Fitch, Walter M. 1971. “Toward Defining the Course of Evolution: Minimum Change for a Specific Tree Topology.” <em>Systematic Zoology</em> 20 (4): 406–16. <a href="https://doi.org/10.2307/2412116">https://doi.org/10.2307/2412116</a>.</p>
</div>
<div id="ref-gentleman_bioconductor_2004">
<p>Gentleman, Robert C, Vincent J Carey, Douglas M Bates, Ben Bolstad, Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004. “Bioconductor: Open Software Development for Computational Biology and Bioinformatics.” <em>Genome Biology</em> 5 (10): R80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80">https://doi.org/10.1186/gb-2004-5-10-r80</a>.</p>
</div>
<div id="ref-goldman_codon-based_1994">
<p>Goldman, N., and Z. Yang. 1994. “A Codon-Based Model of Nucleotide Substitution for Protein-Coding DNA Sequences.” <em>Molecular Biology and Evolution</em> 11 (5): 725–36.</p>
</div>
<div id="ref-grubaugh_genomic_2017">
<p>Grubaugh, Nathan D., Jason T. Ladner, Moritz U. G. Kraemer, Gytis Dudas, Amanda L. Tan, Karthik Gangavarapu, Michael R. Wiley, et al. 2017. “Genomic Epidemiology Reveals Multiple Introductions of Zika Virus into the United States.” <em>Nature</em> 546 (7658): 401–5. <a href="https://doi.org/10.1038/nature22400">https://doi.org/10.1038/nature22400</a>.</p>
</div>
<div id="ref-gupta_using_2015">
<p>Gupta, Ankit, and Vineet K Sharma. 2015. “Using the Taxon-Specific Genes for the Taxonomic Classification of Bacterial Genomes.” <em>BMC Genomics</em> 16 (1). <a href="https://doi.org/10.1186/s12864-015-1542-0">https://doi.org/10.1186/s12864-015-1542-0</a>.</p>
</div>
<div id="ref-he_emergence_2013">
<p>He, Ya-Qing, Long Chen, Wen-Bo Xu, Hong Yang, Han-Zhong Wang, Wen-Ping Zong, Hui-Xia Xian, et al. 2013. “Emergence, Circulation, and Spatiotemporal Phylogenetic Analysis of Coxsackievirus A6- and Coxsackievirus A10-Associated Hand, Foot, and Mouth Disease Infections from 2008 to 2012 in Shenzhen, China.” <em>Journal of Clinical Microbiology</em> 51 (11): 3560–6. <a href="https://doi.org/10.1128/JCM.01231-13">https://doi.org/10.1128/JCM.01231-13</a>.</p>
</div>
<div id="ref-He289">
<p>He, Zhen, Raad Z Gharaibeh, Rachel C Newsome, Jllian L Pope, Michael W Dougherty, Sarah Tomkovich, Benoit Pons, et al. 2019. “Campylobacter Jejuni Promotes Colorectal Tumorigenesis Through the Action of Cytolethal Distending Toxin.” <em>Gut</em> 68 (2): 289–300. <a href="https://doi.org/10.1136/gutjnl-2018-317200">https://doi.org/10.1136/gutjnl-2018-317200</a>.</p>
</div>
<div id="ref-he_evolview_2016">
<p>He, Zilong, Huangkai Zhang, Shenghan Gao, Martin J. Lercher, Wei-Hua Chen, and Songnian Hu. 2016. “Evolview V2: An Online Visualization and Management Tool for Customized and Annotated Phylogenetic Trees.” <em>Nucleic Acids Research</em> 44 (W1): W236–241. <a href="https://doi.org/10.1093/nar/gkw370">https://doi.org/10.1093/nar/gkw370</a>.</p>
</div>
<div id="ref-hohna_probabilistic_2014">
<p>Höhna, Sebastian, Tracy A. Heath, Bastien Boussau, Michael J. Landis, Fredrik Ronquist, and John P. Huelsenbeck. 2014. “Probabilistic Graphical Model Representation in Phylogenetics.” <em>Systematic Biology</em> 63 (5): 753–71. <a href="https://doi.org/10.1093/sysbio/syu039">https://doi.org/10.1093/sysbio/syu039</a>.</p>
</div>
<div id="ref-hohna_revbayes:_2016">
<p>Höhna, Sebastian, Michael J. Landis, Tracy A. Heath, Bastien Boussau, Nicolas Lartillot, Brian R. Moore, John P. Huelsenbeck, and Fredrik Ronquist. 2016. “RevBayes: Bayesian Phylogenetic Inference Using Graphical Models and an Interactive Model-Specification Language.” <em>Systematic Biology</em> 65 (4): 726–36. <a href="https://doi.org/10.1093/sysbio/syw021">https://doi.org/10.1093/sysbio/syw021</a>.</p>
</div>
<div id="ref-huelsenbeck_mrbayes_2001">
<p>Huelsenbeck, J. P., and F. Ronquist. 2001. “MRBAYES: Bayesian Inference of Phylogenetic Trees.” <em>Bioinformatics (Oxford, England)</em> 17 (8): 754–55.</p>
</div>
<div id="ref-huson_dendroscope_2012">
<p>Huson, Daniel H., and Celine Scornavacca. 2012. “Dendroscope 3: An Interactive Tool for Rooted Phylogenetic Trees and Networks.” <em>Systematic Biology</em> 61 (6): 1061–7. <a href="https://doi.org/10.1093/sysbio/sys062">https://doi.org/10.1093/sysbio/sys062</a>.</p>
</div>
<div id="ref-jombart_outbreaktools_2014">
<p>Jombart, Thibaut, David M. Aanensen, Marc Baguelin, Paul Birrell, Simon Cauchemez, Anton Camacho, Caroline Colijn, et al. 2014. “OutbreakTools: A New Platform for Disease Outbreak Analysis Using the R Software.” <em>Epidemics</em> 7 (June): 28–34. <a href="https://doi.org/10.1016/j.epidem.2014.04.003">https://doi.org/10.1016/j.epidem.2014.04.003</a>.</p>
</div>
<div id="ref-kostic2012genomic">
<p>Kostic, Aleksandar D, Dirk Gevers, Chandra Sekhar Pedamallu, Monia Michaud, Fujiko Duke, Ashlee M Earl, Akinyemi I Ojesina, et al. 2012. “Genomic Analysis Identifies Association of Fusobacterium with Colorectal Carcinoma.” <em>Genome Research</em> 22 (2): 292–98. <a href="https://doi.org/10.1101/gr.126573.111">https://doi.org/10.1101/gr.126573.111</a>.</p>
</div>
<div id="ref-kuczynski_using_2011">
<p>Kuczynski, Justin, Jesse Stombaugh, William Anton Walters, Antonio González, J. Gregory Caporaso, and Rob Knight. 2011. “Using QIIME to Analyze 16S rRNA Gene Sequences from Microbial Communities.” <em>Current Protocols in Bioinformatics / Editoral Board, Andreas D. Baxevanis ... [et Al.]</em> CHAPTER (December): Unit10.7. <a href="https://doi.org/10.1002/0471250953.bi1007s36">https://doi.org/10.1002/0471250953.bi1007s36</a>.</p>
</div>
<div id="ref-kumar_mega7_2016">
<p>Kumar, Sudhir, Glen Stecher, and Koichiro Tamura. 2016. “MEGA7: Molecular Evolutionary Genetics Analysis Version 7.0 for Bigger Datasets.” <em>Molecular Biology and Evolution</em> 33 (7): 1870–4. <a href="https://doi.org/10.1093/molbev/msw054">https://doi.org/10.1093/molbev/msw054</a>.</p>
</div>
<div id="ref-kunin_pyrotagger_2010">
<p>Kunin, Victor, and Philip Hugenholtz. 2010. “PyroTagger : A Fast , Accurate Pipeline for Analysis of rRNA Amplicon Pyrosequence Data.” <em>The Open Journal</em>, 1–8. <a href="http://www.theopenjournal.org/toj_articles/1">http://www.theopenjournal.org/toj_articles/1</a>.</p>
</div>
<div id="ref-lam_phylodynamics_2012">
<p>Lam, Tommy Tsan-Yuk, Chung-Chau Hon, Philippe Lemey, Oliver G. Pybus, Mang Shi, Hein Min Tun, Jun Li, Jingwei Jiang, Edward C. Holmes, and Frederick Chi-Ching Leung. 2012. “Phylodynamics of H5N1 Avian Influenza Virus in Indonesia.” <em>Molecular Ecology</em> 21 (12): 3062–77. <a href="https://doi.org/10.1111/j.1365-294X.2012.05577.x">https://doi.org/10.1111/j.1365-294X.2012.05577.x</a>.</p>
</div>
<div id="ref-lam_dissemination_2015">
<p>Lam, Tommy Tsan-Yuk, Boping Zhou, Jia Wang, Yujuan Chai, Yongyi Shen, Xinchun Chen, Chi Ma, et al. 2015. “Dissemination, Divergence and Establishment of H7N9 Influenza Viruses in China.” <em>Nature</em> 522 (7554): 102–5. <a href="https://doi.org/10.1038/nature14348">https://doi.org/10.1038/nature14348</a>.</p>
</div>
<div id="ref-larsen_identification_2019">
<p>Larsen, Frederik T., Bertrand Bed’Hom, Bernt Guldbrandtsen, and Tina S. Dalgaard. 2019. “Identification and Tissue-Expression Profiling of Novel Chicken c-Type Lectin-Like Domain Containing Proteins as Potential Targets for Carbohydrate-Based Vaccine Strategies.” <em>Molecular Immunology</em> 114 (October): 216–25. <a href="https://doi.org/10.1016/j.molimm.2019.07.022">https://doi.org/10.1016/j.molimm.2019.07.022</a>.</p>
</div>
<div id="ref-lemmon_importance_2004">
<p>Lemmon, Alan R., and Emily C. Moriarty. 2004. “The Importance of Proper Model Assumption in Bayesian Phylogenetics.” <em>Systematic Biology</em> 53 (2): 265–77. <a href="https://doi.org/10.1080/10635150490423520">https://doi.org/10.1080/10635150490423520</a>.</p>
</div>
<div id="ref-letunic_interactive_2007">
<p>Letunic, Ivica, and Peer Bork. 2007. “Interactive Tree of Life (iTOL): An Online Tool for Phylogenetic Tree Display and Annotation.” <em>Bioinformatics</em> 23 (1): 127–28. <a href="https://doi.org/10.1093/bioinformatics/btl529">https://doi.org/10.1093/bioinformatics/btl529</a>.</p>
</div>
<div id="ref-liang_expansion_2014">
<p>Liang, Huyi, Tommy Tsan-Yuk Lam, Xiaohui Fan, Xinchun Chen, Yu Zeng, Ji Zhou, Lian Duan, et al. 2014. “Expansion of Genotypic Diversity and Establishment of 2009 H1N1 Pandemic-Origin Internal Genes in Pigs in China.” <em>Journal of Virology</em>, July, JVI.01327–14. <a href="https://doi.org/10.1128/JVI.01327-14">https://doi.org/10.1128/JVI.01327-14</a>.</p>
</div>
<div id="ref-lott_covenntree:_2015">
<p>Lott, Steffen C., Björn Voß, Wolfgang R. Hess, and Claudia Steglich. 2015. “CoVennTree: A New Method for the Comparative Analysis of Large Datasets.” <em>Frontiers in Genetics</em> 6: 43. <a href="https://doi.org/10.3389/fgene.2015.00043">https://doi.org/10.3389/fgene.2015.00043</a>.</p>
</div>
<div id="ref-maddison_nexus:_1997">
<p>Maddison, David R., David L. Swofford, Wayne P. Maddison, and David Cannatella. 1997. “Nexus: An Extensible File Format for Systematic Information.” <em>Systematic Biology</em> 46 (4): 590–621. <a href="https://doi.org/10.1093/sysbio/46.4.590">https://doi.org/10.1093/sysbio/46.4.590</a>.</p>
</div>
<div id="ref-matsen_format_2012">
<p>Matsen, Frederick A., Noah G. Hoffman, Aaron Gallagher, and Alexandros Stamatakis. 2012. “A Format for Phylogenetic Placements.” <em>PLoS ONE</em> 7 (2): e31009. <a href="https://doi.org/10.1371/journal.pone.0031009">https://doi.org/10.1371/journal.pone.0031009</a>.</p>
</div>
<div id="ref-matsen_pplacer:_2010">
<p>Matsen, Frederick A., Robin B. Kodner, and E Virginia Armbrust. 2010. “Pplacer: Linear Time Maximum-Likelihood and Bayesian Phylogenetic Placement of Sequences onto a Fixed Reference Tree.” <em>BMC Bioinformatics</em> 11: 538. <a href="https://doi.org/10.1186/1471-2105-11-538">https://doi.org/10.1186/1471-2105-11-538</a>.</p>
</div>
<div id="ref-matsen_pplacer_2010">
<p>Matsen, Frederick A, Robin B Kodner, and E Virginia Armbrust. 2010. “Pplacer: Linear Time Maximum-Likelihood and Bayesian Phylogenetic Placement of Sequences onto a Fixed Reference Tree.” <em>BMC Bioinformatics</em> 11 (1): 538. <a href="https://doi.org/10.1186/1471-2105-11-538">https://doi.org/10.1186/1471-2105-11-538</a>.</p>
</div>
<div id="ref-mcmurdie_phyloseq_2013">
<p>McMurdie, Paul J., and Susan Holmes. 2013. “Phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.” <em>PLoS ONE</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.</p>
</div>
<div id="ref-michonneau_rotl:_2016">
<p>Michonneau, François, Joseph W. Brown, and David J. Winter. 2016. “Rotl: An R Package to Interact with the Open Tree of Life Data.” <em>Methods in Ecology and Evolution</em> 7 (12): 1476–81. <a href="https://doi.org/10.1111/2041-210X.12593">https://doi.org/10.1111/2041-210X.12593</a>.</p>
</div>
<div id="ref-morgan2013HMP">
<p>Morgan, Xochitl C., Nicola Segata, and Curtis Huttenhower. 2013. “Biodiversity and Functional Genomics in the Human Microbiome.” <em>Trends in Genetics</em> 29 (1): 51–58. <a href="https://doi.org/10.1016/J.TIG.2012.09.005">https://doi.org/10.1016/J.TIG.2012.09.005</a>.</p>
</div>
<div id="ref-neher_prediction_2016">
<p>Neher, Richard A., Trevor Bedford, Rodney S. Daniels, Colin A. Russell, and Boris I. Shraiman. 2016. “Prediction, Dynamics, and Visualization of Antigenic Phenotypes of Seasonal Influenza Viruses.” <em>Proceedings of the National Academy of Sciences</em> 113 (12): E1701–E1709. <a href="https://doi.org/10.1073/pnas.1525578113">https://doi.org/10.1073/pnas.1525578113</a>.</p>
</div>
<div id="ref-page_visualizing_2002">
<p>Page, Roderic D. M. 2002. “Visualizing Phylogenetic Trees Using TreeView.” <em>Current Protocols in Bioinformatics</em> Chapter 6 (August): Unit 6.2. <a href="https://doi.org/10.1002/0471250953.bi0602s01">https://doi.org/10.1002/0471250953.bi0602s01</a>.</p>
</div>
<div id="ref-paradis_ape_2004">
<p>Paradis, Emmanuel, Julien Claude, and Korbinian Strimmer. 2004. “APE: Analyses of Phylogenetics and Evolution in R Language.” <em>Bioinformatics</em> 20 (2): 289–90. <a href="https://doi.org/10.1093/bioinformatics/btg412">https://doi.org/10.1093/bioinformatics/btg412</a>.</p>
</div>
<div id="ref-pond_hyphy:_2005">
<p>Pond, Sergei L. Kosakovsky, Simon D. W. Frost, and Spencer V. Muse. 2005. “HyPhy: Hypothesis Testing Using Phylogenies.” <em>Bioinformatics (Oxford, England)</em> 21 (5): 676–79. <a href="https://doi.org/10.1093/bioinformatics/bti079">https://doi.org/10.1093/bioinformatics/bti079</a>.</p>
</div>
<div id="ref-rannala_probability_1996">
<p>Rannala, B., and Z. Yang. 1996. “Probability Distribution of Molecular Evolutionary Trees: A New Method of Phylogenetic Inference.” <em>Journal of Molecular Evolution</em> 43 (3): 304–11.</p>
</div>
<div id="ref-rstats">
<p>R Core Team. 2016. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
</div>
<div id="ref-retief_phylogenetic_2000">
<p>Retief, J. D. 2000. “Phylogenetic Analysis Using PHYLIP.” <em>Methods in Molecular Biology (Clifton, N.J.)</em> 132: 243–58.</p>
</div>
<div id="ref-revell_phytools_2012">
<p>Revell, Liam J. 2012. “Phytools: An R Package for Phylogenetic Comparative Biology (and Other Things).” <em>Methods in Ecology and Evolution</em> 3 (2): 217–23. <a href="https://doi.org/10.1111/j.2041-210X.2011.00169.x">https://doi.org/10.1111/j.2041-210X.2011.00169.x</a>.</p>
</div>
<div id="ref-sanderson_r8s:_2003">
<p>Sanderson, Michael J. 2003. “R8s: Inferring Absolute Rates of Molecular Evolution and Divergence Times in the Absence of a Molecular Clock.” <em>Bioinformatics</em> 19 (2): 301–2. <a href="https://doi.org/10.1093/bioinformatics/19.2.301">https://doi.org/10.1093/bioinformatics/19.2.301</a>.</p>
</div>
<div id="ref-schliep_phangorn_2011">
<p>Schliep, Klaus Peter. 2011. “Phangorn: Phylogenetic Analysis in R.” <em>Bioinformatics</em> 27 (4): 592–93. <a href="https://doi.org/10.1093/bioinformatics/btq706">https://doi.org/10.1093/bioinformatics/btq706</a>.</p>
</div>
<div id="ref-schloss_introducing_2009">
<p>Schloss, Patrick D., Sarah L. Westcott, Thomas Ryabin, Justine R. Hall, Martin Hartmann, Emily B. Hollister, Ryan A. Lesniewski, et al. 2009. “Introducing Mothur: Open-Source, Platform-Independent, Community-Supported Software for Describing and Comparing Microbial Communities.” <em>Applied and Environmental Microbiology</em> 75 (23): 7537–41. <a href="https://doi.org/10.1128/AEM.01541-09">https://doi.org/10.1128/AEM.01541-09</a>.</p>
</div>
<div id="ref-schmidt_tree-puzzle:_2002">
<p>Schmidt, Heiko A., Korbinian Strimmer, Martin Vingron, and Arndt von Haeseler. 2002. “TREE-PUZZLE: Maximum Likelihood Phylogenetic Analysis Using Quartets and Parallel Computing.” <em>Bioinformatics (Oxford, England)</em> 18 (3): 502–4.</p>
</div>
<div id="ref-schon_age_2015">
<p>Schön, Isa, Rylan Shearn, Koen Martens, Annette Koenders, and Stuart Halse. 2015. “Age and Origin of Australian Bennelongia (Crustacea, Ostracoda).” <em>Hydrobiologia</em> 750 (1): 125–46. <a href="https://doi.org/10.1007/s10750-014-2159-z">https://doi.org/10.1007/s10750-014-2159-z</a>.</p>
</div>
<div id="ref-segata_metagenomic_2011">
<p>Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. “Metagenomic Biomarker Discovery and Explanation.” <em>Genome Biology</em> 12 (6): R60. <a href="https://doi.org/10.1186/gb-2011-12-6-r60">https://doi.org/10.1186/gb-2011-12-6-r60</a>.</p>
</div>
<div id="ref-shoemaker_evidence_1989">
<p>Shoemaker, J. S., and W. M. Fitch. 1989. “Evidence from Nuclear Sequences That Invariable Sites Should Be Considered When Sequence Divergence Is Calculated.” <em>Molecular Biology and Evolution</em> 6 (3): 270–89.</p>
</div>
<div id="ref-stamatakis_raxml_2014">
<p>Stamatakis, Alexandros. 2014. “RAxML Version 8: A Tool for Phylogenetic Analysis and Post-Analysis of Large Phylogenies.” <em>Bioinformatics</em>, January, btu033. <a href="https://doi.org/10.1093/bioinformatics/btu033">https://doi.org/10.1093/bioinformatics/btu033</a>.</p>
</div>
<div id="ref-venkatesh_avian_2018">
<p>Venkatesh, Divya, Marjolein J. Poen, Theo M. Bestebroer, Rachel D. Scheuer, Oanh Vuong, Mzia Chkhaidze, Anna Machablishvili, et al. 2018. “Avian Influenza Viruses in Wild Birds: Virus Evolution in a Multihost Ecosystem.” <em>Journal of Virology</em> 92 (15). <a href="https://doi.org/10.1128/JVI.00433-18">https://doi.org/10.1128/JVI.00433-18</a>.</p>
</div>
<div id="ref-vos_nexml:_2012">
<p>Vos, Rutger A., James P. Balhoff, Jason A. Caravas, Mark T. Holder, Hilmar Lapp, Wayne P. Maddison, Peter E. Midford, et al. 2012. “NeXML: Rich, Extensible, and Verifiable Representation of Comparative Data and Metadata.” <em>Systematic Biology</em> 61 (4): 675–89. <a href="https://doi.org/10.1093/sysbio/sys025">https://doi.org/10.1093/sysbio/sys025</a>.</p>
</div>
<div id="ref-wang_treeio_2020">
<p>Wang, Li-Gen, Tommy Tsan-Yuk Lam, Shuangbin Xu, Zehan Dai, Lang Zhou, Tingze Feng, Pingfan Guo, et al. 2020. “Treeio: An R Package for Phylogenetic Tree Input and Output with Richly Annotated and Associated Data.” <em>Molecular Biology and Evolution</em> 37 (2): 599–603. <a href="https://doi.org/10.1093/molbev/msz240">https://doi.org/10.1093/molbev/msz240</a>.</p>
</div>
<div id="ref-wickham_ggplot2_2016">
<p>Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer. <a href="http://ggplot2.org">http://ggplot2.org</a>.</p>
</div>
<div id="ref-wilgenbusch_inferring_2003">
<p>Wilgenbusch, James C., and David Swofford. 2003. “Inferring Evolutionary Trees with PAUP*.” <em>Current Protocols in Bioinformatics</em> Chapter 6 (February): Unit 6.4. <a href="https://doi.org/10.1002/0471250953.bi0604s00">https://doi.org/10.1002/0471250953.bi0604s00</a>.</p>
</div>
<div id="ref-wilkinson_grammar_2005">
<p>Wilkinson, Leland, D. Wills, D. Rope, A. Norton, and R. Dubbs. 2005. <em>The Grammar of Graphics</em>. 2nd edition. New York: Springer.</p>
</div>
<div id="ref-RN46:_2015">
<p>Wong, Vanessa K., Stephen Baker, Derek J. Pickard, Julian Parkhill, Andrew J. Page, Nicholas A. Feasey, Robert A. Kingsley, et al. 2015. “Phylogeographical Analysis of the Dominant Multidrug-Resistant H58 Clade of Salmonella Typhi Identifies Inter- and Intracontinental Transmission Events.” Journal Article. <em>Nature Genetics</em> 47 (6): 632–39. <a href="https://doi.org/10.1038/ng.3281">https://doi.org/10.1038/ng.3281</a>.</p>
</div>
<div id="ref-wu2013dysbiosis">
<p>Wu, Na, Xi Yang, Ruifen Zhang, Jun Li, Xue Xiao, Yongfei Hu, Yanfei Chen, et al. 2013. “Dysbiosis Signature of Fecal Microbiota in Colorectal Cancer Patients.” <em>Microbial Ecology</em> 66 (2): 462–70. <a href="https://doi.org/10.1007/s00248-013-0245-9">https://doi.org/10.1007/s00248-013-0245-9</a>.</p>
</div>
<div id="ref-yang_maximum_1994">
<p>Yang, Z. 1994. “Maximum Likelihood Phylogenetic Estimation from DNA Sequences with Variable Rates over Sites: Approximate Methods.” <em>Journal of Molecular Evolution</em> 39 (3): 306–14.</p>
</div>
<div id="ref-yang_paml_2007">
<p>Yang, Ziheng. 2007. “PAML 4: Phylogenetic Analysis by Maximum Likelihood.” <em>Molecular Biology and Evolution</em> 24 (8): 1586–91. <a href="https://doi.org/10.1093/molbev/msm088">https://doi.org/10.1093/molbev/msm088</a>.</p>
</div>
<div id="ref-yu_cp_2020">
<p>Yu, Guangchuang. 2020. “Using Ggtree to Visualize Data on Tree-Like Structures.” <em>Current Protocols in Bioinformatics</em> 69 (1): e96. <a href="https://doi.org/10.1002/cpbi.96">https://doi.org/10.1002/cpbi.96</a>.</p>
</div>
<div id="ref-yu_two_2018">
<p>Yu, Guangchuang, Tommy Tsan-Yuk Lam, Huachen Zhu, and Yi Guan. 2018. “Two Methods for Mapping and Visualizing Associated Data on Phylogeny Using Ggtree.” <em>Molecular Biology and Evolution</em> 35 (12): 3041–3. <a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>.</p>
</div>
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
<div id="ref-zmasek_atv:_2001">
<p>Zmasek, Christian M., and Sean R. Eddy. 2001. “ATV: Display and Manipulation of Annotated Phylogenetic Trees.” <em>Bioinformatics</em> 17 (4): 383–84. <a href="https://doi.org/10.1093/bioinformatics/17.4.383">https://doi.org/10.1093/bioinformatics/17.4.383</a>.</p>
</div>
</div>
</div>


























  <div class="chapter-nav">
<div class="prev"><a href="session-info.html"><span class="header-section-number">D</span> Session Info</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references">References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Data Integration, Manipulation and Visualization of Phylogenetic Trees</strong>" was written by Guangchuang Yu. It was last built on 2020-12-18.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
