<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 ggtreeExtra | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 10 ggtreeExtra | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 ggtreeExtra | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 ggtreeExtra | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  



<meta name="date" content="2020-08-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter9.html"/>
<link rel="next" href="other-ggtree-extensions.html"/>
<script src="libs/header-attrs-2.3/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#as-treedata"><i class="fa fa-check"></i><b>1.3.3</b> Converting other tree-like object to <code>phylo</code> or <code>treedata</code> object</a></li>
<li class="chapter" data-level="1.3.4" data-path="chapter1.html"><a href="chapter1.html#get-treedata-data"><i class="fa fa-check"></i><b>1.3.4</b> Getting information from <code>treedata</code> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary1"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary2"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary3"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a>
<ul>
<li class="chapter" data-level="4.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary4"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#colored-strips"><i class="fa fa-check"></i><b>5.2.1</b> Colored strips</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary5"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a>
<ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary6"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a>
<ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Aligning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#composite_plot"><i class="fa fa-check"></i><b>7.5</b> Composite plots</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.6</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.7" data-path="chapter7.html"><a href="chapter7.html#summary7"><i class="fa fa-check"></i><b>7.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a>
<ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-circularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> Emoji in circular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> Emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a>
<ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#igraph"><i class="fa fa-check"></i><b>9.3</b> ggtree for tree graph</a></li>
<li class="chapter" data-level="9.4" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-other-tree-like-structure"><i class="fa fa-check"></i><b>9.4</b> ggtree for other tree-like structure</a></li>
</ul></li>
<li class="part"><span><b>Part III: ggtree extensions</b></span></li>
<li class="chapter" data-level="10" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>10</b> ggtreeExtra</a>
<ul>
<li class="chapter" data-level="10.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="chapter12.html"><a href="chapter12.html#aligning-graphs-to-the-tree-based-on-tree-structure"><i class="fa fa-check"></i><b>10.2</b> Aligning graphs to the tree based on tree structure</a></li>
<li class="chapter" data-level="10.3" data-path="chapter12.html"><a href="chapter12.html#aligning-multiple-graphs-to-the-tree-for-multi-dimensional-data"><i class="fa fa-check"></i><b>10.3</b> Aligning multiple graphs to the tree for multi-dimensional data</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="other-ggtree-extensions.html"><a href="other-ggtree-extensions.html"><i class="fa fa-check"></i><b>11</b> Other ggtree extensions</a></li>
<li class="part"><span><b>Part IV: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="12" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>12</b> ggtree utilities</a>
<ul>
<li class="chapter" data-level="12.1" data-path="chapter10.html"><a href="chapter10.html#facet-utilities"><i class="fa fa-check"></i><b>12.1</b> facet utilities</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="chapter10.html"><a href="chapter10.html#facet_widths"><i class="fa fa-check"></i><b>12.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="12.1.2" data-path="chapter10.html"><a href="chapter10.html#facet_labeller"><i class="fa fa-check"></i><b>12.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="chapter10.html"><a href="chapter10.html#geom2"><i class="fa fa-check"></i><b>12.2</b> Geometric layers</a></li>
<li class="chapter" data-level="12.3" data-path="chapter10.html"><a href="chapter10.html#layout-utilities"><i class="fa fa-check"></i><b>12.3</b> Layout utilities</a></li>
<li class="chapter" data-level="12.4" data-path="chapter10.html"><a href="chapter10.html#scale-utilities"><i class="fa fa-check"></i><b>12.4</b> Scale utilities</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="chapter10.html"><a href="chapter10.html#xlim_expand"><i class="fa fa-check"></i><b>12.4.1</b> xlim_expand</a></li>
<li class="chapter" data-level="12.4.2" data-path="chapter10.html"><a href="chapter10.html#hexpand"><i class="fa fa-check"></i><b>12.4.2</b> hexpand</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>13</b> Gallery of reproducible examples</a>
<ul>
<li class="chapter" data-level="13.1" data-path="chapter11.html"><a href="chapter11.html#hpv58"><i class="fa fa-check"></i><b>13.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="13.2" data-path="chapter11.html"><a href="chapter11.html#symbolic-bootstrap"><i class="fa fa-check"></i><b>13.2</b> Displaying different symbolic points for bootstrap values.</a></li>
<li class="chapter" data-level="13.3" data-path="chapter11.html"><a href="chapter11.html#phylo-grouping"><i class="fa fa-check"></i><b>13.3</b> Highlighting different groups.</a></li>
<li class="chapter" data-level="13.4" data-path="chapter11.html"><a href="chapter11.html#genome-locus"><i class="fa fa-check"></i><b>13.4</b> Phylogenetic tree with genome locus structure</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#branch-setting"><i class="fa fa-check"></i><b>A.5</b> Branch setting</a>
<ul>
<li class="chapter" data-level="A.5.1" data-path="faq.html"><a href="faq.html#change-branch-length-of-outgroup"><i class="fa fa-check"></i><b>A.5.1</b> Change branch length of outgroup</a></li>
<li class="chapter" data-level="A.5.2" data-path="faq.html"><a href="faq.html#bind-tip"><i class="fa fa-check"></i><b>A.5.2</b> Attach a new tip to a tree</a></li>
<li class="chapter" data-level="A.5.3" data-path="faq.html"><a href="faq.html#change-colours-or-line-types-of-arbitrary-selected-branches"><i class="fa fa-check"></i><b>A.5.3</b> Change colours or line types of arbitrary selected branches</a></li>
<li class="chapter" data-level="A.5.4" data-path="faq.html"><a href="faq.html#arbitrary-point"><i class="fa fa-check"></i><b>A.5.4</b> Add an arbitrary point to a branch</a></li>
</ul></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#different-x-axis-labels-for-different-facet-panels"><i class="fa fa-check"></i><b>A.6</b> Different x-axis labels for different facet panels</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.7</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.8</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.9" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.9</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.10" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.10</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="related-tools.html"><a href="related-tools.html"><i class="fa fa-check"></i><b>B</b> Related tools</a>
<ul>
<li class="chapter" data-level="B.1" data-path="related-tools.html"><a href="related-tools.html#rtol"><i class="fa fa-check"></i><b>B.1</b> rtol: An R interface to Open Tree API</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>C</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a>
<ul>
<li class="chapter" data-level="C.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>C.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="C.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>C.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>D</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="E" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>E</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="F" data-path="facet-zoom.html"><a href="facet-zoom.html"><i class="fa fa-check"></i><b>F</b> Zoom in selected portion</a></li>
<li class="chapter" data-level="G" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>G</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="H" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>H</b> Figures and Tables</a></li>
<li class="chapter" data-level="I" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>I</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter12" class="section level1" number="10">
<h1><span class="header-section-number">Chapter 10</span> ggtreeExtra</h1>
<div id="introduction" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Introduction</h2>
<p>Phylogenetic trees can be easily visualized with multiple layout using <a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017" role="doc-biblioref">2017</a>)</span>. It also provides <code>geom_fruit</code> to align the graphs created with associated data to the tree. However, it does not support the trees created using <code>circular</code>, <code>fan</code> or <code>radial</code> layout, which are efficient layouts for visualizing phylogenetic tree and multi-dimensional data. It also provides <code>gheatmap</code> function to visualize heat map plot on the circular tree to interpret the data in phylogenetic context, but it is not universal and only support heat map plot. None R package, to our knowledge, are developed to align multiple layers to the circular trees or other layout trees created in polar coordinate system. To solve the problem, We developed <code>ggtreeExtra</code>, which can align associated graphs to <code>circular</code>, <code>fan</code> or <code>radial</code> and other <code>rectangular</code> layout tree. <code>ggtreeExtra</code> provides function, <code>geom_fruit</code> to align graphs to the tree. But the associated graphs will align in different position. So we also developed <code>geom_fruit_list</code> to add multiple layers in the same position. Furthermore, <code>geom_axis_text</code>, a function to add the axis of associated graphs was developed. These functions are based on <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> using grammar of graphics <span class="citation">(Wickham <a href="#ref-wickham_ggplot2_2009" role="doc-biblioref">2009</a>)</span>.</p>
</div>
<div id="aligning-graphs-to-the-tree-based-on-tree-structure" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Aligning graphs to the tree based on tree structure</h2>
<p>The phylogenetic trees are often visualized with different graph created using associated datasets. Like the <code>geom_facet</code> of <a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017" role="doc-biblioref">2017</a>)</span>, <code>ggtreeExtra</code> also provides <code>geom_fruit</code> layer which accepts an input <code>data.frame</code> and a <code>geom</code> function to plot the input data. The data will be visualized in an additional position of the plot. <code>geom_fruit</code> also is a general function to link graphs to phylogenetic trees. It will re-orders the input data based on the tree structure and displayed the data with the associated <code>geom</code> at specific position.</p>
<p>The <code>geom_fruit</code> is designed to work with most of <code>geom</code> layers defined in <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. It control the position of graphs by <code>position</code> parameter, which was provided the <code>Position</code> object of <code>ggtreeExtra</code>. The default position parameters is ‚Äòauto‚Äô. This means that the <code>geom_bar</code> will use <code>position_stackx()</code>, <code>geom_violin</code> and <code>geom_boxplot</code> will use <code>position_dodgex()</code>, <code>geom_point</code> and <code>geom_tile</code> will use <code>position_identityx()</code>. So if the <code>geom</code> defined in other <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>-based packages has <code>position</code> parameter, which support the result of a call to a position adjustment function, it also can work with <code>geom_fruit</code>, such as <code>geom_star</code> in <a href="https://CRAN.R-project.org/package=ggstar">ggstar</a>, which provides the regular polygon layer for easily discernible shapes based on the grammar of <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. Since the <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> community keeps expanding and more <code>geom</code> layers will be implemented in either <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> or other extensions, <code>geom_fruit</code> also will gain more power to present data in future.</p>


<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="chapter12.html#cb221-1"></a><span class="kw">library</span>(ggtreeExtra)</span>
<span id="cb221-2"><a href="chapter12.html#cb221-2"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb221-3"><a href="chapter12.html#cb221-3"></a><span class="kw">library</span>(phyloseq)</span>
<span id="cb221-4"><a href="chapter12.html#cb221-4"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb221-5"><a href="chapter12.html#cb221-5"></a></span>
<span id="cb221-6"><a href="chapter12.html#cb221-6"></a><span class="kw">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb221-7"><a href="chapter12.html#cb221-7"></a>GP &lt;-<span class="st"> </span>GlobalPatterns</span>
<span id="cb221-8"><a href="chapter12.html#cb221-8"></a>GP &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(GP) <span class="op">&gt;</span><span class="st"> </span><span class="dv">600</span>, GP)</span>
<span id="cb221-9"><a href="chapter12.html#cb221-9"></a><span class="kw">sample_data</span>(GP)<span class="op">$</span>human &lt;-<span class="st"> </span><span class="kw">get_variable</span>(GP, <span class="st">&quot;SampleType&quot;</span>) <span class="op">%in%</span></span>
<span id="cb221-10"><a href="chapter12.html#cb221-10"></a><span class="st">                              </span><span class="kw">c</span>(<span class="st">&quot;Feces&quot;</span>, <span class="st">&quot;Skin&quot;</span>)</span>
<span id="cb221-11"><a href="chapter12.html#cb221-11"></a>mergedGP &lt;-<span class="st"> </span><span class="kw">merge_samples</span>(GP, <span class="st">&quot;SampleType&quot;</span>)</span>
<span id="cb221-12"><a href="chapter12.html#cb221-12"></a>mergedGP &lt;-<span class="st"> </span><span class="kw">rarefy_even_depth</span>(mergedGP,<span class="dt">rngseed=</span><span class="dv">394582</span>)</span>
<span id="cb221-13"><a href="chapter12.html#cb221-13"></a>mergedGP &lt;-<span class="st"> </span><span class="kw">tax_glom</span>(mergedGP,<span class="st">&quot;Order&quot;</span>)</span>
<span id="cb221-14"><a href="chapter12.html#cb221-14"></a></span>
<span id="cb221-15"><a href="chapter12.html#cb221-15"></a>melt_simple &lt;-<span class="st"> </span><span class="kw">psmelt</span>(mergedGP) <span class="op">%&gt;%</span></span>
<span id="cb221-16"><a href="chapter12.html#cb221-16"></a><span class="st">               </span><span class="kw">filter</span>(Abundance <span class="op">&lt;</span><span class="st"> </span><span class="dv">120</span>) <span class="op">%&gt;%</span></span>
<span id="cb221-17"><a href="chapter12.html#cb221-17"></a><span class="st">               </span><span class="kw">select</span>(OTU, <span class="dt">val=</span>Abundance)</span>
<span id="cb221-18"><a href="chapter12.html#cb221-18"></a></span>
<span id="cb221-19"><a href="chapter12.html#cb221-19"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(mergedGP, <span class="dt">layout=</span><span class="st">&quot;fan&quot;</span>, <span class="dt">open.angle=</span><span class="dv">30</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb221-20"><a href="chapter12.html#cb221-20"></a><span class="st">     </span><span class="kw">geom_tippoint</span>(<span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>Phylum), <span class="dt">size=</span><span class="fl">1.5</span>) <span class="op">+</span></span>
<span id="cb221-21"><a href="chapter12.html#cb221-21"></a><span class="st">     </span><span class="kw">scale_color_manual</span>(</span>
<span id="cb221-22"><a href="chapter12.html#cb221-22"></a>         <span class="dt">values=</span>scales<span class="op">::</span><span class="kw">hue_pal</span>()(<span class="dv">24</span>),</span>
<span id="cb221-23"><a href="chapter12.html#cb221-23"></a>         <span class="dt">guide=</span><span class="kw">guide_legend</span>(<span class="dt">keywidth=</span><span class="fl">0.5</span>, <span class="dt">keyheight=</span><span class="fl">0.5</span>, <span class="dt">ncol=</span><span class="dv">1</span>),</span>
<span id="cb221-24"><a href="chapter12.html#cb221-24"></a>     )</span>
<span id="cb221-25"><a href="chapter12.html#cb221-25"></a></span>
<span id="cb221-26"><a href="chapter12.html#cb221-26"></a>p &lt;-<span class="st"> </span>p <span class="op">+</span></span>
<span id="cb221-27"><a href="chapter12.html#cb221-27"></a><span class="st">     </span><span class="kw">geom_fruit</span>(</span>
<span id="cb221-28"><a href="chapter12.html#cb221-28"></a>         <span class="dt">data=</span>melt_simple,</span>
<span id="cb221-29"><a href="chapter12.html#cb221-29"></a>         <span class="dt">geom=</span>geom_boxplot,</span>
<span id="cb221-30"><a href="chapter12.html#cb221-30"></a>         <span class="dt">mapping =</span> <span class="kw">aes</span>(</span>
<span id="cb221-31"><a href="chapter12.html#cb221-31"></a>                     <span class="dt">y=</span>OTU,</span>
<span id="cb221-32"><a href="chapter12.html#cb221-32"></a>                     <span class="dt">x=</span>val,</span>
<span id="cb221-33"><a href="chapter12.html#cb221-33"></a>                     <span class="dt">group=</span>label,</span>
<span id="cb221-34"><a href="chapter12.html#cb221-34"></a>                     <span class="dt">fill=</span>Phylum,</span>
<span id="cb221-35"><a href="chapter12.html#cb221-35"></a>                   ),</span>
<span id="cb221-36"><a href="chapter12.html#cb221-36"></a>         <span class="dt">lwd=</span>.<span class="dv">1</span>,</span>
<span id="cb221-37"><a href="chapter12.html#cb221-37"></a>         <span class="dt">outlier.size=</span><span class="fl">0.5</span>,</span>
<span id="cb221-38"><a href="chapter12.html#cb221-38"></a>         <span class="dt">outlier.stroke=</span><span class="fl">0.08</span>,</span>
<span id="cb221-39"><a href="chapter12.html#cb221-39"></a>         <span class="dt">outlier.shape=</span><span class="dv">21</span>,</span>
<span id="cb221-40"><a href="chapter12.html#cb221-40"></a>         <span class="dt">show.legend=</span><span class="ot">FALSE</span>,</span>
<span id="cb221-41"><a href="chapter12.html#cb221-41"></a>     ) <span class="op">+</span></span>
<span id="cb221-42"><a href="chapter12.html#cb221-42"></a><span class="st">     </span><span class="kw">geom_axis_text</span>(<span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">angle=</span><span class="op">-</span><span class="dv">45</span>, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span></span>
<span id="cb221-43"><a href="chapter12.html#cb221-43"></a><span class="st">     </span><span class="kw">theme</span>(</span>
<span id="cb221-44"><a href="chapter12.html#cb221-44"></a>         <span class="dt">legend.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">9</span>),</span>
<span id="cb221-45"><a href="chapter12.html#cb221-45"></a>         <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">6</span>)</span>
<span id="cb221-46"><a href="chapter12.html#cb221-46"></a>     )</span>
<span id="cb221-47"><a href="chapter12.html#cb221-47"></a>p</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:psbox"></span>
<img src="treedata_files/figure-html/psbox-1.png" alt="Phylogenetic tree with OTU abundance distribution." width="864"  />
<p class="caption">
Figure 10.1: <strong>Phylogenetic tree with OTU abundance distribution</strong>.
</p>
</div>
<p>This example uses microbiome data that provided in <code>phyloseq</code> package and boxplot is employed to visualize species abundance data. The <code>geom_fruit</code> layer automatically re-arranges the abundance data according to the circular tree structure, visualizes the data using the specify <code>geom</code> function.</p>
</div>
<div id="aligning-multiple-graphs-to-the-tree-for-multi-dimensional-data" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> Aligning multiple graphs to the tree for multi-dimensional data</h2>
<p>Circular layout is efficient layout to show the phylogenetic tree and multi-dimensional data. The continuous dataset can be displayed using heat map, bar plot, box plot or dot plot etc. This example reproduce Fig.2 of <span class="citation">(Morgan, Segata, and Huttenhower <a href="#ref-morgan2013HMP" role="doc-biblioref">2013</a>)</span>. The data is provided by GraPhlAn <span class="citation">(Asnicar et al. <a href="#ref-GraPhlAn" role="doc-biblioref">2015</a>)</span>, which contained the relative abundance of microbiome at different body sites. This example demonstrates the abilities of adding multiple layers (dot plot, heat map and bar plot) created with continuous data to a specific panel, and the attributes of tip point also can be extracted to map Figure <a href="chapter12.html#fig:HMPplot">10.2</a>.</p>


<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="chapter12.html#cb222-1"></a><span class="kw">library</span>(ggtreeExtra)</span>
<span id="cb222-2"><a href="chapter12.html#cb222-2"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb222-3"><a href="chapter12.html#cb222-3"></a><span class="kw">library</span>(treeio)</span>
<span id="cb222-4"><a href="chapter12.html#cb222-4"></a><span class="kw">library</span>(tidytree)</span>
<span id="cb222-5"><a href="chapter12.html#cb222-5"></a><span class="kw">library</span>(ggstar)</span>
<span id="cb222-6"><a href="chapter12.html#cb222-6"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb222-7"><a href="chapter12.html#cb222-7"></a><span class="kw">library</span>(ggnewscale)</span>
<span id="cb222-8"><a href="chapter12.html#cb222-8"></a></span>
<span id="cb222-9"><a href="chapter12.html#cb222-9"></a>tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="st">&quot;data/HMP_tree/hmptree.nwk&quot;</span>)</span>
<span id="cb222-10"><a href="chapter12.html#cb222-10"></a><span class="co"># the abundance and types of microbes</span></span>
<span id="cb222-11"><a href="chapter12.html#cb222-11"></a>dat1 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HMP_tree/tippoint_attr.csv&quot;</span>)</span>
<span id="cb222-12"><a href="chapter12.html#cb222-12"></a><span class="co"># the abundance of microbes at different body sites.</span></span>
<span id="cb222-13"><a href="chapter12.html#cb222-13"></a>dat2 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HMP_tree/ringheatmap_attr.csv&quot;</span>)</span>
<span id="cb222-14"><a href="chapter12.html#cb222-14"></a><span class="co"># the abundance of microbes at the body sites of greatest prevalence.</span></span>
<span id="cb222-15"><a href="chapter12.html#cb222-15"></a>dat3 &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;data/HMP_tree/barplot_attr.csv&quot;</span>)</span>
<span id="cb222-16"><a href="chapter12.html#cb222-16"></a></span>
<span id="cb222-17"><a href="chapter12.html#cb222-17"></a><span class="co"># adjust the order</span></span>
<span id="cb222-18"><a href="chapter12.html#cb222-18"></a>dat2<span class="op">$</span>Sites &lt;-<span class="st"> </span><span class="kw">factor</span>(dat2<span class="op">$</span>Sites, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Stool (prevalence)&quot;</span>, <span class="st">&quot;Cheek (prevalence)&quot;</span>,</span>
<span id="cb222-19"><a href="chapter12.html#cb222-19"></a>                                          <span class="st">&quot;Plaque (prevalence)&quot;</span>,<span class="st">&quot;Tongue (prevalence)&quot;</span>,</span>
<span id="cb222-20"><a href="chapter12.html#cb222-20"></a>                                          <span class="st">&quot;Nose (prevalence)&quot;</span>, <span class="st">&quot;Vagina (prevalence)&quot;</span>,</span>
<span id="cb222-21"><a href="chapter12.html#cb222-21"></a>                                          <span class="st">&quot;Skin (prevalence)&quot;</span>))</span>
<span id="cb222-22"><a href="chapter12.html#cb222-22"></a>dat3<span class="op">$</span>Sites &lt;-<span class="st"> </span><span class="kw">factor</span>(dat3<span class="op">$</span>Sites, <span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;Stool (prevalence)&quot;</span>, <span class="st">&quot;Cheek (prevalence)&quot;</span>,</span>
<span id="cb222-23"><a href="chapter12.html#cb222-23"></a>                                          <span class="st">&quot;Plaque (prevalence)&quot;</span>, <span class="st">&quot;Tongue (prevalence)&quot;</span>,</span>
<span id="cb222-24"><a href="chapter12.html#cb222-24"></a>                                          <span class="st">&quot;Nose (prevalence)&quot;</span>, <span class="st">&quot;Vagina (prevalence)&quot;</span>,</span>
<span id="cb222-25"><a href="chapter12.html#cb222-25"></a>                                          <span class="st">&quot;Skin (prevalence)&quot;</span>))</span>
<span id="cb222-26"><a href="chapter12.html#cb222-26"></a><span class="co"># extract the clade label information. Because some nodes of tree are annotated to genera,</span></span>
<span id="cb222-27"><a href="chapter12.html#cb222-27"></a><span class="co"># which can be displayed with high light using ggtree.</span></span>
<span id="cb222-28"><a href="chapter12.html#cb222-28"></a>nodeids &lt;-<span class="st"> </span><span class="kw">nodeid</span>(tree, tree<span class="op">$</span>node.label[<span class="kw">nchar</span>(tree<span class="op">$</span>node.label)<span class="op">&gt;</span><span class="dv">4</span>])</span>
<span id="cb222-29"><a href="chapter12.html#cb222-29"></a>nodedf &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">node=</span>nodeids)</span>
<span id="cb222-30"><a href="chapter12.html#cb222-30"></a>nodelab &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;[</span><span class="ch">\\</span><span class="st">.0-9]&quot;</span>, <span class="st">&quot;&quot;</span>, tree<span class="op">$</span>node.label[<span class="kw">nchar</span>(tree<span class="op">$</span>node.label)<span class="op">&gt;</span><span class="dv">4</span>])</span>
<span id="cb222-31"><a href="chapter12.html#cb222-31"></a><span class="co"># The layers of clade and hightlight</span></span>
<span id="cb222-32"><a href="chapter12.html#cb222-32"></a>hightlight &lt;-<span class="st"> </span><span class="kw">lapply</span>(nodeids, <span class="cf">function</span>(x)<span class="kw">geom_hilight</span>(<span class="dt">node=</span>x, <span class="dt">extendto=</span><span class="fl">6.8</span>, <span class="dt">alpha=</span><span class="fl">0.3</span>,</span>
<span id="cb222-33"><a href="chapter12.html#cb222-33"></a>                                                      <span class="dt">fill=</span><span class="st">&quot;grey&quot;</span>, <span class="dt">color=</span><span class="st">&quot;grey50&quot;</span>, <span class="dt">size=</span><span class="fl">0.05</span>))</span>
<span id="cb222-34"><a href="chapter12.html#cb222-34"></a>poslist &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">1.6</span>, <span class="fl">1.4</span>, <span class="fl">1.6</span>, <span class="fl">0.8</span>, <span class="fl">0.1</span>, <span class="fl">0.25</span>, <span class="fl">1.6</span>, <span class="fl">1.6</span>, <span class="fl">1.2</span>, <span class="fl">0.4</span>,</span>
<span id="cb222-35"><a href="chapter12.html#cb222-35"></a>             <span class="fl">1.2</span>, <span class="fl">1.8</span>, <span class="fl">0.3</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>, <span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.4</span>, <span class="fl">0.4</span>, <span class="fl">0.6</span>,</span>
<span id="cb222-36"><a href="chapter12.html#cb222-36"></a>             <span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.3</span>)</span>
<span id="cb222-37"><a href="chapter12.html#cb222-37"></a>cladelabels &lt;-<span class="st"> </span><span class="kw">mapply</span>(<span class="cf">function</span>(x, y, z){<span class="kw">geom_cladelabel</span>(<span class="dt">node=</span>x, <span class="dt">label=</span>y, <span class="dt">barsize=</span><span class="ot">NA</span>, <span class="dt">extend=</span><span class="dv">0</span>,</span>
<span id="cb222-38"><a href="chapter12.html#cb222-38"></a>                                                    <span class="dt">offset.text=</span>z, <span class="dt">fontsize=</span><span class="fl">1.4</span>, <span class="dt">angle=</span><span class="st">&quot;auto&quot;</span>,</span>
<span id="cb222-39"><a href="chapter12.html#cb222-39"></a>                                                    <span class="dt">hjust=</span><span class="fl">0.5</span>, <span class="dt">horizontal=</span><span class="ot">FALSE</span>, <span class="dt">fontface=</span><span class="st">&quot;italic&quot;</span>)},</span>
<span id="cb222-40"><a href="chapter12.html#cb222-40"></a>                                     nodeids, nodelab, poslist, <span class="dt">SIMPLIFY=</span><span class="ot">FALSE</span>)</span>
<span id="cb222-41"><a href="chapter12.html#cb222-41"></a></span>
<span id="cb222-42"><a href="chapter12.html#cb222-42"></a><span class="co"># The circular layout tree.</span></span>
<span id="cb222-43"><a href="chapter12.html#cb222-43"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;fan&quot;</span>, <span class="dt">size=</span><span class="fl">0.15</span>, <span class="dt">open.angle=</span><span class="dv">5</span>) <span class="op">+</span></span>
<span id="cb222-44"><a href="chapter12.html#cb222-44"></a><span class="st">     </span><span class="kw">geom_hilight</span>(<span class="dt">data=</span>nodedf, <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">node=</span>node),</span>
<span id="cb222-45"><a href="chapter12.html#cb222-45"></a>                  <span class="dt">extendto=</span><span class="fl">6.8</span>, <span class="dt">alpha=</span><span class="fl">0.3</span>, <span class="dt">fill=</span><span class="st">&quot;grey&quot;</span>, <span class="dt">color=</span><span class="st">&quot;grey50&quot;</span>,</span>
<span id="cb222-46"><a href="chapter12.html#cb222-46"></a>                  <span class="dt">size=</span><span class="fl">0.05</span>)</span>
<span id="cb222-47"><a href="chapter12.html#cb222-47"></a>p &lt;-<span class="st"> </span>p <span class="op">%&lt;+%</span><span class="st"> </span>dat1 <span class="op">+</span><span class="st"> </span><span class="kw">geom_fruit</span>(<span class="dt">geom=</span>geom_star,</span>
<span id="cb222-48"><a href="chapter12.html#cb222-48"></a>                              <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">fill=</span>Phylum, <span class="dt">starshape=</span>Type, <span class="dt">size=</span>Size),</span>
<span id="cb222-49"><a href="chapter12.html#cb222-49"></a>                              <span class="dt">position=</span><span class="st">&quot;identity&quot;</span>,<span class="dt">starstroke=</span><span class="fl">0.1</span>)<span class="op">+</span></span>
<span id="cb222-50"><a href="chapter12.html#cb222-50"></a><span class="st">         </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#FFC125&quot;</span>,<span class="st">&quot;#87CEFA&quot;</span>,<span class="st">&quot;#7B68EE&quot;</span>,<span class="st">&quot;#808080&quot;</span>,<span class="st">&quot;#800080&quot;</span>,</span>
<span id="cb222-51"><a href="chapter12.html#cb222-51"></a>                                    <span class="st">&quot;#9ACD32&quot;</span>,<span class="st">&quot;#D15FEE&quot;</span>,<span class="st">&quot;#FFC0CB&quot;</span>,<span class="st">&quot;#EE6A50&quot;</span>,<span class="st">&quot;#8DEEEE&quot;</span>,</span>
<span id="cb222-52"><a href="chapter12.html#cb222-52"></a>                                    <span class="st">&quot;#006400&quot;</span>,<span class="st">&quot;#800000&quot;</span>,<span class="st">&quot;#B0171F&quot;</span>,<span class="st">&quot;#191970&quot;</span>),</span>
<span id="cb222-53"><a href="chapter12.html#cb222-53"></a>                           <span class="dt">guide=</span><span class="kw">guide_legend</span>(<span class="dt">keywidth =</span> <span class="fl">0.5</span>, <span class="dt">keyheight =</span> <span class="fl">0.5</span>, <span class="dt">order=</span><span class="dv">1</span>,</span>
<span id="cb222-54"><a href="chapter12.html#cb222-54"></a>                                              <span class="dt">override.aes=</span><span class="kw">list</span>(<span class="dt">starshape=</span><span class="dv">15</span>)),</span>
<span id="cb222-55"><a href="chapter12.html#cb222-55"></a>                           <span class="dt">na.translate=</span><span class="ot">FALSE</span>)<span class="op">+</span></span>
<span id="cb222-56"><a href="chapter12.html#cb222-56"></a><span class="st">         </span><span class="kw">scale_starshape_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="dv">15</span>, <span class="dv">1</span>),</span>
<span id="cb222-57"><a href="chapter12.html#cb222-57"></a>                                <span class="dt">guide=</span><span class="kw">guide_legend</span>(<span class="dt">keywidth =</span> <span class="fl">0.5</span>, <span class="dt">keyheight =</span> <span class="fl">0.5</span>, <span class="dt">order=</span><span class="dv">2</span>),</span>
<span id="cb222-58"><a href="chapter12.html#cb222-58"></a>                                <span class="dt">na.translate=</span><span class="ot">FALSE</span>)<span class="op">+</span></span>
<span id="cb222-59"><a href="chapter12.html#cb222-59"></a><span class="st">         </span><span class="kw">scale_size_continuous</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="fl">2.5</span>),</span>
<span id="cb222-60"><a href="chapter12.html#cb222-60"></a>                               <span class="dt">guide =</span> <span class="kw">guide_legend</span>(<span class="dt">keywidth =</span> <span class="fl">0.5</span>, <span class="dt">keyheight =</span> <span class="fl">0.5</span>, <span class="dt">order=</span><span class="dv">3</span>,</span>
<span id="cb222-61"><a href="chapter12.html#cb222-61"></a>                                                    <span class="dt">override.aes=</span><span class="kw">list</span>(<span class="dt">starshape=</span><span class="dv">15</span>)))<span class="op">+</span></span>
<span id="cb222-62"><a href="chapter12.html#cb222-62"></a><span class="st">         </span><span class="kw">new_scale_fill</span>()<span class="op">+</span></span>
<span id="cb222-63"><a href="chapter12.html#cb222-63"></a><span class="st">         </span><span class="kw">geom_fruit</span>(<span class="dt">data=</span>dat2, <span class="dt">geom=</span>geom_tile,</span>
<span id="cb222-64"><a href="chapter12.html#cb222-64"></a>                    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">y=</span>ID, <span class="dt">x=</span>Sites, <span class="dt">alpha=</span>Abundance, <span class="dt">fill=</span>Sites),</span>
<span id="cb222-65"><a href="chapter12.html#cb222-65"></a>                    <span class="dt">color =</span> <span class="st">&quot;grey50&quot;</span>, <span class="dt">offset =</span> <span class="fl">0.04</span>,<span class="dt">size =</span> <span class="fl">0.02</span>)<span class="op">+</span></span>
<span id="cb222-66"><a href="chapter12.html#cb222-66"></a><span class="st">         </span><span class="kw">scale_alpha_continuous</span>(<span class="dt">range=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb222-67"><a href="chapter12.html#cb222-67"></a>                             <span class="dt">guide=</span><span class="kw">guide_legend</span>(<span class="dt">keywidth =</span> <span class="fl">0.3</span>, <span class="dt">keyheight =</span> <span class="fl">0.3</span>, <span class="dt">order=</span><span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb222-68"><a href="chapter12.html#cb222-68"></a><span class="st">         </span>cladelabels <span class="op">+</span></span>
<span id="cb222-69"><a href="chapter12.html#cb222-69"></a><span class="st">         </span><span class="kw">geom_fruit</span>(<span class="dt">data=</span>dat3, <span class="dt">geom=</span>geom_bar,</span>
<span id="cb222-70"><a href="chapter12.html#cb222-70"></a>                    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">y=</span>ID, <span class="dt">x=</span>HigherAbundance, <span class="dt">fill=</span>Sites),</span>
<span id="cb222-71"><a href="chapter12.html#cb222-71"></a>                    <span class="dt">pwidth=</span><span class="fl">0.38</span>, <span class="dt">orientation=</span><span class="st">&quot;y&quot;</span>, <span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>)<span class="op">+</span></span>
<span id="cb222-72"><a href="chapter12.html#cb222-72"></a><span class="st">         </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span><span class="kw">c</span>(<span class="st">&quot;#0000FF&quot;</span>,<span class="st">&quot;#FFA500&quot;</span>,<span class="st">&quot;#FF0000&quot;</span>,<span class="st">&quot;#800000&quot;</span>,</span>
<span id="cb222-73"><a href="chapter12.html#cb222-73"></a>                                    <span class="st">&quot;#006400&quot;</span>,<span class="st">&quot;#800080&quot;</span>,<span class="st">&quot;#696969&quot;</span>),</span>
<span id="cb222-74"><a href="chapter12.html#cb222-74"></a>                           <span class="dt">guide=</span><span class="kw">guide_legend</span>(<span class="dt">keywidth =</span> <span class="fl">0.3</span>, <span class="dt">keyheight =</span> <span class="fl">0.3</span>, <span class="dt">order=</span><span class="dv">4</span>))<span class="op">+</span></span>
<span id="cb222-75"><a href="chapter12.html#cb222-75"></a><span class="st">         </span><span class="kw">geom_treescale</span>(<span class="dt">fontsize=</span><span class="fl">1.2</span>, <span class="dt">linesize=</span><span class="fl">0.3</span>, <span class="dt">x=</span><span class="fl">4.9</span>, <span class="dt">y=</span><span class="fl">0.1</span>) <span class="op">+</span></span>
<span id="cb222-76"><a href="chapter12.html#cb222-76"></a><span class="st">         </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(<span class="fl">0.93</span>, <span class="fl">0.5</span>),</span>
<span id="cb222-77"><a href="chapter12.html#cb222-77"></a>               <span class="dt">legend.background=</span><span class="kw">element_rect</span>(<span class="dt">fill=</span><span class="ot">NA</span>),</span>
<span id="cb222-78"><a href="chapter12.html#cb222-78"></a>               <span class="dt">legend.title=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">5</span>),</span>
<span id="cb222-79"><a href="chapter12.html#cb222-79"></a>               <span class="dt">legend.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">4</span>),</span>
<span id="cb222-80"><a href="chapter12.html#cb222-80"></a>               <span class="dt">legend.spacing.y =</span> <span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb222-81"><a href="chapter12.html#cb222-81"></a>p</span></code></pre></div>
<div class="figure" style="text-align: center"><span id="fig:HMPplot"></span>
<img src="treedata_files/figure-html/HMPplot-1.png" alt="Phylogenetic tree about the abundance of microbes at different sites of human." width="2100"  />
<p class="caption">
Figure 10.2: <strong>Phylogenetic tree about the abundance of microbes at different sites of human</strong>.
</p>
</div>
<p>The shape of tip labels indicated the commensal microbes or potential pathogens. The transparency of heat map indicates the abundance of microbes, and colors of heat map indicate the different sites of human. The bar plot indicates the relative abundance at body site of the most abundance. The node labels contain taxonomy information in this example, so it can be highlight using <code>geom_hilight</code>. The datasets disabled with heat map and bar plot are the format of specific <code>geom</code> of <code>ggplot2</code>. If you have short table format datasets, you can use <code>reshape2::melt()</code> or <code>tidyr::pivot_longer()</code> to convert them.</p>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-GraPhlAn">
<p>Asnicar, Francesco, George Weingart, Timothy L Tickle, Curtis Huttenhower, and Nicola Segata. 2015. ‚ÄúCompact Graphical Representation of Phylogenetic Data and Metadata with Graphlan.‚Äù <em>PeerJ</em> 3: e1029. <a href="https://doi.org/10.7717/peerj.1029">https://doi.org/10.7717/peerj.1029</a>.</p>
</div>
<div id="ref-morgan2013HMP">
<p>Morgan, Xochitl C., Nicola Segata, and Curtis Huttenhower. 2013. ‚ÄúBiodiversity and Functional Genomics in the Human Microbiome.‚Äù <em>Trends in Genetics</em> 29 (1): 51‚Äì58. <a href="https://doi.org/10.1016/J.TIG.2012.09.005">https://doi.org/10.1016/J.TIG.2012.09.005</a>.</p>
</div>
<div id="ref-wickham_ggplot2_2009">
<p>Wickham, Hadley. 2009. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. 1st ed. Springer.</p>
</div>
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. ‚ÄúGgtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.‚Äù <em>Methods in Ecology and Evolution</em> 8 (1): 28‚Äì36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter9.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="other-ggtree-extensions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["treedata.pdf", "treedata.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
