<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Exporting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 3 Exporting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.11.3 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Exporting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Exporting tree with data | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu School of Basic Medical Sciences, Southern Medical University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter2.html">
<link rel="next" href="chapter4.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.3.3</b> Getting information from <em>treedata</em> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#layouts-of-phylogenetic-tree"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-scale-evolution-distance"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#displaying-nodestips"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.4</b> Color tree</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.5</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-the-tree-view"><i class="fa fa-check"></i><b>4.3.6</b> Modify compenents of the tree view</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#annotate-clades"><i class="fa fa-check"></i><b>5.2.1</b> Annotate clades</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic"><i class="fa fa-check"></i><b>5.2.2</b> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.3</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.4</b> Taxa connection</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-based-on-tree-classes-defined-in-other-r-packages"><i class="fa fa-check"></i><b>5.3.2</b> Tree annotation based on tree classes defined in other R packages</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.2</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#scaling-selected-clade"><i class="fa fa-check"></i><b>6.3</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#zoom-on-a-portion-of-tree"><i class="fa fa-check"></i><b>6.6</b> Zoom on a portion of tree</a></li>
<li class="chapter" data-level="6.7" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> The <code>%&lt;+%</code> operator</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#visualize-tree-with-associated-matrix"><i class="fa fa-check"></i><b>7.2</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.2.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.2.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#plot-tree-with-associated-data"><i class="fa fa-check"></i><b>7.4</b> Plot tree with associated data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree utilities</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#facet_widths"><i class="fa fa-check"></i><b>9.1</b> facet_widths</a></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#facet_labeller"><i class="fa fa-check"></i><b>9.2</b> facet_labeller</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#xlim_expand"><i class="fa fa-check"></i><b>9.3</b> xlim_expand</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree for other tree objects</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.5</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.6</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.7</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.8</b> Editing tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Image processing using magick package</a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#example-1-remove-background-of-images"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>D</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="E" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>E</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="F" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>F</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter3" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Exporting tree with data</h1>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">3.1</span> Introduction</h2>
<p>The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package supports parsing
various phylogenetic tree file formats including software outputs that contain
evolutionary evidences. Some of the formats are just log file
(<em>e.g.</em> <a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</a>
and <a href="http://ginger.ucdavis.edu/r8s">r8s</a> outputs), while some of the others are
non-standard formats (<em>e.g.</em> <a href="http://beast2.org/">BEAST</a>
and <a href="http://nbisweden.github.io/MrBayes/">MrBayes</a> outputs that introduce square
bracket, which was reserved to store comment in standard Nexus format, to store
inferences). With <a href="https://bioconductor.org/packages/treeio/">treeio</a>, we are
now able to parse these files to extract phylogenetic tree and map associated
data on the tree structure. Exporting tree structure is easy, users can use
<code>as.phyo</code> method defined <a href="https://bioconductor.org/packages/treeio/">treeio</a> to
convert <code>treedata</code> object to <code>phylo</code> object then using <code>write.tree</code> or
<code>write.nexus</code> implemented
in <a href="https://cran.r-project.org/web/packages/ape/index.html">ape</a> package
<span class="citation">(Paradis, Claude, and Strimmer <a href="#ref-paradis_ape_2004" role="doc-biblioref">2004</a>)</span> to export the tree structure as Newick text or Nexus file.
This is quite useful for converting non-standard formats to standard format and
for extracting tree from software outputs, such as log file.</p>
<p>However, exporting tree with associated data is still challenging. These
associated data can be parsed from analysis programs or obtained from external
sources (<em>e.g.</em> phenotypic data, experimental data and clinical data). The major
obstacle here is that there is no standard format that designed for storing
tree with data. <a href="http://www.nexml.org/">NeXML</a> <span class="citation">(Vos et al. <a href="#ref-vos_nexml:_2012" role="doc-biblioref">2012</a>)</span> maybe the most
flexible format, however it is currently not widely supported. Most of the
analysis programs in this field rely extensively on Newick string and Nexus
format. In my opinion, although BEAST Nexus
format<a href="references.html#fn9" class="footnote-ref" id="fnref9"><sup>9</sup></a> may not be the best solution,
it is currently a good approach for storing heterogeneous associated data. The
beauty of the format is that all the annotate elements are stored within square
bracket, which is reserved for comments. So that the file can be parsed as
standard Nexus by ignoring annotate elements and existing programs should be
able to read them.</p>
</div>
<div id="exporting-tree-data-to-beast-nexus-format" class="section level2">
<h2><span class="header-section-number">3.2</span> Exporting Tree Data to <em>BEAST</em> Nexus Format</h2>
<div id="exportingconverting-software-output" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Exporting/converting software output</h3>
<p>The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package provides
<code>write.beast</code> to export <code>treedata</code> object as BEAST Nexus file <span class="citation">(Bouckaert et al. <a href="#ref-bouckaert_beast_2014" role="doc-biblioref">2014</a>)</span>.
With <a href="https://bioconductor.org/packages/treeio/">treeio</a>, it is easy to convert
software output to BEAST format if the output can be parsed
by <a href="https://bioconductor.org/packages/treeio/">treeio</a> (see <a href="chapter1.html#chapter1">Chapter 1</a>). For example, we can
convert NHX file to BEAST file and use NHX tags to color the tree using
FigTree<a href="references.html#fn10" class="footnote-ref" id="fnref10"><sup>10</sup></a> (Figure <a href="chapter3.html#fig:beastFigtree">3.1</a>A) or convert CODEML output and use
<em>d<sub>N</sub>/d<sub>S</sub></em>, <em>d<sub>N</sub></em> or <em>d<sub>S</sub></em> to color the tree in FigTree (Figure <a href="chapter3.html#fig:beastFigtree">3.1</a>B).</p>
<p>Here is an example of converting NHX file to BEAST format:</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="chapter3.html#cb98-1"></a>nhxfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/NHX&quot;</span>, <span class="st">&quot;phyldog.nhx&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb98-2"><a href="chapter3.html#cb98-2"></a>nhx &lt;-<span class="st"> </span><span class="kw">read.nhx</span>(nhxfile)</span>
<span id="cb98-3"><a href="chapter3.html#cb98-3"></a><span class="co"># write.beast(nhx, file = &quot;phyldog.tree&quot;)</span></span>
<span id="cb98-4"><a href="chapter3.html#cb98-4"></a><span class="kw">write.beast</span>(nhx)</span></code></pre></div>
<pre><code>#NEXUS
[R-package treeio, Wed Jul 10 13:17:10 2019]

BEGIN TAXA;
    DIMENSIONS NTAX = 16;
    TAXLABELS
        Prayidae_D27SS7@2825365
        Kephyes_ovata@2606431
        Chuniphyes_multidentata@1277217
        Apolemia_sp_@1353964
        Bargmannia_amoena@263997
        Bargmannia_elongata@946788
        Physonect_sp_@2066767
        Stephalia_dilata@2960089
        Frillagalma_vityazi@1155031
        Resomia_ornicephala@3111757
        Lychnagalma_utricularia@2253871
        Nanomia_bijuga@717864
        Cordagalma_sp_@1525873
        Rhizophysa_filiformis@3073669
        Hydra_magnipapillata@52244
        Ectopleura_larynx@3556167
    ;
END;
BEGIN TREES;
    TRANSLATE
        1   Prayidae_D27SS7@2825365,
        2   Kephyes_ovata@2606431,
        3   Chuniphyes_multidentata@1277217,
        4   Apolemia_sp_@1353964,
        5   Bargmannia_amoena@263997,
        6   Bargmannia_elongata@946788,
        7   Physonect_sp_@2066767,
        8   Stephalia_dilata@2960089,
        9   Frillagalma_vityazi@1155031,
        10  Resomia_ornicephala@3111757,
        11  Lychnagalma_utricularia@2253871,
        12  Nanomia_bijuga@717864,
        13  Cordagalma_sp_@1525873,
        14  Rhizophysa_filiformis@3073669,
        15  Hydra_magnipapillata@52244,
        16  Ectopleura_larynx@3556167
    ;
    TREE * UNTITLED = [&amp;R] (((1[&amp;Ev=S,S=58,ND=0]:0.0682841,(2[&amp;Ev=S,S=69,ND=1]:0.0193941,3[&amp;Ev=S,S=70,ND=2]:0.0121378)[&amp;Ev=S,S=60,ND=3]:0.0217782)[&amp;Ev=S,S=36,ND=4]:0.0607598,((4[&amp;Ev=S,S=31,ND=9]:0.11832,(((5[&amp;Ev=S,S=37,ND=10]:0.0144549,6[&amp;Ev=S,S=38,ND=11]:0.0149723)[&amp;Ev=S,S=33,ND=12]:0.0925388,7[&amp;Ev=S,S=61,ND=13]:0.077429)[&amp;Ev=S,S=24,ND=14]:0.0274637,(8[&amp;Ev=S,S=52,ND=15]:0.0761163,((9[&amp;Ev=S,S=53,ND=16]:0.0906068,10[&amp;Ev=S,S=54,ND=17]:1e-06)[&amp;Ev=S,S=45,ND=18]:1e-06,((11[&amp;Ev=S,S=65,ND=19]:0.120851,12[&amp;Ev=S,S=71,ND=20]:0.133939)[&amp;Ev=S,S=56,ND=21]:1e-06,13[&amp;Ev=S,S=64,ND=22]:0.0693814)[&amp;Ev=S,S=46,ND=23]:1e-06)[&amp;Ev=S,S=40,ND=24]:0.0333823)[&amp;Ev=S,S=35,ND=25]:1e-06)[&amp;Ev=D,S=24,ND=26]:0.0431861)[&amp;Ev=S,S=19,ND=27]:1e-06,14[&amp;Ev=S,S=26,ND=28]:0.22283)[&amp;Ev=S,S=17,ND=29]:0.0292362)[&amp;Ev=D,S=17,ND=8]:0.185603,(15[&amp;Ev=S,S=16,ND=5]:0.0621782,16[&amp;Ev=S,S=15,ND=6]:0.332505)[&amp;Ev=S,S=12,ND=7]:0.185603)[&amp;Ev=S,S=9,ND=30];
END;</code></pre>
<p>Another example of converting CodeML output to BEAST format:</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="chapter3.html#cb100-1"></a>mlcfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;mlc&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb100-2"><a href="chapter3.html#cb100-2"></a>ml &lt;-<span class="st"> </span><span class="kw">read.codeml_mlc</span>(mlcfile)</span>
<span id="cb100-3"><a href="chapter3.html#cb100-3"></a><span class="co"># write.beast(ml, file = &quot;codeml.tree&quot;)</span></span>
<span id="cb100-4"><a href="chapter3.html#cb100-4"></a><span class="kw">write.beast</span>(ml)</span></code></pre></div>
<pre><code>#NEXUS
[R-package treeio, Wed Jul 10 13:17:10 2019]

BEGIN TAXA;
    DIMENSIONS NTAX = 15;
    TAXLABELS
        A
        B
        C
        D
        E
        F
        G
        H
        I
        J
        K
        L
        M
        N
        O
    ;
END;
BEGIN TREES;
    TRANSLATE
        1   A,
        2   B,
        3   C,
        4   D,
        5   E,
        6   F,
        7   G,
        8   H,
        9   I,
        10  J,
        11  K,
        12  L,
        13  M,
        14  N,
        15  O
    ;
    TREE * UNTITLED = [&amp;U] (11[&amp;t=0.082,N=1514.9,S=633.1,dN_vs_dS=0.0224,dN=0.002,dS=0.0878,N_x_dN=3,S_x_dS=55.6]:0.081785,14[&amp;t=0.062,N=1514.9,S=633.1,dN_vs_dS=0.0095,dN=7e-04,dS=0.0689,N_x_dN=1,S_x_dS=43.6]:0.062341,(4[&amp;t=0.082,N=1514.9,S=633.1,dN_vs_dS=0.0385,dN=0.0033,dS=0.0849,N_x_dN=5,S_x_dS=53.8]:0.082021,(12[&amp;t=0.006,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=0.0062,N_x_dN=0,S_x_dS=3.9]:0.005508,(10[&amp;t=0.014,N=1514.9,S=633.1,dN_vs_dS=0.0457,dN=7e-04,dS=0.0143,N_x_dN=1,S_x_dS=9]:0.013996,(7[&amp;t=0.046,N=1514.9,S=633.1,dN_vs_dS=0.1621,dN=0.006,dS=0.0373,N_x_dN=9.2,S_x_dS=23.6]:0.045746,((3[&amp;t=0.028,N=1514.9,S=633.1,dN_vs_dS=0.0461,dN=0.0013,dS=0.0282,N_x_dN=2,S_x_dS=17.9]:0.02773,(5[&amp;t=0.031,N=1514.9,S=633.1,dN_vs_dS=0.0641,dN=0.002,dS=0.0305,N_x_dN=3,S_x_dS=19.3]:0.031104,15[&amp;t=0.048,N=1514.9,S=633.1,dN_vs_dS=0.0538,dN=0.0026,dS=0.0485,N_x_dN=4,S_x_dS=30.7]:0.048389)23[&amp;t=0.008,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=0.0094,N_x_dN=0,S_x_dS=6]:0.008328)22[&amp;t=0.016,N=1514.9,S=633.1,dN_vs_dS=0.0395,dN=7e-04,dS=0.0165,N_x_dN=1,S_x_dS=10.4]:0.015959,(8[&amp;t=0.021,N=1514.9,S=633.1,dN_vs_dS=0.1028,dN=0.002,dS=0.0191,N_x_dN=3,S_x_dS=12.1]:0.021007,(9[&amp;t=0.015,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=0.0167,N_x_dN=0,S_x_dS=10.6]:0.014739,(2[&amp;t=0.032,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=0.0358,N_x_dN=0,S_x_dS=22.7]:0.031643,(1[&amp;t=0.01,N=1514.9,S=633.1,dN_vs_dS=0.0646,dN=7e-04,dS=0.0101,N_x_dN=1,S_x_dS=6.4]:0.01034,(6[&amp;t=0.007,N=1514.9,S=633.1,dN_vs_dS=0.298,dN=0.0013,dS=0.0044,N_x_dN=2,S_x_dS=2.8]:0.006649,13[&amp;t=0.009,N=1514.9,S=633.1,dN_vs_dS=0.0738,dN=7e-04,dS=0.0088,N_x_dN=1,S_x_dS=5.6]:0.009195)28[&amp;t=0.028,N=1514.9,S=633.1,dN_vs_dS=0.0453,dN=0.0013,dS=0.0289,N_x_dN=2,S_x_dS=18.3]:0.028303)27[&amp;t=0.008,N=1514.9,S=633.1,dN_vs_dS=0.0863,dN=7e-04,dS=0.0076,N_x_dN=1,S_x_dS=4.8]:0.008072)26[&amp;t=0.003,N=1514.9,S=633.1,dN_vs_dS=1.5591,dN=0.0013,dS=8e-04,N_x_dN=2,S_x_dS=0.5]:0.0035)25[&amp;t=0.02,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=0.023,N_x_dN=0,S_x_dS=14.6]:0.020359)24[&amp;t=0.001,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=6e-04,N_x_dN=0,S_x_dS=0.4]:0.000555)21[&amp;t=0.024,N=1514.9,S=633.1,dN_vs_dS=0.0549,dN=0.0013,dS=0.0237,N_x_dN=2,S_x_dS=15]:0.023675)20[&amp;t=0.046,N=1514.9,S=633.1,dN_vs_dS=0.0419,dN=0.002,dS=0.047,N_x_dN=3,S_x_dS=29.8]:0.045745)19[&amp;t=0.015,N=1514.9,S=633.1,dN_vs_dS=1e-04,dN=0,dS=0.0166,N_x_dN=0,S_x_dS=10.5]:0.014684)18[&amp;t=0.059,N=1514.9,S=633.1,dN_vs_dS=0.0964,dN=0.0053,dS=0.0545,N_x_dN=8,S_x_dS=34.5]:0.059308)17[&amp;t=0.232,N=1514.9,S=633.1,dN_vs_dS=0.0129,dN=0.0033,dS=0.2541,N_x_dN=5,S_x_dS=160.9]:0.231628)16;
END;</code></pre>


<div class="figure"><span id="fig:beastFigtree"></span>
<img src="treedata_files/figure-html/beastFigtree-1.png" alt="Visualizing BEAST file in FigTree. Directly visualizing NHX file (A) and CodeML output (B) in FigTree is not supported. treeio can convert these files to BEAST compatible NEXUS format which can be directly opened in FigTree and visualized annotated data." width="768" />
<p class="caption">
Figure 3.1: <strong>Visualizing BEAST file in FigTree.</strong> Directly visualizing <code>NHX</code> file (A) and <code>CodeML</code> output (B) in <code>FigTree</code> is not supported. <code>treeio</code> can convert these files to BEAST compatible NEXUS format which can be directly opened in <code>FigTree</code> and visualized annotated data.
</p>
</div>
</div>
<div id="combining-tree-with-external-data" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Combining tree with external data</h3>
<p>Using the utilities provided
by <a href="https://CRAN.R-project.org/package=tidytree">tidytree</a> and <a href="https://bioconductor.org/packages/treeio/">treeio</a>, it is easy to link
external data onto the corresponding phylogeny. The <code>write.beast</code> function enable users
to export the tree with external data to a single tree file.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="chapter3.html#cb102-1"></a>phylo &lt;-<span class="st"> </span><span class="kw">as.phylo</span>(nhx)</span>
<span id="cb102-2"><a href="chapter3.html#cb102-2"></a><span class="co">## print the newick text</span></span>
<span id="cb102-3"><a href="chapter3.html#cb102-3"></a><span class="kw">write.tree</span>(phylo)</span></code></pre></div>
<pre><code>[1] &quot;(((Prayidae_D27SS7@2825365:0.0682841,(Kephyes_ovata@2606431:0.0193941,Chuniphyes_multidentata@1277217:0.0121378):0.0217782):0.0607598,((Apolemia_sp_@1353964:0.11832,(((Bargmannia_amoena@263997:0.0144549,Bargmannia_elongata@946788:0.0149723):0.0925388,Physonect_sp_@2066767:0.077429):0.0274637,(Stephalia_dilata@2960089:0.0761163,((Frillagalma_vityazi@1155031:0.0906068,Resomia_ornicephala@3111757:1e-06):1e-06,((Lychnagalma_utricularia@2253871:0.120851,Nanomia_bijuga@717864:0.133939):1e-06,Cordagalma_sp_@1525873:0.0693814):1e-06):0.0333823):1e-06):0.0431861):1e-06,Rhizophysa_filiformis@3073669:0.22283):0.0292362):0.185603,(Hydra_magnipapillata@52244:0.0621782,Ectopleura_larynx@3556167:0.332505):0.185603);&quot;</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="chapter3.html#cb104-1"></a>N &lt;-<span class="st"> </span><span class="kw">Nnode2</span>(phylo)</span>
<span id="cb104-2"><a href="chapter3.html#cb104-2"></a>fake_data &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">node =</span> <span class="dv">1</span><span class="op">:</span>N, <span class="dt">fake_trait =</span> <span class="kw">rnorm</span>(N), <span class="dt">another_trait =</span> <span class="kw">runif</span>(N))</span>
<span id="cb104-3"><a href="chapter3.html#cb104-3"></a>fake_tree &lt;-<span class="st"> </span><span class="kw">full_join</span>(phylo, fake_data, <span class="dt">by =</span> <span class="st">&quot;node&quot;</span>)</span>
<span id="cb104-4"><a href="chapter3.html#cb104-4"></a><span class="kw">write.beast</span>(fake_tree)</span></code></pre></div>
<pre><code>#NEXUS
[R-package treeio, Wed Jul 10 13:17:12 2019]

BEGIN TAXA;
    DIMENSIONS NTAX = 16;
    TAXLABELS
        Prayidae_D27SS7@2825365
        Kephyes_ovata@2606431
        Chuniphyes_multidentata@1277217
        Apolemia_sp_@1353964
        Bargmannia_amoena@263997
        Bargmannia_elongata@946788
        Physonect_sp_@2066767
        Stephalia_dilata@2960089
        Frillagalma_vityazi@1155031
        Resomia_ornicephala@3111757
        Lychnagalma_utricularia@2253871
        Nanomia_bijuga@717864
        Cordagalma_sp_@1525873
        Rhizophysa_filiformis@3073669
        Hydra_magnipapillata@52244
        Ectopleura_larynx@3556167
    ;
END;
BEGIN TREES;
    TRANSLATE
        1   Prayidae_D27SS7@2825365,
        2   Kephyes_ovata@2606431,
        3   Chuniphyes_multidentata@1277217,
        4   Apolemia_sp_@1353964,
        5   Bargmannia_amoena@263997,
        6   Bargmannia_elongata@946788,
        7   Physonect_sp_@2066767,
        8   Stephalia_dilata@2960089,
        9   Frillagalma_vityazi@1155031,
        10  Resomia_ornicephala@3111757,
        11  Lychnagalma_utricularia@2253871,
        12  Nanomia_bijuga@717864,
        13  Cordagalma_sp_@1525873,
        14  Rhizophysa_filiformis@3073669,
        15  Hydra_magnipapillata@52244,
        16  Ectopleura_larynx@3556167
    ;
    TREE * UNTITLED = [&amp;R] (((1[&amp;fake_trait=-0.118140158479532,another_trait=0.358319966588169]:0.0682841,(2[&amp;fake_trait=-1.02308336346315,another_trait=0.178163790609688]:0.0193941,3[&amp;fake_trait=0.103599672229184,another_trait=0.93473564600572]:0.0121378)[&amp;fake_trait=-0.59581303574252,another_trait=0.922105083474889]:0.0217782)[&amp;fake_trait=-0.300012313113108,another_trait=0.119739749003202]:0.0607598,(14[&amp;fake_trait=0.589053316158453,another_trait=0.565819646697491]:0.22283,(4[&amp;fake_trait=-0.765460180119788,another_trait=0.888429019600153]:0.11832,((7[&amp;fake_trait=-0.306908277375037,another_trait=0.650574740953743]:0.077429,(5[&amp;fake_trait=-0.357587376956114,another_trait=0.547185462666675]:0.0144549,6[&amp;fake_trait=0.582576884787801,another_trait=0.00538489152677357]:0.0149723)[&amp;fake_trait=1.68690137848482,another_trait=0.634304485516623]:0.0925388)[&amp;fake_trait=1.00444824814337,another_trait=0.420116872293875]:0.0274637,(8[&amp;fake_trait=1.10912385375853,another_trait=0.583255837904289]:0.0761163,((9[&amp;fake_trait=-0.707380205642954,another_trait=0.808900895994157]:0.0906068,10[&amp;fake_trait=0.838100693609065,another_trait=0.0142157829832286]:1e-06)[&amp;fake_trait=-0.509864481867074,another_trait=0.939013464609161]:1e-06,(13[&amp;fake_trait=1.56583127248603,another_trait=0.0454649559687823]:0.0693814,(11[&amp;fake_trait=1.45882763141046,another_trait=0.516774294432253]:0.120851,12[&amp;fake_trait=2.43471798636817,another_trait=0.00467536062933505]:0.133939)[&amp;fake_trait=-1.07656365665571,another_trait=0.518969761673361]:1e-06)[&amp;fake_trait=-1.48426105861817,another_trait=0.044428545050323]:1e-06)[&amp;fake_trait=0.409198668366587,another_trait=0.538205695804209]:0.0333823)[&amp;fake_trait=-0.692527146833782,another_trait=0.58515489147976]:1e-06)[&amp;fake_trait=-1.41216094329287,another_trait=0.935482121771201]:0.0431861)[&amp;fake_trait=1.65980403458888,another_trait=0.125082567567006]:1e-06)[&amp;fake_trait=0.476325081878056,another_trait=0.262554442277178]:0.0292362)[&amp;fake_trait=1.15832764983724,another_trait=0.646949481917545]:0.185603,(15[&amp;fake_trait=0.704447323258862,another_trait=0.663062628591433]:0.0621782,16[&amp;fake_trait=-0.659190065346089,another_trait=0.505218442762271]:0.332505)[&amp;fake_trait=-1.04282102019802,another_trait=0.641324875177816]:0.185603)[&amp;fake_trait=-0.479609047683268,another_trait=0.519890103489161];
END;</code></pre>
<p>After merging, the <code>fake_trait</code> and <code>another_trait</code> stored in <code>fake_data</code> will be linked to the tree, <code>phylo</code>, and store in the <code>treedata</code> object, the <code>fake_tree</code>. The <code>write.beast()</code> function export the tree with associated data to a single BEAST format file. The associated data can be used to visualized the tree using <a href="http://bioconductor.org/packages/ggtree">ggtree</a> (Figure <a href="chapter5.html#fig:beast">5.2</a>) or <code>FigTree</code> (Figure <a href="chapter3.html#fig:beastFigtree">3.1</a>).</p>
</div>
<div id="merging-tree-data-from-different-sources" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Merging tree data from different sources</h3>
<p>Not only Newick tree text can be combined with associated data, but also tree
data obtained from software output can be combined with external data, as well
as different tree objects can be merged together (for details, see <a href="chapter2.html#chapter2">Chapter 2</a>).</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="chapter3.html#cb106-1"></a><span class="co">## combine tree object with data</span></span>
<span id="cb106-2"><a href="chapter3.html#cb106-2"></a>tree_with_data &lt;-<span class="st"> </span><span class="kw">full_join</span>(nhx, fake_data, <span class="dt">by =</span> <span class="st">&quot;node&quot;</span>)</span>
<span id="cb106-3"><a href="chapter3.html#cb106-3"></a>tree_with_data</span></code></pre></div>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;/home/ygc/R/library/treeio/extdata/NHX/phyldog.nhx&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 16 tips and 15 internal nodes.
## 
## Tip labels:
##  Prayidae_D27SS7@2825365, Kephyes_ovata@2606431, Chuniphyes_multidentata@1277217, Apolemia_sp_@1353964, Bargmannia_amoena@263997, Bargmannia_elongata@946788, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;Ev&#39;,   &#39;S&#39;,    &#39;ND&#39;,   &#39;fake_trait&#39;,   &#39;another_trait&#39;.</code></pre>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="chapter3.html#cb108-1"></a><span class="co">## merge two tree object</span></span>
<span id="cb108-2"><a href="chapter3.html#cb108-2"></a>tree2 &lt;-<span class="st"> </span><span class="kw">merge_tree</span>(nhx, fake_tree)</span>
<span id="cb108-3"><a href="chapter3.html#cb108-3"></a>tree2</span></code></pre></div>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;/home/ygc/R/library/treeio/extdata/NHX/phyldog.nhx&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 16 tips and 15 internal nodes.
## 
## Tip labels:
##  Prayidae_D27SS7@2825365, Kephyes_ovata@2606431, Chuniphyes_multidentata@1277217, Apolemia_sp_@1353964, Bargmannia_amoena@263997, Bargmannia_elongata@946788, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;Ev&#39;,   &#39;S&#39;,    &#39;ND&#39;,   &#39;fake_trait&#39;,   &#39;another_trait&#39;.</code></pre>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="chapter3.html#cb110-1"></a><span class="kw">identical</span>(tree_with_data, tree2)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>After merging data from different sources, the tree with the associated data can
be exported into a single file.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="chapter3.html#cb112-1"></a><span class="kw">write.beast</span>(tree2)</span></code></pre></div>
<pre><code>#NEXUS
[R-package treeio, Wed Jul 10 13:17:13 2019]

BEGIN TAXA;
    DIMENSIONS NTAX = 16;
    TAXLABELS
        Prayidae_D27SS7@2825365
        Kephyes_ovata@2606431
        Chuniphyes_multidentata@1277217
        Apolemia_sp_@1353964
        Bargmannia_amoena@263997
        Bargmannia_elongata@946788
        Physonect_sp_@2066767
        Stephalia_dilata@2960089
        Frillagalma_vityazi@1155031
        Resomia_ornicephala@3111757
        Lychnagalma_utricularia@2253871
        Nanomia_bijuga@717864
        Cordagalma_sp_@1525873
        Rhizophysa_filiformis@3073669
        Hydra_magnipapillata@52244
        Ectopleura_larynx@3556167
    ;
END;
BEGIN TREES;
    TRANSLATE
        1   Prayidae_D27SS7@2825365,
        2   Kephyes_ovata@2606431,
        3   Chuniphyes_multidentata@1277217,
        4   Apolemia_sp_@1353964,
        5   Bargmannia_amoena@263997,
        6   Bargmannia_elongata@946788,
        7   Physonect_sp_@2066767,
        8   Stephalia_dilata@2960089,
        9   Frillagalma_vityazi@1155031,
        10  Resomia_ornicephala@3111757,
        11  Lychnagalma_utricularia@2253871,
        12  Nanomia_bijuga@717864,
        13  Cordagalma_sp_@1525873,
        14  Rhizophysa_filiformis@3073669,
        15  Hydra_magnipapillata@52244,
        16  Ectopleura_larynx@3556167
    ;
    TREE * UNTITLED = [&amp;R] (((1[&amp;Ev=S,S=58,ND=0,fake_trait=-0.118140158479532,another_trait=0.358319966588169]:0.0682841,(2[&amp;Ev=S,S=69,ND=1,fake_trait=-1.02308336346315,another_trait=0.178163790609688]:0.0193941,3[&amp;Ev=S,S=70,ND=2,fake_trait=0.103599672229184,another_trait=0.93473564600572]:0.0121378)[&amp;Ev=S,S=60,ND=3,fake_trait=-0.59581303574252,another_trait=0.922105083474889]:0.0217782)[&amp;Ev=S,S=36,ND=4,fake_trait=-0.300012313113108,another_trait=0.119739749003202]:0.0607598,((4[&amp;Ev=S,S=31,ND=9,fake_trait=-0.765460180119788,another_trait=0.888429019600153]:0.11832,(((5[&amp;Ev=S,S=37,ND=10,fake_trait=-0.357587376956114,another_trait=0.547185462666675]:0.0144549,6[&amp;Ev=S,S=38,ND=11,fake_trait=0.582576884787801,another_trait=0.00538489152677357]:0.0149723)[&amp;Ev=S,S=33,ND=12,fake_trait=1.68690137848482,another_trait=0.634304485516623]:0.0925388,7[&amp;Ev=S,S=61,ND=13,fake_trait=-0.306908277375037,another_trait=0.650574740953743]:0.077429)[&amp;Ev=S,S=24,ND=14,fake_trait=1.00444824814337,another_trait=0.420116872293875]:0.0274637,(8[&amp;Ev=S,S=52,ND=15,fake_trait=1.10912385375853,another_trait=0.583255837904289]:0.0761163,((9[&amp;Ev=S,S=53,ND=16,fake_trait=-0.707380205642954,another_trait=0.808900895994157]:0.0906068,10[&amp;Ev=S,S=54,ND=17,fake_trait=0.838100693609065,another_trait=0.0142157829832286]:1e-06)[&amp;Ev=S,S=45,ND=18,fake_trait=-0.509864481867074,another_trait=0.939013464609161]:1e-06,((11[&amp;Ev=S,S=65,ND=19,fake_trait=1.45882763141046,another_trait=0.516774294432253]:0.120851,12[&amp;Ev=S,S=71,ND=20,fake_trait=2.43471798636817,another_trait=0.00467536062933505]:0.133939)[&amp;Ev=S,S=56,ND=21,fake_trait=-1.07656365665571,another_trait=0.518969761673361]:1e-06,13[&amp;Ev=S,S=64,ND=22,fake_trait=1.56583127248603,another_trait=0.0454649559687823]:0.0693814)[&amp;Ev=S,S=46,ND=23,fake_trait=-1.48426105861817,another_trait=0.044428545050323]:1e-06)[&amp;Ev=S,S=40,ND=24,fake_trait=0.409198668366587,another_trait=0.538205695804209]:0.0333823)[&amp;Ev=S,S=35,ND=25,fake_trait=-0.692527146833782,another_trait=0.58515489147976]:1e-06)[&amp;Ev=D,S=24,ND=26,fake_trait=-1.41216094329287,another_trait=0.935482121771201]:0.0431861)[&amp;Ev=S,S=19,ND=27,fake_trait=1.65980403458888,another_trait=0.125082567567006]:1e-06,14[&amp;Ev=S,S=26,ND=28,fake_trait=0.589053316158453,another_trait=0.565819646697491]:0.22283)[&amp;Ev=S,S=17,ND=29,fake_trait=0.476325081878056,another_trait=0.262554442277178]:0.0292362)[&amp;Ev=D,S=17,ND=8,fake_trait=1.15832764983724,another_trait=0.646949481917545]:0.185603,(15[&amp;Ev=S,S=16,ND=5,fake_trait=0.704447323258862,another_trait=0.663062628591433]:0.0621782,16[&amp;Ev=S,S=15,ND=6,fake_trait=-0.659190065346089,another_trait=0.505218442762271]:0.332505)[&amp;Ev=S,S=12,ND=7,fake_trait=-1.04282102019802,another_trait=0.641324875177816]:0.185603)[&amp;Ev=S,S=9,ND=30,fake_trait=-0.479609047683268,another_trait=0.519890103489161];
END;</code></pre>
<p>The output BEAST Nexus file can be imported into R using the <code>read.beast</code>
function and all the associated data can be used to annotate the tree
using <a href="https://bioconductor.org/packages/ggtree/">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017" role="doc-biblioref">2017</a>)</span>.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="chapter3.html#cb114-1"></a>outfile &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="dt">fileext =</span> <span class="st">&quot;.tree&quot;</span>)</span>
<span id="cb114-2"><a href="chapter3.html#cb114-2"></a><span class="kw">write.beast</span>(tree2, <span class="dt">file =</span> outfile)</span>
<span id="cb114-3"><a href="chapter3.html#cb114-3"></a><span class="kw">read.beast</span>(outfile)</span></code></pre></div>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;/tmp/Rtmp9l8tHn/file303738b38828.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 16 tips and 15 internal nodes.
## 
## Tip labels:
##  Prayidae_D27SS7@2825365, Kephyes_ovata@2606431, Chuniphyes_multidentata@1277217, Apolemia_sp_@1353964, Bargmannia_amoena@263997, Bargmannia_elongata@946788, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;another_trait&#39;,    &#39;Ev&#39;,   &#39;fake_trait&#39;,   &#39;ND&#39;,   &#39;S&#39;.</code></pre>
</div>
</div>
<div id="write-jtree" class="section level2">
<h2><span class="header-section-number">3.3</span> Exporting Tree Data to <em>jtree</em> Format</h2>
<p>The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package provides
<code>write.beast</code> to export <code>treedata</code> to BEAST Nexus file. This is quite useful
to convert file format, combine tree with data and merging tree data from
different sources as we demonstrated in
<a href="chapter3.html#exporting-tree-data-to-beast-nexus-format">session 3.2</a>.
The <a href="https://bioconductor.org/packages/treeio/">treeio</a> package also supplies
<code>read.beast</code> function to parse output file of <code>write.beast</code>. Although
with <a href="https://bioconductor.org/packages/treeio/">treeio</a>, the R community has the ability to
manipulate BEAST Nexus format and process tree data, there is still lacking
library/package for parsing BEAST file in other programming language.</p>
<p><a href="https://www.json.org/">JSON</a> (JavaScript Object Notation) is a lightweight data-interchange format and
widely supported in almost all modern programming languages. To make it easy
to import tree with data in other programming
languages, <a href="https://bioconductor.org/packages/treeio/">treeio</a> supports
exporting tree with data in jtree format, which is JSON-based and can be easy to parse using any
languages that supports JSON.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="chapter3.html#cb116-1"></a><span class="kw">write.jtree</span>(tree2)</span></code></pre></div>
<pre><code>{
    &quot;tree&quot;: &quot;(((Prayidae_D27SS7@2825365:0.0682841{1},(Kephyes_ovata@2606431:0.0193941{2},Chuniphyes_multidentata@1277217:0.0121378{3}):0.0217782{20}):0.0607598{19},((Apolemia_sp_@1353964:0.11832{4},(((Bargmannia_amoena@263997:0.0144549{5},Bargmannia_elongata@946788:0.0149723{6}):0.0925388{25},Physonect_sp_@2066767:0.077429{7}):0.0274637{24},(Stephalia_dilata@2960089:0.0761163{8},((Frillagalma_vityazi@1155031:0.0906068{9},Resomia_ornicephala@3111757:1{10}e-06):1{28}e-06,((Lychnagalma_utricularia@2253871:0.120851{11},Nanomia_bijuga@717864:0.133939{12}):1{30}e-06,Cordagalma_sp_@1525873:0.0693814{13}):1{29}e-06):0.0333823{27}):1{26}e-06):0.0431861{23}):1{22}e-06,Rhizophysa_filiformis@3073669:0.22283{14}):0.0292362{21}):0.185603{18},(Hydra_magnipapillata@52244:0.0621782{15},Ectopleura_larynx@3556167:0.332505{16}):0.185603{31}){17};&quot;,
    &quot;data&quot;:[
  {
    &quot;edge_num&quot;: 1,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;58&quot;,
    &quot;ND&quot;: 0,
    &quot;fake_trait&quot;: -0.1181,
    &quot;another_trait&quot;: 0.3583
  },
  {
    &quot;edge_num&quot;: 2,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;69&quot;,
    &quot;ND&quot;: 1,
    &quot;fake_trait&quot;: -1.0231,
    &quot;another_trait&quot;: 0.1782
  },
  {
    &quot;edge_num&quot;: 3,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;70&quot;,
    &quot;ND&quot;: 2,
    &quot;fake_trait&quot;: 0.1036,
    &quot;another_trait&quot;: 0.9347
  },
  {
    &quot;edge_num&quot;: 4,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;31&quot;,
    &quot;ND&quot;: 9,
    &quot;fake_trait&quot;: -0.7655,
    &quot;another_trait&quot;: 0.8884
  },
  {
    &quot;edge_num&quot;: 5,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;37&quot;,
    &quot;ND&quot;: 10,
    &quot;fake_trait&quot;: -0.3576,
    &quot;another_trait&quot;: 0.5472
  },
  {
    &quot;edge_num&quot;: 6,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;38&quot;,
    &quot;ND&quot;: 11,
    &quot;fake_trait&quot;: 0.5826,
    &quot;another_trait&quot;: 0.0054
  },
  {
    &quot;edge_num&quot;: 7,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;61&quot;,
    &quot;ND&quot;: 13,
    &quot;fake_trait&quot;: -0.3069,
    &quot;another_trait&quot;: 0.6506
  },
  {
    &quot;edge_num&quot;: 8,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;52&quot;,
    &quot;ND&quot;: 15,
    &quot;fake_trait&quot;: 1.1091,
    &quot;another_trait&quot;: 0.5833
  },
  {
    &quot;edge_num&quot;: 9,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;53&quot;,
    &quot;ND&quot;: 16,
    &quot;fake_trait&quot;: -0.7074,
    &quot;another_trait&quot;: 0.8089
  },
  {
    &quot;edge_num&quot;: 10,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;54&quot;,
    &quot;ND&quot;: 17,
    &quot;fake_trait&quot;: 0.8381,
    &quot;another_trait&quot;: 0.0142
  },
  {
    &quot;edge_num&quot;: 11,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;65&quot;,
    &quot;ND&quot;: 19,
    &quot;fake_trait&quot;: 1.4588,
    &quot;another_trait&quot;: 0.5168
  },
  {
    &quot;edge_num&quot;: 12,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;71&quot;,
    &quot;ND&quot;: 20,
    &quot;fake_trait&quot;: 2.4347,
    &quot;another_trait&quot;: 0.0047
  },
  {
    &quot;edge_num&quot;: 13,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;64&quot;,
    &quot;ND&quot;: 22,
    &quot;fake_trait&quot;: 1.5658,
    &quot;another_trait&quot;: 0.0455
  },
  {
    &quot;edge_num&quot;: 14,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;26&quot;,
    &quot;ND&quot;: 28,
    &quot;fake_trait&quot;: 0.5891,
    &quot;another_trait&quot;: 0.5658
  },
  {
    &quot;edge_num&quot;: 15,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;16&quot;,
    &quot;ND&quot;: 5,
    &quot;fake_trait&quot;: 0.7044,
    &quot;another_trait&quot;: 0.6631
  },
  {
    &quot;edge_num&quot;: 16,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;15&quot;,
    &quot;ND&quot;: 6,
    &quot;fake_trait&quot;: -0.6592,
    &quot;another_trait&quot;: 0.5052
  },
  {
    &quot;edge_num&quot;: 17,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;9&quot;,
    &quot;ND&quot;: 30,
    &quot;fake_trait&quot;: -0.4796,
    &quot;another_trait&quot;: 0.5199
  },
  {
    &quot;edge_num&quot;: 18,
    &quot;Ev&quot;: &quot;D&quot;,
    &quot;S&quot;: &quot;17&quot;,
    &quot;ND&quot;: 8,
    &quot;fake_trait&quot;: 1.1583,
    &quot;another_trait&quot;: 0.6469
  },
  {
    &quot;edge_num&quot;: 19,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;36&quot;,
    &quot;ND&quot;: 4,
    &quot;fake_trait&quot;: -0.3,
    &quot;another_trait&quot;: 0.1197
  },
  {
    &quot;edge_num&quot;: 20,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;60&quot;,
    &quot;ND&quot;: 3,
    &quot;fake_trait&quot;: -0.5958,
    &quot;another_trait&quot;: 0.9221
  },
  {
    &quot;edge_num&quot;: 21,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;17&quot;,
    &quot;ND&quot;: 29,
    &quot;fake_trait&quot;: 0.4763,
    &quot;another_trait&quot;: 0.2626
  },
  {
    &quot;edge_num&quot;: 22,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;19&quot;,
    &quot;ND&quot;: 27,
    &quot;fake_trait&quot;: 1.6598,
    &quot;another_trait&quot;: 0.1251
  },
  {
    &quot;edge_num&quot;: 23,
    &quot;Ev&quot;: &quot;D&quot;,
    &quot;S&quot;: &quot;24&quot;,
    &quot;ND&quot;: 26,
    &quot;fake_trait&quot;: -1.4122,
    &quot;another_trait&quot;: 0.9355
  },
  {
    &quot;edge_num&quot;: 24,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;24&quot;,
    &quot;ND&quot;: 14,
    &quot;fake_trait&quot;: 1.0044,
    &quot;another_trait&quot;: 0.4201
  },
  {
    &quot;edge_num&quot;: 25,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;33&quot;,
    &quot;ND&quot;: 12,
    &quot;fake_trait&quot;: 1.6869,
    &quot;another_trait&quot;: 0.6343
  },
  {
    &quot;edge_num&quot;: 26,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;35&quot;,
    &quot;ND&quot;: 25,
    &quot;fake_trait&quot;: -0.6925,
    &quot;another_trait&quot;: 0.5852
  },
  {
    &quot;edge_num&quot;: 27,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;40&quot;,
    &quot;ND&quot;: 24,
    &quot;fake_trait&quot;: 0.4092,
    &quot;another_trait&quot;: 0.5382
  },
  {
    &quot;edge_num&quot;: 28,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;45&quot;,
    &quot;ND&quot;: 18,
    &quot;fake_trait&quot;: -0.5099,
    &quot;another_trait&quot;: 0.939
  },
  {
    &quot;edge_num&quot;: 29,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;46&quot;,
    &quot;ND&quot;: 23,
    &quot;fake_trait&quot;: -1.4843,
    &quot;another_trait&quot;: 0.0444
  },
  {
    &quot;edge_num&quot;: 30,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;56&quot;,
    &quot;ND&quot;: 21,
    &quot;fake_trait&quot;: -1.0766,
    &quot;another_trait&quot;: 0.519
  },
  {
    &quot;edge_num&quot;: 31,
    &quot;Ev&quot;: &quot;S&quot;,
    &quot;S&quot;: &quot;12&quot;,
    &quot;ND&quot;: 7,
    &quot;fake_trait&quot;: -1.0428,
    &quot;another_trait&quot;: 0.6413
  }
],
    &quot;metadata&quot;: {&quot;info&quot;: &quot;R-package treeio&quot;, &quot;data&quot;: &quot;Wed Jul 10 13:17:13 2019&quot;}
}</code></pre>
<p>The <em>jtree</em> format is based on JSON and can be parsed using JSON parser.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="chapter3.html#cb118-1"></a>jtree_file &lt;-<span class="st"> </span><span class="kw">tempfile</span>(<span class="dt">fileext =</span> <span class="st">&#39;.jtree&#39;</span>)</span>
<span id="cb118-2"><a href="chapter3.html#cb118-2"></a><span class="kw">write.jtree</span>(tree2, <span class="dt">file =</span> jtree_file)</span>
<span id="cb118-3"><a href="chapter3.html#cb118-3"></a>jsonlite<span class="op">::</span><span class="kw">fromJSON</span>(jtree_file)</span></code></pre></div>
<pre><code>$tree
[1] &quot;(((Prayidae_D27SS7@2825365:0.0682841{1},(Kephyes_ovata@2606431:0.0193941{2},Chuniphyes_multidentata@1277217:0.0121378{3}):0.0217782{20}):0.0607598{19},((Apolemia_sp_@1353964:0.11832{4},(((Bargmannia_amoena@263997:0.0144549{5},Bargmannia_elongata@946788:0.0149723{6}):0.0925388{25},Physonect_sp_@2066767:0.077429{7}):0.0274637{24},(Stephalia_dilata@2960089:0.0761163{8},((Frillagalma_vityazi@1155031:0.0906068{9},Resomia_ornicephala@3111757:1{10}e-06):1{28}e-06,((Lychnagalma_utricularia@2253871:0.120851{11},Nanomia_bijuga@717864:0.133939{12}):1{30}e-06,Cordagalma_sp_@1525873:0.0693814{13}):1{29}e-06):0.0333823{27}):1{26}e-06):0.0431861{23}):1{22}e-06,Rhizophysa_filiformis@3073669:0.22283{14}):0.0292362{21}):0.185603{18},(Hydra_magnipapillata@52244:0.0621782{15},Ectopleura_larynx@3556167:0.332505{16}):0.185603{31}){17};&quot;

$data
   edge_num Ev  S ND fake_trait another_trait
1         1  S 58  0    -0.1181        0.3583
2         2  S 69  1    -1.0231        0.1782
3         3  S 70  2     0.1036        0.9347
4         4  S 31  9    -0.7655        0.8884
5         5  S 37 10    -0.3576        0.5472
6         6  S 38 11     0.5826        0.0054
7         7  S 61 13    -0.3069        0.6506
8         8  S 52 15     1.1091        0.5833
9         9  S 53 16    -0.7074        0.8089
10       10  S 54 17     0.8381        0.0142
11       11  S 65 19     1.4588        0.5168
12       12  S 71 20     2.4347        0.0047
13       13  S 64 22     1.5658        0.0455
14       14  S 26 28     0.5891        0.5658
15       15  S 16  5     0.7044        0.6631
16       16  S 15  6    -0.6592        0.5052
17       17  S  9 30    -0.4796        0.5199
18       18  D 17  8     1.1583        0.6469
19       19  S 36  4    -0.3000        0.1197
20       20  S 60  3    -0.5958        0.9221
21       21  S 17 29     0.4763        0.2626
22       22  S 19 27     1.6598        0.1251
23       23  D 24 26    -1.4122        0.9355
24       24  S 24 14     1.0044        0.4201
25       25  S 33 12     1.6869        0.6343
26       26  S 35 25    -0.6925        0.5852
27       27  S 40 24     0.4092        0.5382
28       28  S 45 18    -0.5099        0.9390
29       29  S 46 23    -1.4843        0.0444
30       30  S 56 21    -1.0766        0.5190
31       31  S 12  7    -1.0428        0.6413

$metadata
$metadata$info
[1] &quot;R-package treeio&quot;

$metadata$data
[1] &quot;Wed Jul 10 13:17:13 2019&quot;</code></pre>
<p>The <em>jtree</em> file can be directly imported as a <code>treedata</code> object using
<code>read.jtree</code> provided also
in <a href="https://bioconductor.org/packages/treeio/">treeio</a> package (see also <a href="chapter1.html#jtree">session 1.3</a>).</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="chapter3.html#cb120-1"></a><span class="kw">read.jtree</span>(jtree_file)</span></code></pre></div>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;/tmp/Rtmp9l8tHn/file30371139846d.jtree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 16 tips and 15 internal nodes.
## 
## Tip labels:
##  Prayidae_D27SS7@2825365, Kephyes_ovata@2606431, Chuniphyes_multidentata@1277217, Apolemia_sp_@1353964, Bargmannia_amoena@263997, Bargmannia_elongata@946788, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;Ev&#39;,   &#39;S&#39;,    &#39;ND&#39;,   &#39;fake_trait&#39;,   &#39;another_trait&#39;.</code></pre>
</div>
<div id="summary-2" class="section level2">
<h2><span class="header-section-number">3.4</span> Summary</h2>
<p>Phylogenetic tree associated data is often stored in a separate file and need expertise to map the data to the tree structure. Lacking standardization to store and represent phylogeny and associated data makes it difficult for researchers to access and integrate the phylogenetic data into their studies. The <a href="http://bioconductor.org/packages/treeio">treeio</a> package provides functions to import phylogeny with associated data from a number of sources, including analysis finding from commonly used software and external data such as experimental, clinical or meta data. These tree + data can be exported into a single file as a BEAST or jtree formats, and the output file can be parsed back to R by <a href="http://bioconductor.org/packages/treeio">treeio</a> and the data is easy to access. The input and output utilities supplied by <a href="http://bioconductor.org/packages/treeio">treeio</a> package lay the foundation for phylogenetic data integration for downstream comparative study and visualization.</p>

</div>
</div>



<h3>References</h3>
<div id="refs" class="references">
<div id="ref-bouckaert_beast_2014">
<p>Bouckaert, Remco, Joseph Heled, Denise Kühnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. “BEAST 2: A Software Platform for Bayesian Evolutionary Analysis.” <em>PLoS Comput Biol</em> 10 (4): e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>.</p>
</div>
<div id="ref-paradis_ape_2004">
<p>Paradis, Emmanuel, Julien Claude, and Korbinian Strimmer. 2004. “APE: Analyses of Phylogenetics and Evolution in R Language.” <em>Bioinformatics</em> 20 (2): 289–90. <a href="https://doi.org/10.1093/bioinformatics/btg412">https://doi.org/10.1093/bioinformatics/btg412</a>.</p>
</div>
<div id="ref-vos_nexml:_2012">
<p>Vos, Rutger A., James P. Balhoff, Jason A. Caravas, Mark T. Holder, Hilmar Lapp, Wayne P. Maddison, Peter E. Midford, et al. 2012. “NeXML: Rich, Extensible, and Verifiable Representation of Comparative Data and Metadata.” <em>Systematic Biology</em> 61 (4): 675–89. <a href="https://doi.org/10.1093/sysbio/sys025">https://doi.org/10.1093/sysbio/sys025</a>.</p>
</div>
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
