<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 ggtree utilities | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 12 ggtree utilities | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 ggtree utilities | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 ggtree utilities | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  



<meta name="date" content="2020-10-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter11.html"/>
<link rel="next" href="chapter13.html"/>
<script src="libs/header-attrs-2.4/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/font-awesome-5.3.1/css/fontawesome-all.min.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>üìñ Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>üéØ Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>üìù Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>üìö Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>üíñ Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#as-treedata"><i class="fa fa-check"></i><b>1.3.3</b> Converting other tree-like object to <code>phylo</code> or <code>treedata</code> object</a></li>
<li class="chapter" data-level="1.3.4" data-path="chapter1.html"><a href="chapter1.html#get-treedata-data"><i class="fa fa-check"></i><b>1.3.4</b> Getting information from <code>treedata</code> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary1"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#reroot-treeio"><i class="fa fa-check"></i><b>2.3</b> Rerooting tree</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.4</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.5</b> Subsetting Tree with Data</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.5.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.5.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.5.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.5.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.5.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.6</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.7" data-path="chapter2.html"><a href="chapter2.html#summary2"><i class="fa fa-check"></i><b>2.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary3"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a>
<ul>
<li class="chapter" data-level="4.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="chapter4.html"><a href="chapter4.html#annotate-one-tree-with-values-from-different-variables"><i class="fa fa-check"></i><b>4.4.1</b> Annotate one tree with values from different variables</a></li>
<li class="chapter" data-level="4.4.2" data-path="chapter4.html"><a href="chapter4.html#densitree"><i class="fa fa-check"></i><b>4.4.2</b> DensiTree</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary4"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#colored-strips"><i class="fa fa-check"></i><b>5.2.1</b> Colored strips</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary5"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a>
<ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary6"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a>
<ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Aligning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#composite_plot"><i class="fa fa-check"></i><b>7.5</b> Composite plots</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.6</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.7" data-path="chapter7.html"><a href="chapter7.html#summary7"><i class="fa fa-check"></i><b>7.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a>
<ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-circularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> Emoji in circular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> Emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a>
<ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#igraph"><i class="fa fa-check"></i><b>9.3</b> ggtree for tree graph</a></li>
<li class="chapter" data-level="9.4" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-other-tree-like-structure"><i class="fa fa-check"></i><b>9.4</b> ggtree for other tree-like structure</a></li>
</ul></li>
<li class="part"><span><b>Part III: ggtree extensions</b></span></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtreeExtra</a>
<ul>
<li class="chapter" data-level="10.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="chapter10.html"><a href="chapter10.html#aligning-graphs-to-the-tree-based-on-tree-structure"><i class="fa fa-check"></i><b>10.2</b> Aligning graphs to the tree based on tree structure</a></li>
<li class="chapter" data-level="10.3" data-path="chapter10.html"><a href="chapter10.html#aligning-multiple-graphs-to-the-tree-for-multi-dimensional-data"><i class="fa fa-check"></i><b>10.3</b> Aligning multiple graphs to the tree for multi-dimensional data</a></li>
<li class="chapter" data-level="10.4" data-path="chapter10.html"><a href="chapter10.html#api-to-link-ggtree-and-ggplot2."><i class="fa fa-check"></i><b>10.4</b> API to link <em>ggtree</em> and <em>ggplot2</em>.</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Other ggtree extensions</a>
<ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#MicrobiotaProcess"><i class="fa fa-check"></i><b>11.1</b> Taxonomy annotation using MicrobiotaProcess</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#visualizing-phylogenetic-network"><i class="fa fa-check"></i><b>11.2</b> Visualizing phylogenetic network</a></li>
</ul></li>
<li class="part"><span><b>Part IV: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> ggtree utilities</a>
<ul>
<li class="chapter" data-level="12.1" data-path="chapter12.html"><a href="chapter12.html#facet-utils"><i class="fa fa-check"></i><b>12.1</b> Facet utilities</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="chapter12.html"><a href="chapter12.html#facet_widths"><i class="fa fa-check"></i><b>12.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="12.1.2" data-path="chapter12.html"><a href="chapter12.html#facet_labeller"><i class="fa fa-check"></i><b>12.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="chapter12.html"><a href="chapter12.html#geom2"><i class="fa fa-check"></i><b>12.2</b> Geometric layers</a></li>
<li class="chapter" data-level="12.3" data-path="chapter12.html"><a href="chapter12.html#layout-utilities"><i class="fa fa-check"></i><b>12.3</b> Layout utilities</a></li>
<li class="chapter" data-level="12.4" data-path="chapter12.html"><a href="chapter12.html#scale-utilities"><i class="fa fa-check"></i><b>12.4</b> Scale utilities</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="chapter12.html"><a href="chapter12.html#xlim_expand"><i class="fa fa-check"></i><b>12.4.1</b> Expand x limit for specific panel</a></li>
<li class="chapter" data-level="12.4.2" data-path="chapter12.html"><a href="chapter12.html#ggexpand"><i class="fa fa-check"></i><b>12.4.2</b> Expand plot limit by ratio of plot range</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="chapter12.html"><a href="chapter12.html#tree-data-utilities"><i class="fa fa-check"></i><b>12.5</b> Tree data utilities</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="chapter12.html"><a href="chapter12.html#td_filter"><i class="fa fa-check"></i><b>12.5.1</b> Filter tree data</a></li>
<li class="chapter" data-level="12.5.2" data-path="chapter12.html"><a href="chapter12.html#td_unnest"><i class="fa fa-check"></i><b>12.5.2</b> Flatten list-column tree data</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="chapter12.html"><a href="chapter12.html#tree-utilities"><i class="fa fa-check"></i><b>12.6</b> Tree utilities</a>
<ul>
<li class="chapter" data-level="12.6.1" data-path="chapter12.html"><a href="chapter12.html#tiporder"><i class="fa fa-check"></i><b>12.6.1</b> Extract tip order</a></li>
<li class="chapter" data-level="12.6.2" data-path="chapter12.html"><a href="chapter12.html#padding-taxa-labels"><i class="fa fa-check"></i><b>12.6.2</b> Padding taxa labels</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="chapter12.html"><a href="chapter12.html#identify"><i class="fa fa-check"></i><b>12.7</b> Interactive ggtree annotation</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter13.html"><a href="chapter13.html"><i class="fa fa-check"></i><b>13</b> Gallery of reproducible examples</a>
<ul>
<li class="chapter" data-level="13.1" data-path="chapter13.html"><a href="chapter13.html#hpv58"><i class="fa fa-check"></i><b>13.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="13.2" data-path="chapter13.html"><a href="chapter13.html#symbolic-bootstrap"><i class="fa fa-check"></i><b>13.2</b> Displaying different symbolic points for bootstrap values.</a></li>
<li class="chapter" data-level="13.3" data-path="chapter13.html"><a href="chapter13.html#phylo-grouping"><i class="fa fa-check"></i><b>13.3</b> Highlighting different groups.</a></li>
<li class="chapter" data-level="13.4" data-path="chapter13.html"><a href="chapter13.html#genome-locus"><i class="fa fa-check"></i><b>13.4</b> Phylogenetic tree with genome locus structure</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#branch-setting"><i class="fa fa-check"></i><b>A.5</b> Branch setting</a>
<ul>
<li class="chapter" data-level="A.5.1" data-path="faq.html"><a href="faq.html#change-branch-length-of-outgroup"><i class="fa fa-check"></i><b>A.5.1</b> Change branch length of outgroup</a></li>
<li class="chapter" data-level="A.5.2" data-path="faq.html"><a href="faq.html#bind-tip"><i class="fa fa-check"></i><b>A.5.2</b> Attach a new tip to a tree</a></li>
<li class="chapter" data-level="A.5.3" data-path="faq.html"><a href="faq.html#change-colours-or-line-types-of-arbitrary-selected-branches"><i class="fa fa-check"></i><b>A.5.3</b> Change colours or line types of arbitrary selected branches</a></li>
<li class="chapter" data-level="A.5.4" data-path="faq.html"><a href="faq.html#arbitrary-point"><i class="fa fa-check"></i><b>A.5.4</b> Add an arbitrary point to a branch</a></li>
</ul></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#different-x-axis-labels-for-different-facet-panels"><i class="fa fa-check"></i><b>A.6</b> Different x-axis labels for different facet panels</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.7</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.8</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.9" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.9</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.10" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.10</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="related-tools.html"><a href="related-tools.html"><i class="fa fa-check"></i><b>B</b> Related tools</a>
<ul>
<li class="chapter" data-level="B.1" data-path="chapter11.html"><a href="chapter11.html#MicrobiotaProcess"><i class="fa fa-check"></i><b>B.1</b> MicrobiotaProcess: Convert <em>taxonomy</em> or hierachical clustering result to <em>treedata</em></a></li>
<li class="chapter" data-level="B.2" data-path="related-tools.html"><a href="related-tools.html#rtol"><i class="fa fa-check"></i><b>B.2</b> rtol: An R interface to Open Tree API</a></li>
<li class="chapter" data-level="B.3" data-path="related-tools.html"><a href="related-tools.html#plotly"><i class="fa fa-check"></i><b>B.3</b> Convert ggtree to plotly object</a></li>
<li class="chapter" data-level="B.4" data-path="related-tools.html"><a href="related-tools.html#commicR"><i class="fa fa-check"></i><b>B.4</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="B.5" data-path="related-tools.html"><a href="related-tools.html#ascii-tree"><i class="fa fa-check"></i><b>B.5</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="B.6" data-path="related-tools.html"><a href="related-tools.html#facet-zoom"><i class="fa fa-check"></i><b>B.6</b> Zoom in selected portion</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>C</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a>
<ul>
<li class="chapter" data-level="C.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>C.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="C.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>C.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>D</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="E" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>E</b> Figures and Tables</a></li>
<li class="chapter" data-level="F" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>F</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter12" class="section level1" number="12">
<h1><span class="header-section-number">Chapter 12</span> ggtree utilities</h1>
<div id="facet-utils" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Facet utilities</h2>
<div id="facet_widths" class="section level3" number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> facet_widths</h3>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="chapter12.html#cb231-1" aria-hidden="true"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb231-2"><a href="chapter12.html#cb231-2" aria-hidden="true"></a><span class="kw">library</span>(ggstance)</span>
<span id="cb231-3"><a href="chapter12.html#cb231-3" aria-hidden="true"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb231-4"><a href="chapter12.html#cb231-4" aria-hidden="true"></a><span class="kw">library</span>(reshape2)</span>
<span id="cb231-5"><a href="chapter12.html#cb231-5" aria-hidden="true"></a></span>
<span id="cb231-6"><a href="chapter12.html#cb231-6" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb231-7"><a href="chapter12.html#cb231-7" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)</span>
<span id="cb231-8"><a href="chapter12.html#cb231-8" aria-hidden="true"></a></span>
<span id="cb231-9"><a href="chapter12.html#cb231-9" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="dt">branch.length =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb231-10"><a href="chapter12.html#cb231-10" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&#39;none&#39;</span>)</span>
<span id="cb231-11"><a href="chapter12.html#cb231-11" aria-hidden="true"></a></span>
<span id="cb231-12"><a href="chapter12.html#cb231-12" aria-hidden="true"></a>a &lt;-<span class="st"> </span><span class="kw">runif</span>(<span class="dv">30</span>, <span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb231-13"><a href="chapter12.html#cb231-13" aria-hidden="true"></a>b &lt;-<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>a</span>
<span id="cb231-14"><a href="chapter12.html#cb231-14" aria-hidden="true"></a>df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(tree<span class="op">$</span>tip.label, a, b)</span>
<span id="cb231-15"><a href="chapter12.html#cb231-15" aria-hidden="true"></a>df &lt;-<span class="st"> </span><span class="kw">melt</span>(df, <span class="dt">id =</span> <span class="st">&quot;tree.tip.label&quot;</span>)</span>
<span id="cb231-16"><a href="chapter12.html#cb231-16" aria-hidden="true"></a></span>
<span id="cb231-17"><a href="chapter12.html#cb231-17" aria-hidden="true"></a>p2 &lt;-<span class="st"> </span><span class="kw">facet_plot</span>(p <span class="op">+</span><span class="st"> </span><span class="kw">xlim_tree</span>(<span class="dv">8</span>), <span class="dt">panel =</span> <span class="st">&#39;bar&#39;</span>, <span class="dt">data =</span> df, <span class="dt">geom =</span> geom_barh, </span>
<span id="cb231-18"><a href="chapter12.html#cb231-18" aria-hidden="true"></a>                 <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> value, <span class="dt">fill =</span> <span class="kw">as.factor</span>(variable)), </span>
<span id="cb231-19"><a href="chapter12.html#cb231-19" aria-hidden="true"></a>                 <span class="dt">width =</span> <span class="fl">0.8</span>, <span class="dt">stat=</span><span class="st">&#39;identity&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim_tree</span>(<span class="dv">9</span>)</span>
<span id="cb231-20"><a href="chapter12.html#cb231-20" aria-hidden="true"></a></span>
<span id="cb231-21"><a href="chapter12.html#cb231-21" aria-hidden="true"></a><span class="kw">facet_widths</span>(p2, <span class="dt">widths =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div>
<p>It also supports using name vector to set the widths of specific panels. The following code will display identical figure to Figure <a href="chapter12.html#fig:facetWidth">12.1</a>A.</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="chapter12.html#cb232-1" aria-hidden="true"></a><span class="kw">facet_widths</span>(p2, <span class="kw">c</span>(<span class="dt">Tree =</span> <span class="fl">.5</span>))</span></code></pre></div>
<p>The <code>facet_widths</code> function also work with other <code>ggplot</code> object as demonstrated in Figure <a href="chapter12.html#fig:facetWidth">12.1</a>B.</p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="chapter12.html#cb233-1" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(iris, <span class="kw">aes</span>(Sepal.Width, Petal.Length)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb233-2"><a href="chapter12.html#cb233-2" aria-hidden="true"></a><span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(.<span class="op">~</span>Species)</span>
<span id="cb233-3"><a href="chapter12.html#cb233-3" aria-hidden="true"></a><span class="kw">facet_widths</span>(p, <span class="kw">c</span>(<span class="dt">setosa =</span> <span class="fl">.5</span>))</span></code></pre></div>


<div class="figure"><span id="fig:facetWidth"></span>
<img src="treedata_files/figure-html/facetWidth-1.svg" alt="Adjust relative widths of ggplot facets. The facet_widths function works with ggtree (A) as well as ggplot (B)." width="576" />
<p class="caption">
Figure 12.1: <strong>Adjust relative widths of ggplot facets.</strong> The <code>facet_widths</code> function works with <code>ggtree</code> (A) as well as <code>ggplot</code> (B).
</p>
</div>
</div>
<div id="facet_labeller" class="section level3" number="12.1.2">
<h3><span class="header-section-number">12.1.2</span> facet_labeller</h3>
<p>The <code>facet_labeller</code> function was designed to re-label selected panels, and it currently only works with <code>ggtree</code> object (<em>i.e.</em> <code>facet_plot</code> output).</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="chapter12.html#cb234-1" aria-hidden="true"></a><span class="kw">facet_labeller</span>(p2, <span class="kw">c</span>(<span class="dt">Tree =</span> <span class="st">&quot;phylogeny&quot;</span>, <span class="dt">bar =</span> <span class="st">&quot;HELLO&quot;</span>))</span></code></pre></div>
<p>If you want to combine <code>facet_widths</code> with <code>facet_labeller</code>, you need to call <code>facet_labeller</code> to re-label the panels before using <code>facet_widths</code> to set the relative widths of each panels. Otherwise it wont work since the output of <code>facet_widths</code> is re-drawn from <code>grid</code> object.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="chapter12.html#cb235-1" aria-hidden="true"></a><span class="kw">facet_labeller</span>(p2, <span class="kw">c</span>(<span class="dt">Tree =</span> <span class="st">&quot;phylogeny&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">facet_widths</span>(<span class="kw">c</span>(<span class="dt">Tree =</span> <span class="fl">.4</span>))</span></code></pre></div>


<div class="figure"><span id="fig:facetLab"></span>
<img src="treedata_files/figure-html/facetLab-1.svg" alt="Rename facet labels. Rename multiple labels simultaneously (A) or only for specific one (B) are all supported. facet_labeller can combine with facet_widths to rename facet label and then adjust relative widths (B)." width="576" />
<p class="caption">
Figure 12.2: <strong>Rename facet labels.</strong> Rename multiple labels simultaneously (A) or only for specific one (B) are all supported. <code>facet_labeller</code> can combine with <code>facet_widths</code> to rename facet label and then adjust relative widths (B).
</p>
</div>
</div>
</div>
<div id="geom2" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Geometric layers</h2>
<p>Subsetting is not supported in layers defined in <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>, while it is quite useful in phylogenetic annotation since it allows us to annotate at specific node(s) (e.g.¬†only label bootstrap values that larger than 75).</p>
<p>In <a href="http://bioconductor.org/packages/ggtree">ggtree</a>, we provides modified version of layers defined in <code>ggplot2</code> to support aesthetic mapping of <code>subset</code>, including:</p>
<ul>
<li>geom_segment2</li>
<li>geom_point2</li>
<li>geom_text2</li>
<li>geom_label2</li>
</ul>


<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="chapter12.html#cb236-1" aria-hidden="true"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb236-2"><a href="chapter12.html#cb236-2" aria-hidden="true"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb236-3"><a href="chapter12.html#cb236-3" aria-hidden="true"></a><span class="kw">data</span>(mpg)</span>
<span id="cb236-4"><a href="chapter12.html#cb236-4" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(<span class="dt">data =</span> mpg, <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> displ, <span class="dt">y =</span> hwy)) <span class="op">+</span></span>
<span id="cb236-5"><a href="chapter12.html#cb236-5" aria-hidden="true"></a><span class="st">   </span><span class="kw">geom_point</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">color =</span> class)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb236-6"><a href="chapter12.html#cb236-6" aria-hidden="true"></a><span class="st">   </span><span class="kw">geom_text2</span>(<span class="kw">aes</span>(<span class="dt">label=</span>manufacturer, </span>
<span id="cb236-7"><a href="chapter12.html#cb236-7" aria-hidden="true"></a>                  <span class="dt">subset =</span> hwy <span class="op">&gt;</span><span class="st"> </span><span class="dv">40</span> <span class="op">|</span><span class="st"> </span>displ <span class="op">&gt;</span><span class="st"> </span><span class="fl">6.5</span>), </span>
<span id="cb236-8"><a href="chapter12.html#cb236-8" aria-hidden="true"></a>                  <span class="dt">nudge_y =</span> <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb236-9"><a href="chapter12.html#cb236-9" aria-hidden="true"></a><span class="st">   </span><span class="kw">coord_cartesian</span>(<span class="dt">clip =</span> <span class="st">&quot;off&quot;</span>) <span class="op">+</span></span>
<span id="cb236-10"><a href="chapter12.html#cb236-10" aria-hidden="true"></a><span class="st">   </span><span class="kw">theme_light</span>() <span class="op">+</span></span>
<span id="cb236-11"><a href="chapter12.html#cb236-11" aria-hidden="true"></a><span class="st">   </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(.<span class="dv">85</span>, <span class="fl">.75</span>))          </span>
<span id="cb236-12"><a href="chapter12.html#cb236-12" aria-hidden="true"></a></span>
<span id="cb236-13"><a href="chapter12.html#cb236-13" aria-hidden="true"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(<span class="kw">rtree</span>(<span class="dv">10</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb236-14"><a href="chapter12.html#cb236-14" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_label2</span>(<span class="kw">aes</span>(<span class="dt">subset =</span> node <span class="op">&lt;</span><span class="dv">5</span>, <span class="dt">label =</span> label))</span>
<span id="cb236-15"><a href="chapter12.html#cb236-15" aria-hidden="true"></a><span class="kw">plot_grid</span>(p, p2, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:layer2"></span>
<img src="treedata_files/figure-html/layer2-1.svg" alt="Geometric layers that supports subsetting. Thes layers works with ggplot2 (A) and ggtree (B)." width="1056" />
<p class="caption">
Figure 12.3: <strong>Geometric layers that supports subsetting.</strong> Thes layers works with <code>ggplot2</code> (A) and <code>ggtree</code> (B).
</p>
</div>
</div>
<div id="layout-utilities" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Layout utilities</h2>
<p>In <a href="chapter4.html#tree-layouts">session 4.2.2</a>, we introduce several layouts that supported by <a href="http://bioconductor.org/packages/ggtree">ggtree</a>. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package also provide several layout functions that can transform from one to another. Note that not all layouts are supported (see <a href="chapter12.html#tab:layoutLayerTab">12.1</a>).</p>
<table>
<caption>
<span id="tab:layoutLayerTab">Table 12.1: </span>Layout layers.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Layout
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
layout_circular
</td>
<td style="text-align:left;">
transform rectangular layout to circular layout
</td>
</tr>
<tr>
<td style="text-align:left;">
layout_dendrogram
</td>
<td style="text-align:left;">
transform rectangular layout to dendrogram layout
</td>
</tr>
<tr>
<td style="text-align:left;">
layout_fan
</td>
<td style="text-align:left;">
transform rectangular/circular layout to fan layout
</td>
</tr>
<tr>
<td style="text-align:left;">
layout_rectangular
</td>
<td style="text-align:left;">
transform circular/fan layout to rectangular layout
</td>
</tr>
<tr>
<td style="text-align:left;">
layout_inward_circular
</td>
<td style="text-align:left;">
transform rectangular/circular layout to inward_circular layout
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="chapter12.html#cb237-1" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">2019</span>)</span>
<span id="cb237-2"><a href="chapter12.html#cb237-2" aria-hidden="true"></a>x &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">20</span>)</span>
<span id="cb237-3"><a href="chapter12.html#cb237-3" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(x)</span>
<span id="cb237-4"><a href="chapter12.html#cb237-4" aria-hidden="true"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">layout_dendrogram</span>()</span>
<span id="cb237-5"><a href="chapter12.html#cb237-5" aria-hidden="true"></a><span class="kw">ggtree</span>(x, <span class="dt">layout =</span> <span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">layout_rectangular</span>()</span>
<span id="cb237-6"><a href="chapter12.html#cb237-6" aria-hidden="true"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">layout_circular</span>()</span>
<span id="cb237-7"><a href="chapter12.html#cb237-7" aria-hidden="true"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">layout_fan</span>(<span class="dt">angle=</span><span class="dv">90</span>)</span>
<span id="cb237-8"><a href="chapter12.html#cb237-8" aria-hidden="true"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">layout_inward_circular</span>(<span class="dt">xlim=</span><span class="dv">4</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">hjust=</span><span class="dv">1</span>)</span></code></pre></div>


<div class="figure"><span id="fig:layoutLayer"></span>
<img src="treedata_files/figure-html/layoutLayer-1.svg" alt="Layout layers for transforming among different layouts. Default rectangular layout (A); transform rectangular to dendrogram layout (B); transform circular to rectangular layout (C); transform rectangular to circular layout (D); transform rectangular to fan layout (E); transform rectangular to inward circular layout (F)." width="1036.8" />
<p class="caption">
Figure 12.4: <strong>Layout layers for transforming among different layouts</strong>. Default rectangular layout (A); transform rectangular to dendrogram layout (B); transform circular to rectangular layout (C); transform rectangular to circular layout (D); transform rectangular to fan layout (E); transform rectangular to inward circular layout (F).
</p>
</div>
</div>
<div id="scale-utilities" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> Scale utilities</h2>
<div id="xlim_expand" class="section level3" number="12.4.1">
<h3><span class="header-section-number">12.4.1</span> Expand x limit for specific panel</h3>
<p>Sometimes we need to set <code>xlim</code> for specific panel (<em>e.g.</em> allocate more space for <a href="faq.html#faq-label-truncated">long tip labels</a> at <code>Tree</code> panel). However, the <code>ggplot2::xlim()</code> function applies to all the panels. <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>xlim_expand()</code> to adjust <code>xlim</code> for user specific panel. It accepts two parameters, <code>xlim</code> and <code>panel</code>, and can adjust all individual panels as demonstrated in Figure <a href="chapter12.html#fig:xlimExpand">12.5</a>A. If you only want to adjust <code>xlim</code> of the <code>Tree</code> panel, you can use <code>xlim_tree()</code> as a shortcut.</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="chapter12.html#cb238-1" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">2019-05-02</span>)</span>
<span id="cb238-2"><a href="chapter12.html#cb238-2" aria-hidden="true"></a>x &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)</span>
<span id="cb238-3"><a href="chapter12.html#cb238-3" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(x) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>()</span>
<span id="cb238-4"><a href="chapter12.html#cb238-4" aria-hidden="true"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> x<span class="op">$</span>tip.label, </span>
<span id="cb238-5"><a href="chapter12.html#cb238-5" aria-hidden="true"></a>                <span class="dt">value =</span> <span class="kw">rnorm</span>(<span class="dv">30</span>))</span>
<span id="cb238-6"><a href="chapter12.html#cb238-6" aria-hidden="true"></a>p2 &lt;-<span class="st"> </span><span class="kw">facet_plot</span>(p, <span class="dt">panel =</span> <span class="st">&quot;Dot&quot;</span>, <span class="dt">data =</span> d, </span>
<span id="cb238-7"><a href="chapter12.html#cb238-7" aria-hidden="true"></a>            <span class="dt">geom =</span> geom_point, <span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> value))</span>
<span id="cb238-8"><a href="chapter12.html#cb238-8" aria-hidden="true"></a>p2 <span class="op">+</span><span class="st"> </span><span class="kw">xlim_tree</span>(<span class="dv">6</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim_expand</span>(<span class="kw">c</span>(<span class="op">-</span><span class="dv">10</span>, <span class="dv">10</span>), <span class="st">&#39;Dot&#39;</span>)</span></code></pre></div>
<p>The <code>xlim_expand()</code> function also works with <code>ggplot2::facet_grid()</code>. As demonstrating in Figure <a href="chapter12.html#fig:xlimExpand">12.5</a>B, only the <code>xlim</code> of <em>virginica</em> panel was adjusted by <code>xlim_expand()</code>.</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="chapter12.html#cb239-1" aria-hidden="true"></a>g &lt;-<span class="st"> </span><span class="kw">ggplot</span>(iris, <span class="kw">aes</span>(Sepal.Length, Sepal.Width)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb239-2"><a href="chapter12.html#cb239-2" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_grid</span>(. <span class="op">~</span><span class="st"> </span>Species, <span class="dt">scales =</span> <span class="st">&quot;free_x&quot;</span>) </span>
<span id="cb239-3"><a href="chapter12.html#cb239-3" aria-hidden="true"></a>g <span class="op">+</span><span class="st"> </span><span class="kw">xlim_expand</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="st">&#39;virginica&#39;</span>)</span></code></pre></div>


<div class="figure"><span id="fig:xlimExpand"></span>
<img src="treedata_files/figure-html/xlimExpand-1.svg" alt="Setting xlim for user specific panel. xlim for ggtree::facet_plot (A, Tree and Dot panels), and ggplot2::facet_grid (B, virginica panel)." width="1152" />
<p class="caption">
Figure 12.5: <strong>Setting xlim for user specific panel.</strong> xlim for <code>ggtree::facet_plot</code> (A, Tree and Dot panels), and <code>ggplot2::facet_grid</code> (B, virginica panel).
</p>
</div>
</div>
<div id="ggexpand" class="section level3" number="12.4.2">
<h3><span class="header-section-number">12.4.2</span> Expand plot limit by ratio of plot range</h3>
<p>The <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> package cannot automatically adjust plot limits and it is very common that long text was truncated. Users need to adjust x (y) limits manually via the <code>xlim()</code> (<code>ylim()</code>) command (see also <a href="faq.html#faq-label-truncated">FAQ: Tip label truncated</a>).</p>
<p>The <code>xlim()</code> (<code>ylim()</code>) is a good solution to this issue. However, we can put the thing more simple, by expanding the plot panel by a ratio of the axis range without knowing what the exact value is.</p>
<p>We provide <code>hexpand()</code> function to expand x limit by specifying a fraction of the x range and it works for both direction (<code>direction=1</code> for right hand side and <code>direction=-1</code> for left hand side) (Figure <a href="chapter12.html#fig:hexpand">12.6</a>). Another version <code>vexpand()</code> works with similar behavior for y axis and the <code>ggexpand()</code> function works for both x and y axes (Figure <a href="chapter11.html#fig:phylonetworx">11.2</a>).</p>


<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="chapter12.html#cb240-1" aria-hidden="true"></a>x<span class="op">$</span>tip.label &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;to make the label longer_&#39;</span>, x<span class="op">$</span>tip.label)</span>
<span id="cb240-2"><a href="chapter12.html#cb240-2" aria-hidden="true"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(x) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">hexpand</span>(.<span class="dv">3</span>)</span>
<span id="cb240-3"><a href="chapter12.html#cb240-3" aria-hidden="true"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggplot</span>(iris, <span class="kw">aes</span>(Sepal.Width, Petal.Width)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb240-4"><a href="chapter12.html#cb240-4" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb240-5"><a href="chapter12.html#cb240-5" aria-hidden="true"></a><span class="st">    </span><span class="kw">hexpand</span>(.<span class="dv">2</span>, <span class="dt">direction =</span> <span class="dv">-1</span>) <span class="op">+</span></span>
<span id="cb240-6"><a href="chapter12.html#cb240-6" aria-hidden="true"></a><span class="st">    </span><span class="kw">vexpand</span>(.<span class="dv">2</span>)</span>
<span id="cb240-7"><a href="chapter12.html#cb240-7" aria-hidden="true"></a><span class="kw">plot_grid</span>(p1, p2, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">rel_widths=</span><span class="kw">c</span>(.<span class="dv">6</span>, <span class="fl">.4</span>))</span></code></pre></div>
<div class="figure"><span id="fig:hexpand"></span>
<img src="treedata_files/figure-html/hexpand-1.svg" alt="Expanding plot limits by a fraction of x or y range. expand x limit at right hand side by default (A). expand x limit for left hand side when direction = -1 and expand y limit at upper side (B)." width="1152" />
<p class="caption">
Figure 12.6: <strong>Expanding plot limits by a fraction of x or y range.</strong> expand x limit at right hand side by default (A). expand x limit for left hand side when direction = -1 and expand y limit at upper side (B).
</p>
</div>
</div>
</div>
<div id="tree-data-utilities" class="section level2" number="12.5">
<h2><span class="header-section-number">12.5</span> Tree data utilities</h2>
<div id="td_filter" class="section level3" number="12.5.1">
<h3><span class="header-section-number">12.5.1</span> Filter tree data</h3>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package defined <a href="chapter12.html#geom2">several several geom layers</a> that supports subsetting tree data. However, many other geom layers that didn‚Äôt provide this feature, are defined in <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> and its extensions. To allow filtering tree data with these layers, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides an accompany function, <code>td_filter()</code> that return a function that work similar to <code>dplyr::filter()</code> and can be passed to the <code>data</code> parameter in geom layers to filter ggtree plot data as demonstrated in Figure <a href="chapter12.html#fig:tdFilter">12.7</a>.</p>


<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="chapter12.html#cb241-1" aria-hidden="true"></a><span class="kw">library</span>(tidytree)</span>
<span id="cb241-2"><a href="chapter12.html#cb241-2" aria-hidden="true"></a></span>
<span id="cb241-3"><a href="chapter12.html#cb241-3" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">1997</span>)</span>
<span id="cb241-4"><a href="chapter12.html#cb241-4" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">50</span>)</span>
<span id="cb241-5"><a href="chapter12.html#cb241-5" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) </span>
<span id="cb241-6"><a href="chapter12.html#cb241-6" aria-hidden="true"></a>selected_nodes &lt;-<span class="st"> </span><span class="kw">offspring</span>(p, <span class="dv">67</span>)<span class="op">$</span>node</span>
<span id="cb241-7"><a href="chapter12.html#cb241-7" aria-hidden="true"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>label), </span>
<span id="cb241-8"><a href="chapter12.html#cb241-8" aria-hidden="true"></a>            <span class="dt">data=</span><span class="kw">td_filter</span>(isTip <span class="op">&amp;</span><span class="st"> </span></span>
<span id="cb241-9"><a href="chapter12.html#cb241-9" aria-hidden="true"></a><span class="st">                        </span>node <span class="op">%in%</span><span class="st"> </span>selected_nodes), </span>
<span id="cb241-10"><a href="chapter12.html#cb241-10" aria-hidden="true"></a>            <span class="dt">hjust=</span><span class="dv">0</span>)</span></code></pre></div>
<div class="figure"><span id="fig:tdFilter"></span>
<img src="treedata_files/figure-html/tdFilter-1.svg" alt="Filtering ggtree plot data in geom layers." width="672" />
<p class="caption">
Figure 12.7: <strong>Filtering ggtree plot data in geom layers.</strong>
</p>
</div>
</div>
<div id="td_unnest" class="section level3" number="12.5.2">
<h3><span class="header-section-number">12.5.2</span> Flatten list-column tree data</h3>
<p>The ggtree plot data is a tidy data frame that each row represents a unique node. If multiple values are associated with a node, the data should be stored as nested data (i.e.¬†in a list-column).</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="chapter12.html#cb242-1" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">1997</span>)</span>
<span id="cb242-2"><a href="chapter12.html#cb242-2" aria-hidden="true"></a>tr &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">5</span>)</span>
<span id="cb242-3"><a href="chapter12.html#cb242-3" aria-hidden="true"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">id=</span><span class="kw">rep</span>(tr<span class="op">$</span>tip.label,<span class="dv">2</span>), </span>
<span id="cb242-4"><a href="chapter12.html#cb242-4" aria-hidden="true"></a>                <span class="dt">value=</span><span class="kw">abs</span>(<span class="kw">rnorm</span>(<span class="dv">10</span>, <span class="dv">6</span>, <span class="dv">2</span>)), </span>
<span id="cb242-5"><a href="chapter12.html#cb242-5" aria-hidden="true"></a>                <span class="dt">group=</span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;A&quot;</span>, <span class="dv">5</span>),<span class="kw">rep</span>(<span class="st">&quot;B&quot;</span>,<span class="dv">5</span>)))</span>
<span id="cb242-6"><a href="chapter12.html#cb242-6" aria-hidden="true"></a></span>
<span id="cb242-7"><a href="chapter12.html#cb242-7" aria-hidden="true"></a><span class="kw">require</span>(tidyr)</span>
<span id="cb242-8"><a href="chapter12.html#cb242-8" aria-hidden="true"></a>d2  &lt;-<span class="st"> </span><span class="kw">nest</span>(d, <span class="dt">value =</span>value, <span class="dt">group=</span>group)</span>
<span id="cb242-9"><a href="chapter12.html#cb242-9" aria-hidden="true"></a><span class="co">## d2 is a nested data</span></span>
<span id="cb242-10"><a href="chapter12.html#cb242-10" aria-hidden="true"></a>d2</span></code></pre></div>
<pre><code>## # A tibble: 5 x 3
##   id    value            group           
##   &lt;chr&gt; &lt;list&gt;           &lt;list&gt;          
## 1 t2    &lt;tibble [2 √ó 1]&gt; &lt;tibble [2 √ó 1]&gt;
## 2 t1    &lt;tibble [2 √ó 1]&gt; &lt;tibble [2 √ó 1]&gt;
## 3 t5    &lt;tibble [2 √ó 1]&gt; &lt;tibble [2 √ó 1]&gt;
## 4 t4    &lt;tibble [2 √ó 1]&gt; &lt;tibble [2 √ó 1]&gt;
## 5 t3    &lt;tibble [2 √ó 1]&gt; &lt;tibble [2 √ó 1]&gt;</code></pre>
<p>Neste data is supported by the operator, <code>%&lt;+%</code>, and can be mapped to the tree structure. If a geom layer can‚Äôt directly supports visualizing nested data, we need to flatten the data before applying the geom layer to display it. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package provides a function, <code>td_unnest()</code>, which return a function that works similar to <code>tidyr::unnest()</code> and can be used to flatten ggtree plot data as demonstrated in Figure <a href="chapter12.html#fig:tdUnnest">12.8</a>A.</p>
<p>All tree data utilities provide a <code>.f</code> parameter to pass a function to pre-operate the data. This create the possibility to combine different tree data utilities as demonstrated in Figure <a href="chapter12.html#fig:tdUnnest">12.8</a>B.</p>


<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="chapter12.html#cb244-1" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tr) <span class="op">%&lt;+%</span><span class="st"> </span>d2</span>
<span id="cb244-2"><a href="chapter12.html#cb244-2" aria-hidden="true"></a>p2 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span></span>
<span id="cb244-3"><a href="chapter12.html#cb244-3" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(x, y, <span class="dt">size=</span> value, <span class="dt">colour=</span>group), </span>
<span id="cb244-4"><a href="chapter12.html#cb244-4" aria-hidden="true"></a>            <span class="dt">data =</span> <span class="kw">td_unnest</span>(<span class="kw">c</span>(value, group)), <span class="dt">alpha=</span>.<span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb244-5"><a href="chapter12.html#cb244-5" aria-hidden="true"></a><span class="st">    </span><span class="kw">scale_size</span>(<span class="dt">range=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">10</span>), <span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">10</span>))</span>
<span id="cb244-6"><a href="chapter12.html#cb244-6" aria-hidden="true"></a></span>
<span id="cb244-7"><a href="chapter12.html#cb244-7" aria-hidden="true"></a>p3 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span></span>
<span id="cb244-8"><a href="chapter12.html#cb244-8" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(x, y, <span class="dt">size=</span> value, <span class="dt">colour=</span>group), </span>
<span id="cb244-9"><a href="chapter12.html#cb244-9" aria-hidden="true"></a>            <span class="dt">data =</span> <span class="kw">td_unnest</span>(<span class="kw">c</span>(value, group), </span>
<span id="cb244-10"><a href="chapter12.html#cb244-10" aria-hidden="true"></a>                        <span class="dt">.f =</span> <span class="kw">td_filter</span>(isTip <span class="op">&amp;</span><span class="st"> </span>node<span class="op">==</span><span class="dv">4</span>)), </span>
<span id="cb244-11"><a href="chapter12.html#cb244-11" aria-hidden="true"></a>            <span class="dt">alpha=</span>.<span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb244-12"><a href="chapter12.html#cb244-12" aria-hidden="true"></a><span class="st">    </span><span class="kw">scale_size</span>(<span class="dt">range=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">10</span>), <span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">3</span>, <span class="dv">10</span>))</span>
<span id="cb244-13"><a href="chapter12.html#cb244-13" aria-hidden="true"></a></span>
<span id="cb244-14"><a href="chapter12.html#cb244-14" aria-hidden="true"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p2, p3, <span class="dt">labels=</span>LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>])            </span></code></pre></div>
<div class="figure"><span id="fig:tdUnnest"></span>
<img src="treedata_files/figure-html/tdUnnest-1.svg" alt="Flattening ggtree plot data. (A) list-columns can be flattened by td_unnest(). (B) Different tree data utilites can be combined to work together (e.g.¬†filter data by td_filter() and then flatten it by td_unnest()." width="768" />
<p class="caption">
Figure 12.8: <strong>Flattening ggtree plot data.</strong> (A) list-columns can be flattened by <code>td_unnest()</code>. (B) Different tree data utilites can be combined to work together (e.g.¬†filter data by <code>td_filter()</code> and then flatten it by <code>td_unnest()</code>.
</p>
</div>
</div>
</div>
<div id="tree-utilities" class="section level2" number="12.6">
<h2><span class="header-section-number">12.6</span> Tree utilities</h2>
<div id="tiporder" class="section level3" number="12.6.1">
<h3><span class="header-section-number">12.6.1</span> Extract tip order</h3>
<p>To create <a href="chapter7.html#composite_plot">composite plots</a>, users need to re-order their data manually before they creating tree associated graph. The order of their data should be consistent with tip order presented in ggtree plot. For this purpose, we provide the <code>get_taxa_name()</code> function to extract an ordered vector of tips based on the tree structure plotted by ggtree.</p>


<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="chapter12.html#cb245-1" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb245-2"><a href="chapter12.html#cb245-2" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">10</span>)</span>
<span id="cb245-3"><a href="chapter12.html#cb245-3" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb245-4"><a href="chapter12.html#cb245-4" aria-hidden="true"></a><span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node =</span> <span class="dv">12</span>, <span class="dt">extendto =</span> <span class="fl">2.5</span>)</span>
<span id="cb245-5"><a href="chapter12.html#cb245-5" aria-hidden="true"></a><span class="kw">print</span>(p)</span></code></pre></div>
<div class="figure"><span id="fig:tiporder"></span>
<img src="treedata_files/figure-html/tiporder-1.svg" alt="An example tree for demonstraing get_taxa_name() function." width="480" />
<p class="caption">
Figure 12.9: <strong>An example tree for demonstraing <code>get_taxa_name()</code> function.</strong>
</p>
</div>
<p>The <code>get_taxa_name()</code> function will return a vector of ordered tip labels according to the tree structure displayed on Figure <a href="chapter12.html#fig:tiporder">12.9</a>.</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="chapter12.html#cb246-1" aria-hidden="true"></a><span class="kw">get_taxa_name</span>(p)</span></code></pre></div>
<pre><code>##  [1] &quot;t9&quot;  &quot;t8&quot;  &quot;t3&quot;  &quot;t2&quot;  &quot;t7&quot;  &quot;t10&quot; &quot;t1&quot;  &quot;t5&quot;  &quot;t6&quot; 
## [10] &quot;t4&quot;</code></pre>
<p>If user specific a node, the <code>get_taxa_name()</code> will extract order tips of selected clade (i.e.¬†highlighted region on the Figure <a href="chapter12.html#fig:tiporder">12.9</a>).</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="chapter12.html#cb248-1" aria-hidden="true"></a><span class="kw">get_taxa_name</span>(p, <span class="dt">node =</span> <span class="dv">12</span>)</span></code></pre></div>
<pre><code>## [1] &quot;t5&quot; &quot;t6&quot; &quot;t4&quot;</code></pre>
</div>
<div id="padding-taxa-labels" class="section level3" number="12.6.2">
<h3><span class="header-section-number">12.6.2</span> Padding taxa labels</h3>
<p>The <code>label_pad()</code> function adds padding characters (default is <code>¬∑</code>) to taxa labels.</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="chapter12.html#cb250-1" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">2015-12-21</span>)</span>
<span id="cb250-2"><a href="chapter12.html#cb250-2" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">5</span>)</span>
<span id="cb250-3"><a href="chapter12.html#cb250-3" aria-hidden="true"></a>tree<span class="op">$</span>tip.label[<span class="dv">2</span>] &lt;-<span class="st"> &quot;long string for test&quot;</span></span>
<span id="cb250-4"><a href="chapter12.html#cb250-4" aria-hidden="true"></a></span>
<span id="cb250-5"><a href="chapter12.html#cb250-5" aria-hidden="true"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> tree<span class="op">$</span>tip.label, </span>
<span id="cb250-6"><a href="chapter12.html#cb250-6" aria-hidden="true"></a>                <span class="dt">newlabel =</span> <span class="kw">label_pad</span>(tree<span class="op">$</span>tip.label),</span>
<span id="cb250-7"><a href="chapter12.html#cb250-7" aria-hidden="true"></a>                <span class="dt">newlabel2 =</span> <span class="kw">label_pad</span>(tree<span class="op">$</span>tip.label, <span class="dt">pad =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb250-8"><a href="chapter12.html#cb250-8" aria-hidden="true"></a><span class="kw">print</span>(d)</span></code></pre></div>
<pre><code>##                  label             newlabel
## 1                   t1 ¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑t1
## 2 long string for test long string for test
## 3                   t2 ¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑t2
## 4                   t4 ¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑t4
## 5                   t3 ¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑¬∑t3
##              newlabel2
## 1                   t1
## 2 long string for test
## 3                   t2
## 4                   t4
## 5                   t3</code></pre>
<p>This feature is useful if we want to align tip labels to the end as demonstrated in Figure <a href="chapter12.html#fig:labelpad">12.10</a>. Note that in this case, monospace font should be used to ensure the lengths of the labels displayed in the plot are the same.</p>


<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="chapter12.html#cb252-1" aria-hidden="true"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">%&lt;+%</span><span class="st"> </span>d <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">3</span>)</span>
<span id="cb252-2"><a href="chapter12.html#cb252-2" aria-hidden="true"></a>p1 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>newlabel), </span>
<span id="cb252-3"><a href="chapter12.html#cb252-3" aria-hidden="true"></a>                    <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">family=</span><span class="st">&#39;mono&#39;</span>,</span>
<span id="cb252-4"><a href="chapter12.html#cb252-4" aria-hidden="true"></a>                    <span class="dt">linetype =</span> <span class="st">&quot;dotted&quot;</span>, <span class="dt">linesize =</span> <span class="fl">.7</span>) </span>
<span id="cb252-5"><a href="chapter12.html#cb252-5" aria-hidden="true"></a>p2 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>newlabel2), </span>
<span id="cb252-6"><a href="chapter12.html#cb252-6" aria-hidden="true"></a>                    <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">family=</span><span class="st">&#39;mono&#39;</span>,</span>
<span id="cb252-7"><a href="chapter12.html#cb252-7" aria-hidden="true"></a>                    <span class="dt">linetype =</span> <span class="ot">NULL</span>, <span class="dt">offset=</span><span class="op">-</span>.<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">2</span>)</span>
<span id="cb252-8"><a href="chapter12.html#cb252-8" aria-hidden="true"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>))                            </span></code></pre></div>
<div class="figure"><span id="fig:labelpad"></span>
<img src="treedata_files/figure-html/labelpad-1.svg" alt="Align tip label to the end. With dotted line (A) and without dotted line (B)." width="1401.6" />
<p class="caption">
Figure 12.10: <strong>Align tip label to the end.</strong> With dotted line (A) and without dotted line (B).
</p>
</div>
</div>
</div>
<div id="identify" class="section level2" number="12.7">
<h2><span class="header-section-number">12.7</span> Interactive ggtree annotation</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package supports interactive tree annotation or manipulation by implementing an <code>identify()</code> method. Users can click on a node to highlight a clade, to label or rotate it <em>etc</em>. Users can also use the <a href="https://CRAN.R-project.org/package=plotly">plotly</a> package to convert <code>ggtree</code> to <code>plotly</code> object to quickly create <a href="related-tools.html#plotly">interactive phylogenetic tree</a>.</p>


<div class="figure"><span id="fig:ggtreeidentify"></span>
<img src="img/ggtree-identify.gif" alt="Interactive phylogenetic tree using identify() method. Highlighting, labelling and rotating clades are all supported." width="100%" />
<p class="caption">
Figure 12.11: <strong>Interactive phylogenetic tree using identify() method.</strong> Highlighting, labelling and rotating clades are all supported.
</p>
</div>
<p>Video of using <code>identify()</code> to interactively manipulate phylogenetic tree can be found on Youtube <i class="fab  fa-youtube " style="color:red;"></i> and Youku:</p>
<ul>
<li>Highlighting clades: <a href="https://youtu.be/KcF8Ec38mzI">Youtube <i class="fab  fa-youtube " style="color:red;"></i></a> and <a href="http://v.youku.com/v_show/id_XMTYyMzgyODYyOA">Youku</a>.</li>
<li>Labelling clades: <a href="https://youtu.be/SmcceRD_jxg">Youtube <i class="fab  fa-youtube " style="color:red;"></i></a> and <a href="http://v.youku.com/v_show/id_XMTYyNDIzODA0NA">Youku</a>.</li>
<li>Rotating clades: <a href="https://youtu.be/lKNn4QlPO0E">Youtube <i class="fab  fa-youtube " style="color:red;"></i></a> and <a href="http://v.youku.com/v_show/id_XMTYyMzgyODg2OA">Youku</a>.</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter11.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter13.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["treedata.pdf", "treedata.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
