<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.11 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu School of Basic Medical Sciences, Southern Medical University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter4.html">
<link rel="next" href="chapter6.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.3.3</b> Getting information from <em>treedata</em> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#layouts-of-phylogenetic-tree"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-scale-evolution-distance"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#displaying-nodestips"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.4</b> Color tree</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.5</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-the-tree-view"><i class="fa fa-check"></i><b>4.3.6</b> Modify compenents of the tree view</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#annotate-clades"><i class="fa fa-check"></i><b>5.2.1</b> Annotate clades</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic"><i class="fa fa-check"></i><b>5.2.2</b> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.3</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.4</b> Taxa connection</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-based-on-tree-classes-defined-in-other-r-packages"><i class="fa fa-check"></i><b>5.3.2</b> Tree annotation based on tree classes defined in other R packages</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.2</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#scaling-selected-clade"><i class="fa fa-check"></i><b>6.3</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#zoom-on-a-portion-of-tree"><i class="fa fa-check"></i><b>6.6</b> Zoom on a portion of tree</a></li>
<li class="chapter" data-level="6.7" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> The <code>%&lt;+%</code> operator</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#visualize-tree-with-associated-matrix"><i class="fa fa-check"></i><b>7.2</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.2.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.2.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#plot-tree-with-associated-data"><i class="fa fa-check"></i><b>7.4</b> Plot tree with associated data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree utilities</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#facet_widths"><i class="fa fa-check"></i><b>9.1</b> facet_widths</a></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#facet_labeller"><i class="fa fa-check"></i><b>9.2</b> facet_labeller</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#xlim_expand"><i class="fa fa-check"></i><b>9.3</b> xlim_expand</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree for other tree objects</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.5</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.6</b> Enlarge center space in circular/fan layout tree</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Image processing using magick package</a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#example-1-remove-background-of-images"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="print-ascii-art-rooted-tree.html"><a href="print-ascii-art-rooted-tree.html"><i class="fa fa-check"></i><b>D</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="E" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>E</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="F" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>F</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter5" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Phylogenetic Tree Annotation</h1>
<div id="visualizing-and-annotating-tree-using-grammar-of-graphics" class="section level2">
<h2><span class="header-section-number">5.1</span> Visualizing and Annotating Tree using Grammar of Graphics</h2>
<p>The <em>ggtree</em> is designed for more general purpose or specific type of tree visualization and annotation. It supports grammar of graphics implemented in <em>ggplot2</em> and users can freely visualize/annotate a tree by combining several annotation layers.</p>



<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggtree)
treetext =<span class="st"> &quot;(((ADH2:0.1[&amp;&amp;NHX:S=human], ADH1:0.11[&amp;&amp;NHX:S=human]):</span>
<span class="st">0.05 [&amp;&amp;NHX:S=primates:D=Y:B=100],ADHY:</span>
<span class="st">0.1[&amp;&amp;NHX:S=nematode],ADHX:0.12 [&amp;&amp;NHX:S=insect]):</span>
<span class="st">0.1[&amp;&amp;NHX:S=metazoa:D=N],(ADH4:0.09[&amp;&amp;NHX:S=yeast],</span>
<span class="st">ADH3:0.13[&amp;&amp;NHX:S=yeast], ADH2:0.12[&amp;&amp;NHX:S=yeast],</span>
<span class="st">ADH1:0.11[&amp;&amp;NHX:S=yeast]):0.1[&amp;&amp;NHX:S=Fungi])[&amp;&amp;NHX:D=N];&quot;</span>
tree &lt;-<span class="st"> </span><span class="kw">read.nhx</span>(<span class="kw">textConnection</span>(treetext))
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span>S), <span class="dt">fill=</span><span class="st">&#39;lightgreen&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_label</span>(<span class="kw">aes</span>(<span class="dt">label=</span>D), <span class="dt">fill=</span><span class="st">&#39;steelblue&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>B), <span class="dt">hjust=</span><span class="op">-</span>.<span class="dv">5</span>)</code></pre>
<div class="figure"><span id="fig:ggtreeNHX"></span>
<img src="treedata_files/figure-html/ggtreeNHX-1.png" alt="Annotating tree using grammar of graphics. The NHX tree was annotated using grammar of graphic syntax by combining different layers using + operator. Species information were labelled on the middle of the branches, Duplication events were shown on most recent common ancestor and clade bootstrap value were dispalyed near to it." width="672"  />
<p class="caption">
Figure 5.1: <strong>Annotating tree using grammar of graphics.</strong> The NHX tree was annotated using grammar of graphic syntax by combining different layers using <code>+</code> operator. Species information were labelled on the middle of the branches, Duplication events were shown on most recent common ancestor and clade bootstrap value were dispalyed near to it.
</p>
</div>

<p>Here, as an example, we visualized the tree with several layers to display annotation stored in NHX tags, including a layer of <code>geom_tiplab</code> to display tip labels (gene name in this case), a layer using <code>geom_label</code> to show species information (<code>S</code> tag) colored by lightgreen, a layer of duplication event information (<code>D</code> tag) colored by steelblue and another layer using <em>geom_text</em> to show bootstrap value (<code>B</code> tag).</p>
<p>Layers defined in <em>ggplot2</em> can be applied to <em>ggtree</em> directly as demonstrated in Figure <a href="chapter5.html#fig:ggtreeNHX">5.1</a> of using <em>geom_label</em> and <em>geom_text</em>. But <em>ggplot2</em> does not provide graphic layers that are specific designed for phylogenetic tree annotation. For instance, layers for tip labels, tree branch scale legend, highlight or labeling clade are all unavailable. To make tree annotation more flexible, a number of layers have been implemented in <em>ggtree</em> (Table <a href="chapter5.html#tab:geoms">5.1</a>), enabling different ways of annotation on various parts/components of a phylogenetic tree.</p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:geoms">Table 5.1: </span>Geom layers defined in ggtree.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Layer
</th>
<th style="text-align:left;">
Description
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
geom_balance
</td>
<td style="text-align:left;">
highlights the two direct descendant clades of an internal node
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_cladelabel
</td>
<td style="text-align:left;">
annotate a clade with bar and text label
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_hilight
</td>
<td style="text-align:left;">
highlight a clade with rectangle
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_label2
</td>
<td style="text-align:left;">
modified version of geom_label, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_nodepoint
</td>
<td style="text-align:left;">
annotate internal nodes with symbolic points
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_point2
</td>
<td style="text-align:left;">
modified version of geom_point, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_range
</td>
<td style="text-align:left;">
bar layer to present uncertainty of evolutionary inference
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_rootpoint
</td>
<td style="text-align:left;">
annotate root node with symbolic point
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_segment2
</td>
<td style="text-align:left;">
modified version of geom_segment, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_strip
</td>
<td style="text-align:left;">
annotate associated taxa with bar and (optional) text label
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_taxalink
</td>
<td style="text-align:left;">
associate two related taxa by linking them with a curve
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_text2
</td>
<td style="text-align:left;">
modified version of geom_text, with subsetting supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tiplab
</td>
<td style="text-align:left;">
layer of tip labels
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tiplab2
</td>
<td style="text-align:left;">
layer of tip labels for circular layout
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tippoint
</td>
<td style="text-align:left;">
annotate external nodes with symbolic points
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_tree
</td>
<td style="text-align:left;">
tree structure layer, with multiple layout supported
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_treescale
</td>
<td style="text-align:left;">
tree branch scale legend
</td>
</tr>
</tbody>
</table>
</div>
<div id="layers-for-tree-annotation" class="section level2">
<h2><span class="header-section-number">5.2</span> Layers for Tree Annotation</h2>
<div id="annotate-clades" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Annotate clades</h3>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017" role="doc-biblioref">2017</a>)</span> implements <em><code>geom_cladelabel</code></em> layer to annotate a selected clade with a bar indicating the clade with a corresponding label.</p>
<p>The <em><code>geom_cladelabel</code></em> layer accepts a selected internal node number. To get the internal node number, please refer to <a href="treeManipulation.html#internal-node-number">Tree Manipulation</a> vignette.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">2015-12-21</span>)
tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)
p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">6</span>)

p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<p>Users can set the parameter, <code>align = TRUE</code>, to align the clade label, and use the parameter, <code>offset</code>, to adjust the position.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">offset=</span>.<span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">offset=</span>.<span class="dv">5</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-84-1.png" width="672" /></p>
<p>Users can change the color of the clade label via the parameter <code>color</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-85-1.png" width="672" /></p>
<p>Users can change the <code>angle</code> of the clade label text and relative position from text to bar via the parameter <code>offset.text</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">270</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>, <span class="dt">offset.text=</span>.<span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">45</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-86-1.png" width="672" /></p>
<p>The size of the bar and text can be changed via the parameters <code>barsize</code> and <code>fontsize</code> respectively.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">270</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>, <span class="dt">offset.text=</span>.<span class="dv">5</span>, <span class="dt">barsize=</span><span class="fl">1.5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">45</span>, <span class="dt">fontsize=</span><span class="dv">8</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-87-1.png" width="672" /></p>
<p>Users can also use <code>geom_label</code> to label the text.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">geom=</span><span class="st">&#39;label&#39;</span>, <span class="dt">fill=</span><span class="st">&#39;lightblue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-88-1.png" width="672" /></p>
<div id="annotate-clades-for-unrooted-tree" class="section level4">
<h4><span class="header-section-number">5.2.1.1</span> Annotate clades for unrooted tree</h4>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>geom_clade2</code> for labeling clades of unrooted
layout trees.</p>
<pre class="sourceCode r"><code class="sourceCode r">pg &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>)
pg <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel2</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">angle=</span><span class="dv">10</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel2</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">angle=</span><span class="dv">305</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-89-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</h3>
<p><code>geom_cladelabel</code> is designed for labelling Monophyletic (Clade) while there are related taxa that are not form a clade. <code>ggtree</code> provides <code>geom_strip</code> to add a strip/bar to indicate the association with optional label (see <a href="https://github.com/GuangchuangYu/ggtree/issues/52">the issue</a>).</p>
<pre class="sourceCode r"><code class="sourceCode r">nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sample.nwk&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(nwk)
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_strip</span>(<span class="dv">5</span>, <span class="dv">7</span>, <span class="dt">barsize=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_strip</span>(<span class="dv">6</span>, <span class="dv">12</span>, <span class="dt">barsize=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-90-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="highlight-clades" class="section level3">
<h3><span class="header-section-number">5.2.3</span> Highlight clades</h3>
<p><code>ggtree</code> implements <em><code>geom_hilight</code></em> layer, that accepts an internal node number and add a layer of rectangle to highlight the selected clade.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">21</span>, <span class="dt">fill=</span><span class="st">&quot;steelblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">17</span>, <span class="dt">fill=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-91-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">21</span>, <span class="dt">fill=</span><span class="st">&quot;steelblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">23</span>, <span class="dt">fill=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-92-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Another way to highlight selected clades is setting the clades with different colors and/or line types as demonstrated in <a href="treeManipulation.html#groupclade">Tree Manipulation</a> vignette.</p>
<div id="highlight-balances" class="section level4">
<h4><span class="header-section-number">5.2.3.1</span> Highlight balances</h4>
<p>In addition to <em><code>geom_hilight</code></em>, <code>ggtree</code> also implements <em><code>geom_balance</code></em>
which is designed to highlight neighboring subclades of a given internal node.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_balance</span>(<span class="dt">node=</span><span class="dv">16</span>, <span class="dt">fill=</span><span class="st">&#39;steelblue&#39;</span>, <span class="dt">color=</span><span class="st">&#39;white&#39;</span>, <span class="dt">alpha=</span><span class="fl">0.6</span>, <span class="dt">extend=</span><span class="dv">1</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_balance</span>(<span class="dt">node=</span><span class="dv">19</span>, <span class="dt">fill=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">color=</span><span class="st">&#39;white&#39;</span>, <span class="dt">alpha=</span><span class="fl">0.6</span>, <span class="dt">extend=</span><span class="dv">1</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-93-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="highlight-clades-for-unrooted-tree" class="section level4">
<h4><span class="header-section-number">5.2.3.2</span> Highlight clades for unrooted tree</h4>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>geom_hilight_encircle</code> to support highlight
clades for unrooted layout trees.</p>
<pre class="sourceCode r"><code class="sourceCode r">pg <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight_encircle</span>(<span class="dt">node=</span><span class="dv">45</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight_encircle</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">fill=</span><span class="st">&#39;darkgreen&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-94-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="taxa-connection" class="section level3">
<h3><span class="header-section-number">5.2.4</span> Taxa connection</h3>
<p>Some evolutionary events (e.g. reassortment, horizontal gene transfer) can be modeled by a simple tree. <code>ggtree</code> provides <code>geom_taxalink</code> layer that allows drawing straight or curved lines between any of two nodes in the tree, allow it to represent evolutionary events by connecting taxa.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_taxalink</span>(<span class="st">&#39;A&#39;</span>, <span class="st">&#39;E&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_taxalink</span>(<span class="st">&#39;F&#39;</span>, <span class="st">&#39;K&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">arrow=</span>grid<span class="op">::</span><span class="kw">arrow</span>(<span class="dt">length=</span>grid<span class="op">::</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&quot;npc&quot;</span>)))</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-95-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="tree-annotation-with-output-from-evolution-software" class="section level2">
<h2><span class="header-section-number">5.3</span> Tree annotation with output from evolution software</h2>
<p>The <a href="http://bioconductor.org/packages/treeio">treeio</a> package implemented several parser functions to parse
output from commonly used software in evolutionary biology.</p>
<p>Here, we used <a href="http://beast2.org/">BEAST</a> <span class="citation">(Bouckaert et al. <a href="#ref-bouckaert_beast_2014" role="doc-biblioref">2014</a>)</span> output as an
example. For details, please refer to the
<a href="https://bioconductor.org/packages/devel/bioc/vignettes/treeio/inst/doc/Importer.html">Importer</a> vignette.</p>
<pre class="sourceCode r"><code class="sourceCode r">file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/BEAST&quot;</span>, <span class="st">&quot;beast_mcc.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
beast &lt;-<span class="st"> </span><span class="kw">read.beast</span>(file)
<span class="kw">ggtree</span>(beast, <span class="kw">aes</span>(<span class="dt">color=</span>rate))  <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_range</span>(<span class="dt">range=</span><span class="st">&#39;length_0.95_HPD&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span><span class="kw">round</span>(posterior, <span class="dv">2</span>)), <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">5</span>, <span class="dt">size=</span><span class="dv">3</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_continuous</span>(<span class="dt">low=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">high=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.8</span>))</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-96-1.png" width="480" style="display: block; margin: auto;" /></p>
<div id="tree-annotation-using-data-from-evolutionary-analysis-software" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Tree annotation using data from evolutionary analysis software</h3>
<p><a href="#chapter-treeio">Chapter 2</a> has introduced using <em>treeio</em> packages to parse different tree formats and commonly used software outputs to obtain phylogeny-associated data. These imported data as <code>S4</code> objects can be visualized directly using <em>ggtree</em>. Figure <a href="chapter5.html#fig:ggtreeNHX">5.1</a> demonstrates a tree annotated using the information (species classification, duplication event and bootstrap value) stored in NHX file. <em>PHYLODOG</em> and <em>RevBayes</em> output NHX files that can be parsed by <em>treeio</em> and visualized by <em>ggtree</em> with annotation using their inference data.</p>
<p>Furthermore, the evolutionary data from the inference of <em>BEAST</em>, <em>MrBayes</em> and <em>RevBayes</em>, <em>d<sub>N</sub>/d<sub>S</sub></em> values inferred by <em>CodeML</em>, ancestral sequences inferred by <em>HyPhy</em>, <em>CodeML</em> or <em>BaseML</em> and short read placement by <em>EPA</em> and <em>pplacer</em> can be used to annotate the tree directly.</p>



<pre class="sourceCode r"><code class="sourceCode r">file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/BEAST&quot;</span>, <span class="st">&quot;beast_mcc.tree&quot;</span>, 
                    <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
beast &lt;-<span class="st"> </span><span class="kw">read.beast</span>(file)
<span class="kw">ggtree</span>(beast) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">linetype=</span><span class="st">&#39;dashed&#39;</span>, <span class="dt">linesize=</span>.<span class="dv">3</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_range</span>(<span class="st">&quot;length_0.95_HPD&quot;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">size=</span><span class="dv">2</span>, <span class="dt">alpha=</span>.<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_text2</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">round</span>(<span class="kw">as.numeric</span>(posterior), <span class="dv">2</span>), 
                 <span class="dt">subset=</span><span class="kw">as.numeric</span>(posterior)<span class="op">&gt;</span><span class="st"> </span><span class="fl">0.9</span>, 
                 <span class="dt">x=</span>branch), <span class="dt">vjust=</span><span class="dv">0</span>) </code></pre>
<div class="figure"><span id="fig:beast"></span>
<img src="treedata_files/figure-html/beast-1.png" alt="Annotating BEAST tree with length_95%_HPD and posterior. Branch length credible intervals (95% HPD) were displayed as red horizontal bars and clade posterior values were shown on the middle of branches." width="672"  />
<p class="caption">
Figure 5.2: <strong>Annotating <em>BEAST</em> tree with <em>length_95%_HPD</em> and posterior.</strong> Branch length credible intervals (95% HPD) were displayed as red horizontal bars and clade posterior values were shown on the middle of branches.
</p>
</div>

<p>In Figure <a href="chapter5.html#fig:beast">5.2</a>, the tree was visualized and annotated with posterior &gt; 0.9 and demonstrated length uncertainty (95% Highest Posterior Density (HPD) interval).</p>
<p>Ancestral sequences inferred by <em>HyPhy</em> can be parsed using <a href="chapter1.html#parsing-hyphy-output"><em>treeio</em></a>, whereas the substitutions along each tree branch was automatically computed and stored inside the phylogenetic tree object (<em>i.e.</em>, <code>S4</code> object). The <em>ggtree</em> can utilize this information in the object to annotate the tree, as demonstrated in Figure <a href="chapter5.html#fig:hyphy">5.3</a>.</p>



<pre class="sourceCode r"><code class="sourceCode r">nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/HYPHY&quot;</span>, <span class="st">&quot;labelledtree.tree&quot;</span>, 
                   <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
ancseq &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/HYPHY&quot;</span>, <span class="st">&quot;ancseq.nex&quot;</span>, 
                      <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
tipfas &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;pa.fas&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
hy &lt;-<span class="st"> </span><span class="kw">read.hyphy</span>(nwk, ancseq, tipfas)
<span class="kw">ggtree</span>(hy) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span>AA_subs), <span class="dt">size=</span><span class="dv">2</span>, 
            <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>)</code></pre>
<div class="figure"><span id="fig:hyphy"></span>
<img src="treedata_files/figure-html/hyphy-1.png" alt="Annotating tree with amino acid substitution determined by ancestral sequences inferred by HYPHY. Amino acid substitutions were displayed on the middle of branches." width="748.8"  />
<p class="caption">
Figure 5.3: <strong>Annotating tree with amino acid substitution determined by ancestral sequences inferred by HYPHY.</strong> Amino acid substitutions were displayed on the middle of branches.
</p>
</div>

<p><em>PAML</em>’s <em>BaseML</em> and <em>CodeML</em> can be also used to infer ancestral sequences, whereas <em>CodeML</em> can infer selection pressure. After parsing this information using <a href="chapter1.html#parsing-paml-output"><em>treeio</em></a>, <em>ggtree</em> can integrate this information into the same tree structure and used for annotation as illustrated in Figure <a href="chapter5.html#fig:codeml">5.4</a>.</p>



<pre class="sourceCode r"><code class="sourceCode r">rstfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;rst&quot;</span>, 
                       <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
mlcfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;mlc&quot;</span>, 
                       <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
ml &lt;-<span class="st"> </span><span class="kw">read.codeml</span>(rstfile, mlcfile)
<span class="kw">ggtree</span>(ml, <span class="kw">aes</span>(<span class="dt">color=</span>dN_vs_dS), <span class="dt">branch.length=</span><span class="st">&#39;dN_vs_dS&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_color_continuous</span>(<span class="dt">name=</span><span class="st">&#39;dN/dS&#39;</span>, <span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">1.5</span>),
                         <span class="dt">oob=</span>scales<span class="op">::</span>squish,
                         <span class="dt">low=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">high=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span>AA_subs), 
            <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">5</span>, <span class="dt">color=</span><span class="st">&#39;steelblue&#39;</span>, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme_tree2</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">9</span>, <span class="fl">.3</span>))</code></pre>
<div class="figure"><span id="fig:codeml"></span>
<img src="treedata_files/figure-html/codeml-1.png" alt="Annotating tree with animo acid substitution and dN/dS inferred by CodeML. Branches were rescaled and colored by dN/dS values and amino acid substitutions were displayed on the middle of branches." width="672"  />
<p class="caption">
Figure 5.4: <strong>Annotating tree with animo acid substitution and <em>d<sub>N</sub>/d<sub>S</sub></em> inferred by <em>CodeML</em>.</strong> Branches were rescaled and colored by <em>d<sub>N</sub>/d<sub>S</sub></em> values and amino acid substitutions were displayed on the middle of branches.
</p>
</div>

<p>For more details and examples of annotating tree with evolutionary data inferred by different software packages can be referred to the online vignettes<a href="#fn13" class="footnote-ref" id="fnref13"><sup>13</sup></a>.</p>
</div>
<div id="tree-annotation-based-on-tree-classes-defined-in-other-r-packages" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Tree annotation based on tree classes defined in other R packages</h3>
<p>The <em>ggtree</em> plays a unique role in R ecosystem to facilitate phylogenetic analysis. It serves as a generic tools for tree visualization and annotation with different associated data from various sources. Most of the phylogenetic tree classes defined in R community are supported, including <code>obkData</code>, <code>phyloseq</code>, <code>phylo</code>, <code>multiPhylo</code>, <code>phylo4</code> and <code>phylo4d</code>. Such that <em>ggtree</em> can be easily integrated into their analysis/packages. For instance, <em>phyloseq</em> users will find <em>ggtree</em> useful for visualizing microbiome data and for further annotations, since <em>ggtree</em> supports high-level of annotation using grammar of graphics and some of its features are not available in <em>phyloseq</em>. Here, examples of using <em>ggtree</em> to annotate <code>obkData</code> and <code>phyloseq</code> tree objects are demonstrated. There example data can be found in vignettes of <em>OutbreakTools</em> <span class="citation">(Jombart et al. <a href="#ref-jombart_outbreaktools_2014" role="doc-biblioref">2014</a>)</span> and <em>phyloseq</em> <span class="citation">(McMurdie and Holmes <a href="#ref-mcmurdie_phyloseq_2013" role="doc-biblioref">2013</a>)</span> packages.</p>
<!--

> outbreaktools was removed from CRAN

The `okbData` is defined to store incidence-based outbreak data, including meta data of sampling and information of infected individuals such as age and onset of symptoms. The *ggtree* supports parsing this information which was used to annotate the tree as shown in Figure \@ref(fig:outbreaktools).

(ref:outbreaktoolsscap) Visualizing obkData tree object.

(ref:outbreaktoolscap) **Visualizing obkData tree object.** *x*-axis was scaled by timeline of the outbreak and tips were colored by location of different individuals.

\setstretch{1.2}
r outbreaktools, fig.width=6.3, fig.height=7, fig.cap="(ref:outbreaktoolscap)", fig.scap="(ref:outbreaktoolsscap)", message=FALSE, out.extra=''}
library(OutbreakTools)
data(FluH1N1pdm2009)
attach(FluH1N1pdm2009)

x <- new("obkData", 
         individuals = individuals, 
         dna = dna,
         dna.individualID = samples$individualID, 
         dna.date = samples$date,
         trees = FluH1N1pdm2009$trees)
ggtree(x, mrsd="2009-09-30", as.Date=TRUE, right=TRUE) +
    geom_tippoint(aes(color=location), size=3, alpha=.75) +
    scale_color_brewer("location", palette="Spectral") +
    theme_tree2(legend.position='right')
```
\setstretch{1.5}





The `phyloseq` class that defined in the *phyloseq* package was designed for storing microbiome data, including phylogenetic tree, associated sample data and taxonomy assignment. It can import data from popular pipelines, such as *QIIME* [@kuczynski_using_2011], *mothur* [@schloss_introducing_2009], *DADA2* [@callahan_dada2_2016] and *PyroTagger* [@kunin_pyrotagger_2010], *etc.*. The *ggtree* supports visualizing the phylogenetic tree stored in `phyloseq` object and related data can be used to annotate the tree as demonstrated in Figure \@ref(fig:phyloseq).
 
(ref:phyloseqscap) Visualizing phyloseq tree object.

(ref:phyloseqcap) **Visualizing phyloseq tree object.** Tips were colored by Phylum and corresponding abundance across different samples were visualized as joyplots and sorted according to the tree structure.

\setstretch{1.2} 
r phyloseq, fig.height=8.5, fig.width=9, message=FALSE, fig.cap="(ref:phyloseqcap)", fig.scap="(ref:phyloseqscap)", out.extra='', warning=FALSE}

library(phyloseq)
library(ggridges)
library(dplyr)
library(ggtree)

data("GlobalPatterns")
GP <- GlobalPatterns
GP <- prune_taxa(taxa_sums(GP) > 600, GP)
sample_data(GP)$human <- get_variable(GP, "SampleType") %in% 
  c("Feces", "Skin") 

mergedGP <- merge_samples(GP, "SampleType")
mergedGP <- rarefy_even_depth(mergedGP,rngseed=394582)
mergedGP <- tax_glom(mergedGP,"Order") 

melt_simple <- psmelt(mergedGP) %>% 
  filter(Abundance < 120) %>% 
  select(OTU, val=Abundance)

p <- ggtree(mergedGP) + 
  geom_tippoint(aes(color=Phylum), size=1.5)

facet_plot(p, panel="Abundance", data=melt_simple, 
           geom_joy, mapping = aes(x=val,group=label, 
                                        fill=Phylum), 
           color='grey80', lwd=.3)

\setstretch{1.5}
-->
</div>
</div>
<div id="summary-4" class="section level2">
<h2><span class="header-section-number">5.4</span> Summary</h2>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-bouckaert_beast_2014">
<p>Bouckaert, Remco, Joseph Heled, Denise Kühnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. “BEAST 2: A Software Platform for Bayesian Evolutionary Analysis.” <em>PLoS Comput Biol</em> 10 (4): e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>.</p>
</div>
<div id="ref-jombart_outbreaktools_2014">
<p>Jombart, Thibaut, David M. Aanensen, Marc Baguelin, Paul Birrell, Simon Cauchemez, Anton Camacho, Caroline Colijn, et al. 2014. “OutbreakTools: A New Platform for Disease Outbreak Analysis Using the R Software.” <em>Epidemics</em> 7 (June): 28–34. <a href="https://doi.org/10.1016/j.epidem.2014.04.003">https://doi.org/10.1016/j.epidem.2014.04.003</a>.</p>
</div>
<div id="ref-mcmurdie_phyloseq_2013">
<p>McMurdie, Paul J., and Susan Holmes. 2013. “Phyloseq: An R Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.” <em>PLoS ONE</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.</p>
</div>
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="13">
<li id="fn13"><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/ggtree.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/ggtree/inst/doc/ggtree.html</a><a href="chapter5.html#fnref13" class="footnote-back">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
