<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Annotating Tree with Silhouette Images and Sub-plots | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 8 Annotating Tree with Silhouette Images and Sub-plots | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.13 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Annotating Tree with Silhouette Images and Sub-plots | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Annotating Tree with Silhouette Images and Sub-plots | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu School of Basic Medical Sciences, Southern Medical University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter7.html"/>
<link rel="next" href="chapter9.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.3.3</b> Getting information from <em>treedata</em> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#annotate-clades"><i class="fa fa-check"></i><b>5.2.1</b> Annotate clades</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-based-on-tree-classes-defined-in-other-r-packages"><i class="fa fa-check"></i><b>5.3.2</b> Tree annotation based on tree classes defined in other R packages</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Alignning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.5</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#summary-6"><i class="fa fa-check"></i><b>7.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a><ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
<li class="chapter" data-level="9.1.4" data-path="chapter9.html"><a href="chapter9.html#obkdata"><i class="fa fa-check"></i><b>9.1.4</b> The obkData object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree utilities</a><ul>
<li class="chapter" data-level="10.1" data-path="chapter10.html"><a href="chapter10.html#facet-utilities"><i class="fa fa-check"></i><b>10.1</b> facet utilities</a><ul>
<li class="chapter" data-level="10.1.1" data-path="chapter10.html"><a href="chapter10.html#facet_widths"><i class="fa fa-check"></i><b>10.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="10.1.2" data-path="chapter10.html"><a href="chapter10.html#facet_labeller"><i class="fa fa-check"></i><b>10.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="chapter10.html"><a href="chapter10.html#layout-utilities"><i class="fa fa-check"></i><b>10.2</b> layout utilities</a></li>
<li class="chapter" data-level="10.3" data-path="chapter10.html"><a href="chapter10.html#legend-utilities"><i class="fa fa-check"></i><b>10.3</b> legend utilities</a></li>
<li class="chapter" data-level="10.4" data-path="chapter10.html"><a href="chapter10.html#scale-utilities"><i class="fa fa-check"></i><b>10.4</b> scale utilities</a><ul>
<li class="chapter" data-level="10.4.1" data-path="chapter10.html"><a href="chapter10.html#xlim_expand"><i class="fa fa-check"></i><b>10.4.1</b> xlim_expand</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Gallery of reproducible examples</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#visualizing-pairwise-nucleotide-sequence-distance-with-phylogenetic-tree"><i class="fa fa-check"></i><b>11.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#displaying-different-symbolic-points-for-bootstrap-values."><i class="fa fa-check"></i><b>11.2</b> Displaying different symbolic points for bootstrap values.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.5</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.6</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.7</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.8</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="B.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>B.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>D</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="E" data-path="facet-zoom.html"><a href="facet-zoom.html"><i class="fa fa-check"></i><b>E</b> Zoom in selected portion</a></li>
<li class="chapter" data-level="F" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>F</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="G" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>G</b> Figures and Tables</a></li>
<li class="chapter" data-level="H" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>H</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter8" class="section level1">
<h1><span class="header-section-number">Chapter 8</span> Annotating Tree with Silhouette Images and Sub-plots</h1>
<div id="ggimage" class="section level2">
<h2><span class="header-section-number">8.1</span> Annotating Tree with Images</h2>
<p>We usually use text to label taxa, <em>i.e.</em> displaying taxa names. If the text is image file name (either local or remote), <a href="http://bioconductor.org/packages/ggtree">ggtree</a> can read the image and display the actual image as label of the taxa. The <code>geom_tiplab</code> and <code>geom_nodelab</code> are capable to render silhouette image by internally called <a href="https://CRAN.R-project.org/package=ggimage">ggimage</a> package.</p>
<p>Online tools such as <code>iTOL</code> <span class="citation">(Letunic and Bork <a href="#ref-letunic_interactive_2007" role="doc-biblioref">2007</a>)</span> and <code>EvolView</code> <span class="citation">(He et al. <a href="#ref-he_evolview_2016" role="doc-biblioref">2016</a>)</span> support displaying subplots on phylogenetic tree. However only bar and pie charts are supported by these tools. Users may want to visualize node-associated data with other visualization methods, such as violin plot <span class="citation">(Grubaugh et al. <a href="#ref-grubaugh_genomic_2017" role="doc-biblioref">2017</a>)</span>, venn diagram <span class="citation">(Lott et al. <a href="#ref-lott_covenntree:_2015" role="doc-biblioref">2015</a>)</span>, sequence logo <em>etc</em>, and display them on the tree. In <a href="http://bioconductor.org/packages/ggtree">ggtree</a>, all kinds of subplots are supported as we can export all subplots to image files and use them to label corresponding nodes on the tree.</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="chapter8.html#cb188-1"></a><span class="kw">library</span>(ggimage)</span>
<span id="cb188-2"><a href="chapter8.html#cb188-2"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb188-3"><a href="chapter8.html#cb188-3"></a></span>
<span id="cb188-4"><a href="chapter8.html#cb188-4"></a>nwk &lt;-<span class="st"> &quot;((((bufonidae, dendrobatidae), ceratophryidae), (centrolenidae, leptodactylidae)), hylidae);&quot;</span></span>
<span id="cb188-5"><a href="chapter8.html#cb188-5"></a></span>
<span id="cb188-6"><a href="chapter8.html#cb188-6"></a>x =<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> nwk)</span>
<span id="cb188-7"><a href="chapter8.html#cb188-7"></a><span class="kw">ggtree</span>(x) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">7</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="ot">NA</span>, <span class="fl">6.2</span>) <span class="op">+</span></span>
<span id="cb188-8"><a href="chapter8.html#cb188-8"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">image=</span><span class="kw">paste0</span>(<span class="st">&quot;img/frogs/&quot;</span>, label, <span class="st">&#39;.jpg&#39;</span>)), </span>
<span id="cb188-9"><a href="chapter8.html#cb188-9"></a>                <span class="dt">geom=</span><span class="st">&quot;image&quot;</span>, <span class="dt">offset=</span><span class="dv">2</span>, <span class="dt">align=</span><span class="dv">2</span>, <span class="dt">size=</span>.<span class="dv">2</span>)  <span class="op">+</span><span class="st"> </span></span>
<span id="cb188-10"><a href="chapter8.html#cb188-10"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="dt">geom=</span><span class="st">&#39;label&#39;</span>, <span class="dt">offset=</span><span class="dv">1</span>, <span class="dt">hjust=</span>.<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb188-11"><a href="chapter8.html#cb188-11"></a><span class="st">    </span><span class="kw">geom_image</span>(<span class="dt">x=</span>.<span class="dv">8</span>, <span class="dt">y=</span><span class="fl">5.5</span>, <span class="dt">image=</span><span class="st">&quot;img/frogs/frog.jpg&quot;</span>, <span class="dt">size=</span>.<span class="dv">2</span>)</span></code></pre></div>
<p><img src="treedata_files/figure-html/unnamed-chunk-101-1.png" width="1700" /></p>
</div>
<div id="phylopic" class="section level2">
<h2><span class="header-section-number">8.2</span> Annotating Tree with Phylopic</h2>
<p><a href="phylopic.org">Phylopic</a> contains more than 3200 silhouettes and covers almost all life forms. <strong>ggtree</strong> also supports using phylopic<a href="references.html#fn15" class="footnote-ref" id="fnref15"><sup>15</sup></a> to annotate the tree. Most importantly, we can color the image using numerical/categorical variable.</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="chapter8.html#cb189-1"></a><span class="co">## tree &lt;- ape::read.nexus(&quot;https://raw.githubusercontent.com/rgriff23/Dissertation/master/Chapter_2/data/tree.nex&quot;)</span></span>
<span id="cb189-2"><a href="chapter8.html#cb189-2"></a>tree &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">read.nexus</span>(<span class="st">&quot;data/tree.nex&quot;</span>)</span>
<span id="cb189-3"><a href="chapter8.html#cb189-3"></a>phylopic_info &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">node =</span> <span class="kw">c</span>(<span class="dv">124</span>, <span class="dv">113</span>, <span class="dv">110</span>, <span class="dv">96</span>, <span class="dv">89</span>, <span class="dv">70</span>),</span>
<span id="cb189-4"><a href="chapter8.html#cb189-4"></a>                            <span class="dt">phylopic =</span> <span class="kw">c</span>(<span class="st">&quot;c4ac9ea6-b465-4ed0-a5de-9fe4eebd3ef9&quot;</span>,</span>
<span id="cb189-5"><a href="chapter8.html#cb189-5"></a>                                        <span class="st">&quot;5388a472-994a-48e1-86de-d988c6019e72&quot;</span>,</span>
<span id="cb189-6"><a href="chapter8.html#cb189-6"></a>                                        <span class="st">&quot;615c82cf-fabe-49f8-b571-e8117edfa212&quot;</span>,</span>
<span id="cb189-7"><a href="chapter8.html#cb189-7"></a>                                        <span class="st">&quot;63ff1b70-6b6e-4412-b4d2-b2e2fb51aa95&quot;</span>,</span>
<span id="cb189-8"><a href="chapter8.html#cb189-8"></a>                                        <span class="st">&quot;d65d5abd-df8b-4509-912c-e42f6c34861f&quot;</span>,</span>
<span id="cb189-9"><a href="chapter8.html#cb189-9"></a>                                        <span class="st">&quot;6d54acc7-9e1e-4b79-be88-4904cf58fbb5&quot;</span>),</span>
<span id="cb189-10"><a href="chapter8.html#cb189-10"></a>                            <span class="dt">species =</span> <span class="kw">c</span>(<span class="st">&quot;galagoids&quot;</span>, <span class="st">&quot;lemurs&quot;</span>, <span class="st">&quot;tarsiers&quot;</span>,</span>
<span id="cb189-11"><a href="chapter8.html#cb189-11"></a>                                        <span class="st">&quot;cebids&quot;</span>, <span class="st">&quot;hominoids&quot;</span>, <span class="st">&quot;cercopithecoids&quot;</span>))</span>
<span id="cb189-12"><a href="chapter8.html#cb189-12"></a>pg &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree)</span>
<span id="cb189-13"><a href="chapter8.html#cb189-13"></a>pg <span class="op">%&lt;+%</span><span class="st"> </span>phylopic_info <span class="op">+</span><span class="st"> </span></span>
<span id="cb189-14"><a href="chapter8.html#cb189-14"></a><span class="st">  </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">image=</span>phylopic, <span class="dt">color=</span>species, <span class="dt">subset=</span><span class="op">!</span><span class="kw">is.na</span>(species)), </span>
<span id="cb189-15"><a href="chapter8.html#cb189-15"></a>              <span class="dt">geom=</span><span class="st">&quot;phylopic&quot;</span>, <span class="dt">alpha =</span> <span class="fl">.6</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb189-16"><a href="chapter8.html#cb189-16"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.8</span>))</span></code></pre></div>
<p><img src="treedata_files/figure-html/phylopic-1.png" width="672" /></p>
<p><code>ggimage</code> package supports querying phylopic uid from scientific name, which is very handy for annotating tree with phylopic.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="chapter8.html#cb190-1"></a><span class="kw">require</span>(ggtree)</span>
<span id="cb190-2"><a href="chapter8.html#cb190-2"></a>newick =<span class="st"> &quot;((Pongo_abelii,(Gorilla_gorilla_gorilla,(Pan_paniscus,Pan_troglodytes)Pan,Homo_sapiens)Homininae)Hominidae,Nomascus_leucogenys)Hominoidea;&quot;</span></span>
<span id="cb190-3"><a href="chapter8.html#cb190-3"></a></span>
<span id="cb190-4"><a href="chapter8.html#cb190-4"></a>tree =<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text=</span>newick)</span>
<span id="cb190-5"><a href="chapter8.html#cb190-5"></a></span>
<span id="cb190-6"><a href="chapter8.html#cb190-6"></a>idlist =<span class="st"> </span>ggimage<span class="op">::</span><span class="kw">phylopic_uid</span>(tree<span class="op">$</span>tip.label)</span>
<span id="cb190-7"><a href="chapter8.html#cb190-7"></a>ids &lt;-<span class="st"> </span><span class="kw">sapply</span>(idlist, <span class="cf">function</span>(x) x[<span class="dv">1</span>,<span class="dv">1</span>])</span>
<span id="cb190-8"><a href="chapter8.html#cb190-8"></a></span>
<span id="cb190-9"><a href="chapter8.html#cb190-9"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> tree<span class="op">$</span>tip.label, <span class="dt">uid =</span> ids)</span>
<span id="cb190-10"><a href="chapter8.html#cb190-10"></a></span>
<span id="cb190-11"><a href="chapter8.html#cb190-11"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree)</span>
<span id="cb190-12"><a href="chapter8.html#cb190-12"></a>p <span class="op">%&lt;+%</span><span class="st"> </span>d <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">image=</span>uid, <span class="dt">color=</span>label), <span class="dt">geom=</span><span class="st">&quot;phylopic&quot;</span>, <span class="dt">offset=</span><span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb190-13"><a href="chapter8.html#cb190-13"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>label), <span class="dt">offset =</span> <span class="fl">.2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">7</span>)</span></code></pre></div>
<p><img src="treedata_files/figure-html/ggimage_phylopic-1.png" width="672" /></p>
</div>
<div id="subplots" class="section level2">
<h2><span class="header-section-number">8.3</span> Annotating Tree with Sub-plots</h2>
<p><code>ggtree</code> provides a function, <code>inset</code>, for adding subplots to a phylogenetic tree. The input is a tree graphic object and a named list of ggplot graphic objects (can be any kind of charts), these objects should named by node numbers. To facilitate adding bar and pie charts (e.g. summarized stats of results from ancestral reconstruction) to phylogenetic tree, <em>ggtree</em> provides <code>nodepie</code> and <code>nodebar</code> functions to create a list of pie or bar charts.</p>
<div id="annotate-with-bar-charts" class="section level3">
<h3><span class="header-section-number">8.3.1</span> Annotate with bar charts</h3>


<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="chapter8.html#cb191-1"></a><span class="kw">library</span>(phytools)</span>
<span id="cb191-2"><a href="chapter8.html#cb191-2"></a><span class="kw">data</span>(anoletree)</span>
<span id="cb191-3"><a href="chapter8.html#cb191-3"></a>x &lt;-<span class="st"> </span><span class="kw">getStates</span>(anoletree,<span class="st">&quot;tips&quot;</span>)</span>
<span id="cb191-4"><a href="chapter8.html#cb191-4"></a>tree &lt;-<span class="st"> </span>anoletree</span>
<span id="cb191-5"><a href="chapter8.html#cb191-5"></a></span>
<span id="cb191-6"><a href="chapter8.html#cb191-6"></a>cols &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">palette</span>()[<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(<span class="kw">unique</span>(x))],<span class="kw">sort</span>(<span class="kw">unique</span>(x)))</span>
<span id="cb191-7"><a href="chapter8.html#cb191-7"></a>fitER &lt;-<span class="st"> </span>ape<span class="op">::</span><span class="kw">ace</span>(x,tree,<span class="dt">model=</span><span class="st">&quot;ER&quot;</span>,<span class="dt">type=</span><span class="st">&quot;discrete&quot;</span>)</span>
<span id="cb191-8"><a href="chapter8.html#cb191-8"></a>ancstats &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(fitER<span class="op">$</span>lik.anc)</span>
<span id="cb191-9"><a href="chapter8.html#cb191-9"></a>ancstats<span class="op">$</span>node &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span>tree<span class="op">$</span>Nnode<span class="op">+</span><span class="kw">Ntip</span>(tree)</span>
<span id="cb191-10"><a href="chapter8.html#cb191-10"></a></span>
<span id="cb191-11"><a href="chapter8.html#cb191-11"></a><span class="co">## cols parameter indicate which columns store stats</span></span>
<span id="cb191-12"><a href="chapter8.html#cb191-12"></a>bars &lt;-<span class="st"> </span><span class="kw">nodebar</span>(ancstats, <span class="dt">cols=</span><span class="dv">1</span><span class="op">:</span><span class="dv">6</span>)</span>
<span id="cb191-13"><a href="chapter8.html#cb191-13"></a>bars &lt;-<span class="st"> </span><span class="kw">lapply</span>(bars, <span class="cf">function</span>(g) g<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> cols))</span>
<span id="cb191-14"><a href="chapter8.html#cb191-14"></a></span>
<span id="cb191-15"><a href="chapter8.html#cb191-15"></a>tree2 &lt;-<span class="st"> </span><span class="kw">full_join</span>(tree, <span class="kw">data.frame</span>(<span class="dt">label =</span> <span class="kw">names</span>(x), <span class="dt">stat =</span> x ), <span class="dt">by =</span> <span class="st">&#39;label&#39;</span>)</span>
<span id="cb191-16"><a href="chapter8.html#cb191-16"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree2) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span></span>
<span id="cb191-17"><a href="chapter8.html#cb191-17"></a><span class="st">    </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">color =</span> stat)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb191-18"><a href="chapter8.html#cb191-18"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values =</span> cols) <span class="op">+</span></span>
<span id="cb191-19"><a href="chapter8.html#cb191-19"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;right&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb191-20"><a href="chapter8.html#cb191-20"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">8</span>)</span>
<span id="cb191-21"><a href="chapter8.html#cb191-21"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_inset</span>(bars, <span class="dt">width =</span> <span class="fl">.08</span>, <span class="dt">height =</span> <span class="fl">.05</span>, <span class="dt">x =</span> <span class="st">&quot;branch&quot;</span>) </span></code></pre></div>
<div class="figure"><span id="fig:barInset"></span>
<img src="treedata_files/figure-html/barInset-1.png" alt="Annotate internal nodes with barplots." width="960" />
<p class="caption">
Figure 8.1: <strong>Annotate internal nodes with barplots.</strong>
</p>
</div>
<p>Users can set the color via the parameter <em>color</em>. The <em>x</em> position can be one of ‘node’ or ‘branch’ and can be adjusted by the parameter <em>hjust</em> and <em>vjust</em> for horizontal and vertical adjustment respecitvely.</p>
</div>
<div id="annotate-with-pie-charts" class="section level3">
<h3><span class="header-section-number">8.3.2</span> Annotate with pie charts</h3>
<p>Similarly, users can use <code>nodepie</code> function to generate a list of pie charts and place these charts to annotate corresponding nodes. Both <code>nodebar</code> and <code>nodepie</code> accepts parameter <em>alpha</em> to allow transparency.</p>


<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="chapter8.html#cb192-1"></a>pies &lt;-<span class="st"> </span><span class="kw">nodepie</span>(ancstats, <span class="dt">cols =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>)</span>
<span id="cb192-2"><a href="chapter8.html#cb192-2"></a>pies &lt;-<span class="st"> </span><span class="kw">lapply</span>(pies, <span class="cf">function</span>(g) g<span class="op">+</span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> cols))</span>
<span id="cb192-3"><a href="chapter8.html#cb192-3"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_inset</span>(pies, <span class="dt">width =</span> <span class="fl">.1</span>, <span class="dt">height =</span> <span class="fl">.1</span>) </span></code></pre></div>
<div class="figure"><span id="fig:pieInset"></span>
<img src="treedata_files/figure-html/pieInset-1.png" alt="Annotate internal nodes with pie plots." width="960" />
<p class="caption">
Figure 8.2: <strong>Annotate internal nodes with pie plots.</strong>
</p>
</div>
</div>
<div id="annotate-with-other-types-of-charts" class="section level3">
<h3><span class="header-section-number">8.3.3</span> Annotate with other types of charts</h3>
<p>The <code>inset</code> function accepts a list of ggplot graphic objects and these input objects are not restricted to pie or bar charts. They can be any kinds of charts and hybrid of these charts.</p>


<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="chapter8.html#cb193-1"></a>pies_and_bars &lt;-<span class="st"> </span>pies</span>
<span id="cb193-2"><a href="chapter8.html#cb193-2"></a>i &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">length</span>(pies), <span class="dv">20</span>)</span>
<span id="cb193-3"><a href="chapter8.html#cb193-3"></a>pies_and_bars[i] &lt;-<span class="st"> </span>bars[i]</span>
<span id="cb193-4"><a href="chapter8.html#cb193-4"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_inset</span>(pies_and_bars, <span class="dt">width=</span>.<span class="dv">08</span>, <span class="dt">height=</span>.<span class="dv">05</span>)</span></code></pre></div>
<div class="figure"><span id="fig:barpieInset"></span>
<img src="treedata_files/figure-html/barpieInset-1.png" alt="Annotate internal nodes with different types of subplots (bar and pie)." width="960" />
<p class="caption">
Figure 8.3: <strong>Annotate internal nodes with different types of subplots (bar and pie).</strong>
</p>
</div>
</div>
</div>
<div id="phylomoji" class="section level2">
<h2><span class="header-section-number">8.4</span> Have Fun with Phylomoji</h2>
<p><a href="https://twitter.com/hashtag/phylomoji?src=hash">#phylomoji</a> is fun and it is internally supported by <strong>ggtree</strong> since 2015<a href="references.html#fn16" class="footnote-ref" id="fnref16"><sup>16</sup></a>.</p>
<p>Here, we will use <strong>ggtree</strong> to recreate the figure:</p>
<p><a href="https://twitter.com/OxyMLZ/status/1055586178651451392"><img src="img/phylomoji.png" /></a></p>


<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="chapter8.html#cb194-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb194-2"><a href="chapter8.html#cb194-2"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb194-3"><a href="chapter8.html#cb194-3"></a></span>
<span id="cb194-4"><a href="chapter8.html#cb194-4"></a>tt =<span class="st"> &#39;((snail,mushroom),(((sunflower,evergreen_tree),leaves),green_salad));&#39;</span></span>
<span id="cb194-5"><a href="chapter8.html#cb194-5"></a>tree =<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> tt)</span>
<span id="cb194-6"><a href="chapter8.html#cb194-6"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> <span class="kw">c</span>(<span class="st">&#39;snail&#39;</span>,<span class="st">&#39;mushroom&#39;</span>, <span class="st">&#39;sunflower&#39;</span>,</span>
<span id="cb194-7"><a href="chapter8.html#cb194-7"></a>                          <span class="st">&#39;evergreen_tree&#39;</span>,<span class="st">&#39;leaves&#39;</span>, <span class="st">&#39;green_salad&#39;</span>),</span>
<span id="cb194-8"><a href="chapter8.html#cb194-8"></a>                <span class="dt">group =</span> <span class="kw">c</span>(<span class="st">&#39;animal&#39;</span>, <span class="st">&#39;fungi&#39;</span>, <span class="st">&#39;flowering plant&#39;</span>,</span>
<span id="cb194-9"><a href="chapter8.html#cb194-9"></a>                          <span class="st">&#39;conifers&#39;</span>, <span class="st">&#39;ferns&#39;</span>, <span class="st">&#39;mosses&#39;</span>))</span>
<span id="cb194-10"><a href="chapter8.html#cb194-10"></a></span>
<span id="cb194-11"><a href="chapter8.html#cb194-11"></a><span class="kw">ggtree</span>(tree, <span class="dt">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="dt">size=</span><span class="dv">1</span>, <span class="dt">color=</span><span class="st">&#39;firebrick&#39;</span>) <span class="op">%&lt;+%</span><span class="st"> </span>d <span class="op">+</span><span class="st"> </span></span>
<span id="cb194-12"><a href="chapter8.html#cb194-12"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="fl">4.5</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="fl">0.5</span>, <span class="fl">6.5</span>) <span class="op">+</span></span>
<span id="cb194-13"><a href="chapter8.html#cb194-13"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">parse=</span><span class="st">&quot;emoji&quot;</span>, <span class="dt">size=</span><span class="dv">15</span>, <span class="dt">vjust=</span>.<span class="dv">25</span>) <span class="op">+</span></span>
<span id="cb194-14"><a href="chapter8.html#cb194-14"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label =</span> group), <span class="dt">geom=</span><span class="st">&quot;label&quot;</span>, <span class="dt">x=</span><span class="fl">3.5</span>, <span class="dt">hjust=</span>.<span class="dv">5</span>)</span></code></pre></div>
<div class="figure"><span id="fig:phylomoji1"></span>
<img src="treedata_files/figure-html/phylomoji1-1.png" alt="Parsing label as emoji. Text (e.g. node or tip labels) can be parsed as emoji." width="672" />
<p class="caption">
Figure 8.4: <strong>Parsing label as emoji.</strong> Text (<em>e.g.</em> node or tip labels) can be parsed as emoji.
</p>
</div>
<p>Note that the output may depends on what emoji fonts installed in your system<a href="references.html#fn17" class="footnote-ref" id="fnref17"><sup>17</sup></a>.</p>
<p>With <a href="http://bioconductor.org/packages/ggtree">ggtree</a>, it is easy to generate phylomoji. The emoji is treated as <strong>text</strong>, like abc. We can use emojis to label taxa, clade, color and ratate emoji with any given color and angle. This functionality is internally supported by the <a href="https://CRAN.R-project.org/package=emojifont">emojifont</a> package.</p>
<div id="emoji-in-cirularfan-layout-tree" class="section level3">
<h3><span class="header-section-number">8.4.1</span> emoji in cirular/fan layout tree</h3>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="chapter8.html#cb195-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="dt">layout =</span> <span class="st">&quot;circular&quot;</span>, <span class="dt">size=</span><span class="dv">1</span>) <span class="op">+</span><span class="st">  </span></span>
<span id="cb195-2"><a href="chapter8.html#cb195-2"></a><span class="st">  </span><span class="kw">geom_tiplab2</span>(<span class="dt">parse=</span><span class="st">&quot;emoji&quot;</span>, <span class="dt">size=</span><span class="dv">10</span>, <span class="dt">vjust=</span>.<span class="dv">25</span>)</span>
<span id="cb195-3"><a href="chapter8.html#cb195-3"></a><span class="kw">print</span>(p)</span></code></pre></div>
<p><img src="treedata_files/figure-html/unnamed-chunk-102-1.png" width="576" /></p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="chapter8.html#cb196-1"></a><span class="co">## fan layout  </span></span>
<span id="cb196-2"><a href="chapter8.html#cb196-2"></a>p2 &lt;-<span class="st"> </span><span class="kw">open_tree</span>(p, <span class="dt">angle=</span><span class="dv">200</span>) </span>
<span id="cb196-3"><a href="chapter8.html#cb196-3"></a><span class="kw">print</span>(p2)</span></code></pre></div>
<p><img src="treedata_files/figure-html/unnamed-chunk-104-1.png" width="576" /></p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="chapter8.html#cb197-1"></a>p2 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rotate_tree</span>(<span class="op">-</span><span class="dv">90</span>)</span></code></pre></div>
<p><img src="treedata_files/figure-html/unnamed-chunk-106-1.png" width="336" /></p>
</div>
<div id="emoji-to-label-clades" class="section level3">
<h3><span class="header-section-number">8.4.2</span> emoji to label clades</h3>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="chapter8.html#cb198-1"></a><span class="kw">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb198-2"><a href="chapter8.html#cb198-2"></a>tr &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)</span>
<span id="cb198-3"><a href="chapter8.html#cb198-3"></a></span>
<span id="cb198-4"><a href="chapter8.html#cb198-4"></a><span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="fl">5.2</span>) <span class="op">+</span></span>
<span id="cb198-5"><a href="chapter8.html#cb198-5"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">41</span>, <span class="dt">label=</span><span class="st">&quot;chicken&quot;</span>, <span class="dt">parse=</span><span class="st">&quot;emoji&quot;</span>,</span>
<span id="cb198-6"><a href="chapter8.html#cb198-6"></a>                    <span class="dt">fontsize=</span><span class="dv">12</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>) <span class="op">+</span></span>
<span id="cb198-7"><a href="chapter8.html#cb198-7"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">53</span>, <span class="dt">label=</span><span class="st">&quot;duck&quot;</span>, <span class="dt">parse=</span><span class="st">&quot;emoji&quot;</span>,</span>
<span id="cb198-8"><a href="chapter8.html#cb198-8"></a>                    <span class="dt">fontsize=</span><span class="dv">12</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">color=</span><span class="st">&quot;steelblue&quot;</span>) <span class="op">+</span></span>
<span id="cb198-9"><a href="chapter8.html#cb198-9"></a><span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">48</span>, <span class="dt">label=</span><span class="st">&quot;family&quot;</span>, <span class="dt">parse=</span><span class="st">&quot;emoji&quot;</span>,</span>
<span id="cb198-10"><a href="chapter8.html#cb198-10"></a>                    <span class="dt">fontsize=</span><span class="dv">12</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">color=</span><span class="st">&quot;darkkhaki&quot;</span>)</span></code></pre></div>
<p><img src="treedata_files/figure-html/unnamed-chunk-107-1.png" width="576" /></p>
</div>
<div id="apple-color-emoji" class="section level3">
<h3><span class="header-section-number">8.4.3</span> Apple Color Emoji</h3>
<p>Although <code>R</code>’s graphical devices don’t support <code>AppleColorEmoji</code> font, it’s still possible to use it. We can export the <code>plot</code> to <code>svg</code> file and render it in <code>Safari</code>.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="chapter8.html#cb199-1"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb199-2"><a href="chapter8.html#cb199-2"></a>tree_text &lt;-<span class="st"> &quot;(((((cow, (whale, dolphin)), (pig2, boar)), camel), fish), seedling);&quot;</span></span>
<span id="cb199-3"><a href="chapter8.html#cb199-3"></a>x &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text=</span>tree_text)</span>
<span id="cb199-4"><a href="chapter8.html#cb199-4"></a><span class="kw">library</span>(ggimage)</span>
<span id="cb199-5"><a href="chapter8.html#cb199-5"></a><span class="kw">library</span>(gridSVG)</span>
<span id="cb199-6"><a href="chapter8.html#cb199-6"></a>p &lt;-<span class="st">  </span><span class="kw">ggtree</span>(x, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">20</span>, <span class="dt">parse=</span><span class="st">&#39;emoji&#39;</span>) <span class="op">+</span></span>
<span id="cb199-7"><a href="chapter8.html#cb199-7"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">7</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylim</span>(<span class="ot">NA</span>, <span class="fl">8.5</span>) <span class="op">+</span></span>
<span id="cb199-8"><a href="chapter8.html#cb199-8"></a><span class="st">    </span><span class="kw">geom_phylopic</span>(<span class="dt">image=</span><span class="st">&quot;79ad5f09-cf21-4c89-8e7d-0c82a00ce728&quot;</span>,</span>
<span id="cb199-9"><a href="chapter8.html#cb199-9"></a>                  <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>, <span class="dt">alpha =</span> <span class="fl">.3</span>, <span class="dt">size=</span><span class="ot">Inf</span>)</span>
<span id="cb199-10"><a href="chapter8.html#cb199-10"></a></span>
<span id="cb199-11"><a href="chapter8.html#cb199-11"></a>p</span>
<span id="cb199-12"><a href="chapter8.html#cb199-12"></a>ps =<span class="st"> </span><span class="kw">grid.export</span>(<span class="st">&quot;emoji.svg&quot;</span>, <span class="dt">addClass=</span>T)</span></code></pre></div>
<p><img src="img/apple_emoji2.png" /></p>
</div>
<div id="phylomoji-in-ascii-art" class="section level3">
<h3><span class="header-section-number">8.4.4</span> Phylomoji in ASCII art</h3>
<p>This is also possible, please refer to <a href="ascii-tree.html#ascii-tree">Appendix D</a> for details.</p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-grubaugh_genomic_2017">
<p>Grubaugh, Nathan D., Jason T. Ladner, Moritz U. G. Kraemer, Gytis Dudas, Amanda L. Tan, Karthik Gangavarapu, Michael R. Wiley, et al. 2017. “Genomic Epidemiology Reveals Multiple Introductions of Zika Virus into the United States.” <em>Nature</em> 546 (7658): 401–5. <a href="https://doi.org/10.1038/nature22400">https://doi.org/10.1038/nature22400</a>.</p>
</div>
<div id="ref-he_evolview_2016">
<p>He, Zilong, Huangkai Zhang, Shenghan Gao, Martin J. Lercher, Wei-Hua Chen, and Songnian Hu. 2016. “Evolview V2: An Online Visualization and Management Tool for Customized and Annotated Phylogenetic Trees.” <em>Nucleic Acids Research</em> 44 (W1): W236–241. <a href="https://doi.org/10.1093/nar/gkw370">https://doi.org/10.1093/nar/gkw370</a>.</p>
</div>
<div id="ref-letunic_interactive_2007">
<p>Letunic, Ivica, and Peer Bork. 2007. “Interactive Tree of Life (iTOL): An Online Tool for Phylogenetic Tree Display and Annotation.” <em>Bioinformatics</em> 23 (1): 127–28. <a href="https://doi.org/10.1093/bioinformatics/btl529">https://doi.org/10.1093/bioinformatics/btl529</a>.</p>
</div>
<div id="ref-lott_covenntree:_2015">
<p>Lott, Steffen C., Björn Voß, Wolfgang R. Hess, and Claudia Steglich. 2015. “CoVennTree: A New Method for the Comparative Analysis of Large Datasets.” <em>Frontiers in Genetics</em> 6: 43. <a href="https://doi.org/10.3389/fgene.2015.00043">https://doi.org/10.3389/fgene.2015.00043</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter7.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter9.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
