<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 ggtree for other tree-like objects | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  



<meta name="date" content="2020-05-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter8.html"/>
<link rel="next" href="chapter10.html"/>
<script src="libs/header-attrs-2.1/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i>Motivation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i>Citation</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#book-structure"><i class="fa fa-check"></i>Book structure</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#want-to-help"><i class="fa fa-check"></i>Want to help?</a></li>
</ul></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#as-treedata"><i class="fa fa-check"></i><b>1.3.3</b> Converting other tree object to <code>treedata</code> object</a></li>
<li class="chapter" data-level="1.3.4" data-path="chapter1.html"><a href="chapter1.html#get-treedata-data"><i class="fa fa-check"></i><b>1.3.4</b> Getting information from <code>treedata</code> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a>
<ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#colored-strips"><i class="fa fa-check"></i><b>5.2.1</b> Colored strips</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a>
<ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a>
<ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Aligning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#composite_plot"><i class="fa fa-check"></i><b>7.5</b> Composite plots</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.6</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.7" data-path="chapter7.html"><a href="chapter7.html#summary-6"><i class="fa fa-check"></i><b>7.7</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a>
<ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-circularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> Emoji in circular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> Emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a>
<ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
<li class="chapter" data-level="9.3" data-path="chapter9.html"><a href="chapter9.html#igraph"><i class="fa fa-check"></i><b>9.3</b> ggtree for tree graph</a></li>
<li class="chapter" data-level="9.4" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-other-tree-like-structure"><i class="fa fa-check"></i><b>9.4</b> ggtree for other tree-like structure</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree utilities</a>
<ul>
<li class="chapter" data-level="10.1" data-path="chapter10.html"><a href="chapter10.html#facet-utilities"><i class="fa fa-check"></i><b>10.1</b> facet utilities</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="chapter10.html"><a href="chapter10.html#facet_widths"><i class="fa fa-check"></i><b>10.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="10.1.2" data-path="chapter10.html"><a href="chapter10.html#facet_labeller"><i class="fa fa-check"></i><b>10.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="chapter10.html"><a href="chapter10.html#geom2"><i class="fa fa-check"></i><b>10.2</b> Geometric layers</a></li>
<li class="chapter" data-level="10.3" data-path="chapter10.html"><a href="chapter10.html#layout-utilities"><i class="fa fa-check"></i><b>10.3</b> Layout utilities</a></li>
<li class="chapter" data-level="10.4" data-path="chapter10.html"><a href="chapter10.html#legend-utilities"><i class="fa fa-check"></i><b>10.4</b> Legend utilities</a></li>
<li class="chapter" data-level="10.5" data-path="chapter10.html"><a href="chapter10.html#scale-utilities"><i class="fa fa-check"></i><b>10.5</b> Scale utilities</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="chapter10.html"><a href="chapter10.html#xlim_expand"><i class="fa fa-check"></i><b>10.5.1</b> xlim_expand</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Gallery of reproducible examples</a>
<ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#hpv58"><i class="fa fa-check"></i><b>11.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#symbolic-bootstrap"><i class="fa fa-check"></i><b>11.2</b> Displaying different symbolic points for bootstrap values.</a></li>
<li class="chapter" data-level="11.3" data-path="chapter11.html"><a href="chapter11.html#phylo-grouping"><i class="fa fa-check"></i><b>11.3</b> Highlighting different groups.</a></li>
<li class="chapter" data-level="11.4" data-path="chapter11.html"><a href="chapter11.html#genome-locus"><i class="fa fa-check"></i><b>11.4</b> Phylogenetic tree with genome locus structure</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a>
<ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a>
<ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a>
<ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a>
<ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#branch-setting"><i class="fa fa-check"></i><b>A.5</b> Branch setting</a>
<ul>
<li class="chapter" data-level="A.5.1" data-path="faq.html"><a href="faq.html#change-branch-length-of-outgroup"><i class="fa fa-check"></i><b>A.5.1</b> Change branch length of outgroup</a></li>
<li class="chapter" data-level="A.5.2" data-path="faq.html"><a href="faq.html#bind-tip"><i class="fa fa-check"></i><b>A.5.2</b> Attach a new tip to a tree</a></li>
<li class="chapter" data-level="A.5.3" data-path="faq.html"><a href="faq.html#change-colours-or-line-types-of-arbitrary-selected-branches"><i class="fa fa-check"></i><b>A.5.3</b> Change colours or line types of arbitrary selected branches</a></li>
</ul></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#different-x-axis-labels-for-different-facet-panels"><i class="fa fa-check"></i><b>A.6</b> Different x-axis labels for different facet panels</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.7</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.8</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.9" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.9</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.10" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.10</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="related-tools.html"><a href="related-tools.html"><i class="fa fa-check"></i><b>B</b> Related tools</a>
<ul>
<li class="chapter" data-level="B.1" data-path="related-tools.html"><a href="related-tools.html#rtol"><i class="fa fa-check"></i><b>B.1</b> rtol: An R interface to Open Tree API</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>C</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a>
<ul>
<li class="chapter" data-level="C.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>C.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="C.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>C.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="D" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>D</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="E" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>E</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="F" data-path="facet-zoom.html"><a href="facet-zoom.html"><i class="fa fa-check"></i><b>F</b> Zoom in selected portion</a></li>
<li class="chapter" data-level="G" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>G</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="H" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>H</b> Figures and Tables</a></li>
<li class="chapter" data-level="I" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>I</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter9" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> ggtree for other tree-like objects</h1>
<div id="ggtree-for-phylogenetic-tree-objects" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> ggtree for phylogenetic tree objects</h2>
<p>The <a href="http://bioconductor.org/packages/treeio">treeio</a> packages allows parsing evolutionary inferences from a number of software outputs and linking external data to the tree structure. It serves as an infrastructure to bring evolutionary data to the R community. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package works seamlessly with <a href="http://bioconductor.org/packages/treeio">treeio</a> to visualize tree associated data to annotate the tree. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package is general tool for tree visualization and annotation and it fits the ecosystem of R packages. Most of the S3/S4 tree objects defined by other R packages are also supported by <a href="http://bioconductor.org/packages/ggtree">ggtree</a>, including <code>phylo</code>(<a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree">session 4.2</a>), <code>multiPhylo</code> (<a href="chapter4.html#visualize-a-list-of-trees">session 4.4</a>), <code>phylo4</code>, <code>phylo4d</code>, <code>phyloseq</code> and <code>obkData</code>. With <a href="http://bioconductor.org/packages/ggtree">ggtree</a>, we are able to generate more complex tree graph which is not possible or easy to do with other packages. For example, the visualization of the <code>phyloseq</code> object in Figure <a href="chapter9.html#fig:phyloseq">9.4</a> is not supported by the <a href="http://bioconductor.org/packages/phyloseq">phyloseq</a> package. The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package also extend the possibility of linking external data to these tree object <span class="citation">(Yu et al. <a href="#ref-yu_two_2018" role="doc-biblioref">2018</a>)</span>.</p>
<div id="phylobase" class="section level3" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> The phylo4 and phylo4d objects</h3>
<p>The <code>phylo4</code> and <code>phylo4d</code> are defined in the <a href="https://CRAN.R-project.org/package=phylobase">phylobase</a> package. The <code>phylo4</code> object is a S4 version of <code>phylo</code>, while <code>phylo4d</code> extends <code>phylo4</code> with a data frame that contains trait data. The <a href="https://CRAN.R-project.org/package=phylobase">phylobase</a> package provides <code>plot</code> method, which internally calls the <code>treePlot</code> function, to display the tree with the data. However there are some restrictions of the <code>plot</code> method, it can only plot numeric values for tree-associated data as bubbles and cannot generate figure legend. <code>Phylobase</code> doesn’t implement visualization method to display categorical values. Using associated data as visual characteristics such as color, size and shape, is also not supported. Although it is possible to color the tree using associated data, it requires users to extract the data and map them to color vector manually follow by passing the color vector to the <code>plot</code> method. This is tedious and error-prone since the order of the color vector needs to be consistent with the edge list stored in the object.</p>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package supports <code>phylo4d</code> object and all the associated data stored in the <code>phylo4d</code> object can be used directly to annotate the tree (Fig. ).</p>


<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="chapter9.html#cb204-1"></a><span class="kw">library</span>(phylobase)</span>
<span id="cb204-2"><a href="chapter9.html#cb204-2"></a><span class="kw">data</span>(geospiza_raw)</span>
<span id="cb204-3"><a href="chapter9.html#cb204-3"></a>g1 &lt;-<span class="st"> </span><span class="kw">as</span>(geospiza_raw<span class="op">$</span>tree, <span class="st">&quot;phylo4&quot;</span>)</span>
<span id="cb204-4"><a href="chapter9.html#cb204-4"></a>g2 &lt;-<span class="st"> </span><span class="kw">phylo4d</span>(g1, geospiza_raw<span class="op">$</span>data, <span class="dt">missing.data=</span><span class="st">&quot;warn&quot;</span>)</span>
<span id="cb204-5"><a href="chapter9.html#cb204-5"></a></span>
<span id="cb204-6"><a href="chapter9.html#cb204-6"></a>d1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">seq</span>(<span class="fl">0.93</span>, <span class="fl">1.15</span>, <span class="dt">length.out =</span> <span class="dv">5</span>),</span>
<span id="cb204-7"><a href="chapter9.html#cb204-7"></a>                <span class="dt">lab =</span> <span class="kw">names</span>(geospiza_raw<span class="op">$</span>data))</span>
<span id="cb204-8"><a href="chapter9.html#cb204-8"></a></span>
<span id="cb204-9"><a href="chapter9.html#cb204-9"></a><span class="kw">ggtree</span>(g2) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">size =</span> wingL), <span class="dt">x =</span> d1<span class="op">$</span>x[<span class="dv">1</span>], <span class="dt">shape =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb204-10"><a href="chapter9.html#cb204-10"></a><span class="st">    </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">size =</span> tarsusL), <span class="dt">x =</span> d1<span class="op">$</span>x[<span class="dv">2</span>], <span class="dt">shape =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb204-11"><a href="chapter9.html#cb204-11"></a><span class="st">    </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">size =</span> culmenL), <span class="dt">x =</span> d1<span class="op">$</span>x[<span class="dv">3</span>], <span class="dt">shape =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb204-12"><a href="chapter9.html#cb204-12"></a><span class="st">    </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">size =</span> beakD),   <span class="dt">x =</span> d1<span class="op">$</span>x[<span class="dv">4</span>], <span class="dt">shape =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb204-13"><a href="chapter9.html#cb204-13"></a><span class="st">    </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">size =</span> gonysW),  <span class="dt">x =</span> d1<span class="op">$</span>x[<span class="dv">5</span>], <span class="dt">shape =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb204-14"><a href="chapter9.html#cb204-14"></a><span class="st">    </span><span class="kw">scale_size_continuous</span>(<span class="dt">range =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">12</span>), <span class="dt">name=</span><span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb204-15"><a href="chapter9.html#cb204-15"></a><span class="st">    </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">label =</span> lab), <span class="dt">data =</span> d1, <span class="dt">angle =</span> <span class="dv">90</span>) <span class="op">+</span></span>
<span id="cb204-16"><a href="chapter9.html#cb204-16"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="dt">offset =</span> <span class="fl">.3</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="fl">1.3</span>) <span class="op">+</span></span>
<span id="cb204-17"><a href="chapter9.html#cb204-17"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.75</span>))  </span></code></pre></div>
<div class="figure"><span id="fig:fp4d"></span>
<img src="treedata_files/figure-html/fp4d-1.png" alt="Visualizing phylo4d data using ggtree." width="90%" />
<p class="caption">
Figure 9.1: <strong>Visualizing phylo4d data using ggtree.</strong>
</p>
</div>
</div>
<div id="phylog" class="section level3" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> The phylog object</h3>
<p>The <code>phylog</code> is defined in <a href="https://CRAN.R-project.org/package=ade4">ade4</a> package. The package is designed for analyzing ecological data and provides <code>newick2phylog</code>, <code>hclust2phylog</code> and <code>taxo2phylog</code> functions to create phylogeny from Newick string, hierachical clustering result or a taxonomy. The <code>phylog</code> object is also supported by <code>ggtree</code> as demonstrated in Figure <a href="chapter9.html#fig:phylog">9.2</a>.</p>


<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="chapter9.html#cb205-1"></a><span class="kw">library</span>(ade4)</span>
<span id="cb205-2"><a href="chapter9.html#cb205-2"></a><span class="kw">data</span>(taxo.eg)</span>
<span id="cb205-3"><a href="chapter9.html#cb205-3"></a>tax &lt;-<span class="st"> </span><span class="kw">as.taxo</span>(taxo.eg[[<span class="dv">1</span>]])</span>
<span id="cb205-4"><a href="chapter9.html#cb205-4"></a><span class="kw">print</span>(tax)</span></code></pre></div>
<pre><code>##       genre famille ordre
## esp3     g1    fam1  ORD1
## esp1     g1    fam1  ORD1
## esp2     g1    fam1  ORD1
## esp4     g1    fam1  ORD1
## esp5     g1    fam1  ORD1
## esp6     g1    fam1  ORD1
## esp7     g1    fam1  ORD1
## esp8     g2    fam2  ORD2
## esp9     g3    fam2  ORD2
## esp10    g4    fam3  ORD2
## esp11    g5    fam3  ORD2
## esp12    g6    fam4  ORD2
## esp13    g7    fam4  ORD2
## esp14    g8    fam5  ORD2
## esp15    g8    fam5  ORD2</code></pre>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="chapter9.html#cb207-1"></a>tax.phy &lt;-<span class="st"> </span><span class="kw">taxo2phylog</span>(<span class="kw">as.taxo</span>(taxo.eg[[<span class="dv">1</span>]]))</span>
<span id="cb207-2"><a href="chapter9.html#cb207-2"></a><span class="kw">print</span>(tax.phy)</span></code></pre></div>
<pre><code>## Phylogenetic tree with 15 leaves and 16 nodes
## $class: phylog
## $call: taxo2phylog(taxo = as.taxo(taxo.eg[[1]]))
## $tre: ((((esp3,esp1,esp2,esp4,e...15)l1g8)l2fam5)l3ORD2)Root; 
## 
##         class   length
## $leaves numeric 15    
## $nodes  numeric 16    
## $parts  list    16    
## $paths  list    31    
## $droot  numeric 31    
##         content                                     
## $leaves length of the first preceeding adjacent edge
## $nodes  length of the first preceeding adjacent edge
## $parts  subsets of descendant nodes                 
## $paths  path from root to node or leave             
## $droot  distance to root</code></pre>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="chapter9.html#cb209-1"></a><span class="kw">ggtree</span>(tax.phy) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_nodelab</span>(<span class="dt">geom=</span><span class="st">&#39;label&#39;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:phylog"></span>
<img src="treedata_files/figure-html/phylog-1.png" alt="Visualizing phylog tree object." width="672" />
<p class="caption">
Figure 9.2: <strong>Visualizing phylog tree object.</strong>
</p>
</div>
</div>
<div id="phyloseq" class="section level3" number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> The phyloseq object</h3>
<p>The <code>phyloseq</code> class that defined in the <a href="http://bioconductor.org/packages/phyloseq">phyloseq</a> package was designed for storing microbiome data, including phylogenetic tree, associated sample data and taxonomy assignment. It can import data from popular pipelines, such as <em>QIIME</em> <span class="citation">(Kuczynski et al. <a href="#ref-kuczynski_using_2011" role="doc-biblioref">2011</a>)</span>, <em>mothur</em> <span class="citation">(Schloss et al. <a href="#ref-schloss_introducing_2009" role="doc-biblioref">2009</a>)</span>, <em>DADA2</em> <span class="citation">(Callahan et al. <a href="#ref-callahan_dada2_2016" role="doc-biblioref">2016</a>)</span> and <em>PyroTagger</em> <span class="citation">(Kunin and Hugenholtz <a href="#ref-kunin_pyrotagger_2010" role="doc-biblioref">2010</a>)</span>, <em>etc.</em>. The <em>ggtree</em> supports visualizing the phylogenetic tree stored in <code>phyloseq</code> object and related data can be used to annotate the tree as demonstrated in Figure <a href="chapter9.html#fig:reproducephyloseq">9.3</a> and <a href="chapter9.html#fig:phyloseq">9.4</a>.</p>


<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="chapter9.html#cb210-1"></a><span class="kw">library</span>(phyloseq)</span>
<span id="cb210-2"><a href="chapter9.html#cb210-2"></a><span class="kw">library</span>(scales)</span>
<span id="cb210-3"><a href="chapter9.html#cb210-3"></a></span>
<span id="cb210-4"><a href="chapter9.html#cb210-4"></a><span class="kw">data</span>(GlobalPatterns)</span>
<span id="cb210-5"><a href="chapter9.html#cb210-5"></a>GP &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(GlobalPatterns) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, GlobalPatterns)</span>
<span id="cb210-6"><a href="chapter9.html#cb210-6"></a>GP.chl &lt;-<span class="st"> </span><span class="kw">subset_taxa</span>(GP, Phylum<span class="op">==</span><span class="st">&quot;Chlamydiae&quot;</span>)</span>
<span id="cb210-7"><a href="chapter9.html#cb210-7"></a></span>
<span id="cb210-8"><a href="chapter9.html#cb210-8"></a><span class="kw">ggtree</span>(GP.chl) <span class="op">+</span><span class="st"> </span><span class="kw">geom_text2</span>(<span class="kw">aes</span>(<span class="dt">subset=</span><span class="op">!</span>isTip, <span class="dt">label=</span>label), <span class="dt">hjust=</span><span class="op">-</span>.<span class="dv">2</span>, <span class="dt">size=</span><span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb210-9"><a href="chapter9.html#cb210-9"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>Genus), <span class="dt">hjust=</span><span class="op">-</span>.<span class="dv">3</span>) <span class="op">+</span></span>
<span id="cb210-10"><a href="chapter9.html#cb210-10"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span>x<span class="op">+</span>hjust, <span class="dt">color=</span>SampleType, <span class="dt">shape=</span>Family, <span class="dt">size=</span>Abundance),<span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">+</span></span>
<span id="cb210-11"><a href="chapter9.html#cb210-11"></a><span class="st">  </span><span class="kw">scale_size_continuous</span>(<span class="dt">trans=</span><span class="kw">log_trans</span>(<span class="dv">5</span>)) <span class="op">+</span></span>
<span id="cb210-12"><a href="chapter9.html#cb210-12"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;right&quot;</span>)</span></code></pre></div>
<div class="figure"><span id="fig:reproducephyloseq"></span>
<img src="treedata_files/figure-html/reproducephyloseq-1.png" alt="Visualizing phyloseq tree object." width="768"  />
<p class="caption">
Figure 9.3: <strong>Visualizing phyloseq tree object.</strong>
</p>
</div>
<p>Figure <a href="chapter9.html#fig:reproducephyloseq">9.3</a> reproduce output of <code>phyloseq::plot_tree()</code>. Users of <a href="http://bioconductor.org/packages/phyloseq">phyloseq</a> will find <a href="http://bioconductor.org/packages/ggtree">ggtree</a> useful for visualizing microbiome data and for further annotation, since <a href="http://bioconductor.org/packages/ggtree">ggtree</a> supports high-level of annotation using grammar of graphics and can add tree data layers that are not available in <a href="http://bioconductor.org/packages/phyloseq">phyloseq</a>.</p>


<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="chapter9.html#cb211-1"></a><span class="kw">library</span>(ggridges)</span>
<span id="cb211-2"><a href="chapter9.html#cb211-2"></a></span>
<span id="cb211-3"><a href="chapter9.html#cb211-3"></a><span class="kw">data</span>(<span class="st">&quot;GlobalPatterns&quot;</span>)</span>
<span id="cb211-4"><a href="chapter9.html#cb211-4"></a>GP &lt;-<span class="st"> </span>GlobalPatterns</span>
<span id="cb211-5"><a href="chapter9.html#cb211-5"></a>GP &lt;-<span class="st"> </span><span class="kw">prune_taxa</span>(<span class="kw">taxa_sums</span>(GP) <span class="op">&gt;</span><span class="st"> </span><span class="dv">600</span>, GP)</span>
<span id="cb211-6"><a href="chapter9.html#cb211-6"></a><span class="kw">sample_data</span>(GP)<span class="op">$</span>human &lt;-<span class="st"> </span><span class="kw">get_variable</span>(GP, <span class="st">&quot;SampleType&quot;</span>) <span class="op">%in%</span><span class="st"> </span></span>
<span id="cb211-7"><a href="chapter9.html#cb211-7"></a><span class="st">  </span><span class="kw">c</span>(<span class="st">&quot;Feces&quot;</span>, <span class="st">&quot;Skin&quot;</span>) </span>
<span id="cb211-8"><a href="chapter9.html#cb211-8"></a></span>
<span id="cb211-9"><a href="chapter9.html#cb211-9"></a>mergedGP &lt;-<span class="st"> </span><span class="kw">merge_samples</span>(GP, <span class="st">&quot;SampleType&quot;</span>)</span>
<span id="cb211-10"><a href="chapter9.html#cb211-10"></a>mergedGP &lt;-<span class="st"> </span><span class="kw">rarefy_even_depth</span>(mergedGP,<span class="dt">rngseed=</span><span class="dv">394582</span>)</span>
<span id="cb211-11"><a href="chapter9.html#cb211-11"></a>mergedGP &lt;-<span class="st"> </span><span class="kw">tax_glom</span>(mergedGP,<span class="st">&quot;Order&quot;</span>) </span>
<span id="cb211-12"><a href="chapter9.html#cb211-12"></a></span>
<span id="cb211-13"><a href="chapter9.html#cb211-13"></a>melt_simple &lt;-<span class="st"> </span><span class="kw">psmelt</span>(mergedGP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb211-14"><a href="chapter9.html#cb211-14"></a><span class="st">  </span><span class="kw">filter</span>(Abundance <span class="op">&lt;</span><span class="st"> </span><span class="dv">120</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb211-15"><a href="chapter9.html#cb211-15"></a><span class="st">  </span><span class="kw">select</span>(OTU, <span class="dt">val=</span>Abundance)</span>
<span id="cb211-16"><a href="chapter9.html#cb211-16"></a></span>
<span id="cb211-17"><a href="chapter9.html#cb211-17"></a><span class="kw">ggtree</span>(mergedGP) <span class="op">+</span><span class="st"> </span></span>
<span id="cb211-18"><a href="chapter9.html#cb211-18"></a><span class="st">  </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">color=</span>Phylum), <span class="dt">size=</span><span class="fl">1.5</span>) <span class="op">+</span></span>
<span id="cb211-19"><a href="chapter9.html#cb211-19"></a><span class="st">  </span><span class="kw">geom_facet</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x=</span>val,<span class="dt">group=</span>label, </span>
<span id="cb211-20"><a href="chapter9.html#cb211-20"></a>                           <span class="dt">fill=</span>Phylum),</span>
<span id="cb211-21"><a href="chapter9.html#cb211-21"></a>            <span class="dt">data =</span> melt_simple, </span>
<span id="cb211-22"><a href="chapter9.html#cb211-22"></a>            <span class="dt">geom =</span> geom_density_ridges,</span>
<span id="cb211-23"><a href="chapter9.html#cb211-23"></a>            <span class="dt">panel=</span><span class="st">&quot;Abundance&quot;</span>,  </span>
<span id="cb211-24"><a href="chapter9.html#cb211-24"></a>            <span class="dt">color=</span><span class="st">&#39;grey80&#39;</span>, <span class="dt">lwd=</span>.<span class="dv">3</span>)</span></code></pre></div>
<div class="figure"><span id="fig:phyloseq"></span>
<img src="treedata_files/figure-html/phyloseq-1.png" alt="Phylogenetic tree with OTU abundance densities. Tips were colored by Phylum and corresponding abundance across different samples were visualized as density ridgelines and sorted according to the tree structure." width="864"  />
<p class="caption">
Figure 9.4: <strong>Phylogenetic tree with OTU abundance densities.</strong> Tips were colored by Phylum and corresponding abundance across different samples were visualized as density ridgelines and sorted according to the tree structure.
</p>
</div>
<p>This example uses microbiome data that provided in <code>phyloseq</code> package and density ridgeline is employed to visualize species abundance data. The <code>geom_facet</code> layer automatically re-arranges the abundance data according to the tree structure, visualizes the data using the specify <code>geom</code> function, <em>i.e.</em> <code>geom_density_ridges</code>, and aligns the density curves with the tree as demonstrated in Fig. . Note that data stored in the <code>phyloseq</code> object is visible to <code>ggtree</code> and can be used directly in tree visualization (<code>Phylum</code> was used to color tips and density ridgelines in this example). The source code of this example was firstly published in Supplemental File of <span class="citation">(Yu et al. <a href="#ref-yu_two_2018" role="doc-biblioref">2018</a>)</span>.</p>
<!-- 
### The obkData object {#obkdata}

The `okbData` is defined in the [OutbreakTools](https://CRAN.R-project.org/package=OutbreakTools) package to store incidence-based outbreak data, including meta data of sampling and information of infected individuals such as age and onset of symptoms. The `ggtree` supports the `obkData` object and the information can be used to annotate the tree as shown in Figure \@ref(fig:outbreaktools).

(ref:outbreaktoolsscap) Visualizing obkData tree object.

(ref:outbreaktoolscap) **Visualizing obkData tree object.** *x*-axis was scaled by timeline of the outbreak and tips were colored by location of different individuals.

\setstretch{1.2}
```r outbreaktools, fig.width=6.3, fig.height=7, fig.cap="(ref:outbreaktoolscap)", fig.scap="(ref:outbreaktoolsscap)", message=FALSE, out.extra=''}
library(OutbreakTools)
data(FluH1N1pdm2009)
attach(FluH1N1pdm2009)

x <- new("obkData", 
         individuals = individuals, 
         dna = dna,
         dna.individualID = samples$individualID, 
         dna.date = samples$date,
         trees = FluH1N1pdm2009$trees)
ggtree(x, mrsd="2009-09-30", as.Date=TRUE, right=TRUE) +
    geom_tippoint(aes(color=location), size=3, alpha=.75) +
    scale_color_brewer("location", palette="Spectral") +
    theme_tree2(legend.position='right')
```
\setstretch{1.5}

-->
</div>
</div>
<div id="dendrogram" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> ggtree for dendrograms</h2>
<p>A dendrogram is a tree diagram to display hierachical clustering and classification/regression trees.</p>
<p>We can calculate a hierachical clustering using the function <code>hclust</code></p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="chapter9.html#cb212-1"></a>hc &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">dist</span>(mtcars))</span>
<span id="cb212-2"><a href="chapter9.html#cb212-2"></a>hc</span></code></pre></div>
<pre><code>## 
## Call:
## hclust(d = dist(mtcars))
## 
## Cluster method   : complete 
## Distance         : euclidean 
## Number of objects: 32</code></pre>
<p>The <code>hclust</code> object describes the tree produced by the clustering process. It can be converted to <code>dendrogram</code> object, which stores the tree as deeply-nested lists.</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="chapter9.html#cb214-1"></a>den &lt;-<span class="st"> </span><span class="kw">as.dendrogram</span>(hc)</span>
<span id="cb214-2"><a href="chapter9.html#cb214-2"></a>den</span></code></pre></div>
<pre><code>## &#39;dendrogram&#39; with 2 branches and 32 members total, at height 425.3447</code></pre>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package supports most of the hierarchical clustering objects defined in the R community, including <code>hclust</code> and <code>dendrogram</code> as well as <code>agnes</code>, <code>diana</code> and <code>twins</code> that defined in the <a href="https://CRAN.R-project.org/package=cluster">cluster</a> package. Users can use <code>ggtree(object)</code> to display its tree structure, and user other layers and utilities to customize the graph and of course add annotation to the tree.</p>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>layout_dendrogram</code> to layout the tree top down, and <code>theme_dendrogram</code> to display tree height (similar to <code>theme_tree2</code> for phylogenetic tree) as demonstrated in Figure <a href="chapter9.html#fig:ggtreehclust">9.5</a>.</p>


<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="chapter9.html#cb216-1"></a>clus &lt;-<span class="st"> </span><span class="kw">cutree</span>(hc, <span class="dv">4</span>)</span>
<span id="cb216-2"><a href="chapter9.html#cb216-2"></a>g &lt;-<span class="st"> </span><span class="kw">split</span>(<span class="kw">names</span>(clus), clus)</span>
<span id="cb216-3"><a href="chapter9.html#cb216-3"></a></span>
<span id="cb216-4"><a href="chapter9.html#cb216-4"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(hc, <span class="dt">linetype=</span><span class="st">&#39;dashed&#39;</span>)</span>
<span id="cb216-5"><a href="chapter9.html#cb216-5"></a>clades &lt;-<span class="st"> </span><span class="kw">sapply</span>(g, <span class="cf">function</span>(n) <span class="kw">MRCA</span>(p, n))</span>
<span id="cb216-6"><a href="chapter9.html#cb216-6"></a></span>
<span id="cb216-7"><a href="chapter9.html#cb216-7"></a>p &lt;-<span class="st"> </span><span class="kw">groupClade</span>(p, clades, <span class="dt">group_name=</span><span class="st">&#39;subtree&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">aes</span>(<span class="dt">color=</span>subtree)</span>
<span id="cb216-8"><a href="chapter9.html#cb216-8"></a></span>
<span id="cb216-9"><a href="chapter9.html#cb216-9"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> <span class="kw">names</span>(clus), </span>
<span id="cb216-10"><a href="chapter9.html#cb216-10"></a>                  <span class="dt">cyl =</span> mtcars[<span class="kw">names</span>(clus), <span class="st">&quot;cyl&quot;</span>])</span>
<span id="cb216-11"><a href="chapter9.html#cb216-11"></a></span>
<span id="cb216-12"><a href="chapter9.html#cb216-12"></a>p <span class="op">%&lt;+%</span><span class="st"> </span>d <span class="op">+</span><span class="st"> </span></span>
<span id="cb216-13"><a href="chapter9.html#cb216-13"></a><span class="st">  </span><span class="kw">layout_dendrogram</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb216-14"><a href="chapter9.html#cb216-14"></a><span class="st">  </span><span class="kw">geom_tippoint</span>(<span class="dt">size=</span><span class="dv">5</span>, <span class="dt">shape=</span><span class="dv">21</span>, <span class="kw">aes</span>(<span class="dt">fill=</span><span class="kw">factor</span>(cyl), <span class="dt">x=</span>x<span class="fl">+.5</span>), <span class="dt">color=</span><span class="st">&#39;black&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb216-15"><a href="chapter9.html#cb216-15"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>cyl), <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">hjust=</span>.<span class="dv">5</span>, <span class="dt">color=</span><span class="st">&#39;black&#39;</span>) <span class="op">+</span></span>
<span id="cb216-16"><a href="chapter9.html#cb216-16"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">angle=</span><span class="dv">90</span>, <span class="dt">hjust=</span><span class="dv">1</span>, <span class="dt">offset=</span><span class="op">-</span><span class="dv">10</span>, <span class="dt">show.legend=</span>F) <span class="op">+</span><span class="st"> </span></span>
<span id="cb216-17"><a href="chapter9.html#cb216-17"></a><span class="st">  </span><span class="kw">scale_color_brewer</span>(<span class="dt">palette=</span><span class="st">&#39;Set1&#39;</span>, <span class="dt">breaks=</span><span class="dv">1</span><span class="op">:</span><span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb216-18"><a href="chapter9.html#cb216-18"></a><span class="st">  </span><span class="kw">theme_dendrogram</span>(<span class="dt">plot.margin=</span><span class="kw">margin</span>(<span class="dv">6</span>,<span class="dv">6</span>,<span class="dv">80</span>,<span class="dv">6</span>)) <span class="op">+</span></span>
<span id="cb216-19"><a href="chapter9.html#cb216-19"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">9</span>, <span class="fl">.6</span>))</span></code></pre></div>
<div class="figure"><span id="fig:ggtreehclust"></span>
<img src="treedata_files/figure-html/ggtreehclust-1.png" alt="Visualizing dendrogram." width="864" />
<p class="caption">
Figure 9.5: <strong>Visualizing dendrogram.</strong>
</p>
</div>
</div>
<div id="igraph" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> ggtree for tree graph</h2>
<p><a href="http://bioconductor.org/packages/treeio">treeio</a> supports converting tree graph (as an <code>igraph</code> object) to <code>phylo</code> object and <a href="http://bioconductor.org/packages/ggtree">ggtree</a> supports directly visualizing tree graph as demonstrated in Figure <a href="chapter9.html#fig:treeGraph">9.6</a>.</p>


<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="chapter9.html#cb217-1"></a><span class="kw">library</span>(igraph)</span>
<span id="cb217-2"><a href="chapter9.html#cb217-2"></a>g &lt;-<span class="st"> </span><span class="kw">graph.tree</span>(<span class="dv">40</span>, <span class="dv">3</span>)</span>
<span id="cb217-3"><a href="chapter9.html#cb217-3"></a>arrow_size &lt;-<span class="st"> </span><span class="kw">unit</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">3</span>), <span class="dt">times =</span> <span class="kw">c</span>(<span class="dv">27</span>, <span class="dv">13</span>)), <span class="st">&quot;mm&quot;</span>)</span>
<span id="cb217-4"><a href="chapter9.html#cb217-4"></a><span class="kw">ggtree</span>(g, <span class="dt">layout=</span><span class="st">&#39;slanted&#39;</span>, <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length=</span>arrow_size)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb217-5"><a href="chapter9.html#cb217-5"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size=</span><span class="dv">5</span>, <span class="dt">color=</span><span class="st">&#39;steelblue&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb217-6"><a href="chapter9.html#cb217-6"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="dt">hjust=</span>.<span class="dv">5</span>,<span class="dt">vjust=</span><span class="dv">2</span>) <span class="op">+</span><span class="st"> </span><span class="kw">layout_dendrogram</span>()</span></code></pre></div>
<div class="figure"><span id="fig:treeGraph"></span>
<img src="treedata_files/figure-html/treeGraph-1.png" alt="Visualizing tree graph." width="960" />
<p class="caption">
Figure 9.6: <strong>Visualizing tree graph.</strong>
</p>
</div>
</div>
<div id="ggtree-for-other-tree-like-structure" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> ggtree for other tree-like structure</h2>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package can be used to visualize any data in hierachical structure. Here, we use the GNI (Gross National Income) numbers in 2014 as an example. After preparing an edge list, that is a matrix or data frame that contains two columns indicating the relationship of parent and child nodes, we can use <code>as.phylo</code> provided by <a href="http://bioconductor.org/packages/treeio">treeio</a> to convert the edge list to a phylo object. Then it can be visualized using <a href="http://bioconductor.org/packages/ggtree">ggtree</a> with associated data. In this example, the population was used to scale the size of circle points for each country.</p>


<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="chapter9.html#cb218-1"></a><span class="kw">library</span>(treeio)</span>
<span id="cb218-2"><a href="chapter9.html#cb218-2"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb218-3"><a href="chapter9.html#cb218-3"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb218-4"><a href="chapter9.html#cb218-4"></a></span>
<span id="cb218-5"><a href="chapter9.html#cb218-5"></a><span class="kw">data</span>(<span class="st">&quot;GNI2014&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treemap&quot;</span>)</span>
<span id="cb218-6"><a href="chapter9.html#cb218-6"></a>n &lt;-<span class="st"> </span>GNI2014[, <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">1</span>)]</span>
<span id="cb218-7"><a href="chapter9.html#cb218-7"></a>n[,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw">as.character</span>(n[,<span class="dv">1</span>])</span>
<span id="cb218-8"><a href="chapter9.html#cb218-8"></a>n[,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s</span><span class="ch">\\</span><span class="st">(.*</span><span class="ch">\\</span><span class="st">)&quot;</span>, <span class="st">&quot;&quot;</span>, n[,<span class="dv">1</span>])</span>
<span id="cb218-9"><a href="chapter9.html#cb218-9"></a></span>
<span id="cb218-10"><a href="chapter9.html#cb218-10"></a>w &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="st">&quot;World&quot;</span>, <span class="kw">as.character</span>(<span class="kw">unique</span>(n[,<span class="dv">1</span>])))</span>
<span id="cb218-11"><a href="chapter9.html#cb218-11"></a></span>
<span id="cb218-12"><a href="chapter9.html#cb218-12"></a><span class="kw">colnames</span>(w) &lt;-<span class="st"> </span><span class="kw">colnames</span>(n)</span>
<span id="cb218-13"><a href="chapter9.html#cb218-13"></a>edgelist &lt;-<span class="st"> </span><span class="kw">rbind</span>(n, w)</span>
<span id="cb218-14"><a href="chapter9.html#cb218-14"></a></span>
<span id="cb218-15"><a href="chapter9.html#cb218-15"></a>y &lt;-<span class="st"> </span><span class="kw">as.phylo</span>(edgelist)</span>
<span id="cb218-16"><a href="chapter9.html#cb218-16"></a><span class="kw">ggtree</span>(y, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>) <span class="op">%&lt;+%</span><span class="st"> </span>GNI2014 <span class="op">+</span><span class="st"> </span></span>
<span id="cb218-17"><a href="chapter9.html#cb218-17"></a><span class="st">    </span><span class="kw">aes</span>(<span class="dt">color=</span>continent) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tippoint</span>(<span class="kw">aes</span>(<span class="dt">size=</span>population), <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb218-18"><a href="chapter9.html#cb218-18"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>country), <span class="dt">offset=</span>.<span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb218-19"><a href="chapter9.html#cb218-19"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">plot.margin=</span><span class="kw">margin</span>(<span class="dv">60</span>,<span class="dv">60</span>,<span class="dv">60</span>,<span class="dv">60</span>))</span></code></pre></div>
<div class="figure"><span id="fig:gni2014"></span>
<img src="treedata_files/figure-html/gni2014-1.png" alt="Visualizing data in hierachical structure." width="1152" />
<p class="caption">
Figure 9.7: <strong>Visualizing data in hierachical structure.</strong>
</p>
</div>

</div>
</div>



<h3>References</h3>
<div id="refs" class="references">
<div id="ref-callahan_dada2_2016">
<p>Callahan, Benjamin J., Paul J. McMurdie, Michael J. Rosen, Andrew W. Han, Amy Jo A. Johnson, and Susan P. Holmes. 2016. “DADA2: High-Resolution Sample Inference from Illumina Amplicon Data.” <em>Nature Methods</em> 13 (7): 581–83. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>.</p>
</div>
<div id="ref-kuczynski_using_2011">
<p>Kuczynski, Justin, Jesse Stombaugh, William Anton Walters, Antonio González, J. Gregory Caporaso, and Rob Knight. 2011. “Using QIIME to Analyze 16S rRNA Gene Sequences from Microbial Communities.” <em>Current Protocols in Bioinformatics / Editoral Board, Andreas D. Baxevanis ... [et Al.]</em> CHAPTER (December): Unit10.7. <a href="https://doi.org/10.1002/0471250953.bi1007s36">https://doi.org/10.1002/0471250953.bi1007s36</a>.</p>
</div>
<div id="ref-kunin_pyrotagger_2010">
<p>Kunin, Victor, and Philip Hugenholtz. 2010. “PyroTagger : A Fast , Accurate Pipeline for Analysis of rRNA Amplicon Pyrosequence Data.” <em>The Open Journal</em>, 1–8. <a href="http://www.theopenjournal.org/toj_articles/1">http://www.theopenjournal.org/toj_articles/1</a>.</p>
</div>
<div id="ref-schloss_introducing_2009">
<p>Schloss, Patrick D., Sarah L. Westcott, Thomas Ryabin, Justine R. Hall, Martin Hartmann, Emily B. Hollister, Ryan A. Lesniewski, et al. 2009. “Introducing Mothur: Open-Source, Platform-Independent, Community-Supported Software for Describing and Comparing Microbial Communities.” <em>Applied and Environmental Microbiology</em> 75 (23): 7537–41. <a href="https://doi.org/10.1128/AEM.01541-09">https://doi.org/10.1128/AEM.01541-09</a>.</p>
</div>
<div id="ref-yu_two_2018">
<p>Yu, Guangchuang, Tommy Tsan-Yuk Lam, Huachen Zhu, and Yi Guan. 2018. “Two Methods for Mapping and Visualizing Associated Data on Phylogeny Using Ggtree.” <em>Molecular Biology and Evolution</em> 35 (12): 3041–3. <a href="https://doi.org/10.1093/molbev/msy194">https://doi.org/10.1093/molbev/msy194</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter8.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter10.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["treedata.pdf", "treedata.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
