<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A Frequently asked questions | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="A Frequently asked questions | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta name="generator" content="bookdown 0.13 and GitBook 2.6.7" />

  <meta property="og:title" content="A Frequently asked questions | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A Frequently asked questions | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu School of Basic Medical Sciences, Southern Medical University" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter12.html"/>
<link rel="next" href="ggimage-tips.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<script src="libs/pathseg-1.0.2/pathseg.js"></script>
<script src="libs/comic-0.961/comic.min.js"></script>
<script src="libs/comicR-binding-0.2/comicR.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Part I: Tree data input, output and manipulation</b></span></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Importing Tree with Data</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#pc"><i class="fa fa-check"></i><b>1.1</b> Overview of Phylogenetic Tree Construction</a></li>
<li class="chapter" data-level="1.2" data-path="chapter1.html"><a href="chapter1.html#format"><i class="fa fa-check"></i><b>1.2</b> Phylogenetic Tree Formats</a><ul>
<li class="chapter" data-level="1.2.1" data-path="chapter1.html"><a href="chapter1.html#newick-tree-format"><i class="fa fa-check"></i><b>1.2.1</b> Newick tree format</a></li>
<li class="chapter" data-level="1.2.2" data-path="chapter1.html"><a href="chapter1.html#nexus-tree-format"><i class="fa fa-check"></i><b>1.2.2</b> NEXUS tree format</a></li>
<li class="chapter" data-level="1.2.3" data-path="chapter1.html"><a href="chapter1.html#nhxtext"><i class="fa fa-check"></i><b>1.2.3</b> New Hampshire eXtended format</a></li>
<li class="chapter" data-level="1.2.4" data-path="chapter1.html"><a href="chapter1.html#jplace-format"><i class="fa fa-check"></i><b>1.2.4</b> Jplace format</a></li>
<li class="chapter" data-level="1.2.5" data-path="chapter1.html"><a href="chapter1.html#software-outputs"><i class="fa fa-check"></i><b>1.2.5</b> Software outputs</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="chapter1.html"><a href="chapter1.html#getting-tree-data-with-treeio"><i class="fa fa-check"></i><b>1.3</b> Getting Tree Data with <em>treeio</em></a><ul>
<li class="chapter" data-level="1.3.1" data-path="chapter1.html"><a href="chapter1.html#overview-of-treeio"><i class="fa fa-check"></i><b>1.3.1</b> Overview of <em>treeio</em></a></li>
<li class="chapter" data-level="1.3.2" data-path="chapter1.html"><a href="chapter1.html#function-demonstration"><i class="fa fa-check"></i><b>1.3.2</b> Function Demonstration</a></li>
<li class="chapter" data-level="1.3.3" data-path="chapter1.html"><a href="chapter1.html#getting-information-from-treedata-object"><i class="fa fa-check"></i><b>1.3.3</b> Getting information from <em>treedata</em> object</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="chapter1.html"><a href="chapter1.html#summary"><i class="fa fa-check"></i><b>1.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Manipulating Tree with Data</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#tidytree"><i class="fa fa-check"></i><b>2.1</b> Manipulating tree data using tidy interface</a><ul>
<li class="chapter" data-level="2.1.1" data-path="chapter2.html"><a href="chapter2.html#the-phylo-object"><i class="fa fa-check"></i><b>2.1.1</b> The <code>phylo</code> object</a></li>
<li class="chapter" data-level="2.1.2" data-path="chapter2.html"><a href="chapter2.html#the-treedata-object"><i class="fa fa-check"></i><b>2.1.2</b> The <code>treedata</code> object</a></li>
<li class="chapter" data-level="2.1.3" data-path="chapter2.html"><a href="chapter2.html#accesor-tidytree"><i class="fa fa-check"></i><b>2.1.3</b> Access related nodes</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#data-integration"><i class="fa fa-check"></i><b>2.2</b> Data Integration</a><ul>
<li class="chapter" data-level="2.2.1" data-path="chapter2.html"><a href="chapter2.html#merge-tree"><i class="fa fa-check"></i><b>2.2.1</b> Combining tree data</a></li>
<li class="chapter" data-level="2.2.2" data-path="chapter2.html"><a href="chapter2.html#link-external-data"><i class="fa fa-check"></i><b>2.2.2</b> Linking external data to phylogeny</a></li>
<li class="chapter" data-level="2.2.3" data-path="chapter2.html"><a href="chapter2.html#grouping-taxa"><i class="fa fa-check"></i><b>2.2.3</b> Grouping taxa</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#rescale-treeio"><i class="fa fa-check"></i><b>2.3</b> Rescaling Tree Branches</a></li>
<li class="chapter" data-level="2.4" data-path="chapter2.html"><a href="chapter2.html#subsetting-tree-with-data"><i class="fa fa-check"></i><b>2.4</b> Subsetting Tree with Data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="chapter2.html"><a href="chapter2.html#remove-tip"><i class="fa fa-check"></i><b>2.4.1</b> Removing tips in a phylogenetic tree</a></li>
<li class="chapter" data-level="2.4.2" data-path="chapter2.html"><a href="chapter2.html#subset-tip"><i class="fa fa-check"></i><b>2.4.2</b> Subsetting tree by tip label</a></li>
<li class="chapter" data-level="2.4.3" data-path="chapter2.html"><a href="chapter2.html#subset-node"><i class="fa fa-check"></i><b>2.4.3</b> Subsetting tree by internal node number</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="chapter2.html"><a href="chapter2.html#ggtree-fortify"><i class="fa fa-check"></i><b>2.5</b> Manipulating tree data for visualization</a></li>
<li class="chapter" data-level="2.6" data-path="chapter2.html"><a href="chapter2.html#summary-1"><i class="fa fa-check"></i><b>2.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#exporting-tree-data-to-beast-nexus-format"><i class="fa fa-check"></i><b>3.2</b> Exporting Tree Data to <em>BEAST</em> Nexus Format</a><ul>
<li class="chapter" data-level="3.2.1" data-path="chapter3.html"><a href="chapter3.html#exportingconverting-software-output"><i class="fa fa-check"></i><b>3.2.1</b> Exporting/converting software output</a></li>
<li class="chapter" data-level="3.2.2" data-path="chapter3.html"><a href="chapter3.html#combining-tree-with-external-data"><i class="fa fa-check"></i><b>3.2.2</b> Combining tree with external data</a></li>
<li class="chapter" data-level="3.2.3" data-path="chapter3.html"><a href="chapter3.html#merging-tree-data-from-different-sources"><i class="fa fa-check"></i><b>3.2.3</b> Merging tree data from different sources</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#write-jtree"><i class="fa fa-check"></i><b>3.3</b> Exporting Tree Data to <em>jtree</em> Format</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#summary-2"><i class="fa fa-check"></i><b>3.4</b> Summary</a></li>
</ul></li>
<li class="part"><span><b>Part II: Tree data visualization and annotation</b></span></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#visualizing-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.2</b> Visualizing Phylogenetic Tree with <em>ggtree</em></a><ul>
<li class="chapter" data-level="4.2.1" data-path="chapter4.html"><a href="chapter4.html#basic-tree-visualization"><i class="fa fa-check"></i><b>4.2.1</b> Basic Tree Visualization</a></li>
<li class="chapter" data-level="4.2.2" data-path="chapter4.html"><a href="chapter4.html#tree-layouts"><i class="fa fa-check"></i><b>4.2.2</b> Layouts of phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#displaying-tree-components"><i class="fa fa-check"></i><b>4.3</b> Displaying Tree Components</a><ul>
<li class="chapter" data-level="4.3.1" data-path="chapter4.html"><a href="chapter4.html#geom-trescale"><i class="fa fa-check"></i><b>4.3.1</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.3.2" data-path="chapter4.html"><a href="chapter4.html#geom-nodepoint"><i class="fa fa-check"></i><b>4.3.2</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.3.3" data-path="chapter4.html"><a href="chapter4.html#displaying-labels"><i class="fa fa-check"></i><b>4.3.3</b> Displaying labels</a></li>
<li class="chapter" data-level="4.3.4" data-path="chapter4.html"><a href="chapter4.html#displaying-root-edge"><i class="fa fa-check"></i><b>4.3.4</b> Displaying root edge</a></li>
<li class="chapter" data-level="4.3.5" data-path="chapter4.html"><a href="chapter4.html#color-tree"><i class="fa fa-check"></i><b>4.3.5</b> Color tree</a></li>
<li class="chapter" data-level="4.3.6" data-path="chapter4.html"><a href="chapter4.html#rescale-tree"><i class="fa fa-check"></i><b>4.3.6</b> Rescale tree</a></li>
<li class="chapter" data-level="4.3.7" data-path="chapter4.html"><a href="chapter4.html#modify-compenents-of-a-theme"><i class="fa fa-check"></i><b>4.3.7</b> Modify compenents of a theme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.4</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#summary-3"><i class="fa fa-check"></i><b>4.5</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#visualizing-and-annotating-tree-using-grammar-of-graphics"><i class="fa fa-check"></i><b>5.1</b> Visualizing and Annotating Tree using Grammar of Graphics</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#layers-for-tree-annotation"><i class="fa fa-check"></i><b>5.2</b> Layers for Tree Annotation</a><ul>
<li class="chapter" data-level="5.2.1" data-path="chapter5.html"><a href="chapter5.html#colored-strips"><i class="fa fa-check"></i><b>5.2.1</b> Colored strips</a></li>
<li class="chapter" data-level="5.2.2" data-path="chapter5.html"><a href="chapter5.html#highlight-clades"><i class="fa fa-check"></i><b>5.2.2</b> Highlight clades</a></li>
<li class="chapter" data-level="5.2.3" data-path="chapter5.html"><a href="chapter5.html#taxa-connection"><i class="fa fa-check"></i><b>5.2.3</b> Taxa connection</a></li>
<li class="chapter" data-level="5.2.4" data-path="chapter5.html"><a href="chapter5.html#uncertainty-of-evolutionary-inference"><i class="fa fa-check"></i><b>5.2.4</b> Uncertainty of evolutionary inference</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.3</b> Tree annotation with output from evolution software</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#tree-annotation-using-data-from-evolutionary-analysis-software"><i class="fa fa-check"></i><b>5.3.1</b> Tree annotation using data from evolutionary analysis software</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#summary-4"><i class="fa fa-check"></i><b>5.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#viewing-selected-clade"><i class="fa fa-check"></i><b>6.1</b> Viewing Selected Clade</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#scale-clade"><i class="fa fa-check"></i><b>6.2</b> Scaling Selected Clade</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#collapsing-and-expanding-clade"><i class="fa fa-check"></i><b>6.3</b> Collapsing and Expanding Clade</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#group-taxa-vis"><i class="fa fa-check"></i><b>6.4</b> Grouping Taxa</a></li>
<li class="chapter" data-level="6.5" data-path="chapter6.html"><a href="chapter6.html#exploring-tree-structure"><i class="fa fa-check"></i><b>6.5</b> Exploring tree structure</a></li>
<li class="chapter" data-level="6.6" data-path="chapter6.html"><a href="chapter6.html#summary-5"><i class="fa fa-check"></i><b>6.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.1" data-path="chapter7.html"><a href="chapter7.html#attach-operator"><i class="fa fa-check"></i><b>7.1</b> Mapping data to the tree structure</a></li>
<li class="chapter" data-level="7.2" data-path="chapter7.html"><a href="chapter7.html#facet_plot"><i class="fa fa-check"></i><b>7.2</b> Alignning graph to the tree based on tree structure</a></li>
<li class="chapter" data-level="7.3" data-path="chapter7.html"><a href="chapter7.html#gheatmap"><i class="fa fa-check"></i><b>7.3</b> Visualize tree with associated matrix</a><ul>
<li class="chapter" data-level="7.3.1" data-path="chapter7.html"><a href="chapter7.html#gheatmap-ggnewscale"><i class="fa fa-check"></i><b>7.3.1</b> Visualize tree with multiple associated matrix</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="chapter7.html"><a href="chapter7.html#msaplot"><i class="fa fa-check"></i><b>7.4</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.5" data-path="chapter7.html"><a href="chapter7.html#ggtree_object"><i class="fa fa-check"></i><b>7.5</b> The <code>ggtree</code> object</a></li>
<li class="chapter" data-level="7.6" data-path="chapter7.html"><a href="chapter7.html#summary-6"><i class="fa fa-check"></i><b>7.6</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="chapter8.html"><a href="chapter8.html#ggimage"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="chapter8.html"><a href="chapter8.html#phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="chapter8.html"><a href="chapter8.html#subplots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="chapter8.html"><a href="chapter8.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="chapter8.html"><a href="chapter8.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="chapter8.html"><a href="chapter8.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="chapter8.html"><a href="chapter8.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> Emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="chapter8.html"><a href="chapter8.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> Emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="chapter8.html"><a href="chapter8.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
<li class="chapter" data-level="8.4.4" data-path="chapter8.html"><a href="chapter8.html#phylomoji-in-ascii-art"><i class="fa fa-check"></i><b>8.4.4</b> Phylomoji in ASCII art</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> ggtree for other tree-like objects</a><ul>
<li class="chapter" data-level="9.1" data-path="chapter9.html"><a href="chapter9.html#ggtree-for-phylogenetic-tree-objects"><i class="fa fa-check"></i><b>9.1</b> ggtree for phylogenetic tree objects</a><ul>
<li class="chapter" data-level="9.1.1" data-path="chapter9.html"><a href="chapter9.html#phylobase"><i class="fa fa-check"></i><b>9.1.1</b> The phylo4 and phylo4d objects</a></li>
<li class="chapter" data-level="9.1.2" data-path="chapter9.html"><a href="chapter9.html#phylog"><i class="fa fa-check"></i><b>9.1.2</b> The phylog object</a></li>
<li class="chapter" data-level="9.1.3" data-path="chapter9.html"><a href="chapter9.html#phyloseq"><i class="fa fa-check"></i><b>9.1.3</b> The phyloseq object</a></li>
<li class="chapter" data-level="9.1.4" data-path="chapter9.html"><a href="chapter9.html#obkdata"><i class="fa fa-check"></i><b>9.1.4</b> The obkData object</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="chapter9.html"><a href="chapter9.html#dendrogram"><i class="fa fa-check"></i><b>9.2</b> ggtree for dendrograms</a></li>
</ul></li>
<li class="part"><span><b>Part III: Miscellaneous topics</b></span></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> ggtree utilities</a><ul>
<li class="chapter" data-level="10.1" data-path="chapter10.html"><a href="chapter10.html#facet-utilities"><i class="fa fa-check"></i><b>10.1</b> facet utilities</a><ul>
<li class="chapter" data-level="10.1.1" data-path="chapter10.html"><a href="chapter10.html#facet_widths"><i class="fa fa-check"></i><b>10.1.1</b> facet_widths</a></li>
<li class="chapter" data-level="10.1.2" data-path="chapter10.html"><a href="chapter10.html#facet_labeller"><i class="fa fa-check"></i><b>10.1.2</b> facet_labeller</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="chapter10.html"><a href="chapter10.html#layout-utilities"><i class="fa fa-check"></i><b>10.2</b> layout utilities</a></li>
<li class="chapter" data-level="10.3" data-path="chapter10.html"><a href="chapter10.html#legend-utilities"><i class="fa fa-check"></i><b>10.3</b> legend utilities</a></li>
<li class="chapter" data-level="10.4" data-path="chapter10.html"><a href="chapter10.html#scale-utilities"><i class="fa fa-check"></i><b>10.4</b> scale utilities</a><ul>
<li class="chapter" data-level="10.4.1" data-path="chapter10.html"><a href="chapter10.html#xlim_expand"><i class="fa fa-check"></i><b>10.4.1</b> xlim_expand</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Gallery of reproducible examples</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#visualizing-pairwise-nucleotide-sequence-distance-with-phylogenetic-tree"><i class="fa fa-check"></i><b>11.1</b> Visualizing pairwise nucleotide sequence distance with phylogenetic tree</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#displaying-different-symbolic-points-for-bootstrap-values."><i class="fa fa-check"></i><b>11.2</b> Displaying different symbolic points for bootstrap values.</a></li>
<li class="chapter" data-level="11.3" data-path="chapter11.html"><a href="chapter11.html#genome-locus"><i class="fa fa-check"></i><b>11.3</b> Phylogenetic tree with genome locus structure</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> ggtree extensions</a></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>A</b> Frequently asked questions</a><ul>
<li class="chapter" data-level="A.1" data-path="faq.html"><a href="faq.html#installation"><i class="fa fa-check"></i><b>A.1</b> Installation</a></li>
<li class="chapter" data-level="A.2" data-path="faq.html"><a href="faq.html#faq-r"><i class="fa fa-check"></i><b>A.2</b> Basic R related</a><ul>
<li class="chapter" data-level="A.2.1" data-path="faq.html"><a href="faq.html#faq-local-file"><i class="fa fa-check"></i><b>A.2.1</b> Use your local file</a></li>
</ul></li>
<li class="chapter" data-level="A.3" data-path="faq.html"><a href="faq.html#faq-aes-mapping"><i class="fa fa-check"></i><b>A.3</b> Aesthetic mapping</a><ul>
<li class="chapter" data-level="A.3.1" data-path="faq.html"><a href="faq.html#faq-inherit-aes"><i class="fa fa-check"></i><b>A.3.1</b> Inherit <em>aes</em></a></li>
<li class="chapter" data-level="A.3.2" data-path="faq.html"><a href="faq.html#faq-dollar-aes"><i class="fa fa-check"></i><b>A.3.2</b> Never use <code>$</code> in aes</a></li>
</ul></li>
<li class="chapter" data-level="A.4" data-path="faq.html"><a href="faq.html#faq-text"><i class="fa fa-check"></i><b>A.4</b> Text &amp; Label</a><ul>
<li class="chapter" data-level="A.4.1" data-path="faq.html"><a href="faq.html#faq-label-truncated"><i class="fa fa-check"></i><b>A.4.1</b> Tip label truncated</a></li>
<li class="chapter" data-level="A.4.2" data-path="faq.html"><a href="faq.html#faq-modify-label"><i class="fa fa-check"></i><b>A.4.2</b> Modify (tip) labels</a></li>
<li class="chapter" data-level="A.4.3" data-path="faq.html"><a href="faq.html#faq-formatting-label"><i class="fa fa-check"></i><b>A.4.3</b> Formatting (tip) labels</a></li>
<li class="chapter" data-level="A.4.4" data-path="faq.html"><a href="faq.html#faq-ggrepel"><i class="fa fa-check"></i><b>A.4.4</b> Avoid overlapping text labels</a></li>
<li class="chapter" data-level="A.4.5" data-path="faq.html"><a href="faq.html#faq-bootstrap"><i class="fa fa-check"></i><b>A.4.5</b> Bootstrap values from newick format</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="faq.html"><a href="faq.html#faq-under-the-tree"><i class="fa fa-check"></i><b>A.5</b> Plot something behind the phylogeny</a></li>
<li class="chapter" data-level="A.6" data-path="faq.html"><a href="faq.html#faq-enlarge-center-space"><i class="fa fa-check"></i><b>A.6</b> Enlarge center space in circular/fan layout tree</a></li>
<li class="chapter" data-level="A.7" data-path="faq.html"><a href="faq.html#use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale"><i class="fa fa-check"></i><b>A.7</b> Use the most distant tip from the root as the origin of the time scale</a></li>
<li class="chapter" data-level="A.8" data-path="faq.html"><a href="faq.html#export-edit"><i class="fa fa-check"></i><b>A.8</b> Edit tree graphic details</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="ggimage-tips.html"><a href="ggimage-tips.html"><i class="fa fa-check"></i><b>B</b> Tips for using <code>ggtree</code> with <code>ggimage</code></a><ul>
<li class="chapter" data-level="B.1" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-rm-image-bg"><i class="fa fa-check"></i><b>B.1</b> Example 1: Remove background of images</a></li>
<li class="chapter" data-level="B.2" data-path="ggimage-tips.html"><a href="ggimage-tips.html#ggimage-bgimage"><i class="fa fa-check"></i><b>B.2</b> Example 2: Plot tree on a background image</a></li>
</ul></li>
<li class="chapter" data-level="C" data-path="commicR.html"><a href="commicR.html"><i class="fa fa-check"></i><b>C</b> Comic (xkcd-like) phylogenetic tree</a></li>
<li class="chapter" data-level="D" data-path="ascii-tree.html"><a href="ascii-tree.html"><i class="fa fa-check"></i><b>D</b> Print ASCII-art rooted tree</a></li>
<li class="chapter" data-level="E" data-path="facet-zoom.html"><a href="facet-zoom.html"><i class="fa fa-check"></i><b>E</b> Zoom in selected portion</a></li>
<li class="chapter" data-level="F" data-path="run-ggtree-in-jupyter-notebook.html"><a href="run-ggtree-in-jupyter-notebook.html"><i class="fa fa-check"></i><b>F</b> Run ggtree in Jupyter notebook</a></li>
<li class="chapter" data-level="G" data-path="figures-and-tables.html"><a href="figures-and-tables.html"><i class="fa fa-check"></i><b>G</b> Figures and Tables</a></li>
<li class="chapter" data-level="H" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>H</b> Session Info</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="faq" class="section level1">
<h1><span class="header-section-number">A</span> Frequently asked questions</h1>
<div id="installation" class="section level2">
<h2><span class="header-section-number">A.1</span> Installation</h2>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> is released within the Bioconductor project, you need to use <a href="https://CRAN.R-project.org/package=BiocManager">BiocManager</a> to install it.</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="faq.html#cb230-1"></a><span class="co">## you need to install BiocManager before using it</span></span>
<span id="cb230-2"><a href="faq.html#cb230-2"></a><span class="co">## install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb230-3"><a href="faq.html#cb230-3"></a><span class="kw">library</span>(BiocManager)</span>
<span id="cb230-4"><a href="faq.html#cb230-4"></a><span class="kw">install</span>(<span class="st">&quot;ggtree&quot;</span>)</span></code></pre></div>
<p>Bioconductor release is adhere to specific R version. Please make sure you are using latest version of R if you want to install the latest release of Bioconductor packages, including <code>ggtree</code>. Beware that bugs will only be fixed in current release and develop branches. If you find a bug, please follow the guide<a href="references.html#fn19" class="footnote-ref" id="fnref19"><sup>19</sup></a> to report it.</p>
</div>
<div id="faq-r" class="section level2">
<h2><span class="header-section-number">A.2</span> Basic R related</h2>
<div id="faq-local-file" class="section level3">
<h3><span class="header-section-number">A.2.1</span> Use your local file</h3>
<p>If you are new to <code>R</code> and want to use <code>ggtree</code> for tree visualization, please do
learn some basic <code>R</code> and <code>ggplot2</code>.</p>
<p>A very common issue is that users always copy-paste command without looking at
the function’s behavior. <code>system.file()</code> was used in the <code>treeio</code> and <code>ggtree</code> package documentation to find files in the packages.</p>
<pre><code>system.file                package:base                R Documentation

Find Names of R System Files

Description:

     Finds the full file names of files in packages etc.

Usage:

     system.file(..., package = &quot;base&quot;, lib.loc = NULL,
                 mustWork = FALSE)</code></pre>
<p>For users who want to use their own files, please just use relative or absolute file path (<em>e.g.</em> <code>f = "your/folder/filename"</code>).</p>
</div>
</div>
<div id="faq-aes-mapping" class="section level2">
<h2><span class="header-section-number">A.3</span> Aesthetic mapping</h2>
<div id="faq-inherit-aes" class="section level3">
<h3><span class="header-section-number">A.3.1</span> Inherit <em>aes</em></h3>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="faq.html#cb232-1"></a><span class="kw">ggtree</span>(<span class="kw">rtree</span>(<span class="dv">30</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>()</span></code></pre></div>
<p>For example, we can add symbolic points to nodes with <code>geom_point()</code> directly.
The magic here is we don’t need to map <code>x</code> and <code>y</code> position of the points by providing <code>aes(x, y)</code> to <code>geom_point()</code> since it was already mapped by <code>ggtree</code> function and it serves as a global mapping for all layers.</p>
<p>But what if we provide a <code>dataset</code> in a layer and the <code>dataset</code> doesn’t contain column of <code>x</code> and/or <code>y</code>,
the layer function also try to map <code>x</code> and <code>y</code> and also others if you map them in <code>ggtree</code> function.
As these variable is not available in your <code>dataset</code>, you will get the following error:</p>
<pre><code>Error in eval(expr, envir, enclos) : object &#39;x&#39; not found</code></pre>
<p>This can be fixed by using parameter <code>inherit.aes=FALSE</code> which will disable inheriting mapping from <code>ggtree</code> function.</p>
</div>
<div id="faq-dollar-aes" class="section level3">
<h3><span class="header-section-number">A.3.2</span> Never use <code>$</code> in aes</h3>
<p>NEVER DO THIS<a href="references.html#fn20" class="footnote-ref" id="fnref20"><sup>20</sup></a>.</p>
<p>See the explaination in the <a href="https://github.com/hadley/ggplot2-book/blob/master/layers.rmd#aesthetic-mappings-secaes">ggplot2 book 2ed</a>:</p>
<blockquote>
<p>Never refer to a variable with <code>$</code> (e.g., <code>diamonds$carat</code>) in <code>aes()</code>. This breaks containment, so that the plot no longer contains everything it needs, and causes problems if ggplot2 changes the order of the rows, as it does when facetting.</p>
</blockquote>
</div>
</div>
<div id="faq-text" class="section level2">
<h2><span class="header-section-number">A.4</span> Text &amp; Label</h2>
<div id="faq-label-truncated" class="section level3">
<h3><span class="header-section-number">A.4.1</span> Tip label truncated</h3>
<p><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> can’t auto adjust <code>xlim</code> based on added text<a href="references.html#fn21" class="footnote-ref" id="fnref21"><sup>21</sup></a>.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="faq.html#cb234-1"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb234-2"><a href="faq.html#cb234-2"></a><span class="co">## example tree from https://support.bioconductor.org/p/72398/</span></span>
<span id="cb234-3"><a href="faq.html#cb234-3"></a>tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text=</span> <span class="kw">paste</span>(<span class="st">&quot;(Organism1.006G249400.1:0.03977,(Organism2.022118m:0.01337,&quot;</span>, </span>
<span id="cb234-4"><a href="faq.html#cb234-4"></a>             <span class="st">&quot;(Organism3.J34265.1:0.00284,Organism4.G02633.1:0.00468)0.51:0.0104):0.02469);&quot;</span>))</span>
<span id="cb234-5"><a href="faq.html#cb234-5"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>()  </span></code></pre></div>
<p>In this example, the tip labels displayed on Figure <a href="faq.html#fig:truncatedTip">A.1</a>A are truncated. This is because the units are in two different spaces (data and pixel). Users can use <code>xlim</code> to allocate more spaces for tip labels (Figure <a href="faq.html#fig:truncatedTip">A.1</a>B).</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="faq.html#cb235-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="fl">0.08</span>)</span></code></pre></div>
<p>Another solution is to set <code>clip = "off"</code> to allow drawing outside of the plot panel. We may also need to set <code>plot.margin</code> to allocate more spaces for margin (Figure <a href="faq.html#fig:truncatedTip">A.1</a>C).</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="faq.html#cb236-1"></a>p <span class="op">+</span><span class="st"> </span><span class="kw">coord_cartesian</span>(<span class="dt">clip =</span> <span class="st">&#39;off&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb236-2"><a href="faq.html#cb236-2"></a><span class="st">  </span><span class="kw">theme_tree2</span>(<span class="dt">plot.margin=</span><span class="kw">margin</span>(<span class="dv">6</span>, <span class="dv">120</span>, <span class="dv">6</span>, <span class="dv">6</span>))</span></code></pre></div>


<div class="figure"><span id="fig:truncatedTip"></span>
<img src="treedata_files/figure-html/truncatedTip-1.png" alt="Allocating more spaces for truncated tip lables. Long tip lables may be truncated (A). One solution is to allocate more spaces for plot panel (B) and another solution is to allow plotting labels outside the plot panel (C)." width="1152" />
<p class="caption">
Figure A.1: <strong>Allocating more spaces for truncated tip lables.</strong> Long tip lables may be truncated (A). One solution is to allocate more spaces for plot panel (B) and another solution is to allow plotting labels outside the plot panel (C).
</p>
</div>
</div>
<div id="faq-modify-label" class="section level3">
<h3><span class="header-section-number">A.4.2</span> Modify (tip) labels</h3>
<p>If you want to modify tip labels of the tree, you can use <code>treeio::rename_taxa()</code> to rename a <code>phylo</code> or <code>treedata</code> object.</p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="faq.html#cb237-1"></a>tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> <span class="st">&quot;((A, B), (C, D));&quot;</span>)</span>
<span id="cb237-2"><a href="faq.html#cb237-2"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], </span>
<span id="cb237-3"><a href="faq.html#cb237-3"></a>                <span class="dt">label2 =</span> <span class="kw">c</span>(<span class="st">&quot;sunflower&quot;</span>, <span class="st">&quot;tree&quot;</span>, <span class="st">&quot;snail&quot;</span>, <span class="st">&quot;mushroom&quot;</span>))</span>
<span id="cb237-4"><a href="faq.html#cb237-4"></a></span>
<span id="cb237-5"><a href="faq.html#cb237-5"></a><span class="co">## rename_taxa use 1st column as key and 2nd column as value by default                </span></span>
<span id="cb237-6"><a href="faq.html#cb237-6"></a><span class="co">## rename_taxa(tree, d)</span></span>
<span id="cb237-7"><a href="faq.html#cb237-7"></a><span class="kw">rename_taxa</span>(tree, d, label, label2) <span class="op">%&gt;%</span><span class="st"> </span>write.tree</span></code></pre></div>
<pre><code>## [1] &quot;((sunflower,tree),(snail,mushroom));&quot;</code></pre>
<p>If the input tree object is a <code>treedata</code> instance, you can use <code>write.beast()</code> to export the tree with with associated data to a BEAST compatible NEXUS file.</p>
<p>Renaming phylogeny tip labels seems not be a good idea, since it may introduce problems when mapping the original sequence alignment to the tree. Personally, I recommend to store the new labels as a tip annotation in <code>treedata</code> object.</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="faq.html#cb239-1"></a>tree2 &lt;-<span class="st"> </span><span class="kw">full_join</span>(tree, d, <span class="dt">by =</span> <span class="st">&quot;label&quot;</span>)</span>
<span id="cb239-2"><a href="faq.html#cb239-2"></a>tree2</span></code></pre></div>
<pre><code>## &#39;treedata&#39; S4 object&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 4 tips and 3 internal nodes.
## 
## Tip labels:
## [1] &quot;A&quot; &quot;B&quot; &quot;C&quot; &quot;D&quot;
## 
## Rooted; no branch lengths.
## 
## with the following features available:
##  &#39;label2&#39;.</code></pre>
<p>If you just want to show different or additional information when plotting the tree, you don’t need to modify tip labels. This could be easily done via the <code>%&lt;+%</code> operator to attach the modified version of the labels and than use <code>geom_tiplab</code> to display
the modified version (Figure <a href="faq.html#fig:renameTip">A.2</a>).</p>


<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="faq.html#cb241-1"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">3</span>)</span>
<span id="cb241-2"><a href="faq.html#cb241-2"></a>p1 &lt;-<span class="st"> </span>p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>()</span>
<span id="cb241-3"><a href="faq.html#cb241-3"></a></span>
<span id="cb241-4"><a href="faq.html#cb241-4"></a><span class="co">## the following command will produce identical figure of p2</span></span>
<span id="cb241-5"><a href="faq.html#cb241-5"></a><span class="co">## ggtree(tree2) + geom_tiplab(aes(label = label2))</span></span>
<span id="cb241-6"><a href="faq.html#cb241-6"></a>p2 &lt;-<span class="st"> </span>p <span class="op">%&lt;+%</span><span class="st"> </span>d <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span>label2))</span>
<span id="cb241-7"><a href="faq.html#cb241-7"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:renameTip"></span>
<img src="treedata_files/figure-html/renameTip-1.png" alt="Alternative tip labels. Original tip lables (A) and modified version (B)." width="768" />
<p class="caption">
Figure A.2: <strong>Alternative tip labels.</strong> Original tip lables (A) and modified version (B).
</p>
</div>
</div>
<div id="faq-formatting-label" class="section level3">
<h3><span class="header-section-number">A.4.3</span> Formatting (tip) labels</h3>
<p>If you want to format labels, you need to set <code>parse=TRUE</code> in <code>geom_text</code>/<code>geom_tiplab</code> and the <code>label</code> should be string that can be parsed into expression and displayed as described in <code>?plotmath</code>.</p>
<p>For example, the tip labels contains two parts, species name and accession number and we want to display species name in <em>italic</em>, we can use command like this to format specific tip/node label (Figure <a href="faq.html#fig:formatTip">A.3</a>A):</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="faq.html#cb242-1"></a><span class="kw">set.seed</span>(<span class="dv">2019-06-24</span>)</span>
<span id="cb242-2"><a href="faq.html#cb242-2"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)</span>
<span id="cb242-3"><a href="faq.html#cb242-3"></a>p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span></span>
<span id="cb242-4"><a href="faq.html#cb242-4"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">subset=</span>node<span class="op">==</span><span class="dv">35</span>), </span>
<span id="cb242-5"><a href="faq.html#cb242-5"></a>                  <span class="dt">label=</span><span class="st">&#39;paste(italic(&quot;species name&quot;), </span></span>
<span id="cb242-6"><a href="faq.html#cb242-6"></a><span class="st">                              &quot; accession number&quot;)&#39;</span>, </span>
<span id="cb242-7"><a href="faq.html#cb242-7"></a>              <span class="dt">parse=</span>T) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">6</span>)</span></code></pre></div>
<p>Another example for formating all tip labels is demonstrated in Figure <a href="faq.html#fig:formatTip">A.3</a>B:</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="faq.html#cb243-1"></a>p2 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span></span>
<span id="cb243-2"><a href="faq.html#cb243-2"></a><span class="st">  </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">paste0</span>(<span class="st">&#39;bold(&#39;</span>, label, </span>
<span id="cb243-3"><a href="faq.html#cb243-3"></a>                  <span class="st">&#39;)~italic(&#39;</span>, node, <span class="st">&#39;)&#39;</span>)), </span>
<span id="cb243-4"><a href="faq.html#cb243-4"></a>            <span class="dt">parse=</span><span class="ot">TRUE</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="dv">0</span>, <span class="dv">5</span>)</span></code></pre></div>
<p>The <code>label</code> can be provided by a <code>data.frame</code> that contains related information
of the taxa (Figure <a href="faq.html#fig:formatTip">A.3</a>C).</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="faq.html#cb244-1"></a>tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="dt">text =</span> <span class="st">&quot;((a,(b,c)),d);&quot;</span>)</span>
<span id="cb244-2"><a href="faq.html#cb244-2"></a>genus &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Gorilla&quot;</span>, <span class="st">&quot;Pan&quot;</span>, <span class="st">&quot;Homo&quot;</span>, <span class="st">&quot;Pongo&quot;</span>)</span>
<span id="cb244-3"><a href="faq.html#cb244-3"></a>species &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;gorilla&quot;</span>, <span class="st">&quot;spp.&quot;</span>, <span class="st">&quot;sapiens&quot;</span>, <span class="st">&quot;pygmaeus&quot;</span>)</span>
<span id="cb244-4"><a href="faq.html#cb244-4"></a>geo &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Africa&quot;</span>, <span class="st">&quot;Africa&quot;</span>, <span class="st">&quot;World&quot;</span>, <span class="st">&quot;Asia&quot;</span>)</span>
<span id="cb244-5"><a href="faq.html#cb244-5"></a>d &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">label =</span> tree<span class="op">$</span>tip.label, <span class="dt">genus =</span> genus,</span>
<span id="cb244-6"><a href="faq.html#cb244-6"></a>                <span class="dt">species =</span> species, <span class="dt">geo =</span> geo)</span>
<span id="cb244-7"><a href="faq.html#cb244-7"></a></span>
<span id="cb244-8"><a href="faq.html#cb244-8"></a>p3 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">%&lt;+%</span><span class="st"> </span>d <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">6</span>) <span class="op">+</span></span>
<span id="cb244-9"><a href="faq.html#cb244-9"></a><span class="st">    </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">paste0</span>(<span class="st">&#39;italic(&#39;</span>, genus, </span>
<span id="cb244-10"><a href="faq.html#cb244-10"></a>                  <span class="st">&#39;)~bolditalic(&#39;</span>, species, <span class="st">&#39;)~&#39;</span>, geo)), </span>
<span id="cb244-11"><a href="faq.html#cb244-11"></a>                <span class="dt">parse=</span>T)</span>
<span id="cb244-12"><a href="faq.html#cb244-12"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p1, p2, p3, <span class="dt">ncol=</span><span class="dv">3</span>, <span class="dt">labels =</span> LETTERS[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])                </span></code></pre></div>


<div class="figure"><span id="fig:formatTip"></span>
<img src="treedata_files/figure-html/formatTip-1.png" alt="Formatting labels. Formatting specific tip/node label (A), all tip labels (B &amp; C)." width="1152" />
<p class="caption">
Figure A.3: <strong>Formatting labels.</strong> Formatting specific tip/node label (A), all tip labels (B &amp; C).
</p>
</div>
</div>
<div id="faq-ggrepel" class="section level3">
<h3><span class="header-section-number">A.4.4</span> Avoid overlapping text labels</h3>
<p>User can use <a href="https://cran.r-project.org/web/packages/ggrepel/">ggrepel</a> package to repel overlapping text labels<a href="references.html#fn22" class="footnote-ref" id="fnref22"><sup>22</sup></a>.
.</p>
<p>For example:</p>


<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="faq.html#cb245-1"></a><span class="kw">library</span>(ggrepel)</span>
<span id="cb245-2"><a href="faq.html#cb245-2"></a><span class="kw">library</span>(ggtree)</span>
<span id="cb245-3"><a href="faq.html#cb245-3"></a>raxml_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>, <span class="st">&quot;RAxML_bipartitionsBranchLabels.H3&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb245-4"><a href="faq.html#cb245-4"></a>raxml &lt;-<span class="st"> </span><span class="kw">read.raxml</span>(raxml_file)</span>
<span id="cb245-5"><a href="faq.html#cb245-5"></a><span class="kw">ggtree</span>(raxml) <span class="op">+</span><span class="st"> </span><span class="kw">geom_label_repel</span>(<span class="kw">aes</span>(<span class="dt">label=</span>bootstrap, <span class="dt">fill=</span>bootstrap)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb245-6"><a href="faq.html#cb245-6"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.8</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">scale_fill_viridis_c</span>()</span></code></pre></div>
<div class="figure"><span id="fig:repelTip"></span>
<img src="treedata_files/figure-html/repelTip-1.png" alt="Repel labels. Repel labels to avoid overlapping." width="1152" />
<p class="caption">
Figure A.4: <strong>Repel labels.</strong> Repel labels to avoid overlapping.
</p>
</div>
</div>
<div id="faq-bootstrap" class="section level3">
<h3><span class="header-section-number">A.4.5</span> Bootstrap values from newick format</h3>
<p>It’s quite command to store <code>bootstrap</code> value as node label in <code>newick</code> format. Visualizing node label is easy using <code>geom_text2(aes(subset = !isTip, label=label))</code>.</p>
<p>If you want to only display a subset of <code>bootstrap</code> (e.g. bootstrap &gt; 80), you can’t simply using <code>geom_text2(subset= (label &gt; 80), label=label)</code> since <code>label</code> is a character vector, which contains node label (bootstrap value) and tip label (taxa name). If we use <code>geom_text2(subset=(as.numeric(label) &gt; 80), label=label)</code>, it will also fail since <code>NAs</code> were introduced by coercion. We need to convert <code>NAs</code> to logical <code>FALSE</code>, this can be done by the following code:</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="faq.html#cb246-1"></a>nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>,<span class="st">&quot;RAxML_bipartitions.H3&quot;</span>, <span class="dt">package=</span><span class="st">&#39;ggtree&#39;</span>)</span>
<span id="cb246-2"><a href="faq.html#cb246-2"></a>tr &lt;-<span class="st"> </span><span class="kw">read.tree</span>(nwk)</span>
<span id="cb246-3"><a href="faq.html#cb246-3"></a><span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">geom_text2</span>(<span class="kw">aes</span>(<span class="dt">label=</span>label, <span class="dt">subset =</span> <span class="op">!</span><span class="kw">is.na</span>(<span class="kw">as.numeric</span>(label)) <span class="op">&amp;</span><span class="st"> </span><span class="kw">as.numeric</span>(label) <span class="op">&gt;</span><span class="st"> </span><span class="dv">80</span>))</span></code></pre></div>
<p>Another solution is converting the bootstrap value outside <code>ggtree</code>.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="faq.html#cb247-1"></a>q &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tr)</span>
<span id="cb247-2"><a href="faq.html#cb247-2"></a>d &lt;-<span class="st"> </span>q<span class="op">$</span>data</span>
<span id="cb247-3"><a href="faq.html#cb247-3"></a>d &lt;-<span class="st"> </span>d[<span class="op">!</span>d<span class="op">$</span>isTip,]</span>
<span id="cb247-4"><a href="faq.html#cb247-4"></a>d<span class="op">$</span>label &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(d<span class="op">$</span>label)</span>
<span id="cb247-5"><a href="faq.html#cb247-5"></a>d &lt;-<span class="st"> </span>d[d<span class="op">$</span>label <span class="op">&gt;</span><span class="st"> </span><span class="dv">80</span>,]</span>
<span id="cb247-6"><a href="faq.html#cb247-6"></a></span>
<span id="cb247-7"><a href="faq.html#cb247-7"></a>q <span class="op">+</span><span class="st"> </span><span class="kw">geom_text</span>(<span class="dt">data=</span>d, <span class="kw">aes</span>(<span class="dt">label=</span>label))</span></code></pre></div>
</div>
</div>
<div id="faq-under-the-tree" class="section level2">
<h2><span class="header-section-number">A.5</span> Plot something behind the phylogeny</h2>
<p>The <code>ggtree</code> function plot the tree structure and normally we add layers on top of the tree.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="faq.html#cb248-1"></a><span class="kw">set.seed</span>(<span class="dv">1982</span>)</span>
<span id="cb248-2"><a href="faq.html#cb248-2"></a>x &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">5</span>)</span>
<span id="cb248-3"><a href="faq.html#cb248-3"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(x) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dv">7</span>, <span class="dt">alpha=</span><span class="dv">1</span>)</span></code></pre></div>
<p>If we want the layers behind the tree layer, we can reverse the order of all the layers.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="faq.html#cb249-1"></a>p<span class="op">$</span>layers &lt;-<span class="st"> </span><span class="kw">rev</span>(p<span class="op">$</span>layers)</span></code></pre></div>
<p>Another solution is to use <code>ggplot()</code> instead of <code>ggtree()</code> and <code>+ geom_tree()</code> to add the layer of tree structure at the correct position of layer stack.</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="faq.html#cb250-1"></a><span class="kw">ggplot</span>(x) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dv">7</span>, <span class="dt">alpha=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tree</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>()     </span></code></pre></div>


<div class="figure"><span id="fig:treeLayerOrder"></span>
<img src="treedata_files/figure-html/treeLayerOrder-1.png" alt="Add layers behind tree structure. A layer on top of the tree structure (A). Reverse layer order of A (B). Add layer behind the tree layer (C)." width="576" />
<p class="caption">
Figure A.5: <strong>Add layers behind tree structure.</strong> A layer on top of the tree structure (A). Reverse layer order of A (B). Add layer behind the tree layer (C).
</p>
</div>
</div>
<div id="faq-enlarge-center-space" class="section level2">
<h2><span class="header-section-number">A.6</span> Enlarge center space in circular/fan layout tree</h2>
<p>This question was asked several times<a href="references.html#fn23" class="footnote-ref" id="fnref23"><sup>23</sup></a>, and a published example can be found in <a href="https://www.ncbi.nlm.nih.gov/pubmed/27605062" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/27605062</a>. Increasing percentage of center white space in circular tree is useful to avoid overlapping tip labels and to increase readibility of the tree by moving all nodes and branches further out. This can be done simply by using <code>+xlim()</code> to allocate more space, just like in Figure <a href="chapter4.html#fig:layout2">4.3</a>G, or assign a long root branch that is similar to the “Root Length” parameter in FigTree.</p>


<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="faq.html#cb251-1"></a><span class="kw">set.seed</span>(<span class="dv">1982</span>)</span>
<span id="cb251-2"><a href="faq.html#cb251-2"></a>tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)</span>
<span id="cb251-3"><a href="faq.html#cb251-3"></a><span class="kw">plot_grid</span>(</span>
<span id="cb251-4"><a href="faq.html#cb251-4"></a>  <span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">10</span>, <span class="ot">NA</span>),</span>
<span id="cb251-5"><a href="faq.html#cb251-5"></a>  <span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_rootedge</span>(<span class="dv">5</span>),</span>
<span id="cb251-6"><a href="faq.html#cb251-6"></a>  <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>, <span class="dt">ncol=</span><span class="dv">2</span>)</span>
<span id="cb251-7"><a href="faq.html#cb251-7"></a>)</span></code></pre></div>
<div class="figure"><span id="fig:circular-space"></span>
<img src="treedata_files/figure-html/circular-space-1.png" alt="Enlarge center space in circular tree. Allocate more space by xlim (A) or long root branch (B)." width="768" />
<p class="caption">
Figure A.6: <strong>Enlarge center space in circular tree.</strong> Allocate more space by <code>xlim</code> (A) or long root branch (B).
</p>
</div>
</div>
<div id="use-the-most-distant-tip-from-the-root-as-the-origin-of-the-time-scale" class="section level2">
<h2><span class="header-section-number">A.7</span> Use the most distant tip from the root as the origin of the time scale</h2>
<p>The <code>revts</code> will reverse the x-axis by setting the most recent tip to 0. We can use <code>scale_x_continuous(labels=abs)</code> to label x-axis using absolute values.</p>


<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="faq.html#cb252-1"></a>tr &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">10</span>)</span>
<span id="cb252-2"><a href="faq.html#cb252-2"></a>p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</span>
<span id="cb252-3"><a href="faq.html#cb252-3"></a>p2 &lt;-<span class="st"> </span><span class="kw">revts</span>(p) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_continuous</span>(<span class="dt">labels=</span>abs)</span>
<span id="cb252-4"><a href="faq.html#cb252-4"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(p, p2, <span class="dt">ncol=</span><span class="dv">2</span>, <span class="dt">labels=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<div class="figure"><span id="fig:distantTip"></span>
<img src="treedata_files/figure-html/distantTip-1.png" alt="Origin of the time scale. Forward: from the root to tips (A). Backward: from the most distant tip to the root (B)." width="576" />
<p class="caption">
Figure A.7: <strong>Origin of the time scale.</strong> Forward: from the root to tips (A). Backward: from the most distant tip to the root (B).
</p>
</div>
</div>
<div id="export-edit" class="section level2">
<h2><span class="header-section-number">A.8</span> Edit tree graphic details</h2>
<p>It can be hard to modify plot details for ordinary users using <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>/<a href="http://bioconductor.org/packages/ggtree">ggtree</a>. We recommend using the <a href="https://CRAN.R-project.org/package=export">export</a> package to export <a href="http://bioconductor.org/packages/ggtree">ggtree</a> output to ‘Microsoft Office’ Document and edit the tree graphic in ‘PowerPoint’.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter12.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ggimage-tips.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
