<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Phylogenetic Tree Annotation | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="phylogenetic-tree-visualization.html">
<link rel="next" href="visual-exploration-of-phylogenetic-tree.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html"><i class="fa fa-check"></i><b>1</b> Importing tree with data</a></li>
<li class="chapter" data-level="2" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html"><i class="fa fa-check"></i><b>2</b> Manipulating tree with data</a></li>
<li class="chapter" data-level="3" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a></li>
<li class="chapter" data-level="4" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#viewing-a-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.1</b> Viewing a phylogenetic tree with <em>ggtree</em></a></li>
<li class="chapter" data-level="4.2" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#layout"><i class="fa fa-check"></i><b>4.2</b> Layout</a><ul>
<li class="chapter" data-level="4.2.1" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#phylogram-and-cladogram"><i class="fa fa-check"></i><b>4.2.1</b> Phylogram and Cladogram</a></li>
<li class="chapter" data-level="4.2.2" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#time-scaled-tree"><i class="fa fa-check"></i><b>4.2.2</b> Time-scaled tree</a></li>
<li class="chapter" data-level="4.2.3" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#two-dimensional-tree"><i class="fa fa-check"></i><b>4.2.3</b> Two dimensional tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-tree-scale-evolution-distance"><i class="fa fa-check"></i><b>4.3</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.4" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-nodestips"><i class="fa fa-check"></i><b>4.4</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.5" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-labels"><i class="fa fa-check"></i><b>4.5</b> Displaying labels</a></li>
<li class="chapter" data-level="4.6" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#update-tree-view-with-a-new-tree"><i class="fa fa-check"></i><b>4.6</b> Update tree view with a new tree</a></li>
<li class="chapter" data-level="4.7" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#theme"><i class="fa fa-check"></i><b>4.7</b> Theme</a></li>
<li class="chapter" data-level="4.8" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.8</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.9" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#rescale-tree"><i class="fa fa-check"></i><b>4.9</b> Rescale tree</a></li>
<li class="chapter" data-level="4.10" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#zoom-on-a-portion-of-tree"><i class="fa fa-check"></i><b>4.10</b> Zoom on a portion of tree</a></li>
<li class="chapter" data-level="4.11" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#color-tree"><i class="fa fa-check"></i><b>4.11</b> Color tree</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#annotate-clades"><i class="fa fa-check"></i><b>5.1</b> Annotate clades</a><ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#annotate-clades-for-unrooted-tree"><i class="fa fa-check"></i><b>5.1.1</b> Annotate clades for unrooted tree</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic"><i class="fa fa-check"></i><b>5.2</b> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-clades"><i class="fa fa-check"></i><b>5.3</b> Highlight clades</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-balances"><i class="fa fa-check"></i><b>5.3.1</b> Highlight balances</a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-clades-for-unrooted-tree"><i class="fa fa-check"></i><b>5.3.2</b> Highlight clades for unrooted tree</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#taxa-connection"><i class="fa fa-check"></i><b>5.4</b> Taxa connection</a></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.5</b> Tree annotation with output from evolution software</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#internal-node-number"><i class="fa fa-check"></i><b>6.1</b> Internal node number</a></li>
<li class="chapter" data-level="6.2" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#view-clade"><i class="fa fa-check"></i><b>6.2</b> View Clade</a></li>
<li class="chapter" data-level="6.3" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#group-clades"><i class="fa fa-check"></i><b>6.3</b> Group Clades</a></li>
<li class="chapter" data-level="6.4" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#group-otus"><i class="fa fa-check"></i><b>6.4</b> Group OTUs</a></li>
<li class="chapter" data-level="6.5" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#collapse-clade"><i class="fa fa-check"></i><b>6.5</b> Collapse clade</a></li>
<li class="chapter" data-level="6.6" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#expand-collapsed-clade"><i class="fa fa-check"></i><b>6.6</b> Expand collapsed clade</a></li>
<li class="chapter" data-level="6.7" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#scale-clade"><i class="fa fa-check"></i><b>6.7</b> Scale clade</a></li>
<li class="chapter" data-level="6.8" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#rotate-clade"><i class="fa fa-check"></i><b>6.8</b> Rotate clade</a></li>
<li class="chapter" data-level="6.9" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#flip-clade"><i class="fa fa-check"></i><b>6.9</b> Flip clade</a></li>
<li class="chapter" data-level="6.10" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#open-tree"><i class="fa fa-check"></i><b>6.10</b> Open tree</a></li>
<li class="chapter" data-level="6.11" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#rotate-tree"><i class="fa fa-check"></i><b>6.11</b> Rotate tree</a></li>
<li class="chapter" data-level="6.12" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#interactive-tree-manipulation"><i class="fa fa-check"></i><b>6.12</b> Interactive tree manipulation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.0.1" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#the-operator"><i class="fa fa-check"></i><b>7.0.1</b> The <code>%&lt;+%</code> operator</a></li>
<li class="chapter" data-level="7.1" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#visualize-tree-with-associated-matrix"><i class="fa fa-check"></i><b>7.1</b> Visualize tree with associated matrix</a></li>
<li class="chapter" data-level="7.2" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#visualize-tree-with-multiple-sequence-alignment"><i class="fa fa-check"></i><b>7.2</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.3" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#plot-tree-with-associated-data"><i class="fa fa-check"></i><b>7.3</b> Plot tree with associated data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-images"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-sub-plots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#have-fun-with-phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-tree-annotation" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Phylogenetic Tree Annotation</h1>
<div id="annotate-clades" class="section level2">
<h2><span class="header-section-number">5.1</span> Annotate clades</h2>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span> implements <em><code>geom_cladelabel</code></em> layer to annotate a selected clade with a bar indicating the clade with a corresponding label.</p>
<p>The <em><code>geom_cladelabel</code></em> layer accepts a selected internal node number. To get the internal node number, please refer to <a href="treeManipulation.html#internal-node-number">Tree Manipulation</a> vignette.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">2015-12-21</span>)
tree &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">30</span>)
p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">6</span>)

p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<p>Users can set the parameter, <code>align = TRUE</code>, to align the clade label, and use the parameter, <code>offset</code>, to adjust the position.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">offset=</span>.<span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span><span class="ot">TRUE</span>, <span class="dt">offset=</span>.<span class="dv">5</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
<p>Users can change the color of the clade label via the parameter <code>color</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<p>Users can change the <code>angle</code> of the clade label text and relative position from text to bar via the parameter <code>offset.text</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">270</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>, <span class="dt">offset.text=</span>.<span class="dv">5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">45</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
<p>The size of the bar and text can be changed via the parameters <code>barsize</code> and <code>fontsize</code> respectively.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">270</span>, <span class="dt">hjust=</span><span class="st">&#39;center&#39;</span>, <span class="dt">offset.text=</span>.<span class="dv">5</span>, <span class="dt">barsize=</span><span class="fl">1.5</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">angle=</span><span class="dv">45</span>, <span class="dt">fontsize=</span><span class="dv">8</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-41-1.png" width="672" /></p>
<p>Users can also use <code>geom_label</code> to label the text.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">align=</span>T, <span class="dt">geom=</span><span class="st">&#39;label&#39;</span>, <span class="dt">fill=</span><span class="st">&#39;lightblue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<div id="annotate-clades-for-unrooted-tree" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Annotate clades for unrooted tree</h3>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>geom_clade2</code> for labeling clades of unrooted
layout trees.</p>
<pre class="sourceCode r"><code class="sourceCode r">pg &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>)
pg <span class="op">+</span><span class="st"> </span><span class="kw">geom_cladelabel2</span>(<span class="dt">node=</span><span class="dv">45</span>, <span class="dt">label=</span><span class="st">&quot;test label&quot;</span>, <span class="dt">angle=</span><span class="dv">10</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_cladelabel2</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">label=</span><span class="st">&quot;another clade&quot;</span>, <span class="dt">angle=</span><span class="dv">305</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic" class="section level2">
<h2><span class="header-section-number">5.2</span> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</h2>
<p><code>geom_cladelabel</code> is designed for labelling Monophyletic (Clade) while there are related taxa that are not form a clade. <code>ggtree</code> provides <code>geom_strip</code> to add a strip/bar to indicate the association with optional label (see <a href="https://github.com/GuangchuangYu/ggtree/issues/52">the issue</a>).</p>
<pre class="sourceCode r"><code class="sourceCode r">nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sample.nwk&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(nwk)
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_strip</span>(<span class="dv">5</span>, <span class="dv">7</span>, <span class="dt">barsize=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_strip</span>(<span class="dv">6</span>, <span class="dv">12</span>, <span class="dt">barsize=</span><span class="dv">2</span>, <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-44-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="highlight-clades" class="section level2">
<h2><span class="header-section-number">5.3</span> Highlight clades</h2>
<p><code>ggtree</code> implements <em><code>geom_hilight</code></em> layer, that accepts an internal node number and add a layer of rectangle to highlight the selected clade.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">21</span>, <span class="dt">fill=</span><span class="st">&quot;steelblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">17</span>, <span class="dt">fill=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-45-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">21</span>, <span class="dt">fill=</span><span class="st">&quot;steelblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_hilight</span>(<span class="dt">node=</span><span class="dv">23</span>, <span class="dt">fill=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-46-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Another way to highlight selected clades is setting the clades with different colors and/or line types as demonstrated in <a href="treeManipulation.html#groupclade">Tree Manipulation</a> vignette.</p>
<div id="highlight-balances" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Highlight balances</h3>
<p>In addition to <em><code>geom_hilight</code></em>, <code>ggtree</code> also implements <em><code>geom_balance</code></em>
which is designed to highlight neighboring subclades of a given internal node.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_balance</span>(<span class="dt">node=</span><span class="dv">16</span>, <span class="dt">fill=</span><span class="st">&#39;steelblue&#39;</span>, <span class="dt">color=</span><span class="st">&#39;white&#39;</span>, <span class="dt">alpha=</span><span class="fl">0.6</span>, <span class="dt">extend=</span><span class="dv">1</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_balance</span>(<span class="dt">node=</span><span class="dv">19</span>, <span class="dt">fill=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">color=</span><span class="st">&#39;white&#39;</span>, <span class="dt">alpha=</span><span class="fl">0.6</span>, <span class="dt">extend=</span><span class="dv">1</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-47-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="highlight-clades-for-unrooted-tree" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Highlight clades for unrooted tree</h3>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides <code>geom_hilight_encircle</code> to support highlight
clades for unrooted layout trees.</p>
<pre class="sourceCode r"><code class="sourceCode r">pg <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight_encircle</span>(<span class="dt">node=</span><span class="dv">45</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_hilight_encircle</span>(<span class="dt">node=</span><span class="dv">34</span>, <span class="dt">fill=</span><span class="st">&#39;darkgreen&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-48-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="taxa-connection" class="section level2">
<h2><span class="header-section-number">5.4</span> Taxa connection</h2>
<p>Some evolutionary events (e.g. reassortment, horizontal gene transfer) can be modeled by a simple tree. <code>ggtree</code> provides <code>geom_taxalink</code> layer that allows drawing straight or curved lines between any of two nodes in the tree, allow it to represent evolutionary events by connecting taxa.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_taxalink</span>(<span class="st">&#39;A&#39;</span>, <span class="st">&#39;E&#39;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_taxalink</span>(<span class="st">&#39;F&#39;</span>, <span class="st">&#39;K&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">arrow=</span>grid<span class="op">::</span><span class="kw">arrow</span>(<span class="dt">length=</span>grid<span class="op">::</span><span class="kw">unit</span>(<span class="fl">0.02</span>, <span class="st">&quot;npc&quot;</span>)))</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-49-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="tree-annotation-with-output-from-evolution-software" class="section level2">
<h2><span class="header-section-number">5.5</span> Tree annotation with output from evolution software</h2>
<p>The <a href="http://bioconductor.org/packages/treeio">treeio</a> package implemented several parser functions to parse
output from commonly used software in evolutionary biology.</p>
<p>Here, we used <a href="http://beast2.org/">BEAST</a> <span class="citation">(Bouckaert et al. <a href="#ref-bouckaert_beast_2014">2014</a>)</span> output as an
example. For details, please refer to the
<a href="https://bioconductor.org/packages/devel/bioc/vignettes/treeio/inst/doc/Importer.html">Importer</a> vignette.</p>
<pre class="sourceCode r"><code class="sourceCode r">file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/BEAST&quot;</span>, <span class="st">&quot;beast_mcc.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
beast &lt;-<span class="st"> </span><span class="kw">read.beast</span>(file)
<span class="kw">ggtree</span>(beast, <span class="kw">aes</span>(<span class="dt">color=</span>rate))  <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_range</span>(<span class="dt">range=</span><span class="st">&#39;length_0.95_HPD&#39;</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>, <span class="dt">alpha=</span>.<span class="dv">6</span>, <span class="dt">size=</span><span class="dv">2</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_nodelab</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch, <span class="dt">label=</span><span class="kw">round</span>(posterior, <span class="dv">2</span>)), <span class="dt">vjust=</span><span class="op">-</span>.<span class="dv">5</span>, <span class="dt">size=</span><span class="dv">3</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_continuous</span>(<span class="dt">low=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="dt">high=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.8</span>))</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-50-1.png" width="480" style="display: block; margin: auto;" /></p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
<div id="ref-bouckaert_beast_2014">
<p>Bouckaert, Remco, Joseph Heled, Denise Kühnert, Tim Vaughan, Chieh-Hsi Wu, Dong Xie, Marc A. Suchard, Andrew Rambaut, and Alexei J. Drummond. 2014. “BEAST 2: A Software Platform for Bayesian Evolutionary Analysis.” <em>PLoS Comput Biol</em> 10 (4): e1003537. <a href="https://doi.org/10.1371/journal.pcbi.1003537">https://doi.org/10.1371/journal.pcbi.1003537</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="phylogenetic-tree-visualization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visual-exploration-of-phylogenetic-tree.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
