<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees</title>
  <meta name="description" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Phylogenetic Tree Visualization | Data Integration, Manipulation and Visualization of Phylogenetic Trees" />
  
  
  

<meta name="author" content="Guangchuang Yu">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="exporting-tree-with-data.html">
<link rel="next" href="phylogenetic-tree-annotation.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="importing-tree-with-data.html"><a href="importing-tree-with-data.html"><i class="fa fa-check"></i><b>1</b> Importing tree with data</a></li>
<li class="chapter" data-level="2" data-path="manipulating-tree-with-data.html"><a href="manipulating-tree-with-data.html"><i class="fa fa-check"></i><b>2</b> Manipulating tree with data</a></li>
<li class="chapter" data-level="3" data-path="exporting-tree-with-data.html"><a href="exporting-tree-with-data.html"><i class="fa fa-check"></i><b>3</b> Exporting tree with data</a></li>
<li class="chapter" data-level="4" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html"><i class="fa fa-check"></i><b>4</b> Phylogenetic Tree Visualization</a><ul>
<li class="chapter" data-level="4.1" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#viewing-a-phylogenetic-tree-with-ggtree"><i class="fa fa-check"></i><b>4.1</b> Viewing a phylogenetic tree with <em>ggtree</em></a></li>
<li class="chapter" data-level="4.2" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#layout"><i class="fa fa-check"></i><b>4.2</b> Layout</a><ul>
<li class="chapter" data-level="4.2.1" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#phylogram-and-cladogram"><i class="fa fa-check"></i><b>4.2.1</b> Phylogram and Cladogram</a></li>
<li class="chapter" data-level="4.2.2" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#time-scaled-tree"><i class="fa fa-check"></i><b>4.2.2</b> Time-scaled tree</a></li>
<li class="chapter" data-level="4.2.3" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#two-dimensional-tree"><i class="fa fa-check"></i><b>4.2.3</b> Two dimensional tree</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-tree-scale-evolution-distance"><i class="fa fa-check"></i><b>4.3</b> Displaying tree scale (evolution distance)</a></li>
<li class="chapter" data-level="4.4" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-nodestips"><i class="fa fa-check"></i><b>4.4</b> Displaying nodes/tips</a></li>
<li class="chapter" data-level="4.5" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#displaying-labels"><i class="fa fa-check"></i><b>4.5</b> Displaying labels</a></li>
<li class="chapter" data-level="4.6" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#update-tree-view-with-a-new-tree"><i class="fa fa-check"></i><b>4.6</b> Update tree view with a new tree</a></li>
<li class="chapter" data-level="4.7" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#theme"><i class="fa fa-check"></i><b>4.7</b> Theme</a></li>
<li class="chapter" data-level="4.8" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#visualize-a-list-of-trees"><i class="fa fa-check"></i><b>4.8</b> Visualize a list of trees</a></li>
<li class="chapter" data-level="4.9" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#rescale-tree"><i class="fa fa-check"></i><b>4.9</b> Rescale tree</a></li>
<li class="chapter" data-level="4.10" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#zoom-on-a-portion-of-tree"><i class="fa fa-check"></i><b>4.10</b> Zoom on a portion of tree</a></li>
<li class="chapter" data-level="4.11" data-path="phylogenetic-tree-visualization.html"><a href="phylogenetic-tree-visualization.html#color-tree"><i class="fa fa-check"></i><b>4.11</b> Color tree</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html"><i class="fa fa-check"></i><b>5</b> Phylogenetic Tree Annotation</a><ul>
<li class="chapter" data-level="5.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#annotate-clades"><i class="fa fa-check"></i><b>5.1</b> Annotate clades</a><ul>
<li class="chapter" data-level="5.1.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#annotate-clades-for-unrooted-tree"><i class="fa fa-check"></i><b>5.1.1</b> Annotate clades for unrooted tree</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#labelling-associated-taxa-monophyletic-polyphyletic-or-paraphyletic"><i class="fa fa-check"></i><b>5.2</b> Labelling associated taxa (Monophyletic, Polyphyletic or Paraphyletic)</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-clades"><i class="fa fa-check"></i><b>5.3</b> Highlight clades</a><ul>
<li class="chapter" data-level="5.3.1" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-balances"><i class="fa fa-check"></i><b>5.3.1</b> Highlight balances</a></li>
<li class="chapter" data-level="5.3.2" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#highlight-clades-for-unrooted-tree"><i class="fa fa-check"></i><b>5.3.2</b> Highlight clades for unrooted tree</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#taxa-connection"><i class="fa fa-check"></i><b>5.4</b> Taxa connection</a></li>
<li class="chapter" data-level="5.5" data-path="phylogenetic-tree-annotation.html"><a href="phylogenetic-tree-annotation.html#tree-annotation-with-output-from-evolution-software"><i class="fa fa-check"></i><b>5.5</b> Tree annotation with output from evolution software</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html"><i class="fa fa-check"></i><b>6</b> Visual Exploration of Phylogenetic Tree</a><ul>
<li class="chapter" data-level="6.1" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#internal-node-number"><i class="fa fa-check"></i><b>6.1</b> Internal node number</a></li>
<li class="chapter" data-level="6.2" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#view-clade"><i class="fa fa-check"></i><b>6.2</b> View Clade</a></li>
<li class="chapter" data-level="6.3" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#group-clades"><i class="fa fa-check"></i><b>6.3</b> Group Clades</a></li>
<li class="chapter" data-level="6.4" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#group-otus"><i class="fa fa-check"></i><b>6.4</b> Group OTUs</a></li>
<li class="chapter" data-level="6.5" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#collapse-clade"><i class="fa fa-check"></i><b>6.5</b> Collapse clade</a></li>
<li class="chapter" data-level="6.6" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#expand-collapsed-clade"><i class="fa fa-check"></i><b>6.6</b> Expand collapsed clade</a></li>
<li class="chapter" data-level="6.7" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#scale-clade"><i class="fa fa-check"></i><b>6.7</b> Scale clade</a></li>
<li class="chapter" data-level="6.8" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#rotate-clade"><i class="fa fa-check"></i><b>6.8</b> Rotate clade</a></li>
<li class="chapter" data-level="6.9" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#flip-clade"><i class="fa fa-check"></i><b>6.9</b> Flip clade</a></li>
<li class="chapter" data-level="6.10" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#open-tree"><i class="fa fa-check"></i><b>6.10</b> Open tree</a></li>
<li class="chapter" data-level="6.11" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#rotate-tree"><i class="fa fa-check"></i><b>6.11</b> Rotate tree</a></li>
<li class="chapter" data-level="6.12" data-path="visual-exploration-of-phylogenetic-tree.html"><a href="visual-exploration-of-phylogenetic-tree.html#interactive-tree-manipulation"><i class="fa fa-check"></i><b>6.12</b> Interactive tree manipulation</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html"><i class="fa fa-check"></i><b>7</b> Plotting tree with data</a><ul>
<li class="chapter" data-level="7.0.1" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#the-operator"><i class="fa fa-check"></i><b>7.0.1</b> The <code>%&lt;+%</code> operator</a></li>
<li class="chapter" data-level="7.1" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#visualize-tree-with-associated-matrix"><i class="fa fa-check"></i><b>7.1</b> Visualize tree with associated matrix</a></li>
<li class="chapter" data-level="7.2" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#visualize-tree-with-multiple-sequence-alignment"><i class="fa fa-check"></i><b>7.2</b> Visualize tree with multiple sequence alignment</a></li>
<li class="chapter" data-level="7.3" data-path="plotting-tree-with-data.html"><a href="plotting-tree-with-data.html#plot-tree-with-associated-data"><i class="fa fa-check"></i><b>7.3</b> Plot tree with associated data</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html"><i class="fa fa-check"></i><b>8</b> Annotating Tree with Silhouette Images and Sub-plots</a><ul>
<li class="chapter" data-level="8.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-images"><i class="fa fa-check"></i><b>8.1</b> Annotating Tree with Images</a></li>
<li class="chapter" data-level="8.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-phylopic"><i class="fa fa-check"></i><b>8.2</b> Annotating Tree with Phylopic</a></li>
<li class="chapter" data-level="8.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotating-tree-with-sub-plots"><i class="fa fa-check"></i><b>8.3</b> Annotating Tree with Sub-plots</a><ul>
<li class="chapter" data-level="8.3.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-bar-charts"><i class="fa fa-check"></i><b>8.3.1</b> Annotate with bar charts</a></li>
<li class="chapter" data-level="8.3.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-pie-charts"><i class="fa fa-check"></i><b>8.3.2</b> Annotate with pie charts</a></li>
<li class="chapter" data-level="8.3.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#annotate-with-other-types-of-charts"><i class="fa fa-check"></i><b>8.3.3</b> Annotate with other types of charts</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#have-fun-with-phylomoji"><i class="fa fa-check"></i><b>8.4</b> Have Fun with Phylomoji</a><ul>
<li class="chapter" data-level="8.4.1" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#emoji-in-cirularfan-layout-tree"><i class="fa fa-check"></i><b>8.4.1</b> emoji in cirular/fan layout tree</a></li>
<li class="chapter" data-level="8.4.2" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#emoji-to-label-clades"><i class="fa fa-check"></i><b>8.4.2</b> emoji to label clades</a></li>
<li class="chapter" data-level="8.4.3" data-path="annotating-tree-with-silhouette-images-and-sub-plots.html"><a href="annotating-tree-with-silhouette-images-and-sub-plots.html#apple-color-emoji"><i class="fa fa-check"></i><b>8.4.3</b> Apple Color Emoji</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Integration, Manipulation and Visualization of Phylogenetic Trees</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="phylogenetic-tree-visualization" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Phylogenetic Tree Visualization</h1>
<p>To view a phylogenetic tree, we first need to parse the tree file into <em>R</em>. The
<a href="http://bioconductor.org/packages/ggtree">ggtree</a> <span class="citation">(Yu et al. <a href="#ref-yu_ggtree:_2017">2017</a>)</span> package supports many file formats via the <a href="http://bioconductor.org/packages/treeio">treeio</a> package, including output files of commonly used software
packages in evolutionary biology. For more details, plase refer to
the <a href="https://bioconductor.org/packages/devel/bioc/vignettes/treeio/inst/doc/Importer.html">treeio vignette</a>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;treeio&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;ggtree&quot;</span>)

nwk &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;sample.nwk&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(nwk)</code></pre>
<div id="viewing-a-phylogenetic-tree-with-ggtree" class="section level2">
<h2><span class="header-section-number">4.1</span> Viewing a phylogenetic tree with <em>ggtree</em></h2>
<p>The <a href="http://bioconductor.org/packages/ggtree">ggtree</a> package extends <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a> <span class="citation">(Wickham <a href="#ref-wickham_ggplot2_2009">2009</a>)</span> package to support viewing phylogenetic tree.
It implements <code>geom_tree</code> layer for displaying phylogenetic tree, as shown below:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(tree, <span class="kw">aes</span>(x, y)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tree</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-3-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>The function, <code>ggtree</code>, was implemented as a short cut to visualize a tree, and it works exactly the same as shown above.</p>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> takes all the advantages of <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>. For example, we can change the color, size and type of the lines as we do with <a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">color=</span><span class="st">&quot;firebrick&quot;</span>, <span class="dt">size=</span><span class="dv">1</span>, <span class="dt">linetype=</span><span class="st">&quot;dotted&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-4-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>By default, the tree is viewed in ladderize form, user can set the parameter <em>ladderize = FALSE</em> to disable it.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">ladderize=</span><span class="ot">FALSE</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-5-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>The <em>branch.length</em> is used to scale the edge, user can set the parameter <em>branch.length = “none”</em> to only view the tree topology (cladogram) or other numerical variable to scale the tree (<em>e.g.</em> <em>d<sub>N</sub>/d<sub>S</sub></em>, see also in <a href="treeAnnotation.html">Tree Annotation</a> vignette).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">branch.length=</span><span class="st">&quot;none&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-6-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="layout" class="section level2">
<h2><span class="header-section-number">4.2</span> Layout</h2>
<p>Currently, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> supports several layouts, including:</p>
<ul>
<li>rectangular (by default)</li>
<li>slanted</li>
<li>circular</li>
<li>fan</li>
</ul>
<p>for <em>phylogram</em> (by default) and <em>cladogram</em> if user explicitly setting
<em>branch.length=‘none’</em>. Unrooted (equal angle and daylight methods), time-scaled
and 2-dimensional layouts are also supported.</p>
<div id="phylogram-and-cladogram" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Phylogram and Cladogram</h3>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggtree)
<span class="kw">set.seed</span>(<span class="dv">2017-02-16</span>)
tr &lt;-<span class="st"> </span><span class="kw">rtree</span>(<span class="dv">50</span>)
<span class="kw">ggtree</span>(tr)
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&quot;slanted&quot;</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&quot;fan&quot;</span>, <span class="dt">open.angle=</span><span class="dv">120</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&quot;equal_angle&quot;</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>)
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&quot;daylight&quot;</span>, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-8-1.png" width="768" /></p>
<p>There are also other possible layouts that can be drawn by modifying
scales/coordination, for examples, <a href="https://github.com/GuangchuangYu/ggtree/issues/87">reverse label of time
scale</a>, <a href="https://groups.google.com/d/msgid/bioc-ggtree/6db25ec7-ffb1-4776-bfe4-4a1ba239c8e8%40googlegroups.com">repropotion
circular/fan tree</a>, <em>etc.</em>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>()
<span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>()
<span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>() <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>()
<span class="kw">print</span>(<span class="kw">ggtree</span>(tr), <span class="dt">newpage=</span><span class="ot">TRUE</span>, <span class="dt">vp=</span>grid<span class="op">::</span><span class="kw">viewport</span>(<span class="dt">angle=</span><span class="op">-</span><span class="dv">30</span>, <span class="dt">width=</span>.<span class="dv">9</span>, <span class="dt">height=</span>.<span class="dv">9</span>))
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&#39;slanted&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">coord_flip</span>()
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&#39;slanted&#39;</span>, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">coord_flip</span>() <span class="op">+</span><span class="st"> </span><span class="kw">scale_y_reverse</span>() <span class="op">+</span><span class="kw">scale_x_reverse</span>()
<span class="kw">ggtree</span>(tr, <span class="dt">layout=</span><span class="st">&#39;circular&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">10</span>, <span class="ot">NA</span>)
<span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>() <span class="op">+</span><span class="st"> </span><span class="kw">coord_polar</span>(<span class="dt">theta=</span><span class="st">&#39;y&#39;</span>)
<span class="kw">ggtree</span>(tr) <span class="op">+</span><span class="st"> </span><span class="kw">scale_x_reverse</span>(<span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">0</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">coord_polar</span>(<span class="dt">theta=</span><span class="st">&#39;y&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-11-1.png" width="768" /></p>
</div>
<div id="time-scaled-tree" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Time-scaled tree</h3>
<p>A phylogenetic tree can be scaled by time (time-scaled tree) by specifying the parameter, <em>mrsd</em> (most recent sampling date).</p>
<pre class="sourceCode r"><code class="sourceCode r">tree2d &lt;-<span class="st"> </span><span class="kw">read.beast</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;twoD.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>))
<span class="kw">ggtree</span>(tree2d, <span class="dt">mrsd=</span><span class="st">&quot;2014-05-01&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-12-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="two-dimensional-tree" class="section level3">
<h3><span class="header-section-number">4.2.3</span> Two dimensional tree</h3>
<p><a href="http://bioconductor.org/packages/ggtree">ggtree</a> implemented two dimensional tree. It accepts parameter <em>yscale</em> to scale the y-axis based on the selected tree attribute. The attribute should be numerical variable. If it is <em>character</em>/<em>category</em> variable, user should provides a name vector of mapping the variable to numeric by passing it to parameter <em>yscale_mapping</em>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree2d, <span class="dt">mrsd=</span><span class="st">&quot;2014-05-01&quot;</span>,
       <span class="dt">yscale=</span><span class="st">&quot;NGS&quot;</span>, <span class="dt">yscale_mapping=</span><span class="kw">c</span>(<span class="dt">N2=</span><span class="dv">2</span>, <span class="dt">N3=</span><span class="dv">3</span>, <span class="dt">N4=</span><span class="dv">4</span>, <span class="dt">N5=</span><span class="dv">5</span>, <span class="dt">N6=</span><span class="dv">6</span>, <span class="dt">N7=</span><span class="dv">7</span>)) <span class="op">+</span>
<span class="st">           </span><span class="kw">theme_classic</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.line.x=</span><span class="kw">element_line</span>(), <span class="dt">axis.line.y=</span><span class="kw">element_line</span>()) <span class="op">+</span>
<span class="st">               </span><span class="kw">theme</span>(<span class="dt">panel.grid.major.x=</span><span class="kw">element_line</span>(<span class="dt">color=</span><span class="st">&quot;grey20&quot;</span>, <span class="dt">linetype=</span><span class="st">&quot;dotted&quot;</span>, <span class="dt">size=</span>.<span class="dv">3</span>),
                     <span class="dt">panel.grid.major.y=</span><span class="kw">element_blank</span>()) <span class="op">+</span>
<span class="st">                         </span><span class="kw">scale_y_continuous</span>(<span class="dt">labels=</span><span class="kw">paste0</span>(<span class="st">&quot;N&quot;</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">7</span>))</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-13-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>In this example, the figure demonstrates the quantity of <strong><em>y</em></strong> increase along the trunk. User can highlight the trunk with different line size or color using the functions described in <a href="treeManipulation.html">Tree Manipulation</a> vignette.</p>
</div>
</div>
<div id="displaying-tree-scale-evolution-distance" class="section level2">
<h2><span class="header-section-number">4.3</span> Displaying tree scale (evolution distance)</h2>
<p>To show tree scale, user can use <code>geom_treescale()</code> layer.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_treescale</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-14-1.png" width="384" style="display: block; margin: auto;" /></p>
<p><code>geom_treescale()</code> supports the following parameters:</p>
<ul>
<li><em>x</em> and <em>y</em> for tree scale position</li>
<li><em>width</em> for the length of the tree scale</li>
<li><em>fontsize</em> for the size of the text</li>
<li><em>linesize</em> for the size of the line</li>
<li><em>offset</em> for relative position of the line and the text</li>
<li><em>color</em> for color of the tree scale</li>
</ul>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_treescale</span>(<span class="dt">x=</span><span class="dv">0</span>, <span class="dt">y=</span><span class="dv">12</span>, <span class="dt">width=</span><span class="dv">6</span>, <span class="dt">color=</span><span class="st">&#39;red&#39;</span>)
<span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_treescale</span>(<span class="dt">fontsize=</span><span class="dv">8</span>, <span class="dt">linesize=</span><span class="dv">2</span>, <span class="dt">offset=</span><span class="op">-</span><span class="dv">1</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-16-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>We can also use <code>theme_tree2()</code> to display the tree scale by adding <em>x axis</em>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-17-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>Tree scale is not restricted to evolution distance, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> can re-scale the tree with other numerical variable. More details can be found in the <a href="treeAnnotation.html">Tree Annotation</a> vignette.</p>
</div>
<div id="displaying-nodestips" class="section level2">
<h2><span class="header-section-number">4.4</span> Displaying nodes/tips</h2>
<p>Showing all the internal nodes and tips in the tree can be done by adding a layer of points using <code>geom_nodepoint</code>, <code>geom_tippoint</code> or <code>geom_point</code>.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">shape=</span>isTip, <span class="dt">color=</span>isTip), <span class="dt">size=</span><span class="dv">3</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-18-1.png" width="288" style="display: block; margin: auto;" /></p>
<pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(tree) <span class="op">+</span><span class="st"> </span><span class="kw">geom_nodepoint</span>(<span class="dt">color=</span><span class="st">&quot;#b5e521&quot;</span>, <span class="dt">alpha=</span><span class="dv">1</span><span class="op">/</span><span class="dv">4</span>, <span class="dt">size=</span><span class="dv">10</span>)
p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tippoint</span>(<span class="dt">color=</span><span class="st">&quot;#FDAC4F&quot;</span>, <span class="dt">shape=</span><span class="dv">8</span>, <span class="dt">size=</span><span class="dv">3</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-19-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="displaying-labels" class="section level2">
<h2><span class="header-section-number">4.5</span> Displaying labels</h2>
<p>Users can use <code>geom_text</code> or <code>geom_label</code> to display the node (if available) and tip labels simultaneously or <code>geom_tiplab</code> to only display tip labels:</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;purple&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-20-1.png" width="288" style="display: block; margin: auto;" /></p>
<p><code>geom_tiplab</code> not only supports using <em>text</em> or <em>label</em> geom to display labels,
it also supports <em>image</em> geom to label tip with image files. A corresponding
geom, <code>geom_nodelab</code> is also provided for displaying node labels. For details of
label nodes with images, please refer to
the vignette,
<a href="https://guangchuangyu.github.io/software/ggtree/vignettes/ggtree-ggimage.html">Annotating phylogenetic tree with images</a>.</p>
<p>For <em>circular</em> and <em>unrooted</em> layout, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> supports rotating node labels according to the angles of the branches.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">angle=</span>angle), <span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-21-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>To make it more readable for human eye, <a href="http://bioconductor.org/packages/ggtree">ggtree</a> provides a <code>geom_tiplab2</code> for <code>circular</code> layout (see post <a href="https://groups.google.com/forum/?utm_medium=email&amp;utm_source=footer#!topic/bioc-ggtree/o35PV3iHO-0">1</a> and <a href="https://groups.google.com/forum/#!topic/bioc-ggtree/p42R5l8J-14">2</a>).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(tree, <span class="dt">layout=</span><span class="st">&quot;circular&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab2</span>(<span class="dt">color=</span><span class="st">&#39;blue&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-22-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>By default, the positions are based on the node positions, we can change them to based on the middle of the branch/edge.</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>(<span class="kw">aes</span>(<span class="dt">x=</span>branch), <span class="dt">size=</span><span class="dv">3</span>, <span class="dt">color=</span><span class="st">&quot;purple&quot;</span>, <span class="dt">vjust=</span><span class="op">-</span><span class="fl">0.3</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-23-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>Based on the middle of branch is very useful when annotating transition from parent node to child node.</p>
</div>
<div id="update-tree-view-with-a-new-tree" class="section level2">
<h2><span class="header-section-number">4.6</span> Update tree view with a new tree</h2>
<p>In previous example, we have a <em><code>p</code></em> object that stored the tree viewing of 13 tips and internal nodes highlighted with specific colored big dots. If users want to apply this pattern (we can imaging a more complex one) to a new tree, you don’t need to build the tree step by step. <code>ggtree</code> provides an operator, <em><code>%&lt;%</code></em>, for applying the visualization pattern to a new tree.</p>
<p>For example, the pattern in the <em><code>p</code></em> object will be applied to a new tree with 50 tips as shown below:</p>
<pre class="sourceCode r"><code class="sourceCode r">p <span class="op">%&lt;%</span><span class="st"> </span><span class="kw">rtree</span>(<span class="dv">50</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-24-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="theme" class="section level2">
<h2><span class="header-section-number">4.7</span> Theme</h2>
<p><code>theme_tree()</code> defined a totally blank canvas, while <em><code>theme_tree2()</code></em> adds
phylogenetic distance (via x-axis). These two themes all accept a parameter of
<em><code>bgcolor</code></em> that defined the background color. Users can pass
any <a href="http://ggplot2.tidyverse.org/reference/theme.html">theme components</a> to the
<code>theme_tree()</code> function to modify them.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(<span class="kw">rtree</span>(<span class="dv">30</span>), <span class="dt">color=</span><span class="st">&quot;red&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>(<span class="st">&quot;steelblue&quot;</span>)
<span class="kw">ggtree</span>(<span class="kw">rtree</span>(<span class="dv">20</span>), <span class="dt">color=</span><span class="st">&quot;white&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree</span>(<span class="st">&quot;black&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-26-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-a-list-of-trees" class="section level2">
<h2><span class="header-section-number">4.8</span> Visualize a list of trees</h2>
<p><code>ggtree</code> supports <code>multiPhylo</code> object and a list of trees can be viewed simultaneously.</p>
<pre class="sourceCode r"><code class="sourceCode r">trees &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">40</span>), rtree)
<span class="kw">class</span>(trees) &lt;-<span class="st"> &quot;multiPhylo&quot;</span>
<span class="kw">ggtree</span>(trees) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>.id, <span class="dt">scale=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-27-1.png" width="1152" /></p>
<p>One hundred bootstrap trees can also be view simultaneously.</p>
<pre class="sourceCode r"><code class="sourceCode r">btrees &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>, <span class="st">&quot;RAxML_bootstrap.H3&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>))
<span class="kw">ggtree</span>(btrees) <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>.id, <span class="dt">ncol=</span><span class="dv">10</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-28-1.png" width="1920" /></p>
<p>Another way to view the bootstrap trees is to merge them together to form a density tree. We can add a layer of the best tree on the top of the density tree.</p>
<pre class="sourceCode r"><code class="sourceCode r">p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(btrees, <span class="dt">layout=</span><span class="st">&quot;rectangular&quot;</span>, <span class="dt">color=</span><span class="st">&quot;lightblue&quot;</span>, <span class="dt">alpha=</span>.<span class="dv">3</span>)

best_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/RAxML&quot;</span>, <span class="st">&quot;RAxML_bipartitionsBranchLabels.H3&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>))
df &lt;-<span class="st"> </span><span class="kw">fortify</span>(best_tree, <span class="dt">branch.length=</span><span class="st">&#39;none&#39;</span>)
p<span class="op">+</span><span class="kw">geom_tree</span>(<span class="dt">data=</span>df, <span class="dt">color=</span><span class="st">&#39;firebrick&#39;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
</div>
<div id="rescale-tree" class="section level2">
<h2><span class="header-section-number">4.9</span> Rescale tree</h2>
<p>Most of the phylogenetic trees are scaled by evolutionary distance (substitution/site). In <code>ggtree</code>, users can re-scale a phylogenetic tree by any numerical variable inferred by evolutionary analysis (e.g. <em>dN/dS</em>).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;treeio&quot;</span>)
beast_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;examples/MCC_FluA_H3.tree&quot;</span>, <span class="dt">package=</span><span class="st">&quot;ggtree&quot;</span>)
beast_tree &lt;-<span class="st"> </span><span class="kw">read.beast</span>(beast_file)
beast_tree</code></pre>
<pre><code>## &#39;treedata&#39; S4 object that stored information of
##  &#39;C:/Users/YGC/Library/R/library/ggtree/examples/MCC_FluA_H3.tree&#39;.
## 
## ...@ phylo: 
## Phylogenetic tree with 76 tips and 75 internal nodes.
## 
## Tip labels:
##  A/Hokkaido/30-1-a/2013, A/New_York/334/2004, A/New_York/463/2005, A/New_York/452/1999, A/New_York/238/2005, A/New_York/523/1998, ...
## 
## Rooted; includes branch lengths.
## 
## with the following features available:
##  &#39;height&#39;,   &#39;height_0.95_HPD&#39;,  &#39;height_median&#39;,
##  &#39;height_range&#39;, &#39;length&#39;,   &#39;length_0.95_HPD&#39;,
##  &#39;length_median&#39;,    &#39;length_range&#39;, &#39;posterior&#39;,    &#39;rate&#39;,
##  &#39;rate_0.95_HPD&#39;,    &#39;rate_median&#39;,  &#39;rate_range&#39;.</code></pre>
<pre class="sourceCode r"><code class="sourceCode r">p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(beast_tree, <span class="dt">mrsd=</span><span class="st">&#39;2013-01-01&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Divergence time&quot;</span>)
p2 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(beast_tree, <span class="dt">branch.length=</span><span class="st">&#39;rate&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Substitution rate&quot;</span>)

<span class="kw">library</span>(cowplot)
<span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-30-1.png" width="960" /></p>
<pre class="sourceCode r"><code class="sourceCode r">mlcfile &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/PAML_Codeml&quot;</span>, <span class="st">&quot;mlc&quot;</span>, <span class="dt">package=</span><span class="st">&quot;treeio&quot;</span>)
mlc_tree &lt;-<span class="st"> </span><span class="kw">read.codeml_mlc</span>(mlcfile)
p1 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(mlc_tree) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;nucleotide substitutions per codon&quot;</span>)
p2 &lt;-<span class="st"> </span><span class="kw">ggtree</span>(mlc_tree, <span class="dt">branch.length=</span><span class="st">&#39;dN_vs_dS&#39;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;dN/dS tree&quot;</span>)
<span class="kw">plot_grid</span>(p1, p2, <span class="dt">ncol=</span><span class="dv">2</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-31-1.png" width="960" /></p>
<p>In addition to specify <code>branch.length</code> in tree visualization, users can change branch length stored in tree object by using <code>rescale_tree</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">beast_tree2 &lt;-<span class="st"> </span><span class="kw">rescale_tree</span>(beast_tree, <span class="dt">branch.length=</span><span class="st">&#39;rate&#39;</span>)
<span class="kw">ggtree</span>(beast_tree2) <span class="op">+</span><span class="st"> </span><span class="kw">theme_tree2</span>()</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
</div>
<div id="zoom-on-a-portion-of-tree" class="section level2">
<h2><span class="header-section-number">4.10</span> Zoom on a portion of tree</h2>
<p><code>ggtree</code> provides <em><code>gzoom</code></em> function that similar to <em><code>zoom</code></em> function provided in <code>ape</code>. This function plots simultaneously a whole phylogenetic tree and a portion of it. It aims at exploring very large trees.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;ape&quot;</span>)
<span class="kw">data</span>(chiroptera)
<span class="kw">library</span>(<span class="st">&quot;ggtree&quot;</span>)
<span class="kw">gzoom</span>(chiroptera, <span class="kw">grep</span>(<span class="st">&quot;Plecotus&quot;</span>, chiroptera<span class="op">$</span>tip.label))</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-33-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>Zoom in selected clade of a tree that was already annotated with <code>ggtree</code> is also supported.</p>
<pre class="sourceCode r"><code class="sourceCode r">groupInfo &lt;-<span class="st"> </span><span class="kw">split</span>(chiroptera<span class="op">$</span>tip.label, <span class="kw">gsub</span>(<span class="st">&quot;_</span><span class="ch">\\</span><span class="st">w+&quot;</span>, <span class="st">&quot;&quot;</span>, chiroptera<span class="op">$</span>tip.label))
chiroptera &lt;-<span class="st"> </span><span class="kw">groupOTU</span>(chiroptera, groupInfo)
p &lt;-<span class="st"> </span><span class="kw">ggtree</span>(chiroptera, <span class="kw">aes</span>(<span class="dt">color=</span>group)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tiplab</span>() <span class="op">+</span><span class="st"> </span><span class="kw">xlim</span>(<span class="ot">NA</span>, <span class="dv">23</span>)
<span class="kw">gzoom</span>(p, <span class="kw">grep</span>(<span class="st">&quot;Plecotus&quot;</span>, chiroptera<span class="op">$</span>tip.label), <span class="dt">xmax_adjust=</span><span class="dv">2</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-34-1.png" width="864" /></p>
</div>
<div id="color-tree" class="section level2">
<h2><span class="header-section-number">4.11</span> Color tree</h2>
<p>In <code>ggtree</code>, coloring phylogenetic tree is easy, by using <code>aes(color=VAR)</code> to map the color of tree based on a specific variable (numeric and category are both supported).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggtree</span>(beast_tree, <span class="kw">aes</span>(<span class="dt">color=</span>rate)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_color_continuous</span>(<span class="dt">low=</span><span class="st">&#39;darkgreen&#39;</span>, <span class="dt">high=</span><span class="st">&#39;red&#39;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position=</span><span class="st">&quot;right&quot;</span>)</code></pre>
<p><img src="treedata_files/figure-html/unnamed-chunk-35-1.png" width="480" /></p>
<p>User can use any feature (if available), including clade posterior and <em>dN/dS</em> <em>etc.</em>, to scale the color of the tree.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-yu_ggtree:_2017">
<p>Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2017. “Ggtree: An R Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.” <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210X.12628">https://doi.org/10.1111/2041-210X.12628</a>.</p>
</div>
<div id="ref-wickham_ggplot2_2009">
<p>Wickham, Hadley. 2009. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. 1st ed. Springer.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exporting-tree-with-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenetic-tree-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["treedata.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
